<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Connor Jops">
<meta name="dcterms.date" content="2023-01-31">

<title>Dev Brain Iso-Seq - Figure S9 - Correlation plots</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-sidebar floating">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Figure S9 - Correlation plots</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Dev Brain Iso-Seq</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/gandallab/Dev_Brain_IsoSeq" title="" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure1_BulkTxomeAnalysis.html" class="sidebar-item-text sidebar-link">Figure 1 - BulkTxomeAnalysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure1_TableS1.html" class="sidebar-item-text sidebar-link">Figure 1 - Table S1</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../code/Fig1_MetaCoverage.html" class="sidebar-item-text sidebar-link">Figure 1 - MetaCoverage</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_isoFraction.html" class="sidebar-item-text sidebar-link">Figure 2 - Isoform Fraction analysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_plotNovelGenes.html" class="sidebar-item-text sidebar-link">Figure 2 - Plot Novel Genes</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_PlotGene.html" class="sidebar-item-text sidebar-link">Figure 2 - Plot One Gene</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_novel_exons.html" class="sidebar-item-text sidebar-link">Figure 2 - Novel Exons</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_NF1Locus.html" class="sidebar-item-text sidebar-link">Figure 2 - NF1 Locus</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_PeptideSupport.html" class="sidebar-item-text sidebar-link">Figure 2 - Peptide Support</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_externalSupport.html" class="sidebar-item-text sidebar-link">Figure 2 - External Support</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_IsoformSwitchAnalyzeR.html" class="sidebar-item-text sidebar-link">Figure 3 - IsoformSwitchAnalyzeR</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_DTU_enrichments.html" class="sidebar-item-text sidebar-link">Figure 3 - DTU enrichments</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_RBPEnrichments.html" class="sidebar-item-text sidebar-link">Figure 3 - RBP Enrichments</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_APA.html" class="sidebar-item-text sidebar-link">Figure 3 - APA</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure4_combinedNetworks.html" class="sidebar-item-text sidebar-link">Figure 4 - Combined Networks</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure4_Networks_Enrichments.html" class="sidebar-item-text sidebar-link">Figure 4 - Networks Enrichments</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_SeuratClustering.html" class="sidebar-item-text sidebar-link">Figure 5 - Seurat Clustering</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_MarkerGenesFigs.html" class="sidebar-item-text sidebar-link">Figure 5 - Marker Genes</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_TableS5D.html" class="sidebar-item-text sidebar-link">Figure 5 - Table S5D</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_PseudotimeAnalysis.html" class="sidebar-item-text sidebar-link">Figure 5 - Pseudotime Analysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_Sankey.html" class="sidebar-item-text sidebar-link">Figure 5 - Sankey</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure6_rareVariant.html" class="sidebar-item-text sidebar-link">Figure 6 - Rare Variant</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure6_VEPConsequences.html" class="sidebar-item-text sidebar-link">Figure 6 - VEP Consequences</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure6_TrackPlots.html" class="sidebar-item-text sidebar-link">Figure 6 - Track Plots</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/FigureS9_cor_plots.html" class="sidebar-item-text sidebar-link active">Figure S9 - Correlation plots</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/FigureS9_GH.html" class="sidebar-item-text sidebar-link">Figure S9 - G, H</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#gene-expr" id="toc-gene-expr" class="nav-link active" data-scroll-target="#gene-expr">Gene expr</a></li>
  <li><a href="#umi" id="toc-umi" class="nav-link" data-scroll-target="#umi">UMI</a></li>
  <li><a href="#unique-transcripts-per-cell" id="toc-unique-transcripts-per-cell" class="nav-link" data-scroll-target="#unique-transcripts-per-cell">Unique transcripts per cell</a></li>
  <li><a href="#donor-v-donor-in-iso-seq" id="toc-donor-v-donor-in-iso-seq" class="nav-link" data-scroll-target="#donor-v-donor-in-iso-seq">Donor v donor in iso-seq</a>
  <ul class="collapse">
  <li><a href="#filtered-for-isoforms-present-in-3-or-more-cells" id="toc-filtered-for-isoforms-present-in-3-or-more-cells" class="nav-link" data-scroll-target="#filtered-for-isoforms-present-in-3-or-more-cells">Filtered for isoforms present in 3 or more cells</a></li>
  <li><a href="#digression---gene-expression-is-correlated-to-gene-length" id="toc-digression---gene-expression-is-correlated-to-gene-length" class="nav-link" data-scroll-target="#digression---gene-expression-is-correlated-to-gene-length">Digression - Gene expression is correlated to gene length</a></li>
  <li><a href="#unfiltered" id="toc-unfiltered" class="nav-link" data-scroll-target="#unfiltered">Unfiltered</a></li>
  <li><a href="#final-plots" id="toc-final-plots" class="nav-link" data-scroll-target="#final-plots">Final plots</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Figure S9 - Correlation plots</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Connor Jops </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 31, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2 ──
✔ ggplot2 3.3.6      ✔ purrr   0.3.4 
✔ tibble  3.1.8      ✔ dplyr   1.0.10
✔ tidyr   1.2.0      ✔ stringr 1.5.0 
✔ readr   2.1.2      ✔ forcats 0.5.2 
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rtracklayer)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges
Loading required package: stats4
Loading required package: BiocGenerics

Attaching package: 'BiocGenerics'

The following objects are masked from 'package:dplyr':

    combine, intersect, setdiff, union

The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs

The following objects are masked from 'package:base':

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min

Loading required package: S4Vectors

Attaching package: 'S4Vectors'

The following objects are masked from 'package:dplyr':

    first, rename

The following object is masked from 'package:tidyr':

    expand

The following objects are masked from 'package:base':

    expand.grid, I, unname

Loading required package: IRanges

Attaching package: 'IRanges'

The following objects are masked from 'package:dplyr':

    collapse, desc, slice

The following object is masked from 'package:purrr':

    reduce

Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.setenv</span>(<span class="st">"VROOM_CONNECTION_SIZE"</span> <span class="ot">=</span> <span class="dv">524288</span>) <span class="co"># 512kb - default is 128kb. entire line must fit inside this buffer and we have very long lines</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>base_size_pt <span class="ot">=</span> <span class="dv">16</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>(<span class="at">base_size =</span> base_size_pt))</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_update</span>(</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.2</span>), <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="dv">1</span>)),</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size =</span> <span class="fu">rel</span>(<span class="dv">1</span>)),</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="dv">1</span>)),</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size =</span> <span class="fu">rel</span>(<span class="dv">1</span>)),</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size =</span> <span class="fu">rel</span>(<span class="dv">1</span>))</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>base_size_mm <span class="ot">=</span> base_size_pt <span class="sc">*</span> <span class="fl">25.4</span> <span class="sc">/</span> <span class="fl">72.27</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>isoseq <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">"data/AllBarcode_newID_geneLevel.txt.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 19158 Columns: 4282
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr    (1): gene_name
dbl (4281): ACGGCATAGTAC, ACTGACTGATCC, ACTGGCACCGAG, AGCTAGCCCATG, AGTTTAGC...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>isoseq</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 19,158 × 4,282
   gene_name ACGGCATAG…¹ ACTGA…² ACTGG…³ AGCTA…⁴ AGTTT…⁵ ATCCG…⁶ CAGAC…⁷ CCCCC…⁸
   &lt;chr&gt;           &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
 1 A1BG                0       0       0       0       0       0       0       0
 2 A1BG-AS1            0       0       0       0       0       0       0       0
 3 A2M                 0       0       0       1       0       0       0       0
 4 AAAS                0       0       0       0       0       0       0       0
 5 AACS                1       0       0       0       0       0       0       0
 6 AADAT               0       0       0       0       0       0       0       0
 7 AAGAB               0       0       0       0       0       0       0       0
 8 AAK1                0       0       0       0       0       0       0       0
 9 AAMDC               0       0       0       0       0       0       0       0
10 AAMP                0       0       0       0       0       0       0       0
# … with 19,148 more rows, 4,273 more variables: GCCGTAAAACTC &lt;dbl&gt;,
#   GGGCTGCTTATA &lt;dbl&gt;, GGTTGCTGTACG &lt;dbl&gt;, GTAGCGAGATGC &lt;dbl&gt;,
#   TCAGCTCACCCT &lt;dbl&gt;, TGACATTCTATT &lt;dbl&gt;, TGGTGAACCAGA &lt;dbl&gt;,
#   TTCTATGCCCGC &lt;dbl&gt;, AAAAATAAATTG &lt;dbl&gt;, AAAACCGCGCAG &lt;dbl&gt;,
#   AAAAGCAACGTG &lt;dbl&gt;, AAAGGCGCGCGG &lt;dbl&gt;, AAATCAACTGTA &lt;dbl&gt;,
#   AACCACGTTTAA &lt;dbl&gt;, AACGTACTATCG &lt;dbl&gt;, AACTGCCTTGTA &lt;dbl&gt;,
#   AAGATCCAAGGC &lt;dbl&gt;, AAGGCAGAATCT &lt;dbl&gt;, AAGGCTGACCGA &lt;dbl&gt;, …</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>short_read <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">"data/single_cell/Raw_Single_Cell_4281.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 30119 Columns: 4282
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr    (1): gene_name
dbl (4281): GTATAACATGAA, CGAATATTCTGT, TGGAACAGCAAT, TACCCTGTAAGA, CTCACTGA...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>short_read</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 30,119 × 4,282
   gene_name GTATAACAT…¹ CGAAT…² TGGAA…³ TACCC…⁴ CTCAC…⁵ ACAGC…⁶ AGTCC…⁷ AGGCT…⁸
   &lt;chr&gt;           &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
 1 TSPAN6              0       1       0       1       0       1       1       0
 2 DPM1                1       0       0       0       0       1       3       2
 3 SCYL3               0       0       0       0       0       0       0       0
 4 C1orf112            0       0       0       0       1       0       1       0
 5 CFH                 0       0       0       0       0       0       0       0
 6 FUCA2               0       0       0       0       0       0       0       0
 7 GCLC                0       0       0       0       0       0       0       0
 8 NFYA                1       0       0       0       0       0       0       0
 9 STPG1               0       0       0       0       0       0       0       0
10 NIPAL3              0       0       0       0       0       0       0       0
# … with 30,109 more rows, 4,273 more variables: CCATCTATCAGG &lt;dbl&gt;,
#   AAGAAGTCCATT &lt;dbl&gt;, GCCGCAGAGTCA &lt;dbl&gt;, AACCTTCGCAAC &lt;dbl&gt;,
#   CTCTCCCCAATC &lt;dbl&gt;, TTTTATTTATAA &lt;dbl&gt;, CGTATCAGTATT &lt;dbl&gt;,
#   GTACTCTACAGA &lt;dbl&gt;, CAGTTTAGGAAA &lt;dbl&gt;, GTACCACCCTTC &lt;dbl&gt;,
#   ATGTGCACAGCA &lt;dbl&gt;, GTAGGTTAACTA &lt;dbl&gt;, CAGGGCCAGGCA &lt;dbl&gt;,
#   CGGGAACCCAAA &lt;dbl&gt;, GTGATTAGCCCA &lt;dbl&gt;, ATCTCCTTGTTA &lt;dbl&gt;,
#   CGGCTTATCGTA &lt;dbl&gt;, ACCCATCAAAGG &lt;dbl&gt;, GGCTGAGAGTCA &lt;dbl&gt;, …</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>cell_metadata <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">"ref/polioudakis_neuron2020/TableS3_Cell_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 33976 Columns: 14
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (7): Cell, Cluster, Subcluster, Layer, Index, Library, Phase
dbl (7): Donor, Gestation_week, Number_genes_detected, Number_UMI, Percentag...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>cell_metadata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 33,976 × 14
   Cell        Cluster Subcl…¹ Donor Layer Gesta…² Index Library Numbe…³ Numbe…⁴
   &lt;chr&gt;       &lt;chr&gt;   &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;     &lt;dbl&gt;   &lt;dbl&gt;
 1 TGCTAATACT… vRG     vRG_0     368 CP         17 N701  Plath      1366    2736
 2 TTCACGATTT… vRG     vRG_2     368 GZ         17 N702  Plath      3012    7318
 3 CTGTCAGAAT… vRG     vRG_2     368 GZ         17 N702  Plath      1597    3451
 4 CATAATATGT… vRG     vRG_0     368 GZ         17 N702  Plath      1551    2941
 5 TGCCAATCCG… vRG     vRG_0     368 GZ         17 N702  Plath      1569    2769
 6 TCGACTATTT… vRG     vRG_2     368 GZ         17 N702  Plath      1610    2994
 7 GTTACCCGGC… vRG     vRG_1     368 GZ         17 N702  Plath      1227    2092
 8 GCCACCGATA… vRG     vRG_1     368 GZ         17 N702  Plath      1197    2057
 9 GGGCAGATGT… vRG     vRG_1     368 GZ         17 N702  Plath      1237    2221
10 TTGAGCGGTA… vRG     vRG_0     368 GZ         17 N702  Plath      1344    2393
# … with 33,966 more rows, 4 more variables: Percentage_mitochondrial &lt;dbl&gt;,
#   S_phase_score &lt;dbl&gt;, G2M_phase_score &lt;dbl&gt;, Phase &lt;chr&gt;, and abbreviated
#   variable names ¹​Subcluster, ²​Gestation_week, ³​Number_genes_detected,
#   ⁴​Number_UMI</code></pre>
</div>
</div>
<section id="gene-expr" class="level2">
<h2 class="anchored" data-anchor-id="gene-expr">Gene expr</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>gene_counts <span class="ot">=</span> <span class="fu">inner_join</span>(</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  isoseq <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">cts_isoseq =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric))),</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">.keep =</span> <span class="st">"unused"</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  short_read <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">cts_short_read =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric))),</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">.keep =</span> <span class="st">"unused"</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span> <span class="st">"gene_name"</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">if_any</span>(<span class="fu">starts_with</span>(<span class="st">"cts_"</span>), <span class="sc">~</span>.x <span class="sc">==</span> <span class="dv">0</span>))</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>gene_counts</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 13,946 × 3
   gene_name cts_isoseq cts_short_read
   &lt;chr&gt;          &lt;dbl&gt;          &lt;dbl&gt;
 1 A1BG              18              6
 2 A1BG-AS1           3              6
 3 A2M               22             57
 4 AAAS             169            288
 5 AACS              69            198
 6 AADAT             79            174
 7 AAGAB             62             82
 8 AAK1              64            479
 9 AAMDC             95            189
10 AAMP             298            401
# … with 13,936 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>gene_counts_logRPM <span class="ot">=</span> gene_counts <span class="sc">%&gt;%</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">starts_with</span>(<span class="st">"cts_"</span>),</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span><span class="fu">log2</span>(<span class="dv">1</span> <span class="sc">+</span> (.x <span class="sc">/</span> (<span class="fu">sum</span>(.x) <span class="sc">/</span> <span class="dv">1000000</span>)))</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>gene_counts_logRPM</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 13,946 × 3
   gene_name cts_isoseq cts_short_read
   &lt;chr&gt;          &lt;dbl&gt;          &lt;dbl&gt;
 1 A1BG           2.77           0.916
 2 A1BG-AS1       0.976          0.916
 3 A2M            3.02           3.24 
 4 AAAS           5.79           5.44 
 5 AACS           4.54           4.92 
 6 AADAT          4.73           4.74 
 7 AAGAB          4.39           3.71 
 8 AAK1           4.43           6.17 
 9 AAMDC          4.98           4.85 
10 AAMP           6.60           5.91 
# … with 13,936 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>cor <span class="ot">=</span> <span class="fu">cor.test</span>(gene_counts_logRPM<span class="sc">$</span>cts_isoseq, gene_counts_logRPM<span class="sc">$</span>cts_short_read, <span class="at">method =</span> <span class="st">"pearson"</span>)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>cor</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Pearson's product-moment correlation

data:  gene_counts_logRPM$cts_isoseq and gene_counts_logRPM$cts_short_read
t = 242.12, df = 13944, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.8955646 0.9019444
sample estimates:
      cor 
0.8988021 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  gene_counts_logRPM,</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> cts_short_read, <span class="at">y =</span> cts_isoseq)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">color =</span> <span class="st">"#08519c"</span>) <span class="sc">+</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_abline</span>() <span class="sc">+</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"text"</span>,</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">str_c</span>(<span class="st">"r = "</span>, <span class="fu">round</span>(cor<span class="sc">$</span>estimate[[<span class="dv">1</span>]], <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">", p\uadvalue &lt; 2.2e-16"</span>),</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> base_size_mm,</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">16</span>,</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="st">"middle"</span>, <span class="at">hjust =</span> <span class="st">"left"</span>,</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">parse =</span> F</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Gene Expr (RNA\uadSeq)</span><span class="sc">\n</span><span class="st">log2(TPM+1)"</span>,</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Gene Expr (Iso\uadSeq)</span><span class="sc">\n</span><span class="st">log2(TPM+1)"</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS9_cor_plots_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"output/figures/supplement/FigS9_GeneExprCor.pdf"</span>, <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="umi" class="level2">
<h2 class="anchored" data-anchor-id="umi">UMI</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>UMI_counts <span class="ot">=</span> <span class="fu">inner_join</span>(</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  short_read <span class="sc">%&gt;%</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>gene_name) <span class="sc">%&gt;%</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), sum)) <span class="sc">%&gt;%</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="fu">everything</span>(), <span class="at">names_to =</span> <span class="st">"UMI"</span>, <span class="at">values_to =</span> <span class="st">"cts_short_read"</span>),</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  isoseq <span class="sc">%&gt;%</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>gene_name) <span class="sc">%&gt;%</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), sum)) <span class="sc">%&gt;%</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="fu">everything</span>(), <span class="at">names_to =</span> <span class="st">"UMI"</span>, <span class="at">values_to =</span> <span class="st">"cts_isoseq"</span>),</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span> <span class="st">"UMI"</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>UMI_counts</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4,281 × 3
   UMI          cts_short_read cts_isoseq
   &lt;chr&gt;                 &lt;dbl&gt;      &lt;dbl&gt;
 1 GTATAACATGAA           7756       3123
 2 CGAATATTCTGT           6478       2604
 3 TGGAACAGCAAT           5334       2524
 4 TACCCTGTAAGA           6227       2721
 5 CTCACTGAATGT           6731       3021
 6 ACAGCTCAGATA           5432       2452
 7 AGTCCCCCACTC           6623       3923
 8 AGGCTCCCATTT           5286       2998
 9 CCATCTATCAGG           5827       2671
10 AAGAAGTCCATT           4599       2187
# … with 4,271 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>cor <span class="ot">=</span> <span class="fu">cor.test</span>(UMI_counts<span class="sc">$</span>cts_isoseq, UMI_counts<span class="sc">$</span>cts_short_read, <span class="at">method =</span> <span class="st">"pearson"</span>)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>cor</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Pearson's product-moment correlation

data:  UMI_counts$cts_isoseq and UMI_counts$cts_short_read
t = 154.02, df = 4279, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.9157149 0.9248780
sample estimates:
      cor 
0.9204228 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  UMI_counts,</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> cts_short_read, <span class="at">y =</span> cts_isoseq)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"text"</span>,</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">str_c</span>(<span class="st">"r = "</span>, <span class="fu">round</span>(cor<span class="sc">$</span>estimate[[<span class="dv">1</span>]], <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">", p\uadvalue &lt; 2.2e-16"</span>),</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> base_size_mm,</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">5500</span>,</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="st">"middle"</span>, <span class="at">hjust =</span> <span class="st">"left"</span>,</span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">parse =</span> F</span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">8000</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">8000</span>, <span class="at">by =</span> <span class="dv">2000</span>)) <span class="sc">+</span></span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">5500</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">5000</span>, <span class="at">by =</span> <span class="dv">1000</span>)) <span class="sc">+</span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Total UMI count in RNA\uadSeq"</span>,</span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Total UMI count in Iso\uadSeq"</span></span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS9_cor_plots_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"output/figures/supplement/FigS9_UMICountCor.pdf"</span>, <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="unique-transcripts-per-cell" class="level2">
<h2 class="anchored" data-anchor-id="unique-transcripts-per-cell">Unique transcripts per cell</h2>
<p>mean = 538.6</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>isoseq_tx <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">"data/AllBarcode_newID_4281.txt.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 143490 Columns: 4283
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr    (2): gene_name, transcript_id
dbl (4281): ACGGCATAGTAC, ACTGACTGATCC, ACTGGCACCGAG, AGCTAGCCCATG, AGTTTAGC...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>isoseq_tx</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 143,490 × 4,283
   gene_name  transcri…¹ ACGGC…² ACTGA…³ ACTGG…⁴ AGCTA…⁵ AGTTT…⁶ ATCCG…⁷ CAGAC…⁸
   &lt;chr&gt;      &lt;chr&gt;        &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
 1 AL669831.3 ENST00000…       0       0       0       0       0       0       0
 2 MTND1P23   ENST00000…       0       0       0       0       0       0       0
 3 MTND2P28   ENST00000…       0       0       0       0       0       0       0
 4 MTCO1P12   ENST00000…       0       0       0       0       1       0       1
 5 MTCO2P12   ENST00000…       0       0       0       0       0       0       0
 6 MTATP6P1   ENST00000…       0       0       0       0       0       0       0
 7 LINC01409  ENST00000…       0       0       0       0       0       0       0
 8 LINC01409  ENST00000…       0       0       0       0       0       0       0
 9 LINC01128  ENST00000…       0       0       0       0       0       0       0
10 LINC01128  ENST00000…       0       0       0       0       0       0       0
# … with 143,480 more rows, 4,274 more variables: CCCCCACTTAGC &lt;dbl&gt;,
#   GCCGTAAAACTC &lt;dbl&gt;, GGGCTGCTTATA &lt;dbl&gt;, GGTTGCTGTACG &lt;dbl&gt;,
#   GTAGCGAGATGC &lt;dbl&gt;, TCAGCTCACCCT &lt;dbl&gt;, TGACATTCTATT &lt;dbl&gt;,
#   TGGTGAACCAGA &lt;dbl&gt;, TTCTATGCCCGC &lt;dbl&gt;, AAAAATAAATTG &lt;dbl&gt;,
#   AAAACCGCGCAG &lt;dbl&gt;, AAAAGCAACGTG &lt;dbl&gt;, AAAGGCGCGCGG &lt;dbl&gt;,
#   AAATCAACTGTA &lt;dbl&gt;, AACCACGTTTAA &lt;dbl&gt;, AACGTACTATCG &lt;dbl&gt;,
#   AACTGCCTTGTA &lt;dbl&gt;, AAGATCCAAGGC &lt;dbl&gt;, AAGGCAGAATCT &lt;dbl&gt;, …</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>tx_per_cell <span class="ot">=</span> isoseq_tx <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">!</span>(<span class="fu">c</span>(gene_name, transcript_id))) <span class="sc">%&gt;%</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">sum</span>(.x <span class="sc">&gt;</span> <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">everything</span>())</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>tx_per_cell</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4,281 × 2
   name         value
   &lt;chr&gt;        &lt;int&gt;
 1 ACGGCATAGTAC   198
 2 ACTGACTGATCC   147
 3 ACTGGCACCGAG   129
 4 AGCTAGCCCATG   226
 5 AGTTTAGCGGTC   930
 6 ATCCGACCATCA   103
 7 CAGACGCGACAT   253
 8 CCCCCACTTAGC   115
 9 GCCGTAAAACTC    65
10 GGGCTGCTTATA   201
# … with 4,271 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(tx_per_cell)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     name               value       
 Length:4281        Min.   :  10.0  
 Class :character   1st Qu.: 262.0  
 Mode  :character   Median : 452.0  
                    Mean   : 538.6  
                    3rd Qu.: 720.0  
                    Max.   :2785.0  </code></pre>
</div>
</div>
</section>
<section id="donor-v-donor-in-iso-seq" class="level2">
<h2 class="anchored" data-anchor-id="donor-v-donor-in-iso-seq">Donor v donor in iso-seq</h2>
<section id="filtered-for-isoforms-present-in-3-or-more-cells" class="level3">
<h3 class="anchored" data-anchor-id="filtered-for-isoforms-present-in-3-or-more-cells">Filtered for isoforms present in 3 or more cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>cpgz <span class="ot">=</span> rtracklayer<span class="sc">::</span><span class="fu">import</span>(<span class="st">"data/cp_vz_sc_unified_talon.gtf.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>donor_counts_filt <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">"data/AllBarcode_newID_byDonor_min3cells.txt.gz"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="sc">~</span><span class="fu">str_c</span>(<span class="st">"donor_"</span>, .x), <span class="at">.cols =</span> <span class="fu">where</span>(is.numeric))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 84478 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): gene_name, transcript_id
dbl (3): 370, 371, 372

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>donor_counts_filt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 84,478 × 5
   gene_name transcript_id   donor_370 donor_371 donor_372
   &lt;chr&gt;     &lt;chr&gt;               &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
 1 MTND1P23  ENST00000416931         0         3         0
 2 MTND2P28  ENST00000457540         3        52        38
 3 MTCO1P12  ENST00000414273         2         9        10
 4 LINC00115 ENST00000473798         4         6         9
 5 NOC2L     ENST00000327044        20        16        28
 6 NOC2L     ENST00000469563         7         9        14
 7 KLHL17    ENST00000338591         0         3         1
 8 HES4      ENST00000428771         7        28        36
 9 HES4      ENST00000304952        22        42        47
10 HES4      ENST00000481869        17        35        37
# … with 84,468 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>donor_counts_filt <span class="ot">=</span> donor_counts_filt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(transcript_id <span class="sc">%in%</span> <span class="fu">str_extract</span>(cpgz<span class="sc">$</span>transcript_id, <span class="st">"^[^</span><span class="sc">\\</span><span class="st">.]*"</span>))</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>donor_counts_filt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 84,224 × 5
   gene_name transcript_id   donor_370 donor_371 donor_372
   &lt;chr&gt;     &lt;chr&gt;               &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
 1 MTND1P23  ENST00000416931         0         3         0
 2 MTND2P28  ENST00000457540         3        52        38
 3 MTCO1P12  ENST00000414273         2         9        10
 4 LINC00115 ENST00000473798         4         6         9
 5 NOC2L     ENST00000327044        20        16        28
 6 NOC2L     ENST00000469563         7         9        14
 7 KLHL17    ENST00000338591         0         3         1
 8 HES4      ENST00000428771         7        28        36
 9 HES4      ENST00000304952        22        42        47
10 HES4      ENST00000481869        17        35        37
# … with 84,214 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>cpgz.exons <span class="ot">=</span> cpgz[cpgz<span class="sc">$</span>type <span class="sc">==</span> <span class="st">"exon"</span> <span class="sc">&amp;</span> <span class="fu">str_extract</span>(cpgz<span class="sc">$</span>transcript_id, <span class="st">"^[^</span><span class="sc">\\</span><span class="st">.]*"</span>) <span class="sc">%in%</span> donor_counts_filt<span class="sc">$</span>transcript_id]</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>cpgz.exons.by.gene <span class="ot">=</span> <span class="fu">split</span>(cpgz.exons, cpgz.exons<span class="sc">$</span>gene_name)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>donor_counts_filt <span class="ot">=</span> donor_counts_filt <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">enframe</span>(</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sum</span>(<span class="fu">width</span>(<span class="fu">reduce</span>(<span class="fu">ranges</span>(cpgz.exons.by.gene)))), <span class="co"># calculate exonic width</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"gene_name"</span>,</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">value =</span> <span class="st">"gene_length"</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "gene_name"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>donor_counts_filt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 84,224 × 6
   gene_name transcript_id   donor_370 donor_371 donor_372 gene_length
   &lt;chr&gt;     &lt;chr&gt;               &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;       &lt;int&gt;
 1 MTND1P23  ENST00000416931         0         3         0         372
 2 MTND2P28  ENST00000457540         3        52        38        1044
 3 MTCO1P12  ENST00000414273         2         9        10        1543
 4 LINC00115 ENST00000473798         4         6         9        1317
 5 NOC2L     ENST00000327044        20        16        28        4572
 6 NOC2L     ENST00000469563         7         9        14        4572
 7 KLHL17    ENST00000338591         0         3         1        2567
 8 HES4      ENST00000428771         7        28        36        1675
 9 HES4      ENST00000304952        22        42        47        1675
10 HES4      ENST00000481869        17        35        37        1675
# … with 84,214 more rows</code></pre>
</div>
</div>
<p>wrong but matches old plots:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>total <span class="ot">=</span> <span class="fu">sum</span>(donor_counts_filt <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"donor_"</span>)))</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>donor_counts_filt_logRPM_wrong <span class="ot">=</span> donor_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">starts_with</span>(<span class="st">"donor_"</span>),</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span><span class="fu">log10</span>(<span class="dv">1</span> <span class="sc">+</span> (.x <span class="sc">/</span> (total <span class="sc">/</span> <span class="dv">1000000</span>)))</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>donor_counts_filt_logRPM_wrong</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 84,224 × 6
   gene_name transcript_id   donor_370 donor_371 donor_372 gene_length
   &lt;chr&gt;     &lt;chr&gt;               &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;       &lt;int&gt;
 1 MTND1P23  ENST00000416931     0         0.288     0             372
 2 MTND2P28  ENST00000457540     0.288     1.24      1.11         1044
 3 MTCO1P12  ENST00000414273     0.212     0.583     0.617        1543
 4 LINC00115 ENST00000473798     0.354     0.460     0.583        1317
 5 NOC2L     ENST00000327044     0.863     0.780     0.991        4572
 6 NOC2L     ENST00000469563     0.505     0.583     0.732        4572
 7 KLHL17    ENST00000338591     0         0.288     0.119        2567
 8 HES4      ENST00000428771     0.505     0.991     1.09         1675
 9 HES4      ENST00000304952     0.898     1.15      1.20         1675
10 HES4      ENST00000481869     0.802     1.08      1.10         1675
# … with 84,214 more rows</code></pre>
</div>
</div>
<p>correctly calculate logRPM:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>donor_counts_filt_logRPM <span class="ot">=</span> donor_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">starts_with</span>(<span class="st">"donor_"</span>),</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span><span class="fu">log10</span>(<span class="dv">1</span> <span class="sc">+</span> (.x <span class="sc">/</span> (<span class="fu">sum</span>(.x) <span class="sc">/</span> <span class="dv">1000000</span>)))</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>donor_counts_filt_logRPM</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 84,224 × 6
   gene_name transcript_id   donor_370 donor_371 donor_372 gene_length
   &lt;chr&gt;     &lt;chr&gt;               &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;       &lt;int&gt;
 1 MTND1P23  ENST00000416931     0         0.565     0             372
 2 MTND2P28  ENST00000457540     0.816     1.68      1.42         1044
 3 MTCO1P12  ENST00000414273     0.672     0.955     0.880        1543
 4 LINC00115 ENST00000473798     0.924     0.803     0.841        1317
 5 NOC2L     ENST00000327044     1.58      1.18      1.29         4572
 6 NOC2L     ENST00000469563     1.14      0.955     1.01         4572
 7 KLHL17    ENST00000338591     0         0.565     0.220        2567
 8 HES4      ENST00000428771     1.14      1.41      1.39         1675
 9 HES4      ENST00000304952     1.62      1.58      1.50         1675
10 HES4      ENST00000481869     1.51      1.51      1.40         1675
# … with 84,214 more rows</code></pre>
</div>
</div>
<p>correctly calculate logTPM:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>donor_counts_filt_logTPM <span class="ot">=</span> donor_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">starts_with</span>(<span class="st">"donor_"</span>),</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span><span class="fu">log10</span>(<span class="dv">1</span> <span class="sc">+</span> ((.x <span class="sc">/</span> (gene_length <span class="sc">/</span> <span class="dv">1000</span>)) <span class="sc">/</span> (<span class="fu">sum</span>(.x) <span class="sc">/</span> <span class="dv">1000000</span>)))</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>donor_counts_filt_logTPM</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 84,224 × 6
   gene_name transcript_id   donor_370 donor_371 donor_372 gene_length
   &lt;chr&gt;     &lt;chr&gt;               &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;       &lt;int&gt;
 1 MTND1P23  ENST00000416931     0         0.913    0              372
 2 MTND2P28  ENST00000457540     0.800     1.66     1.40          1044
 3 MTCO1P12  ENST00000414273     0.531     0.792    0.722         1543
 4 LINC00115 ENST00000473798     0.820     0.704    0.740         1317
 5 NOC2L     ENST00000327044     0.958     0.615    0.702         4572
 6 NOC2L     ENST00000469563     0.583     0.440    0.480         4572
 7 KLHL17    ENST00000338591     0         0.310    0.0992        2567
 8 HES4      ENST00000428771     0.941     1.20     1.18          1675
 9 HES4      ENST00000304952     1.40      1.37     1.29          1675
10 HES4      ENST00000481869     1.30      1.29     1.19          1675
# … with 84,214 more rows</code></pre>
</div>
</div>
</section>
<section id="digression---gene-expression-is-correlated-to-gene-length" class="level3">
<h3 class="anchored" data-anchor-id="digression---gene-expression-is-correlated-to-gene-length">Digression - Gene expression is correlated to gene length</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">weighted.mean</span>(donor_counts_filt<span class="sc">$</span>gene_length, donor_counts_filt<span class="sc">$</span>donor_370)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 6988.761</code></pre>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">weighted.mean</span>(donor_counts_filt<span class="sc">$</span>gene_length, donor_counts_filt<span class="sc">$</span>donor_371)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 6925.062</code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">weighted.mean</span>(donor_counts_filt<span class="sc">$</span>gene_length, donor_counts_filt<span class="sc">$</span>donor_372)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 7101.579</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>cpgz.exons <span class="ot">=</span> cpgz[cpgz<span class="sc">$</span>type <span class="sc">==</span> <span class="st">"exon"</span> <span class="sc">&amp;</span> <span class="fu">str_extract</span>(cpgz<span class="sc">$</span>transcript_id, <span class="st">"^[^</span><span class="sc">\\</span><span class="st">.]*"</span>) <span class="sc">%in%</span> donor_counts_filt<span class="sc">$</span>transcript_id]</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>cpgz.exons.by.gene <span class="ot">=</span> <span class="fu">split</span>(cpgz.exons, cpgz.exons<span class="sc">$</span>gene_name)</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>donor_counts_filt <span class="ot">=</span> donor_counts_filt <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">enframe</span>(</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sum</span>(<span class="fu">width</span>(<span class="fu">reduce</span>(<span class="fu">ranges</span>(cpgz.exons.by.gene)))), <span class="co"># calculate exonic width</span></span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"gene_name"</span>,</span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">value =</span> <span class="st">"gene_length"</span></span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = c("gene_name", "gene_length")</code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>donor_counts_filt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 84,224 × 6
   gene_name transcript_id   donor_370 donor_371 donor_372 gene_length
   &lt;chr&gt;     &lt;chr&gt;               &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;       &lt;int&gt;
 1 MTND1P23  ENST00000416931         0         3         0         372
 2 MTND2P28  ENST00000457540         3        52        38        1044
 3 MTCO1P12  ENST00000414273         2         9        10        1543
 4 LINC00115 ENST00000473798         4         6         9        1317
 5 NOC2L     ENST00000327044        20        16        28        4572
 6 NOC2L     ENST00000469563         7         9        14        4572
 7 KLHL17    ENST00000338591         0         3         1        2567
 8 HES4      ENST00000428771         7        28        36        1675
 9 HES4      ENST00000304952        22        42        47        1675
10 HES4      ENST00000481869        17        35        37        1675
# … with 84,214 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>donor_gene_counts <span class="ot">=</span> donor_counts_filt <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene_name) <span class="sc">%&gt;%</span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">starts_with</span>(<span class="st">"donor_"</span>), sum), <span class="at">gene_length =</span> dplyr<span class="sc">::</span><span class="fu">first</span>(gene_length)) <span class="sc">%&gt;%</span></span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_length =</span> gene_length <span class="sc">/</span> <span class="dv">1000</span>,</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">across</span>(<span class="fu">starts_with</span>(<span class="st">"donor_"</span>), <span class="sc">~</span>.x <span class="sc">/</span> (<span class="fu">sum</span>(.x) <span class="sc">/</span> <span class="dv">1000000</span>))</span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span><span class="fu">log10</span>(.x <span class="sc">+</span> <span class="dv">1</span>))</span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a>  ) <span class="co">#%&gt;%</span></span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a><span class="co">#  mutate(across(starts_with("donor_"), ~na_if(.x, 0)))</span></span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a>donor_gene_counts</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 14,177 × 5
   gene_name donor_370 donor_371 donor_372 gene_length
   &lt;chr&gt;         &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;       &lt;dbl&gt;
 1 A1BG          0.816     0.565     0.950       0.496
 2 A2M           0.672     0.910     0.695       0.488
 3 AAAS          1.71      1.81      1.65        0.717
 4 AACS          1.40      1.44      1.19        0.995
 5 AADAT         1.25      1.41      1.36        1.01 
 6 AAGAB         1.14      1.44      1.17        0.972
 7 AAK1          1.40      1.25      1.29        0.939
 8 AAMDC         1.58      1.44      1.42        0.322
 9 AAMP          1.82      1.93      2.00        0.646
10 AAR2          1.49      1.18      1.34        0.568
# … with 14,167 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>  donor_gene_counts <span class="sc">%&gt;%</span></span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"donor_"</span>), <span class="at">names_to =</span> <span class="st">"donor"</span>, <span class="at">values_to =</span> <span class="st">"count"</span>),</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> count, <span class="at">y =</span> gene_length)</span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_point() +</span></span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_rug(alpha = 0.01) +</span></span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_hex(bins = 20) +</span></span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">abline</span>() <span class="sc">+</span></span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb70-14"><a href="#cb70-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Gene expression (logRPM)"</span>,</span>
<span id="cb70-15"><a href="#cb70-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Gene length (logKB)"</span></span>
<span id="cb70-16"><a href="#cb70-16" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb70-17"><a href="#cb70-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(donor))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"output/figures/supplement/FigS9_GeneLengthCor.pdf"</span>, <span class="at">width =</span> <span class="dv">14</span>, <span class="at">height =</span> <span class="dv">7</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(donor_gene_counts<span class="sc">$</span>donor_370, donor_gene_counts<span class="sc">$</span>gene_length, <span class="at">method =</span> <span class="st">"pearson"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Pearson's product-moment correlation

data:  donor_gene_counts$donor_370 and donor_gene_counts$gene_length
t = 61.234, df = 14175, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.4442555 0.4702924
sample estimates:
     cor 
0.457372 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(donor_gene_counts<span class="sc">$</span>donor_371, donor_gene_counts<span class="sc">$</span>gene_length, <span class="at">method =</span> <span class="st">"pearson"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Pearson's product-moment correlation

data:  donor_gene_counts$donor_371 and donor_gene_counts$gene_length
t = 61.173, df = 14175, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.4438894 0.4699372
sample estimates:
      cor 
0.4570113 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(donor_gene_counts<span class="sc">$</span>donor_372, donor_gene_counts<span class="sc">$</span>gene_length, <span class="at">method =</span> <span class="st">"pearson"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Pearson's product-moment correlation

data:  donor_gene_counts$donor_372 and donor_gene_counts$gene_length
t = 64.128, df = 14175, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.4613527 0.4868732
sample estimates:
      cor 
0.4742125 </code></pre>
</div>
</div>
</section>
<section id="unfiltered" class="level3">
<h3 class="anchored" data-anchor-id="unfiltered">Unfiltered</h3>
<p>We will use this but instead filter out isoforms with 0 counts in either donor from each comparison.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>donor_counts <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">"data/AllBarcode_newID_byDonor.txt.gz"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="sc">~</span><span class="fu">str_c</span>(<span class="st">"donor_"</span>, .x), <span class="at">.cols =</span> <span class="fu">where</span>(is.numeric))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 143490 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): gene_name, transcript_id
dbl (3): 370, 371, 372

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>donor_counts</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 143,490 × 5
   gene_name  transcript_id   donor_370 donor_371 donor_372
   &lt;chr&gt;      &lt;chr&gt;               &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
 1 AL669831.3 ENST00000447954         0         1         1
 2 MTND1P23   ENST00000416931         0         3         0
 3 MTND2P28   ENST00000457540         3        52        38
 4 MTCO1P12   ENST00000414273         2         9        10
 5 MTCO2P12   ENST00000427426         0         0         1
 6 MTATP6P1   ENST00000514057         0         0         0
 7 LINC01409  ENST00000665867         0         0         1
 8 LINC01409  ENST00000591702         0         0         1
 9 LINC01128  ENST00000445118         1         0         1
10 LINC01128  ENST00000669922         0         1         0
# … with 143,480 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>donor_counts <span class="ot">=</span> donor_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(transcript_id <span class="sc">%in%</span> <span class="fu">str_extract</span>(cpgz<span class="sc">$</span>transcript_id, <span class="st">"^[^</span><span class="sc">\\</span><span class="st">.]*"</span>))</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>donor_counts</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 142,938 × 5
   gene_name  transcript_id   donor_370 donor_371 donor_372
   &lt;chr&gt;      &lt;chr&gt;               &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
 1 AL669831.3 ENST00000447954         0         1         1
 2 MTND1P23   ENST00000416931         0         3         0
 3 MTND2P28   ENST00000457540         3        52        38
 4 MTCO1P12   ENST00000414273         2         9        10
 5 MTCO2P12   ENST00000427426         0         0         1
 6 MTATP6P1   ENST00000514057         0         0         0
 7 LINC01409  ENST00000665867         0         0         1
 8 LINC01409  ENST00000591702         0         0         1
 9 LINC01128  ENST00000445118         1         0         1
10 LINC01128  ENST00000669922         0         1         0
# … with 142,928 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>cpgz.exons <span class="ot">=</span> cpgz[cpgz<span class="sc">$</span>type <span class="sc">==</span> <span class="st">"exon"</span> <span class="sc">&amp;</span> <span class="fu">str_extract</span>(cpgz<span class="sc">$</span>transcript_id, <span class="st">"^[^</span><span class="sc">\\</span><span class="st">.]*"</span>) <span class="sc">%in%</span> donor_counts<span class="sc">$</span>transcript_id]</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>cpgz.exons.by.gene <span class="ot">=</span> <span class="fu">split</span>(cpgz.exons, cpgz.exons<span class="sc">$</span>gene_name)</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>donor_counts <span class="ot">=</span> donor_counts <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">enframe</span>(</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sum</span>(<span class="fu">width</span>(<span class="fu">reduce</span>(<span class="fu">ranges</span>(cpgz.exons.by.gene)))), <span class="co"># calculate exonic width</span></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"gene_name"</span>,</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">value =</span> <span class="st">"gene_length"</span></span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "gene_name"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>donor_counts</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 142,938 × 6
   gene_name  transcript_id   donor_370 donor_371 donor_372 gene_length
   &lt;chr&gt;      &lt;chr&gt;               &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;       &lt;int&gt;
 1 AL669831.3 ENST00000447954         0         1         1         355
 2 MTND1P23   ENST00000416931         0         3         0         372
 3 MTND2P28   ENST00000457540         3        52        38        1044
 4 MTCO1P12   ENST00000414273         2         9        10        1543
 5 MTCO2P12   ENST00000427426         0         0         1         682
 6 MTATP6P1   ENST00000514057         0         0         0         681
 7 LINC01409  ENST00000665867         0         0         1        2157
 8 LINC01409  ENST00000591702         0         0         1        2157
 9 LINC01128  ENST00000445118         1         0         1       13584
10 LINC01128  ENST00000669922         0         1         0       13584
# … with 142,928 more rows</code></pre>
</div>
</div>
<p>correctly calculate logRPM:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>donor_counts_logRPM <span class="ot">=</span> donor_counts <span class="sc">%&gt;%</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">starts_with</span>(<span class="st">"donor_"</span>),</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span><span class="fu">log2</span>(<span class="dv">1</span> <span class="sc">+</span> (.x <span class="sc">/</span> (<span class="fu">sum</span>(.x) <span class="sc">/</span> <span class="dv">1000000</span>)))</span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>donor_counts_logRPM</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 142,938 × 6
   gene_name  transcript_id   donor_370 donor_371 donor_372 gene_length
   &lt;chr&gt;      &lt;chr&gt;               &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;       &lt;int&gt;
 1 AL669831.3 ENST00000447954      0        0.901     0.713         355
 2 MTND1P23   ENST00000416931      0        1.85      0             372
 3 MTND2P28   ENST00000457540      2.67     5.53      4.66         1044
 4 MTCO1P12   ENST00000414273      2.20     3.14      2.88         1543
 5 MTCO2P12   ENST00000427426      0        0         0.713         682
 6 MTATP6P1   ENST00000514057      0        0         0             681
 7 LINC01409  ENST00000665867      0        0         0.713        2157
 8 LINC01409  ENST00000591702      0        0         0.713        2157
 9 LINC01128  ENST00000445118      1.48     0         0.713       13584
10 LINC01128  ENST00000669922      0        0.901     0           13584
# … with 142,928 more rows</code></pre>
</div>
</div>
<p>correctly calculate logTPM:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>donor_counts_logTPM <span class="ot">=</span> donor_counts <span class="sc">%&gt;%</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">starts_with</span>(<span class="st">"donor_"</span>),</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span><span class="fu">log10</span>(<span class="dv">1</span> <span class="sc">+</span> ((.x <span class="sc">/</span> (gene_length <span class="sc">/</span> <span class="dv">1000</span>)) <span class="sc">/</span> (<span class="fu">sum</span>(.x) <span class="sc">/</span> <span class="dv">1000000</span>)))</span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>donor_counts_logTPM</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 142,938 × 6
   gene_name  transcript_id   donor_370 donor_371 donor_372 gene_length
   &lt;chr&gt;      &lt;chr&gt;               &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;       &lt;int&gt;
 1 AL669831.3 ENST00000447954    0         0.537     0.447          355
 2 MTND1P23   ENST00000416931    0         0.903     0              372
 3 MTND2P28   ENST00000457540    0.789     1.65      1.38          1044
 4 MTCO1P12   ENST00000414273    0.521     0.782     0.711         1543
 5 MTCO2P12   ENST00000427426    0         0         0.287          682
 6 MTATP6P1   ENST00000514057    0         0         0              681
 7 LINC01409  ENST00000665867    0         0         0.113         2157
 8 LINC01409  ENST00000591702    0         0         0.113         2157
 9 LINC01128  ENST00000445118    0.0538    0         0.0200       13584
10 LINC01128  ENST00000669922    0         0.0269    0            13584
# … with 142,928 more rows</code></pre>
</div>
</div>
</section>
<section id="final-plots" class="level3">
<h3 class="anchored" data-anchor-id="final-plots">Final plots</h3>
<section id="donor-370-donor-371" class="level4">
<h4 class="anchored" data-anchor-id="donor-370-donor-371">Donor 370 + Donor 371</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>cor <span class="ot">=</span> <span class="fu">cor.test</span>(<span class="fu">na_if</span>(donor_counts_logRPM<span class="sc">$</span>donor_370, <span class="dv">0</span>), <span class="fu">na_if</span>(donor_counts_logRPM<span class="sc">$</span>donor_371, <span class="dv">0</span>), <span class="at">method =</span> <span class="st">"pearson"</span>)</span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>cor</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Pearson's product-moment correlation

data:  na_if(donor_counts_logRPM$donor_370, 0) and na_if(donor_counts_logRPM$donor_371, 0)
t = 397.42, df = 57300, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.8544174 0.8587770
sample estimates:
      cor 
0.8566125 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>  donor_counts_logRPM <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">starts_with</span>(<span class="st">"donor_"</span>), na_if, <span class="dv">0</span>)),</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> donor_370, <span class="at">y =</span> donor_371)</span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">color =</span> <span class="st">"#08519c"</span>) <span class="sc">+</span></span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_abline</span>() <span class="sc">+</span></span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(</span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"text"</span>,</span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">str_c</span>(<span class="st">"r = "</span>, <span class="fu">round</span>(cor<span class="sc">$</span>estimate[[<span class="dv">1</span>]], <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">", p\uadvalue &lt; 2.2e-16"</span>),</span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> base_size_mm,</span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">16</span>,</span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="st">"middle"</span>, <span class="at">hjust =</span> <span class="st">"left"</span>,</span>
<span id="cb94-13"><a href="#cb94-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">parse =</span> F</span>
<span id="cb94-14"><a href="#cb94-14" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb94-15"><a href="#cb94-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb94-16"><a href="#cb94-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb94-17"><a href="#cb94-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb94-18"><a href="#cb94-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Iso Expr (Donor 370)</span><span class="sc">\n</span><span class="st">log2(TPM+1)"</span>,</span>
<span id="cb94-19"><a href="#cb94-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Iso Expr (Donor 371)</span><span class="sc">\n</span><span class="st">log2(TPM+1)"</span></span>
<span id="cb94-20"><a href="#cb94-20" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 85636 rows containing missing values (geom_point).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="FigureS9_cor_plots_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"output/figures/supplement/FigS9_DonorCor_RPMno0_370_371.pdf"</span>, <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 85636 rows containing missing values (geom_point).</code></pre>
</div>
</div>
</section>
<section id="donor-370-donor-372" class="level4">
<h4 class="anchored" data-anchor-id="donor-370-donor-372">Donor 370 + Donor 372</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>cor <span class="ot">=</span> <span class="fu">cor.test</span>(<span class="fu">na_if</span>(donor_counts_logRPM<span class="sc">$</span>donor_370, <span class="dv">0</span>), <span class="fu">na_if</span>(donor_counts_logRPM<span class="sc">$</span>donor_372, <span class="dv">0</span>), <span class="at">method =</span> <span class="st">"pearson"</span>)</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>cor</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Pearson's product-moment correlation

data:  na_if(donor_counts_logRPM$donor_370, 0) and na_if(donor_counts_logRPM$donor_372, 0)
t = 418.37, df = 60683, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.8596590 0.8637556
sample estimates:
      cor 
0.8617213 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>  donor_counts_logRPM <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">starts_with</span>(<span class="st">"donor_"</span>), na_if, <span class="dv">0</span>)),</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> donor_370, <span class="at">y =</span> donor_372)</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">color =</span> <span class="st">"#08519c"</span>) <span class="sc">+</span></span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_abline</span>() <span class="sc">+</span></span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(</span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"text"</span>,</span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">str_c</span>(<span class="st">"r = "</span>, <span class="fu">round</span>(cor<span class="sc">$</span>estimate[[<span class="dv">1</span>]], <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">", p\uadvalue &lt; 2.2e-16"</span>),</span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> base_size_mm,</span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">16</span>,</span>
<span id="cb100-12"><a href="#cb100-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="st">"middle"</span>, <span class="at">hjust =</span> <span class="st">"left"</span>,</span>
<span id="cb100-13"><a href="#cb100-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">parse =</span> F</span>
<span id="cb100-14"><a href="#cb100-14" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb100-15"><a href="#cb100-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb100-16"><a href="#cb100-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb100-17"><a href="#cb100-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb100-18"><a href="#cb100-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Iso Expr (Donor 370)</span><span class="sc">\n</span><span class="st">log2(TPM+1)"</span>,</span>
<span id="cb100-19"><a href="#cb100-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Iso Expr (Donor 372)</span><span class="sc">\n</span><span class="st">log2(TPM+1)"</span></span>
<span id="cb100-20"><a href="#cb100-20" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 82253 rows containing missing values (geom_point).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="FigureS9_cor_plots_files/figure-html/unnamed-chunk-40-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"output/figures/supplement/FigS9_DonorCor_RPMno0_370_372.pdf"</span>, <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 82253 rows containing missing values (geom_point).</code></pre>
</div>
</div>
</section>
<section id="donor-371-donor-372" class="level4">
<h4 class="anchored" data-anchor-id="donor-371-donor-372">Donor 371 + Donor 372</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>cor <span class="ot">=</span> <span class="fu">cor.test</span>(<span class="fu">na_if</span>(donor_counts_logRPM<span class="sc">$</span>donor_371, <span class="dv">0</span>), <span class="fu">na_if</span>(donor_counts_logRPM<span class="sc">$</span>donor_372, <span class="dv">0</span>), <span class="at">method =</span> <span class="st">"pearson"</span>)</span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>cor</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Pearson's product-moment correlation

data:  na_if(donor_counts_logRPM$donor_371, 0) and na_if(donor_counts_logRPM$donor_372, 0)
t = 538.97, df = 80515, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.8833523 0.8863504
sample estimates:
      cor 
0.8848605 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>  donor_counts_logRPM <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">starts_with</span>(<span class="st">"donor_"</span>), na_if, <span class="dv">0</span>)),</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> donor_371, <span class="at">y =</span> donor_372)</span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">color =</span> <span class="st">"#08519c"</span>) <span class="sc">+</span></span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_abline</span>() <span class="sc">+</span></span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(</span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"text"</span>,</span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">str_c</span>(<span class="st">"r = "</span>, <span class="fu">round</span>(cor<span class="sc">$</span>estimate[[<span class="dv">1</span>]], <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">", p\uadvalue &lt; 2.2e-16"</span>),</span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> base_size_mm,</span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">16</span>,</span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="st">"middle"</span>, <span class="at">hjust =</span> <span class="st">"left"</span>,</span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">parse =</span> F</span>
<span id="cb106-14"><a href="#cb106-14" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb106-15"><a href="#cb106-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb106-16"><a href="#cb106-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb106-17"><a href="#cb106-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb106-18"><a href="#cb106-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Iso Expr (Donor 371)</span><span class="sc">\n</span><span class="st">log2(TPM+1)"</span>,</span>
<span id="cb106-19"><a href="#cb106-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Iso Expr (Donor 372)</span><span class="sc">\n</span><span class="st">log2(TPM+1)"</span></span>
<span id="cb106-20"><a href="#cb106-20" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 62421 rows containing missing values (geom_point).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="FigureS9_cor_plots_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"output/figures/supplement/FigS9_DonorCor_RPMno0_371_372.pdf"</span>, <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 62421 rows containing missing values (geom_point).</code></pre>
</div>
</div>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>