<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Michael Gandal">
<meta name="dcterms.date" content="2023-01-19">

<title>Dev Brain Iso-Seq - Figure 3 - DTU enrichments</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-sidebar floating">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Figure 3 - DTU enrichments</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Dev Brain Iso-Seq</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/gandallab/Dev_Brain_IsoSeq" title="" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure1_BulkTxomeAnalysis.html" class="sidebar-item-text sidebar-link">Figure 1 - BulkTxomeAnalysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure1_TableS1.html" class="sidebar-item-text sidebar-link">Figure 1 - Table S1</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../code/Fig1_MetaCoverage.html" class="sidebar-item-text sidebar-link">Figure 1 - MetaCoverage</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_isoFraction.html" class="sidebar-item-text sidebar-link">Figure 2 - Isoform Fraction analysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_plotNovelGenes.html" class="sidebar-item-text sidebar-link">Figure 2 - Plot Novel Genes</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_PlotGene.html" class="sidebar-item-text sidebar-link">Figure 2 - Plot One Gene</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_novel_exons.html" class="sidebar-item-text sidebar-link">Figure 2 - Novel Exons</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_NF1Locus.html" class="sidebar-item-text sidebar-link">Figure 2 - NF1 Locus</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_PeptideSupport.html" class="sidebar-item-text sidebar-link">Figure 2 - Peptide Support</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_IsoformSwitchAnalyzeR.html" class="sidebar-item-text sidebar-link">Figure 3 - IsoformSwitchAnalyzeR</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_DTU_enrichments.html" class="sidebar-item-text sidebar-link active">Figure 3 - DTU enrichments</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_RBPEnrichments.html" class="sidebar-item-text sidebar-link">Figure 3 - RBP Enrichments</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_APA.html" class="sidebar-item-text sidebar-link">Figure 3 - APA</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure4_combinedNetworks.html" class="sidebar-item-text sidebar-link">Figure 4 - Combined Networks</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure4_Networks_Enrichments.html" class="sidebar-item-text sidebar-link">Figure 4 - Networks Enrichments</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_SeuratClustering.html" class="sidebar-item-text sidebar-link">Figure 5 - Seurat Clustering</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_MarkerGenesFigs.html" class="sidebar-item-text sidebar-link">Figure 5 - Marker Genes</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_TableS5D.html" class="sidebar-item-text sidebar-link">Figure 5 - Table S5D</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_PseudotimeAnalysis.html" class="sidebar-item-text sidebar-link">Figure 5 - Pseudotime Analysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_Sankey.html" class="sidebar-item-text sidebar-link">Figure 5 - Sankey</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure6_rareVariant.html" class="sidebar-item-text sidebar-link">Figure 6 - Rare Variant</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure6_VEPConsequences.html" class="sidebar-item-text sidebar-link">Figure 6 - VEP Consequences</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure6_TrackPlots.html" class="sidebar-item-text sidebar-link">Figure 6 - Track Plots</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/FigureS9_cor_plots.html" class="sidebar-item-text sidebar-link">Figure S9 - Correlation plots</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#load-data" id="toc-load-data" class="nav-link active" data-scroll-target="#load-data">Load Data</a></li>
  <li><a href="#fig3d-dtu-pathway" id="toc-fig3d-dtu-pathway" class="nav-link" data-scroll-target="#fig3d-dtu-pathway">Fig3d: DTU pathway</a>
  <ul class="collapse">
  <li><a href="#dtu.up-and-dtu.down" id="toc-dtu.up-and-dtu.down" class="nav-link" data-scroll-target="#dtu.up-and-dtu.down">DTU.up and DTU.down</a></li>
  <li><a href="#dtu-all-pathways" id="toc-dtu-all-pathways" class="nav-link" data-scroll-target="#dtu-all-pathways">DTU all pathways</a></li>
  <li><a href="#also-dtu-not-dge-pathway" id="toc-also-dtu-not-dge-pathway" class="nav-link" data-scroll-target="#also-dtu-not-dge-pathway">Also: DTU not DGE pathway</a></li>
  </ul></li>
  <li><a href="#ewce" id="toc-ewce" class="nav-link" data-scroll-target="#ewce"><del>EWCE</del></a></li>
  <li><a href="#cell-type-makers" id="toc-cell-type-makers" class="nav-link" data-scroll-target="#cell-type-makers">Cell-type makers</a></li>
  <li><a href="#rare-variant-logit" id="toc-rare-variant-logit" class="nav-link" data-scroll-target="#rare-variant-logit">Rare variant logit</a>
  <ul class="collapse">
  <li><a href="#isoform-level" id="toc-isoform-level" class="nav-link" data-scroll-target="#isoform-level"><del>Isoform-level</del></a></li>
  <li><a href="#gene-level" id="toc-gene-level" class="nav-link" data-scroll-target="#gene-level">Gene-level</a></li>
  </ul></li>
  <li><a href="#rbp-enrichments" id="toc-rbp-enrichments" class="nav-link" data-scroll-target="#rbp-enrichments">RBP Enrichments</a>
  <ul class="collapse">
  <li><a href="#celines-rbp-targets" id="toc-celines-rbp-targets" class="nav-link" data-scroll-target="#celines-rbp-targets">Celine’s RBP Targets</a></li>
  <li><a href="#rbp-motif" id="toc-rbp-motif" class="nav-link" data-scroll-target="#rbp-motif">RBP motif:</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Figure 3 - DTU enrichments</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Michael Gandal </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 19, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="load-data" class="level2">
<h2 class="anchored" data-anchor-id="load-data">Load Data</h2>
<div class="cell" data-hash="Figure3_DTU_enrichments_cache/html/unnamed-chunk-1_3cdf78e3feb047ec708f0dc44c669cb3">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(rtracklayer)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggrepel)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(biomaRt)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">fig.width=</span><span class="dv">12</span>, <span class="at">fig.height=</span><span class="dv">8</span>) </span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"code/fisher_overlap.R"</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>tableS3.isoform <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"output/tables/TableS3_v3.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 102319 Columns: 17
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (5): isoform_id, gene_id, gene_name, condition_1, condition_2
dbl (9): DTU_dIF, DTU_pval, DTU_qval, DTE_log2FC, DTE_pval, DTE_qval, DGE_lo...
lgl (3): DTU, DTE, DGE

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>tableS3.isoform<span class="sc">$</span>ensg <span class="ot">=</span> <span class="fu">substr</span>(tableS3.isoform<span class="sc">$</span>gene_id,<span class="dv">1</span>,<span class="dv">15</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>tableS3.gene <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"output/tables/TableS3b_geneLevel.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 10809 Columns: 11
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): gene_name, gene_id
dbl (6): DTU_qval_min, DTU_pval_min, DTE_qval_min, DTE_pval_min, DGE_pval, D...
lgl (3): DTU, DTE, DGE

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">=</span> <span class="fu">read_table</span>(<span class="st">"data/cp_vz_0.75_min_7_recovery_talon_abundance_filtered.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  annot_gene_id = col_character(),
  annot_transcript_id = col_character(),
  annot_gene_name = col_character(),
  annot_transcript_name = col_character(),
  gene_novelty = col_character(),
  transcript_novelty = col_character(),
  ISM_subtype = col_character()
)
ℹ Use `spec()` for the full column specifications.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>geneAnno <span class="ot">=</span> rtracklayer<span class="sc">::</span><span class="fu">import</span>(<span class="st">"ref/gencode.v33lift37.annotation.gtf.gz"</span>) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(type<span class="sc">==</span><span class="st">'gene'</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>geneAnno<span class="sc">$</span>ensg <span class="ot">=</span> <span class="fu">substr</span>(geneAnno<span class="sc">$</span>gene_id,<span class="dv">1</span>,<span class="dv">15</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>datAnno <span class="ot">=</span> tableS3.isoform <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">ensg =</span> <span class="fu">substr</span>(gene_id,<span class="dv">1</span>,<span class="dv">15</span>))</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>datAnno <span class="ot">&lt;-</span> datAnno <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(cts <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">isoform_id =</span> annot_transcript_id, length))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "isoform_id"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>geneSetsForLDSC <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">ensg =</span> <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg), <span class="at">set=</span> <span class="st">"background"</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>geneSetsForLDSC <span class="ot">=</span> <span class="fu">rbind</span>(geneSetsForLDSC, <span class="fu">data.frame</span>(<span class="at">set =</span> <span class="st">"DTU"</span>, </span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ensg =</span> <span class="fu">unique</span>(tableS3.isoform<span class="sc">$</span>ensg[tableS3.isoform<span class="sc">$</span>DTU <span class="sc">==</span><span class="cn">TRUE</span>])))</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>geneSetsForLDSC <span class="ot">=</span> <span class="fu">rbind</span>(geneSetsForLDSC, <span class="fu">data.frame</span>(<span class="at">set =</span> <span class="st">"DTU.up"</span>, </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ensg =</span> <span class="fu">unique</span>(tableS3.isoform<span class="sc">$</span>ensg[tableS3.isoform<span class="sc">$</span>DTU <span class="sc">==</span><span class="cn">TRUE</span> <span class="sc">&amp;</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>                                                                      tableS3.isoform<span class="sc">$</span>DTU_dIF <span class="sc">&gt;</span> <span class="dv">0</span>])))</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>geneSetsForLDSC <span class="ot">=</span> <span class="fu">rbind</span>(geneSetsForLDSC, <span class="fu">data.frame</span>(<span class="at">set =</span> <span class="st">"DTU.down"</span>, </span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ensg =</span> <span class="fu">unique</span>(tableS3.isoform<span class="sc">$</span>ensg[tableS3.isoform<span class="sc">$</span>DTU <span class="sc">==</span><span class="cn">TRUE</span> <span class="sc">&amp;</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>                                                                      tableS3.isoform<span class="sc">$</span>DTU_dIF <span class="sc">&lt;</span> <span class="dv">0</span>])))</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>geneSetsForLDSC <span class="ot">=</span> <span class="fu">rbind</span>(geneSetsForLDSC, <span class="fu">data.frame</span>(<span class="at">set =</span> <span class="st">"DTU.not.DGE"</span>, </span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ensg =</span> <span class="fu">unique</span>(tableS3.isoform<span class="sc">$</span>ensg[tableS3.isoform<span class="sc">$</span>DTU <span class="sc">==</span><span class="cn">TRUE</span> <span class="sc">&amp;</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>                                                                      tableS3.isoform<span class="sc">$</span>DGE <span class="sc">==</span> <span class="cn">FALSE</span>])))</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>geneSetsForLDSC <span class="ot">=</span> <span class="fu">rbind</span>(geneSetsForLDSC, <span class="fu">data.frame</span>(<span class="at">set =</span> <span class="st">"DTU.not.DTE"</span>, </span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ensg =</span> <span class="fu">unique</span>(tableS3.isoform<span class="sc">$</span>ensg[tableS3.isoform<span class="sc">$</span>DTU <span class="sc">==</span><span class="cn">TRUE</span> <span class="sc">&amp;</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>                                                                      tableS3.isoform<span class="sc">$</span>DTE <span class="sc">==</span> <span class="cn">FALSE</span>])))</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>geneSetsForLDSC <span class="ot">=</span> <span class="fu">rbind</span>(geneSetsForLDSC, <span class="fu">data.frame</span>(<span class="at">set =</span> <span class="st">"DTE"</span>, </span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ensg =</span> <span class="fu">unique</span>(tableS3.isoform<span class="sc">$</span>ensg[tableS3.isoform<span class="sc">$</span>DTE <span class="sc">==</span><span class="cn">TRUE</span>])))</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>geneSetsForLDSC <span class="ot">=</span> <span class="fu">rbind</span>(geneSetsForLDSC, <span class="fu">data.frame</span>(<span class="at">set =</span> <span class="st">"DTE.up"</span>, </span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ensg =</span> <span class="fu">unique</span>(tableS3.isoform<span class="sc">$</span>ensg[tableS3.isoform<span class="sc">$</span>DTE <span class="sc">==</span><span class="cn">TRUE</span> <span class="sc">&amp;</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>                                                                      tableS3.isoform<span class="sc">$</span>DTE_log2FC <span class="sc">&gt;</span> <span class="dv">0</span>])))</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>geneSetsForLDSC <span class="ot">=</span> <span class="fu">rbind</span>(geneSetsForLDSC, <span class="fu">data.frame</span>(<span class="at">set =</span> <span class="st">"DTE.down"</span>, </span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ensg =</span> <span class="fu">unique</span>(tableS3.isoform<span class="sc">$</span>ensg[tableS3.isoform<span class="sc">$</span>DTE <span class="sc">==</span><span class="cn">TRUE</span> <span class="sc">&amp;</span></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>                                                                      tableS3.isoform<span class="sc">$</span>DTE_log2FC <span class="sc">&lt;</span> <span class="dv">0</span>])))</span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>geneSetsForLDSC <span class="ot">=</span> <span class="fu">rbind</span>(geneSetsForLDSC, <span class="fu">data.frame</span>(<span class="at">set =</span> <span class="st">"DGE"</span>, </span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ensg =</span> <span class="fu">unique</span>(tableS3.isoform<span class="sc">$</span>ensg[tableS3.isoform<span class="sc">$</span>DGE <span class="sc">==</span><span class="cn">TRUE</span>])))</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>geneSetsForLDSC <span class="ot">=</span> <span class="fu">rbind</span>(geneSetsForLDSC, <span class="fu">data.frame</span>(<span class="at">set =</span> <span class="st">"DGE.up"</span>, </span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ensg =</span> <span class="fu">unique</span>(tableS3.isoform<span class="sc">$</span>ensg[tableS3.isoform<span class="sc">$</span>DGE <span class="sc">==</span><span class="cn">TRUE</span> <span class="sc">&amp;</span></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>                                                                      tableS3.isoform<span class="sc">$</span>DGE_log2FC <span class="sc">&gt;</span> <span class="dv">0</span>])))</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>geneSetsForLDSC <span class="ot">=</span> <span class="fu">rbind</span>(geneSetsForLDSC, <span class="fu">data.frame</span>(<span class="at">set =</span> <span class="st">"DGE.down"</span>, </span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ensg =</span> <span class="fu">unique</span>(tableS3.isoform<span class="sc">$</span>ensg[tableS3.isoform<span class="sc">$</span>DGE <span class="sc">==</span><span class="cn">TRUE</span> <span class="sc">&amp;</span></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>                                                                      tableS3.isoform<span class="sc">$</span>DGE_log2FC <span class="sc">&lt;</span> <span class="dv">0</span>])))</span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(geneSetsForLDSC, <span class="at">file=</span><span class="st">'output/tables/TableS3_DTU_DGE_DTE_forLDSC.tsv'</span>)</span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>geneSetsForLDSC<span class="sc">$</span>gene <span class="ot">=</span> datAnno<span class="sc">$</span>gene_name[<span class="fu">match</span>(geneSetsForLDSC<span class="sc">$</span>ensg, datAnno<span class="sc">$</span>ensg)]</span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(geneSetsForLDSC[,<span class="fu">c</span>(<span class="st">"gene"</span>, <span class="st">"set"</span>)], <span class="at">file=</span><span class="st">'output/tables/TableS3_DTU_DGE_DTE_forTransite.tsv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="fig3d-dtu-pathway" class="level2">
<h2 class="anchored" data-anchor-id="fig3d-dtu-pathway">Fig3d: DTU pathway</h2>
<div class="cell" data-hash="Figure3_DTU_enrichments_cache/html/unnamed-chunk-2_ed2c35d23fbb4c8cfbac2a2e2bb87b0a">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>DTU <span class="ot">&lt;-</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU<span class="sc">==</span>T) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval_min) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>DTU.bg <span class="ot">=</span> tableS3.isoform<span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id)<span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Pathway analysis with gProfileR</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Note: ordered query here because genes are ranked by DTU significance. Usually this will be F</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Always use the matching background gene set</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>path <span class="ot">=</span> gprofiler2<span class="sc">::</span><span class="fu">gost</span>(<span class="at">query=</span>DTU,<span class="at">ordered_query =</span> T,<span class="at">correction_method =</span> <span class="st">'fdr'</span>,</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">custom_bg =</span> DTU.bg, <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">"GO"</span>,<span class="st">"KEGG"</span>, <span class="st">"REACTOME"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Detected custom background input, domain scope is set to 'custom'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Filter results for terms between 5-3000 genes. Also here I remove results where a child GO term is also included, just because there were a lot of results. Can remove this </span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>df_path <span class="ot">=</span> <span class="fu">as_tibble</span>(path<span class="sc">$</span>result) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term_size <span class="sc">&lt;</span> <span class="dv">3000</span>, path<span class="sc">$</span>result<span class="sc">$</span>term_size<span class="sc">&gt;</span><span class="dv">5</span>) </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>df_path <span class="ot">&lt;-</span> df_path <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>term_id <span class="sc">%in%</span> <span class="fu">unlist</span>(df_path<span class="sc">$</span>parents))</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot top 5 results per database</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>Fig3d <span class="ot">&lt;-</span> df_path <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(source) <span class="sc">%&gt;%</span> <span class="fu">slice_min</span>(p_value,<span class="at">n=</span><span class="dv">5</span>,<span class="at">with_ties =</span> T) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(term_name, <span class="sc">-</span>p_value), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p_value), <span class="at">fill=</span>source)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position=</span><span class="fu">position_identity</span>()) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>)  <span class="sc">+</span> </span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(source<span class="sc">~</span>., <span class="at">space =</span> <span class="st">'free'</span>, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>)</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>Fig3d</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_DTU_enrichments_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(Fig3d, file="output/figures/Fig3d.pdf",width=4.5,height=3)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="dtu.up-and-dtu.down" class="level3">
<h3 class="anchored" data-anchor-id="dtu.up-and-dtu.down">DTU.up and DTU.down</h3>
<div class="cell" data-hash="Figure3_DTU_enrichments_cache/html/unnamed-chunk-3_674151fb70839ea46fa21cddd647e6e7">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>DTU.up <span class="ot">&lt;-</span> tableS3.isoform <span class="sc">%&gt;%</span>  <span class="fu">filter</span>(DTU_dIF <span class="sc">&gt;</span> <span class="dv">0</span>, DTU_qval <span class="sc">&lt;</span> .<span class="dv">05</span>)  <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval) <span class="sc">%&gt;%</span> </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id)<span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>) </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>DTU.down <span class="ot">&lt;-</span> tableS3.isoform <span class="sc">%&gt;%</span>  <span class="fu">filter</span>(DTU_dIF <span class="sc">&lt;</span> <span class="dv">0</span>, DTU_qval <span class="sc">&lt;</span> .<span class="dv">05</span>)  <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval) <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id)<span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>DTU.bg <span class="ot">=</span> tableS3.isoform  <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id)<span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Pathway analysis with gProfileR</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Note: ordered query here because genes are ranked by DTU significance. Usually this will be F</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Always use the matching background gene set</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>path.up <span class="ot">=</span> gprofiler2<span class="sc">::</span><span class="fu">gost</span>(<span class="at">query=</span>DTU.up,<span class="at">ordered_query =</span> T,<span class="at">correction_method =</span> <span class="st">'fdr'</span>,</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>                        <span class="at">custom_bg =</span> DTU.bg, <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">"GO"</span>,<span class="st">"KEGG"</span>, <span class="st">"REACTOME"</span>, <span class="st">"CORUM"</span>, <span class="st">"WP"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Detected custom background input, domain scope is set to 'custom'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Filter results for terms between 5-3000 genes. Also here I remove results where a child GO term is also included, just because there were a lot of results. Can remove this </span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>df_path.up <span class="ot">=</span> <span class="fu">as_tibble</span>(path.up<span class="sc">$</span>result) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term_size <span class="sc">&lt;</span> <span class="dv">3000</span>, term_size<span class="sc">&gt;</span><span class="dv">5</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">direction=</span><span class="st">'up'</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>df_path.up <span class="ot">&lt;-</span> df_path.up <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>term_id <span class="sc">%in%</span> <span class="fu">unlist</span>(df_path.up<span class="sc">$</span>parents))</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot top 5 results per database</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>Fig3d.up <span class="ot">&lt;-</span> df_path.up <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(source) <span class="sc">%&gt;%</span> <span class="fu">slice_min</span>(p_value,<span class="at">n=</span><span class="dv">8</span>) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(term_name, <span class="sc">-</span>p_value), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p_value), <span class="at">fill=</span>source)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>, <span class="at">position=</span><span class="fu">position_identity</span>()) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>)  <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"DTU.up"</span>) <span class="sc">+</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(source<span class="sc">~</span>., <span class="at">space =</span> <span class="st">'free'</span>, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>)</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>path.down <span class="ot">=</span> gprofiler2<span class="sc">::</span><span class="fu">gost</span>(<span class="at">query=</span>DTU.down,<span class="at">ordered_query =</span> T,<span class="at">correction_method =</span> <span class="st">'fdr'</span>,</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>                        <span class="at">custom_bg =</span> DTU.bg, <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">"GO"</span>,<span class="st">"KEGG"</span>, <span class="st">"REACTOME"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Detected custom background input, domain scope is set to 'custom'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Filter results for terms between 5-3000 genes. Also here I remove results where a child GO term is also included, just because there were a lot of results. Can remove this </span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>df_path.down <span class="ot">=</span> <span class="fu">as_tibble</span>(path.down<span class="sc">$</span>result) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term_size <span class="sc">&lt;</span> <span class="dv">3000</span>, term_size<span class="sc">&gt;</span><span class="dv">5</span>)  <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">direction=</span><span class="st">'down'</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>df_path.down <span class="ot">&lt;-</span> df_path.down <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>term_id <span class="sc">%in%</span> <span class="fu">unlist</span>(df_path<span class="sc">$</span>parents))</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot top 5 results per database</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>Fig3d.down <span class="ot">&lt;-</span> df_path.down <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(source) <span class="sc">%&gt;%</span> <span class="fu">slice_min</span>(p_value,<span class="at">n=</span><span class="dv">8</span>) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(term_name, <span class="sc">-</span>p_value), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p_value), <span class="at">fill=</span>source)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>, <span class="at">position=</span><span class="fu">position_identity</span>()) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>)  <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"DTU.down"</span>) <span class="sc">+</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(source<span class="sc">~</span>., <span class="at">space =</span> <span class="st">'free'</span>, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>)</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(Fig3d.down, Fig3d.up,<span class="at">ncol=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_DTU_enrichments_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="768"></p>
</div>
</div>
</section>
<section id="dtu-all-pathways" class="level3">
<h3 class="anchored" data-anchor-id="dtu-all-pathways">DTU all pathways</h3>
<div class="cell" data-hash="Figure3_DTU_enrichments_cache/html/unnamed-chunk-4_8823a7e7a61aaf8b81bff6f6278fee1c">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>DTU <span class="ot">&lt;-</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU<span class="sc">==</span>T) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval_min) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>DTU.bg <span class="ot">=</span> tableS3.gene<span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id)<span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>path <span class="ot">=</span> gprofiler2<span class="sc">::</span><span class="fu">gost</span>(<span class="at">query=</span>DTU,<span class="at">ordered_query =</span> T,<span class="at">correction_method =</span> <span class="st">'fdr'</span>,</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">custom_bg =</span> DTU.bg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Detected custom background input, domain scope is set to 'custom'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>df_path <span class="ot">=</span> <span class="fu">as_tibble</span>(path<span class="sc">$</span>result) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term_size <span class="sc">&lt;</span> <span class="dv">3000</span>, path<span class="sc">$</span>result<span class="sc">$</span>term_size<span class="sc">&gt;</span><span class="dv">5</span>) </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>df_path <span class="ot">&lt;-</span> df_path <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>term_id <span class="sc">%in%</span> <span class="fu">unlist</span>(df_path<span class="sc">$</span>parents))</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>df_path <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(source) <span class="sc">%&gt;%</span> <span class="fu">slice_min</span>(p_value,<span class="at">n=</span><span class="dv">5</span>,<span class="at">with_ties =</span> F) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(term_name, <span class="sc">-</span>p_value), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p_value), <span class="at">fill=</span>source)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>, <span class="at">position=</span><span class="fu">position_identity</span>()) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>)  <span class="sc">+</span> </span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(source<span class="sc">~</span>., <span class="at">space =</span> <span class="st">'free'</span>, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_DTU_enrichments_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="also-dtu-not-dge-pathway" class="level3">
<h3 class="anchored" data-anchor-id="also-dtu-not-dge-pathway">Also: DTU not DGE pathway</h3>
<div class="cell" data-hash="Figure3_DTU_enrichments_cache/html/unnamed-chunk-5_9cd9f54214d6ea002744720e66d912ac">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>DTU <span class="ot">&lt;-</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU<span class="sc">==</span>T, DGE_pval<span class="sc">&gt;</span>.<span class="dv">05</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval_min) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>DTU.bg <span class="ot">=</span> tableS3.gene<span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id)<span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>path <span class="ot">=</span> gprofiler2<span class="sc">::</span><span class="fu">gost</span>(<span class="at">query=</span>DTU,<span class="at">ordered_query =</span> T,<span class="at">correction_method =</span> <span class="st">'fdr'</span>,</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">custom_bg =</span> DTU.bg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Detected custom background input, domain scope is set to 'custom'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>df_path <span class="ot">=</span> <span class="fu">as_tibble</span>(path<span class="sc">$</span>result) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term_size <span class="sc">&lt;</span> <span class="dv">3000</span>, path<span class="sc">$</span>result<span class="sc">$</span>term_size<span class="sc">&gt;</span><span class="dv">5</span>) </span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>df_path <span class="ot">&lt;-</span> df_path <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>term_id <span class="sc">%in%</span> <span class="fu">unlist</span>(df_path<span class="sc">$</span>parents))</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>df_path <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(source) <span class="sc">%&gt;%</span> <span class="fu">slice_min</span>(p_value,<span class="at">n=</span><span class="dv">30</span>) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(term_name, <span class="sc">-</span>p_value), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p_value), <span class="at">fill=</span>source)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>)  <span class="sc">+</span> </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(source<span class="sc">~</span>., <span class="at">space =</span> <span class="st">'free'</span>, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_DTU_enrichments_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="ewce" class="level2">
<h2 class="anchored" data-anchor-id="ewce"><del>EWCE</del></h2>
<p>Using fisher ORA instead</p>
<div class="cell" data-hash="Figure3_DTU_enrichments_cache/html/unnamed-chunk-6_14898c78806609b86cf93b94668e0a6f">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="cn">FALSE</span>) {</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">"ref/EWCE/CellTypeData_DamonNeuralFetalOnly.rda"</span>)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  DTU.up <span class="ot">&lt;-</span> tableS3.isoform <span class="sc">%&gt;%</span>  <span class="fu">filter</span>(DTU_dIF <span class="sc">&gt;</span> <span class="dv">0</span>, DTU_qval <span class="sc">&lt;</span> .<span class="dv">05</span>)  <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval) <span class="sc">%&gt;%</span> </span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name)<span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>) </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  DTU.down <span class="ot">&lt;-</span> tableS3.isoform <span class="sc">%&gt;%</span>  <span class="fu">filter</span>(DTU_dIF <span class="sc">&lt;</span> <span class="dv">0</span>, DTU_qval <span class="sc">&lt;</span> .<span class="dv">05</span>)  <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval) <span class="sc">%&gt;%</span> </span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name)<span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  DTU <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">c</span>(DTU.up, DTU.down))</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  DTU.notDGE <span class="ot">&lt;-</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU<span class="sc">==</span>T, DGE_pval<span class="sc">&gt;</span>.<span class="dv">05</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval_min) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  DTU.bg <span class="ot">=</span> tableS3.isoform  <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name)<span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() </span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>  gene_sets <span class="ot">=</span> <span class="fu">list</span>(DTU,DTU.notDGE)</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(gene_sets) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"DTU"</span>, <span class="st">"DTU.notDGE"</span>)</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>  df_ewce <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene_sets)) {</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(i)</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>    these_genes <span class="ot">=</span> gene_sets[[i]]</span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>    res<span class="ot">=</span>EWCE<span class="sc">::</span><span class="fu">bootstrap_enrichment_test</span>(ctd, <span class="at">hits =</span> these_genes, <span class="at">bg=</span>DTU.bg,</span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">genelistSpecies =</span> <span class="st">'human'</span>,<span class="at">sctSpecies =</span> <span class="st">'human'</span>,</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">annotLevel =</span><span class="dv">2</span>,<span class="at">verbose =</span> F,<span class="at">no_cores =</span> <span class="dv">10</span>)</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>    df_ewce <span class="ot">=</span> <span class="fu">rbind</span>(df_ewce,<span class="fu">data.frame</span>(<span class="at">list=</span><span class="fu">names</span>(gene_sets)[i], res<span class="sc">$</span>results))</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>  df_ewce<span class="sc">$</span>label <span class="ot">=</span> <span class="fu">signif</span>(<span class="sc">-</span><span class="fu">log10</span>(df_ewce<span class="sc">$</span>q<span class="fl">+1e-5</span>),<span class="dv">2</span>)</span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>  df_ewce<span class="sc">$</span>label[df_ewce<span class="sc">$</span>q<span class="sc">&gt;</span>.<span class="dv">05</span>] <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(df_ewce, <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(list),<span class="at">y=</span>CellType, <span class="at">label=</span>label, <span class="at">fill=</span>(sd_from_mean))) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>, <span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="sc">-</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">0</span>)) <span class="sc">+</span> <span class="fu">scale_fill_gradient2</span>()</span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="cell-type-makers" class="level2">
<h2 class="anchored" data-anchor-id="cell-type-makers">Cell-type makers</h2>
<div class="cell" data-hash="Figure3_DTU_enrichments_cache/html/unnamed-chunk-7_c45c68720faf47444c69567a93508697">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load poliodakis et al markers (Table S4)</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="co"># see: https://doi.org/10.1016/j.neuron.2019.06.011</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>celltypemarkers <span class="ot">&lt;-</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="st">'https://www.cell.com/cms/10.1016/j.neuron.2019.06.011/attachment/508ae008-b926-487a-b871-844a12acc1f8/mmc5'</span>, <span class="at">sheet=</span><span class="st">'Cluster enriched genes'</span>) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>celltypemarkers.tableS5 <span class="ot">=</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="st">'https://www.cell.com/cms/10.1016/j.neuron.2019.06.011/attachment/ec1863fa-0dfc-405c-928c-c5ca96115667/mmc6'</span>, <span class="at">sheet=</span><span class="st">'Sub-cluster enriched genes'</span>) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>celltypemarkers.bg <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"ref/polioudakis_neuron2020/single_cell_background_MJG221228.csv"</span>)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>celltypemarkers.bg <span class="ot">=</span> <span class="fu">unique</span>(celltypemarkers.bg<span class="sc">$</span>ensembl_gene_id)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>DTU <span class="ot">&lt;-</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU<span class="sc">==</span>T) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval_min) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>DTE <span class="ot">&lt;-</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTE<span class="sc">==</span>T) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval_min) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>DTU_notDGE <span class="ot">&lt;-</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU<span class="sc">==</span>T, DGE_pval<span class="sc">&gt;</span>.<span class="dv">05</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval_min) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DGE<span class="sc">==</span>T) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval_min) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>DTU_up <span class="ot">&lt;-</span> tableS3.isoform <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU<span class="sc">==</span>T,DTU_dIF<span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>DTU_down <span class="ot">&lt;-</span> tableS3.isoform <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU<span class="sc">==</span>T,DTU_dIF<span class="sc">&lt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(DTU_qval) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>DTU.bg <span class="ot">=</span> tableS3.gene<span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id)<span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">0</span>,<span class="dv">15</span>)</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>  df_fisher <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster)) {</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>    marker_genes <span class="ot">=</span> celltypemarkers <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Ensembl) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>    enrichment <span class="ot">=</span> <span class="fu">ORA</span>(DTU, marker_genes, DTU.bg, celltypemarkers.bg)</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>    df_fisher <span class="ot">=</span> <span class="fu">rbind</span>(df_fisher, <span class="fu">data.frame</span>(<span class="at">set=</span><span class="st">"DTU"</span>, <span class="at">cell=</span>this_cell, <span class="fu">as.data.frame</span>(<span class="fu">t</span>(enrichment))))</span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>    enrichment2 <span class="ot">=</span> <span class="fu">ORA</span>(DTU_notDGE, marker_genes, DTU.bg, celltypemarkers.bg)</span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>    df_fisher <span class="ot">=</span> <span class="fu">rbind</span>(df_fisher, <span class="fu">data.frame</span>(<span class="at">set=</span><span class="st">"DTU-not-DGE"</span>, <span class="at">cell=</span>this_cell, <span class="fu">as.data.frame</span>(<span class="fu">t</span>(enrichment2))))</span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>     enrichment3 <span class="ot">=</span> <span class="fu">ORA</span>(DGE, marker_genes, DTU.bg, celltypemarkers.bg)</span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a>     df_fisher <span class="ot">=</span> <span class="fu">rbind</span>(df_fisher, <span class="fu">data.frame</span>(<span class="at">set=</span><span class="st">"DGE"</span>, <span class="at">cell=</span>this_cell, <span class="fu">as.data.frame</span>(<span class="fu">t</span>(enrichment3))))</span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>     enrichment4 <span class="ot">=</span> <span class="fu">ORA</span>(DTE, marker_genes, DTU.bg, celltypemarkers.bg)</span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a>     df_fisher <span class="ot">=</span> <span class="fu">rbind</span>(df_fisher, <span class="fu">data.frame</span>(<span class="at">set=</span><span class="st">"DTE"</span>, <span class="at">cell=</span>this_cell, <span class="fu">as.data.frame</span>(<span class="fu">t</span>(enrichment4))))</span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>    <span class="co"># </span></span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># enrichment5 = ORA(DTU_down, marker_genes, DTU.bg, celltypemarkers.bg)</span></span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>    <span class="co"># df_fisher = rbind(df_fisher, data.frame(set="DTU_down", cell=this_cell, as.data.frame(t(enrichment5))))</span></span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>Fisher.p <span class="ot">=</span> <span class="fu">p.adjust</span>(<span class="fu">as.numeric</span>(df_fisher<span class="sc">$</span>Fisher.p),<span class="st">'fdr'</span>)</span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>OR <span class="ot">=</span> <span class="fu">as.numeric</span>(df_fisher<span class="sc">$</span>OR)</span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>CellClass <span class="ot">=</span> <span class="st">"Other"</span></span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>CellClass[<span class="fu">grep</span>(<span class="st">"Ex"</span>,df_fisher<span class="sc">$</span>cell)] <span class="ot">=</span> <span class="st">"Excitatory Neurons"</span></span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>CellClass[<span class="fu">grep</span>(<span class="st">"In"</span>,df_fisher<span class="sc">$</span>cell)] <span class="ot">=</span> <span class="st">"Interneurons"</span></span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>CellClass[df_fisher<span class="sc">$</span>cell <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"vRG"</span>, <span class="st">"oRG"</span>, <span class="st">"IP"</span>, <span class="st">"PgS"</span>, <span class="st">"PgG2M"</span>)] <span class="ot">=</span> <span class="st">"Progenitors"</span></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-46"><a href="#cb26-46" aria-hidden="true" tabindex="-1"></a>  Fig3f<span class="ot">=</span><span class="fu">ggplot</span>(df_fisher,<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(cell, <span class="sc">-</span>Fisher.p), <span class="at">y=</span> <span class="sc">-</span><span class="fu">log10</span>(Fisher.p), <span class="at">size=</span>OR, <span class="at">color=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p))) <span class="sc">+</span></span>
<span id="cb26-47"><a href="#cb26-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">facet_grid</span>(CellClass<span class="sc">~</span>set, <span class="at">scales =</span> <span class="st">'free'</span>, <span class="at">space=</span><span class="st">'free_y'</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb26-48"><a href="#cb26-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>,<span class="at">lty=</span><span class="dv">2</span>,<span class="at">color=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">y=</span><span class="st">'Enrichment (-log10 q-value)'</span>,<span class="at">x=</span><span class="st">''</span>)</span>
<span id="cb26-49"><a href="#cb26-49" aria-hidden="true" tabindex="-1"></a>  Fig3f</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_DTU_enrichments_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(Fig3f, <span class="at">file=</span><span class="st">"output/figures/Fig3/Fig3f.pdf"</span>,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height=</span><span class="dv">4</span>)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>Fig3f<span class="fl">.2</span> <span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher,<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(cell, <span class="sc">-</span>Fisher.p), <span class="at">y=</span> <span class="sc">-</span><span class="fu">log10</span>(Fisher.p), <span class="at">size=</span>OR, <span class="at">color=</span> set)) <span class="sc">+</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">facet_grid</span>(CellClass<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">'free'</span>, <span class="at">space=</span><span class="st">'free_y'</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>,<span class="at">lty=</span><span class="dv">2</span>,<span class="at">color=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">y=</span><span class="st">'Enrichment (-log10 q-value)'</span>,<span class="at">x=</span><span class="st">''</span>)</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(Fig3f<span class="fl">.2</span>, <span class="at">file=</span><span class="st">"output/figures/Fig3/Fig3f_2.pdf"</span>,<span class="at">width=</span><span class="dv">4</span>,<span class="at">height=</span><span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="rare-variant-logit" class="level2">
<h2 class="anchored" data-anchor-id="rare-variant-logit">Rare variant logit</h2>
<section id="isoform-level" class="level3">
<h3 class="anchored" data-anchor-id="isoform-level"><del>Isoform-level</del></h3>
<p>using gene level results to be most comparable to DGE</p>
<div class="cell" data-hash="Figure3_DTU_enrichments_cache/html/unnamed-chunk-8_07926d409054415958375e678d6f28db">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="cn">FALSE</span>) {</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  fu<span class="ot">=</span>openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>((<span class="st">'https://static-content.springer.com/esm/art%3A10.1038%2Fs41588-022-01104-0/MediaObjects/41588_2022_1104_MOESM3_ESM.xlsx'</span>),<span class="st">'Supplementary Table 11'</span>)</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  fu<span class="sc">$</span>p_TADA_ASD[fu<span class="sc">$</span>p_TADA_ASD<span class="sc">==</span><span class="dv">0</span>] <span class="ot">=</span> <span class="fu">min</span>(fu<span class="sc">$</span>p_TADA_ASD[fu<span class="sc">$</span>p_TADA_ASD <span class="sc">&gt;</span><span class="dv">0</span>],<span class="at">na.rm=</span>T)</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  fu<span class="sc">$</span>p_TADA_NDD[fu<span class="sc">$</span>p_TADA_NDD<span class="sc">==</span><span class="dv">0</span>] <span class="ot">=</span> <span class="fu">min</span>(fu<span class="sc">$</span>p_TADA_NDD[fu<span class="sc">$</span>p_TADA_NDD <span class="sc">&gt;</span><span class="dv">0</span>],<span class="at">na.rm=</span>T)</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>  datAnno.logit <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">ASD_fuTADA=</span> <span class="sc">-</span><span class="fu">log10</span>(fu<span class="sc">$</span>p_TADA_ASD)[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg, fu<span class="sc">$</span>gene_id)])</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>  datAnno.logit<span class="sc">$</span>NDD_fuTADA <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(fu<span class="sc">$</span>p_TADA_NDD)[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg, fu<span class="sc">$</span>gene_id)]</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>  SCZ.schema <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/SCHEMA_gene_results.tsv'</span>)</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>  datAnno.logit<span class="sc">$</span>SCZ_schema <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(SCZ.schema<span class="sc">$</span><span class="st">`</span><span class="at">P meta</span><span class="st">`</span>[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg, SCZ.schema<span class="sc">$</span>gene_id)])</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>  BIP.bipex <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/BipEx_gene_results.tsv'</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(group<span class="sc">==</span><span class="st">"Bipolar Disorder"</span>)</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>  datAnno.logit<span class="sc">$</span>BIP.bipex <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(BIP.bipex<span class="sc">$</span>ptv_fisher_gnom_non_psych_pval[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg,BIP.bipex<span class="sc">$</span>gene_id)])</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>  EPI.epi25 <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/Epi25_gene_results.tsv'</span>)  <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">pval =</span> <span class="fu">min</span>(pval))</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>  datAnno.logit<span class="sc">$</span>EPI.epi25 <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(EPI.epi25<span class="sc">$</span>pval[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg,EPI.epi25<span class="sc">$</span>gene_id)])</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>  df_logit<span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>  binaryVec <span class="ot">=</span> <span class="fu">list</span>(<span class="st">"DTU"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>DTU_qval<span class="sc">&lt;</span>.<span class="dv">05</span>), <span class="st">"DTE"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>DTE_qval<span class="sc">&lt;</span>.<span class="dv">05</span>),</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"DTU_up"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>DTU_qval<span class="sc">&lt;</span>.<span class="dv">05</span> <span class="sc">&amp;</span> datAnno<span class="sc">$</span>DTU_dIF<span class="sc">&gt;</span><span class="dv">0</span>),</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"DTU_down"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>DTU_qval<span class="sc">&lt;</span>.<span class="dv">05</span> <span class="sc">&amp;</span> datAnno<span class="sc">$</span>DTU_dIF<span class="sc">&lt;</span><span class="dv">0</span>),</span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"DTE_up"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>DTE_qval<span class="sc">&lt;</span>.<span class="dv">05</span> <span class="sc">&amp;</span> datAnno<span class="sc">$</span>DTE_log2FC<span class="sc">&gt;</span><span class="dv">0</span>),</span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"DTE_down"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>DTE_qval<span class="sc">&lt;</span>.<span class="dv">05</span> <span class="sc">&amp;</span> datAnno<span class="sc">$</span>DTE_log2FC<span class="sc">&lt;</span><span class="dv">0</span>),</span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"DTU_notDGE"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>DTU_qval<span class="sc">&lt;</span>.<span class="dv">05</span> <span class="sc">&amp;</span> datAnno<span class="sc">$</span>DGE_pval<span class="sc">&gt;</span><span class="fl">0.05</span>))</span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(binaryVec)) {</span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(this_rare_var <span class="cf">in</span> <span class="fu">colnames</span>(datAnno.logit)) {</span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a>      this_glm <span class="ot">=</span> <span class="fu">summary</span>(<span class="fu">glm</span>(binaryVec[[i]] <span class="sc">~</span> datAnno.logit[,this_rare_var] <span class="sc">+</span> length <span class="sc">+</span> <span class="fu">log10</span>(length), <span class="at">data=</span>datAnno, <span class="at">family=</span><span class="st">'binomial'</span>))</span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a>      df_logit <span class="ot">=</span> <span class="fu">rbind</span>(df_logit, <span class="fu">data.frame</span>(<span class="at">feature =</span> <span class="fu">names</span>(binaryVec)[[i]], <span class="at">set =</span> this_rare_var, <span class="fu">t</span>(this_glm<span class="sc">$</span>coefficients[<span class="dv">2</span>,])))</span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a>  }}</span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a>  df_logit<span class="sc">$</span>OR <span class="ot">=</span> <span class="fu">exp</span>(df_logit<span class="sc">$</span>Estimate)</span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a>  df_logit<span class="sc">$</span>P <span class="ot">=</span> df_logit<span class="sc">$</span>Pr...z..</span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a>  df_logit<span class="sc">$</span>P[df_logit<span class="sc">$</span>OR <span class="sc">&lt;</span> <span class="dv">1</span>] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a>  df_logit<span class="sc">$</span>Q <span class="ot">=</span> <span class="fu">p.adjust</span>(df_logit<span class="sc">$</span>P,<span class="st">'fdr'</span>)</span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a>  df_logit<span class="sc">$</span>class <span class="ot">=</span> <span class="st">"DGE"</span></span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a>  df_logit<span class="sc">$</span>class[<span class="fu">grepl</span>(<span class="st">"DTE"</span>, df_logit<span class="sc">$</span>feature)] <span class="ot">=</span> <span class="st">"DTE"</span></span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a>  df_logit<span class="sc">$</span>class[<span class="fu">grepl</span>(<span class="st">"DTU"</span>, df_logit<span class="sc">$</span>feature)] <span class="ot">=</span> <span class="st">"DTU"</span></span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(df_logit, <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(set,<span class="sc">-</span>Q), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Q), <span class="at">fill=</span>OR)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> </span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">y=</span><span class="st">"-log10(qvalue)"</span>,<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">facet_wrap</span>(feature<span class="sc">~</span>.,<span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-48"><a href="#cb28-48" aria-hidden="true" tabindex="-1"></a>  g1 <span class="ot">=</span> <span class="fu">ggplot</span>(df_logit <span class="sc">%&gt;%</span> <span class="fu">filter</span>(feature <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"DTU"</span>, <span class="st">"DTE"</span>, <span class="st">"DTU_notDGE"</span>)), <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(set,<span class="sc">-</span>Q), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Q), <span class="at">fill=</span>OR)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> </span>
<span id="cb28-49"><a href="#cb28-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">y=</span><span class="st">"-log10(qvalue)"</span>,<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">facet_grid</span>(feature<span class="sc">~</span>.)</span>
<span id="cb28-50"><a href="#cb28-50" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="gene-level" class="level3">
<h3 class="anchored" data-anchor-id="gene-level">Gene-level</h3>
<div class="cell" data-hash="Figure3_DTU_enrichments_cache/html/unnamed-chunk-9_948b8e77df456b6829c8f0bc6a7d7d24">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>tableS3.isoform<span class="sc">$</span>ensg <span class="ot">=</span> <span class="fu">substr</span>(tableS3.isoform<span class="sc">$</span>gene_id,<span class="dv">1</span>,<span class="dv">15</span>)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>datAnno <span class="ot">=</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">ensg =</span> <span class="fu">substr</span>(gene_id,<span class="dv">1</span>,<span class="dv">15</span>))</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>datAnno <span class="ot">&lt;-</span> datAnno <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(geneAnno <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensg, <span class="at">length=</span>width))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "ensg"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>fu<span class="ot">=</span>openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>((<span class="st">'https://static-content.springer.com/esm/art%3A10.1038%2Fs41588-022-01104-0/MediaObjects/41588_2022_1104_MOESM3_ESM.xlsx'</span>),<span class="st">'Supplementary Table 11'</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>fu<span class="sc">$</span>p_TADA_ASD[fu<span class="sc">$</span>p_TADA_ASD<span class="sc">==</span><span class="dv">0</span>] <span class="ot">=</span> <span class="fu">min</span>(fu<span class="sc">$</span>p_TADA_ASD[fu<span class="sc">$</span>p_TADA_ASD <span class="sc">&gt;</span><span class="dv">0</span>],<span class="at">na.rm=</span>T)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>fu<span class="sc">$</span>p_TADA_NDD[fu<span class="sc">$</span>p_TADA_NDD<span class="sc">==</span><span class="dv">0</span>] <span class="ot">=</span> <span class="fu">min</span>(fu<span class="sc">$</span>p_TADA_NDD[fu<span class="sc">$</span>p_TADA_NDD <span class="sc">&gt;</span><span class="dv">0</span>],<span class="at">na.rm=</span>T)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>datAnno.logit <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">ASD_fuTADA=</span> <span class="sc">-</span><span class="fu">log10</span>(fu<span class="sc">$</span>p_TADA_ASD)[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg, fu<span class="sc">$</span>gene_id)])</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>datAnno.logit<span class="sc">$</span>NDD_fuTADA <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(fu<span class="sc">$</span>p_TADA_NDD)[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg, fu<span class="sc">$</span>gene_id)]</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>SCZ.schema <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/SCHEMA_gene_results.tsv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: One or more parsing issues, see `problems()` for details</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 18321 Columns: 26
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (8): gene_id, group, OR (PTV), OR (Class I), OR (Class II), OR (PTV) up...
dbl (16): Case PTV, Ctrl PTV, Case mis3, Ctrl mis3, Case mis2, Ctrl mis2, P ...
lgl  (2): De novo mis3, De novo mis2

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>datAnno.logit<span class="sc">$</span>SCZ_schema <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(SCZ.schema<span class="sc">$</span><span class="st">`</span><span class="at">P meta</span><span class="st">`</span>[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg, SCZ.schema<span class="sc">$</span>gene_id)])</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>BIP.bipex <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/BipEx_gene_results.tsv'</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(group<span class="sc">==</span><span class="st">"Bipolar Disorder"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 119958 Columns: 20
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (4): gene_id, group, damaging_missense_fisher_gnom_non_psych_OR, ptv_fi...
dbl (16): n_cases, n_controls, damaging_missense_case_count, damaging_missen...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>datAnno.logit<span class="sc">$</span>BIP.bipex <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(BIP.bipex<span class="sc">$</span>ptv_fisher_gnom_non_psych_pval[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg,BIP.bipex<span class="sc">$</span>gene_id)])</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>EPI.epi25 <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/Epi25_gene_results.tsv'</span>)  <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">pval =</span> <span class="fu">min</span>(pval))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 71456 Columns: 12
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): gene_id, group
dbl (9): xcase_lof, xctrl_lof, pval_lof, xcase_mpc, xctrl_mpc, pval_mpc, xca...
lgl (1): pval_infrIndel

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>datAnno.logit<span class="sc">$</span>EPI.epi25 <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(EPI.epi25<span class="sc">$</span>pval[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg,EPI.epi25<span class="sc">$</span>gene_id)])</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>df_logit<span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>binaryVec <span class="ot">=</span> <span class="fu">list</span>(<span class="st">"DTU"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>DTU), <span class="st">"DTE"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>DTE), <span class="st">"DGE"</span><span class="ot">=</span><span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>DGE),</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"DTU-not-DGE"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>DTU <span class="sc">&amp;</span> <span class="sc">!</span>datAnno<span class="sc">$</span>DGE), </span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"DTU-not-DTE"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>DTU <span class="sc">&amp;</span> <span class="sc">!</span>datAnno<span class="sc">$</span>DTE),</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"DTU.up"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>ensg <span class="sc">%in%</span> (tableS3.isoform <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU, DTU_dIF<span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensg) <span class="sc">%&gt;%</span> <span class="fu">pull</span>())),</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"DTU.down"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>ensg <span class="sc">%in%</span> (tableS3.isoform <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU, DTU_dIF<span class="sc">&lt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensg) <span class="sc">%&gt;%</span> <span class="fu">pull</span>())),</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"DTE.up"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>ensg <span class="sc">%in%</span> (tableS3.isoform <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTE, DTE_log2FC<span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensg) <span class="sc">%&gt;%</span> <span class="fu">pull</span>())),</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"DTE.down"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>ensg <span class="sc">%in%</span> (tableS3.isoform <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTE, DTE_log2FC<span class="sc">&lt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensg) <span class="sc">%&gt;%</span> <span class="fu">pull</span>())),</span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"DGE.up"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>ensg <span class="sc">%in%</span> (tableS3.isoform <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DGE, DGE_log2FC<span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensg) <span class="sc">%&gt;%</span> <span class="fu">pull</span>())),</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"DGE.down"</span> <span class="ot">=</span> <span class="fu">as.numeric</span>(datAnno<span class="sc">$</span>ensg <span class="sc">%in%</span> (tableS3.isoform <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DGE, DGE_log2FC<span class="sc">&lt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensg) <span class="sc">%&gt;%</span> <span class="fu">pull</span>())))</span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(binaryVec)) {</span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_rare_var <span class="cf">in</span> <span class="fu">colnames</span>(datAnno.logit)) {</span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a>    this_glm <span class="ot">=</span> <span class="fu">summary</span>(<span class="fu">glm</span>(datAnno.logit[,this_rare_var] <span class="sc">~</span> binaryVec[[i]] <span class="sc">+</span> length <span class="sc">+</span> <span class="fu">log10</span>(length), <span class="at">data=</span>datAnno))</span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a>    df_logit <span class="ot">=</span> <span class="fu">rbind</span>(df_logit, <span class="fu">data.frame</span>(<span class="at">feature =</span> <span class="fu">names</span>(binaryVec)[[i]], <span class="at">set =</span> this_rare_var, <span class="fu">t</span>(this_glm<span class="sc">$</span>coefficients[<span class="dv">2</span>,])))</span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a>}}</span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>OR <span class="ot">=</span> <span class="fu">exp</span>(df_logit<span class="sc">$</span>Estimate)</span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>P <span class="ot">=</span> df_logit<span class="sc">$</span>Pr...z..</span>
<span id="cb38-23"><a href="#cb38-23" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>Q <span class="ot">=</span> <span class="fu">p.adjust</span>(df_logit<span class="sc">$</span>P,<span class="st">'fdr'</span>)</span>
<span id="cb38-24"><a href="#cb38-24" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>class <span class="ot">=</span> <span class="st">"DGE"</span></span>
<span id="cb38-25"><a href="#cb38-25" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>class[<span class="fu">grepl</span>(<span class="st">"DTE"</span>, df_logit<span class="sc">$</span>feature)] <span class="ot">=</span> <span class="st">"DTE"</span></span>
<span id="cb38-26"><a href="#cb38-26" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>class[<span class="fu">grepl</span>(<span class="st">"DTU"</span>, df_logit<span class="sc">$</span>feature)] <span class="ot">=</span> <span class="st">"DTU"</span></span>
<span id="cb38-27"><a href="#cb38-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-28"><a href="#cb38-28" aria-hidden="true" tabindex="-1"></a>Fig3H<span class="ot">=</span><span class="fu">ggplot</span>(df_logit <span class="sc">%&gt;%</span> <span class="fu">filter</span>(feature <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"DTE"</span>, <span class="st">"DGE"</span>, <span class="st">"DTU"</span>, <span class="st">"DTU-not-DGE"</span>)), <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(set,<span class="sc">-</span>Q), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Q), <span class="at">color=</span>feature, <span class="at">alpha =</span> OR <span class="sc">&gt;</span> <span class="dv">1</span>,<span class="at">size=</span>OR)) <span class="sc">+</span></span>
<span id="cb38-29"><a href="#cb38-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>), <span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>) <span class="sc">+</span>  </span>
<span id="cb38-30"><a href="#cb38-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_alpha_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(.<span class="dv">5</span>, <span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb38-31"><a href="#cb38-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">y=</span><span class="st">"-log10(qvalue)"</span>,<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">'Gene level rare variant enrichment'</span>)</span>
<span id="cb38-32"><a href="#cb38-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-33"><a href="#cb38-33" aria-hidden="true" tabindex="-1"></a>Fig3H</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_DTU_enrichments_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(Fig3H,<span class="at">file=</span><span class="st">'output/figures/Fig3/Fig3H.pdf'</span>,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height=</span><span class="fl">2.5</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>class[<span class="fu">grepl</span>(<span class="st">"-not-"</span>, df_logit<span class="sc">$</span>feature)] <span class="ot">=</span> <span class="st">"DTU-not-DGE/DTE"</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>feature <span class="ot">=</span> <span class="fu">factor</span>(df_logit<span class="sc">$</span>feature, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"DGE"</span>, <span class="st">"DGE.down"</span>, <span class="st">"DGE.up"</span>,<span class="st">"DTE"</span>, <span class="st">"DTE.down"</span>, <span class="st">"DTE.up"</span>,<span class="st">"DTU"</span>, <span class="st">"DTU.down"</span>, <span class="st">"DTU.up"</span>,<span class="st">"DTU-not-DGE"</span>, <span class="st">"DTU-not-DTE"</span>))</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>FigS4B<span class="ot">=</span><span class="fu">ggplot</span>(df_logit, <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(set,<span class="sc">-</span>Q), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Q), <span class="at">color=</span>class, <span class="at">size=</span>OR, <span class="at">alpha=</span>OR<span class="sc">&gt;</span><span class="dv">1</span>)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>), <span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">y=</span><span class="st">"-log10(qvalue)"</span>,<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">facet_wrap</span>(feature<span class="sc">~</span>.,<span class="at">ncol =</span> <span class="dv">3</span>) <span class="sc">+</span> <span class="fu">scale_alpha_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(.<span class="dv">5</span>, <span class="dv">1</span>))</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(FigS4B,<span class="at">file=</span><span class="st">'output/figures/supplement/FigS4B.pdf'</span>,<span class="at">width=</span><span class="dv">7</span>,<span class="at">height=</span><span class="dv">5</span>)</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>FigS4B</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_DTU_enrichments_files/figure-html/unnamed-chunk-9-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(df_logit, aes(x=reorder(feature,-Q), y=-log10(Q), fill=OR)) + geom_bar(stat='identity') + </span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="co">#   coord_flip() + geom_hline(yintercept = -log10(.05), lty=2,col='red') + labs(y="-log10(qvalue)",x="") + theme_bw() +   facet_grid(set~.,scales = 'free')</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="rbp-enrichments" class="level2">
<h2 class="anchored" data-anchor-id="rbp-enrichments">RBP Enrichments</h2>
<section id="celines-rbp-targets" class="level3">
<h3 class="anchored" data-anchor-id="celines-rbp-targets">Celine’s RBP Targets</h3>
<div class="cell" data-hash="Figure3_DTU_enrichments_cache/html/unnamed-chunk-10_6a995719ff610c850f7f88fefa8f63ee">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>rbp_targets <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"ref/RBPs/RBP_targets_v2.csv"</span>)</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>rbp_targets<span class="sc">$</span>regulation[<span class="fu">is.na</span>(rbp_targets<span class="sc">$</span>regulation)]<span class="ot">=</span><span class="st">""</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>mart <span class="ot">=</span> <span class="fu">useMart</span>(<span class="st">"ENSEMBL_MART_ENSEMBL"</span>,<span class="st">"mmusculus_gene_ensembl"</span>)</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>f <span class="ot">=</span> <span class="fu">listFilters</span>(mart); a <span class="ot">=</span> <span class="fu">listAttributes</span>(mart)</span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>featuresToGet <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>, <span class="st">"external_gene_name"</span>, <span class="st">"hsapiens_homolog_ensembl_gene"</span>, <span class="st">"hsapiens_homolog_associated_gene_name"</span>,<span class="st">"hsapiens_homolog_orthology_type"</span>)</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>mouseHumanHomologs <span class="ot">=</span> <span class="fu">getBM</span>(<span class="at">attributes =</span> featuresToGet,<span class="at">mart =</span> mart)</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>human_mouse_bg <span class="ot">=</span> mouseHumanHomologs <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hsapiens_homolog_orthology_type <span class="sc">==</span> <span class="st">"ortholog_one2one"</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">"hsapiens_homolog_ensembl_gene"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>genesets <span class="ot">=</span> <span class="fu">list</span>(<span class="st">"DTU"</span><span class="ot">=</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_id =</span> <span class="fu">substr</span>(gene_id,<span class="dv">1</span>,<span class="dv">15</span>)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(),</span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"DTE"</span> <span class="ot">=</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTE) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_id =</span> <span class="fu">substr</span>(gene_id,<span class="dv">1</span>,<span class="dv">15</span>)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(),</span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"DGE"</span> <span class="ot">=</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTE) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_id =</span> <span class="fu">substr</span>(gene_id,<span class="dv">1</span>,<span class="dv">15</span>)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(),</span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"DTUnotDGE"</span> <span class="ot">=</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU,DGE_pval<span class="sc">&gt;</span>.<span class="dv">05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_id =</span> <span class="fu">substr</span>(gene_id,<span class="dv">1</span>,<span class="dv">15</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>())</span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a>DTU.bg <span class="ot">=</span> tableS3.gene <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_id =</span> <span class="fu">substr</span>(gene_id,<span class="dv">1</span>,<span class="dv">15</span>)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-21"><a href="#cb41-21" aria-hidden="true" tabindex="-1"></a>df_fisher <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb41-22"><a href="#cb41-22" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(genesets)) {</span>
<span id="cb41-23"><a href="#cb41-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_dataset <span class="cf">in</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(rbp_targets<span class="sc">$</span>dataset))) {</span>
<span id="cb41-24"><a href="#cb41-24" aria-hidden="true" tabindex="-1"></a>    this_rbp <span class="ot">=</span> rbp_targets <span class="sc">%&gt;%</span> <span class="fu">filter</span>(dataset <span class="sc">==</span> this_dataset) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">target =</span> <span class="fu">paste0</span>(RBP, <span class="st">"_"</span>, data.type, <span class="st">"_"</span>, msORhs)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(target) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()  <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb41-25"><a href="#cb41-25" aria-hidden="true" tabindex="-1"></a>    target_genes <span class="ot">=</span> rbp_targets <span class="sc">%&gt;%</span> <span class="fu">filter</span>(dataset <span class="sc">==</span> this_dataset) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ENSG) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb41-26"><a href="#cb41-26" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb41-27"><a href="#cb41-27" aria-hidden="true" tabindex="-1"></a> <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">"Human"</span>,this_rbp)) {</span>
<span id="cb41-28"><a href="#cb41-28" aria-hidden="true" tabindex="-1"></a>   this_or <span class="ot">=</span> <span class="fu">ORA</span>(genesets[[i]], target_genes, DTU.bg, DTU.bg)</span>
<span id="cb41-29"><a href="#cb41-29" aria-hidden="true" tabindex="-1"></a> } <span class="cf">else</span> {</span>
<span id="cb41-30"><a href="#cb41-30" aria-hidden="true" tabindex="-1"></a>  this_or <span class="ot">=</span> <span class="fu">ORA</span>(genesets[[i]], target_genes, DTU.bg, human_mouse_bg)</span>
<span id="cb41-31"><a href="#cb41-31" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb41-32"><a href="#cb41-32" aria-hidden="true" tabindex="-1"></a> df_fisher <span class="ot">=</span> <span class="fu">rbind</span>(df_fisher, <span class="fu">data.frame</span>(<span class="at">set =</span> <span class="fu">names</span>(genesets)[[i]], <span class="at">dataset =</span> this_dataset, <span class="at">target =</span> this_rbp, <span class="fu">t</span>(this_or)))</span>
<span id="cb41-33"><a href="#cb41-33" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb41-34"><a href="#cb41-34" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb41-35"><a href="#cb41-35" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>OR <span class="ot">=</span> <span class="fu">as.numeric</span>(df_fisher<span class="sc">$</span>OR)</span>
<span id="cb41-36"><a href="#cb41-36" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>Fisher.p[df_fisher<span class="sc">$</span>OR<span class="sc">&lt;</span><span class="dv">1</span>] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb41-37"><a href="#cb41-37" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>Fisher.p <span class="ot">=</span> <span class="fu">p.adjust</span>(<span class="fu">as.numeric</span>(df_fisher<span class="sc">$</span>Fisher.p),<span class="st">'fdr'</span>)</span>
<span id="cb41-38"><a href="#cb41-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb41-39"><a href="#cb41-39" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>target <span class="ot">=</span> <span class="fu">factor</span>(df_fisher<span class="sc">$</span>target, <span class="at">levels=</span> df_fisher <span class="sc">%&gt;%</span> <span class="fu">filter</span>(set<span class="sc">==</span><span class="st">"DTU"</span>) <span class="sc">%&gt;%</span>  <span class="fu">group_by</span>(target) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">pmin =</span> <span class="fu">min</span>(Fisher.p)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(pmin) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(target) <span class="sc">%&gt;%</span> <span class="fu">pull</span>())</span>
<span id="cb41-40"><a href="#cb41-40" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>org <span class="ot">=</span> <span class="st">"Mouse"</span>; df_fisher<span class="sc">$</span>org[<span class="fu">grep</span>(<span class="st">"Human"</span>,df_fisher<span class="sc">$</span>target)] <span class="ot">=</span> <span class="st">"Human"</span></span>
<span id="cb41-41"><a href="#cb41-41" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>data.type <span class="ot">=</span> rbp_targets<span class="sc">$</span>data.type[<span class="fu">match</span>(df_fisher<span class="sc">$</span>dataset, rbp_targets<span class="sc">$</span>dataset)]</span>
<span id="cb41-42"><a href="#cb41-42" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>cell.type <span class="ot">=</span> rbp_targets<span class="sc">$</span>cell.type[<span class="fu">match</span>(df_fisher<span class="sc">$</span>dataset, rbp_targets<span class="sc">$</span>dataset)]</span>
<span id="cb41-43"><a href="#cb41-43" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>label <span class="ot">=</span> <span class="fu">signif</span>(df_fisher<span class="sc">$</span>OR,<span class="dv">1</span>)</span>
<span id="cb41-44"><a href="#cb41-44" aria-hidden="true" tabindex="-1"></a>  df_fisher<span class="sc">$</span>label[df_fisher<span class="sc">$</span>Fisher.p<span class="sc">&gt;</span>.<span class="dv">05</span>] <span class="ot">=</span> <span class="st">''</span></span>
<span id="cb41-45"><a href="#cb41-45" aria-hidden="true" tabindex="-1"></a>  Fig3H<span class="fl">.1</span><span class="ot">=</span><span class="fu">ggplot</span>(df_fisher <span class="sc">%&gt;%</span> <span class="fu">filter</span>(set<span class="sc">==</span><span class="st">"DTU"</span>),<span class="fu">aes</span>(<span class="at">x=</span>target, <span class="at">y=</span> <span class="sc">-</span><span class="fu">log10</span>(Fisher.p), <span class="at">fill=</span>OR)) <span class="sc">+</span></span>
<span id="cb41-46"><a href="#cb41-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb41-47"><a href="#cb41-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>,<span class="at">lty=</span><span class="dv">2</span>,<span class="at">color=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">y=</span><span class="st">'Enrichment</span><span class="sc">\n</span><span class="st">(-log10 q-value)'</span>,<span class="at">x=</span><span class="st">''</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>,<span class="at">vjust =</span> .<span class="dv">5</span>, <span class="at">hjust=</span><span class="dv">1</span>))</span>
<span id="cb41-48"><a href="#cb41-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb41-49"><a href="#cb41-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(df_fisher,<span class="fu">aes</span>(<span class="at">x=</span>target, <span class="at">y=</span> set, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p),<span class="at">label=</span>label)) <span class="sc">+</span> <span class="fu">geom_tile</span>(<span class="at">color=</span><span class="st">'grey60'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb41-50"><a href="#cb41-50" aria-hidden="true" tabindex="-1"></a>     <span class="fu">labs</span>(<span class="at">y=</span><span class="st">''</span>,<span class="at">x=</span><span class="st">''</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>,<span class="at">vjust =</span> .<span class="dv">5</span>, <span class="at">hjust=</span><span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb41-51"><a href="#cb41-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>, <span class="at">high=</span><span class="st">'red'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_DTU_enrichments_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>    Fig3H<span class="fl">.2</span><span class="ot">=</span><span class="fu">ggplot</span>(df_fisher <span class="sc">%&gt;%</span> <span class="fu">filter</span>(set<span class="sc">==</span><span class="st">"DTU"</span>),<span class="fu">aes</span>(<span class="at">x=</span>target, <span class="at">label=</span>cell.type)) <span class="sc">+</span> <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(<span class="dv">1</span>),<span class="at">fill=</span>data.type)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(<span class="dv">1</span>), <span class="at">shape=</span>cell.type),<span class="at">position=</span><span class="fu">position_dodge2</span>()) <span class="sc">+</span> <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>)) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">""</span>)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pdf</span>(<span class="at">file=</span><span class="st">"output/figures/Fig3/Fig3H.pdf"</span>,<span class="at">width =</span> <span class="dv">8</span>,<span class="at">height=</span><span class="dv">5</span>)</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>    gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(<span class="at">grobs=</span><span class="fu">list</span>(Fig3H<span class="fl">.1</span>,Fig3H<span class="fl">.2</span>),<span class="at">layout_matrix=</span><span class="fu">cbind</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Width not defined. Set with `position_dodge2(width = ?)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>  Fig3H_supplement<span class="ot">=</span><span class="fu">ggplot</span>(df_fisher,<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(target, <span class="sc">-</span>Fisher.p), <span class="at">y=</span> <span class="sc">-</span><span class="fu">log10</span>(Fisher.p), <span class="at">fill=</span>OR)) <span class="sc">+</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">facet_grid</span>(<span class="sc">~</span>set) <span class="sc">+</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>,<span class="at">lty=</span><span class="dv">2</span>,<span class="at">color=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">y=</span><span class="st">'Enrichment (-log10 q-value)'</span>,<span class="at">x=</span><span class="st">''</span>)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>Fig3H_supplement</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_DTU_enrichments_files/figure-html/unnamed-chunk-10-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(Fig3H_supplement, <span class="at">file=</span><span class="st">"output/figures/supplement/Fig3H_supp.pdf"</span>,<span class="at">width =</span> <span class="dv">8</span>,<span class="at">height=</span><span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="rbp-motif" class="level3">
<h3 class="anchored" data-anchor-id="rbp-motif">RBP motif:</h3>
<div class="cell" data-hash="Figure3_DTU_enrichments_cache/html/unnamed-chunk-11_d9575ca946ad267f832f0506a456ad32">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># library(transite)</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="co"># library(biomaRt)</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="co"># mart = useMart('ENSEMBL_MART_ENSEMBL', 'hsapiens_gene_ensembl',host = 'https://grch37.ensembl.org')</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="co"># a=listAttributes(mart)</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a><span class="co"># ensembl = getBM(attributes=c("ensembl_gene_id", "hgnc_symbol",'refseq_mrna'),mart=mart)</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a><span class="co"># out = run_kmer_tsma(foreground_sets = tableS3.gene$gene_name[tableS3.gene$DTU], background_set = tableS3.gene$gene_name)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>