<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Celine Vuong">

<title>Dev Brain Iso-Seq - Figure 4 - Networks Enrichments</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-sidebar floating">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Figure 4 - Networks Enrichments</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Dev Brain Iso-Seq</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/gandallab/Dev_Brain_IsoSeq" title="" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure1_BulkTxomeAnalysis.html" class="sidebar-item-text sidebar-link">Figure 1 - BulkTxomeAnalysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure1_BulkTxomeAnalysis_revision1.html" class="sidebar-item-text sidebar-link">Figure 1 - BulkTxomeAnalysis REVISION1</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure1_TableS1.html" class="sidebar-item-text sidebar-link">Figure 1 - Table S1</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../code/Fig1_MetaCoverage.html" class="sidebar-item-text sidebar-link">Figure 1 - MetaCoverage</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_isoFraction.html" class="sidebar-item-text sidebar-link">Figure 2 - Isoform Fraction analysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_plotNovelGenes.html" class="sidebar-item-text sidebar-link">Figure 2 - Plot Novel Genes</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_PlotGene.html" class="sidebar-item-text sidebar-link">Figure 2 - Plot One Gene</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_novel_exons.html" class="sidebar-item-text sidebar-link">Figure 2 - Novel Exons</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_NF1Locus.html" class="sidebar-item-text sidebar-link">Figure 2 - NF1 Locus</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_PeptideSupport.html" class="sidebar-item-text sidebar-link">Figure 2 - Peptide Support</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_IsoformSwitchAnalyzeR.html" class="sidebar-item-text sidebar-link">Figure 3 - IsoformSwitchAnalyzeR</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_DTU_enrichments.html" class="sidebar-item-text sidebar-link">Figure 3 - DTU enrichments</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_RBPEnrichments.html" class="sidebar-item-text sidebar-link">Figure 3 - RBP Enrichments</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_APA.html" class="sidebar-item-text sidebar-link">Figure 3 - APA</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure4_combinedNetworks.html" class="sidebar-item-text sidebar-link">Figure 4 - Combined Networks</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure4_Networks_Enrichments.html" class="sidebar-item-text sidebar-link active">Figure 4 - Networks Enrichments</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_SeuratClustering.html" class="sidebar-item-text sidebar-link">Figure 5 - Seurat Clustering</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_MarkerGenesFigs.html" class="sidebar-item-text sidebar-link">Figure 5 - Marker Genes</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_TableS5D.html" class="sidebar-item-text sidebar-link">Figure 5 - Table S5D</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_PseudotimeAnalysis.html" class="sidebar-item-text sidebar-link">Figure 5 - Pseudotime Analysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_Sankey.html" class="sidebar-item-text sidebar-link">Figure 5 - Sankey</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure6_rareVariant.html" class="sidebar-item-text sidebar-link">Figure 6 - Rare Variant</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure6_rareVariant_revision1.html" class="sidebar-item-text sidebar-link">Figure 6 - Rare Variant REVISION1</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure6_VEPConsequences.html" class="sidebar-item-text sidebar-link">Figure 6 - VEP Consequences</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure6_TrackPlots.html" class="sidebar-item-text sidebar-link">Figure 6 - Track Plots</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/FigureS9_cor_plots.html" class="sidebar-item-text sidebar-link">Figure S9 - Correlation plots</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/FigureS9H_revision1.html" class="sidebar-item-text sidebar-link">Fig S9H REVISION1</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#load-data" id="toc-load-data" class="nav-link active" data-scroll-target="#load-data">Load Data</a></li>
  <li><a href="#network-dendrogram-plots" id="toc-network-dendrogram-plots" class="nav-link" data-scroll-target="#network-dendrogram-plots">Network Dendrogram Plots</a>
  <ul class="collapse">
  <li><a href="#fig4a" id="toc-fig4a" class="nav-link" data-scroll-target="#fig4a">Fig4A</a></li>
  <li><a href="#figs5a" id="toc-figs5a" class="nav-link" data-scroll-target="#figs5a">FigS5A</a></li>
  <li><a href="#figs5b" id="toc-figs5b" class="nav-link" data-scroll-target="#figs5b">FigS5B</a></li>
  <li><a href="#figs5c" id="toc-figs5c" class="nav-link" data-scroll-target="#figs5c">FigS5C</a></li>
  </ul></li>
  <li><a href="#compare-networks" id="toc-compare-networks" class="nav-link" data-scroll-target="#compare-networks">Compare Networks</a>
  <ul class="collapse">
  <li><a href="#cell-type-enrichment" id="toc-cell-type-enrichment" class="nav-link" data-scroll-target="#cell-type-enrichment">Cell-type enrichment</a></li>
  <li><a href="#figs6a" id="toc-figs6a" class="nav-link" data-scroll-target="#figs6a">FigS6A</a></li>
  <li><a href="#figs6b" id="toc-figs6b" class="nav-link" data-scroll-target="#figs6b">FigS6B</a></li>
  <li><a href="#figs6c" id="toc-figs6c" class="nav-link" data-scroll-target="#figs6c">FigS6C</a></li>
  <li><a href="#rare-var-enrichments" id="toc-rare-var-enrichments" class="nav-link" data-scroll-target="#rare-var-enrichments">Rare Var Enrichments</a></li>
  <li><a href="#gwas-enrichments" id="toc-gwas-enrichments" class="nav-link" data-scroll-target="#gwas-enrichments">GWAS Enrichments</a></li>
  <li><a href="#rbp-enrichments" id="toc-rbp-enrichments" class="nav-link" data-scroll-target="#rbp-enrichments">RBP Enrichments</a></li>
  <li><a href="#tables4d" id="toc-tables4d" class="nav-link" data-scroll-target="#tables4d">TableS4D</a></li>
  <li><a href="#figs6b-1" id="toc-figs6b-1" class="nav-link" data-scroll-target="#figs6b-1">FigS6B</a></li>
  <li><a href="#fig4b" id="toc-fig4b" class="nav-link" data-scroll-target="#fig4b">Fig4B</a></li>
  </ul></li>
  <li><a href="#annotate-individual-modules" id="toc-annotate-individual-modules" class="nav-link" data-scroll-target="#annotate-individual-modules">Annotate Individual Modules</a>
  <ul class="collapse">
  <li><a href="#fig4cd-and-figs7" id="toc-fig4cd-and-figs7" class="nav-link" data-scroll-target="#fig4cd-and-figs7">Fig4C,D and FigS7</a></li>
  <li><a href="#isousage-network" id="toc-isousage-network" class="nav-link" data-scroll-target="#isousage-network">IsoUsage Network</a></li>
  <li><a href="#isoexpr-network" id="toc-isoexpr-network" class="nav-link" data-scroll-target="#isoexpr-network">IsoExpr Network</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Figure 4 - Networks Enrichments</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Celine Vuong </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-1_dbca7a79bfd5c6b73084de15cb00d378">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(edgeR)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(WGCNA)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(biomaRt)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(openxlsx)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(readxl)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(gprofiler2)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(gridExtra)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(cowplot)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(paletteer)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"code/fisher_overlap.R"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>colorVector <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Known"</span> <span class="ot">=</span> <span class="st">"#009E73"</span>,</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ISM"</span>   <span class="ot">=</span> <span class="st">"#0072B2"</span>,</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ISM_Prefix"</span> <span class="ot">=</span> <span class="st">"#005996"</span>,</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ISM_Suffix"</span> <span class="ot">=</span> <span class="st">"#378bcc"</span>,</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  <span class="st">"NIC"</span>   <span class="ot">=</span> <span class="st">"#D55E00"</span>,</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>  <span class="st">"NNC"</span>   <span class="ot">=</span> <span class="st">"#E69F00"</span>,</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Other"</span> <span class="ot">=</span> <span class="st">"#000000"</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>colorVector_ismSplit <span class="ot">=</span> colorVector[<span class="sc">-</span><span class="dv">2</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="load-data" class="level2">
<h2 class="anchored" data-anchor-id="load-data">Load Data</h2>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-2_17ec12223e40882ab166fcd12f5899a0">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">"data/cp_vz_0.75_min_7_recovery_talon_abundance_filtered.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 214516 Columns: 35
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (7): annot_gene_id, annot_transcript_id, annot_gene_name, annot_transcr...
dbl (28): gene_ID, transcript_ID, n_exons, length, 209_1_VZ, 209_2_VZ, 209_3...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"ref/EWCE/CellTypeData_DamonNeuralFetalOnly.rda"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>celltypemarkers <span class="ot">&lt;-</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="st">'https://www.cell.com/cms/10.1016/j.neuron.2019.06.011/attachment/508ae008-b926-487a-b871-844a12acc1f8/mmc5'</span>, <span class="at">sheet=</span><span class="st">'Cluster enriched genes'</span>) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>celltypemarkers_tableS5 <span class="ot">=</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="st">'https://ars.els-cdn.com/content/image/1-s2.0-S0896627319305616-mmc6.xlsx'</span>,<span class="at">sheet=</span><span class="dv">2</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>celltypemarkers.bg <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"ref/polioudakis_neuron2020/single_cell_background_MJG221228.csv"</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensembl_gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>celltypemarkers.bg <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">c</span>(celltypemarkers.bg, celltypemarkers<span class="sc">$</span>Ensembl))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-3_fdd6eae7b49ffc8571fd93c61a2a76c4">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>datExpr.isoCounts <span class="ot">=</span> <span class="fu">as.data.frame</span>(cts[,<span class="dv">12</span><span class="sc">:</span><span class="dv">35</span>])</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(datExpr.isoCounts) <span class="ot">=</span> cts<span class="sc">$</span>annot_transcript_id</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>datMeta <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">sample=</span><span class="fu">colnames</span>(datExpr.isoCounts))</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>datMeta<span class="sc">$</span>Region <span class="ot">=</span> <span class="fu">substr</span>(datMeta<span class="sc">$</span>sample, <span class="dv">7</span>,<span class="dv">9</span>)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>datMeta<span class="sc">$</span>Subject <span class="ot">=</span> <span class="fu">substr</span>(datMeta<span class="sc">$</span>sample, <span class="dv">1</span>,<span class="dv">3</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>datMeta<span class="sc">$</span>batch <span class="ot">=</span> <span class="fu">substr</span>(datMeta<span class="sc">$</span>sample, <span class="dv">5</span>,<span class="dv">5</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>datAnno <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">gene_id =</span> cts<span class="sc">$</span>annot_gene_id, <span class="at">transcript_id=</span>cts<span class="sc">$</span>annot_transcript_id,</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">transcript_name=</span>cts<span class="sc">$</span>annot_transcript_name, <span class="at">length=</span>cts<span class="sc">$</span>length,</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">gene_name =</span> cts<span class="sc">$</span>annot_gene_name, <span class="at">ensg =</span> <span class="fu">substr</span>(cts<span class="sc">$</span>annot_gene_id,<span class="dv">1</span>,<span class="dv">15</span>), <span class="at">novelty=</span>cts<span class="sc">$</span>transcript_novelty)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>brainRBPs <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"data/RBP_Data/CSVs/RBP_targets_v5.csv"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>brainRBPs <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">select</span>(brainRBPs, <span class="sc">-</span><span class="fu">c</span>(MGI.symbol, ENSMUSG))</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>encodeRBPs <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"data/RBP_Data/CSVs/RBP_targets_ENCODE.csv"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>encodeRBPs <span class="ot">=</span> encodeRBPs <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cell.type<span class="sc">==</span><span class="st">"HepG2"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="st">"HGNC.symbol"</span><span class="ot">=</span><span class="st">"hgnc_symbol"</span>, <span class="st">"ENSG"</span><span class="ot">=</span><span class="st">"ensembl_gene_id"</span>)</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>rbp_targets <span class="ot">=</span> <span class="fu">rbind</span>(brainRBPs,encodeRBPs)</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>rbp_targets<span class="sc">$</span>regulation[<span class="fu">is.na</span>(rbp_targets<span class="sc">$</span>regulation)]<span class="ot">=</span><span class="st">""</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>mart <span class="ot">=</span> <span class="fu">useMart</span>(<span class="st">"ENSEMBL_MART_ENSEMBL"</span>,<span class="st">"mmusculus_gene_ensembl"</span>)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>f <span class="ot">=</span> <span class="fu">listFilters</span>(mart); a <span class="ot">=</span> <span class="fu">listAttributes</span>(mart)</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>featuresToGet <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>, <span class="st">"external_gene_name"</span>, <span class="st">"hsapiens_homolog_ensembl_gene"</span>, <span class="st">"hsapiens_homolog_associated_gene_name"</span>,<span class="st">"hsapiens_homolog_orthology_type"</span>)</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>mouseHumanHomologs <span class="ot">=</span> <span class="fu">getBM</span>(<span class="at">attributes =</span> featuresToGet,<span class="at">mart =</span> mart)</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>human_mouse_bg <span class="ot">=</span> mouseHumanHomologs <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hsapiens_homolog_orthology_type <span class="sc">==</span> <span class="st">"ortholog_one2one"</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">"hsapiens_homolog_ensembl_gene"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-4_c9580418ab051b6662804d1ac210858c">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Load 3 Networks</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>datExpr.isoFr <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">"data/working/WGCNA/final/datExpr.localIF_batchCorrected_103k.rds"</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>net.isoFr <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">"data/working/WGCNA/final/WGCNA_isoformFraction_top103k_signed_sft14_net.rds"</span>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>net.isoFr<span class="sc">$</span>colors <span class="ot">=</span> <span class="fu">labels2colors</span>(net.isoFr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(net.isoFr<span class="sc">$</span>colors) <span class="ot">=</span> <span class="fu">names</span>(net.isoFr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>net.isoFr<span class="sc">$</span>module.number <span class="ot">=</span> net.isoFr<span class="sc">$</span>cut2<span class="sc">$</span>labels</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>net.isoFr<span class="sc">$</span>MEs <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(<span class="fu">t</span>(datExpr.isoFr), <span class="at">colors=</span>net.isoFr<span class="sc">$</span>colors)</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>net.isoFr<span class="sc">$</span>kMEtable <span class="ot">=</span> <span class="fu">signedKME</span>(<span class="fu">t</span>(datExpr.isoFr), <span class="at">datME =</span>net.isoFr<span class="sc">$</span>MEs<span class="sc">$</span>eigengenes,<span class="at">corFnc =</span> <span class="st">"bicor"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bicor(datExpr, datME, , use = "p"): bicor: zero MAD in variable 'x'.
Pearson correlation was used for individual columns with zero (or missing) MAD.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>datExpr.isoExpr <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="at">file=</span><span class="st">'data/working/WGCNA/final/datExpr.isoExpr_batchCorrected_92k.rds'</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>net.isoExpr <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">'data/working/WGCNA/final/WGCNA_isoExpression_top92k_signed_sft9_net.rds'</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>net.isoExpr<span class="sc">$</span>colors <span class="ot">=</span> <span class="fu">labels2colors</span>(net.isoExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(net.isoExpr<span class="sc">$</span>colors) <span class="ot">=</span> <span class="fu">names</span>(net.isoExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>net.isoExpr<span class="sc">$</span>module.number <span class="ot">=</span> net.isoExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>net.isoExpr<span class="sc">$</span>MEs <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(<span class="fu">t</span>(datExpr.isoExpr), <span class="at">colors=</span>net.isoExpr<span class="sc">$</span>colors)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>net.isoExpr<span class="sc">$</span>kMEtable <span class="ot">=</span> <span class="fu">signedKME</span>(<span class="fu">t</span>(datExpr.isoExpr), <span class="at">datME =</span>net.isoExpr<span class="sc">$</span>MEs<span class="sc">$</span>eigengenes,<span class="at">corFnc =</span> <span class="st">"bicor"</span>)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>datExpr.geneExpr <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="at">file=</span><span class="st">'data/working/WGCNA/final/datExpr.geneExpr_batchCorrected_16k.rds'</span>)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>net.geneExpr <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="at">file=</span><span class="st">'data/working/WGCNA/final/WGCNA_geneExpression_top16k_signed_sft14_net.rds'</span>)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>net.geneExpr<span class="sc">$</span>colors <span class="ot">=</span> <span class="fu">labels2colors</span>(net.geneExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(net.geneExpr<span class="sc">$</span>colors) <span class="ot">=</span> <span class="fu">names</span>(net.geneExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>net.geneExpr<span class="sc">$</span>module.number <span class="ot">=</span> net.geneExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>net.geneExpr<span class="sc">$</span>MEs <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(<span class="fu">t</span>(datExpr.geneExpr), <span class="at">colors=</span>net.geneExpr<span class="sc">$</span>colors)</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>net.geneExpr<span class="sc">$</span>kMEtable <span class="ot">=</span> <span class="fu">signedKME</span>(<span class="fu">t</span>(datExpr.geneExpr), <span class="at">datME =</span>net.geneExpr<span class="sc">$</span>MEs<span class="sc">$</span>eigengenes,<span class="at">corFnc =</span> <span class="st">"bicor"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="network-dendrogram-plots" class="level2">
<h2 class="anchored" data-anchor-id="network-dendrogram-plots">Network Dendrogram Plots</h2>
<section id="fig4a" class="level3">
<h3 class="anchored" data-anchor-id="fig4a">Fig4A</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-5_266da92caa291402e277462a761c7341">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>this_tree <span class="ot">=</span> net.isoFr<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]]</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>this_anno <span class="ot">=</span> datAnno[<span class="fu">match</span>(<span class="fu">rownames</span>(datExpr.isoFr), datAnno<span class="sc">$</span>transcript_id),]</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>these_mods <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">IsoUsage=</span>net.isoFr<span class="sc">$</span>colors)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>isoExpr <span class="ot">=</span> net.isoExpr<span class="sc">$</span>colors[this_anno<span class="sc">$</span>transcript_id]</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>geneExpr <span class="ot">=</span> net.geneExpr<span class="sc">$</span>colors[this_anno<span class="sc">$</span>gene_id]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-6_3eba4595c459dda8f729badd9fd2932e">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>cell_anno <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="st">'white'</span>, <span class="at">nrow=</span><span class="fu">nrow</span>(this_anno), <span class="at">ncol =</span> <span class="fu">length</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))))</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cell_anno) <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster)) {</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  marker_genes <span class="ot">=</span> celltypemarkers <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Ensembl) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  cell_anno[this_anno<span class="sc">$</span>ensg <span class="sc">%in%</span> marker_genes, this_cell] <span class="ot">=</span> <span class="st">"red"</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-7_1a1774250439d97547992b6534b62a10">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>rbp.comp <span class="ot">=</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="st">"ref/RBPCompilation_Sundararaman_MolCell2016_TableS1.xlsx"</span>, <span class="st">'Sheet2_1072_RBP_compilation'</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>RBP <span class="ot">=</span> <span class="st">'white'</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>RBP[this_anno<span class="sc">$</span>gene_name <span class="sc">%in%</span> rbp.comp<span class="sc">$</span>Gene.Symbol] <span class="ot">=</span> <span class="st">'black'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-8_8ca7417444653d92724bd1d56247ec1d">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_tree, <span class="at">colors=</span><span class="fu">cbind</span>(these_mods), <span class="at">dendroLabels =</span> F, <span class="at">cex.colorLabels =</span> <span class="fl">0.4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="576"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-9_be354d73a5e30b302a42f5c23402dbec">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/Fig4/Fig4A_IsoExprDendro_wRBPComp.pdf"</span>, <span class="at">width=</span><span class="dv">6</span>, <span class="at">height=</span><span class="fl">3.5</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_tree, <span class="at">colors=</span><span class="fu">cbind</span>(these_mods), <span class="at">dendroLabels =</span> F, <span class="at">cex.colorLabels =</span> <span class="fl">0.4</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
<section id="figs5a" class="level3">
<h3 class="anchored" data-anchor-id="figs5a">FigS5A</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-10_af5a5953c1eb76d50fa61c7a7f70b35c">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>this_tree <span class="ot">=</span> net.geneExpr<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]]</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>this_anno <span class="ot">=</span> datAnno[<span class="fu">match</span>(<span class="fu">rownames</span>(datExpr.geneExpr), datAnno<span class="sc">$</span>gene_id),]</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>these_mods <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">geneExpr=</span>net.geneExpr<span class="sc">$</span>colors)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>IsoExpr <span class="ot">=</span> net.isoExpr<span class="sc">$</span>colors[this_anno<span class="sc">$</span>transcript_id]</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>IsoUsage <span class="ot">=</span> net.isoFr<span class="sc">$</span>colors[this_anno<span class="sc">$</span>transcript_id]</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>cell_anno <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="st">'grey'</span>, <span class="at">nrow=</span><span class="fu">nrow</span>(this_anno), <span class="at">ncol =</span> <span class="fu">length</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))))</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cell_anno) <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster)) {</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  marker_genes <span class="ot">=</span> celltypemarkers <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Ensembl) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  cell_anno[this_anno<span class="sc">$</span>ensg <span class="sc">%in%</span> marker_genes, this_cell] <span class="ot">=</span> <span class="st">"red"</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>rbp.comp <span class="ot">=</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="st">"ref/RBPCompilation_Sundararaman_MolCell2016_TableS1.xlsx"</span>, <span class="st">'Sheet2_1072_RBP_compilation'</span>)</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>RBP <span class="ot">=</span> <span class="st">'white'</span></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>RBP[this_anno<span class="sc">$</span>gene_name <span class="sc">%in%</span> rbp.comp<span class="sc">$</span>Gene.Symbol] <span class="ot">=</span> <span class="st">'black'</span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_tree, <span class="at">colors=</span><span class="fu">cbind</span>(these_mods,cell_anno), <span class="at">dendroLabels =</span> F, <span class="at">cex.colorLabels =</span> <span class="fl">0.4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="576"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-11_1f2335b79127accc557735f6a0a1d3f3">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/supplement/FigS5A_Dendro_geneExpr_wCellAnno_wRBP.pdf"</span>, <span class="at">width=</span><span class="dv">8</span>,<span class="at">height=</span><span class="dv">6</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_tree, <span class="at">colors=</span><span class="fu">cbind</span>(these_mods,cell_anno), <span class="at">dendroLabels =</span> F, <span class="at">cex.colorLabels =</span> <span class="fl">0.8</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
<section id="figs5b" class="level3">
<h3 class="anchored" data-anchor-id="figs5b">FigS5B</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-12_1c4e3900fa44ac1c19c8df1a23d35dac">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>this_tree <span class="ot">=</span> net.isoExpr<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]]</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>this_anno <span class="ot">=</span> datAnno[<span class="fu">match</span>(<span class="fu">rownames</span>(datExpr.isoExpr), datAnno<span class="sc">$</span>transcript_id),]</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>these_mods <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">isoExpr=</span>net.isoExpr<span class="sc">$</span>colors)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>GeneExpr <span class="ot">=</span> net.geneExpr<span class="sc">$</span>colors[this_anno<span class="sc">$</span>gene_id]</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>IsoUsage <span class="ot">=</span> net.isoFr<span class="sc">$</span>colors[this_anno<span class="sc">$</span>transcript_id]</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>cell_anno <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="st">'grey'</span>, <span class="at">nrow=</span><span class="fu">nrow</span>(this_anno), <span class="at">ncol =</span> <span class="fu">length</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))))</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cell_anno) <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster)) {</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  marker_genes <span class="ot">=</span> celltypemarkers <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Ensembl) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  cell_anno[this_anno<span class="sc">$</span>ensg <span class="sc">%in%</span> marker_genes, this_cell] <span class="ot">=</span> <span class="st">"red"</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>rbp.comp <span class="ot">=</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="st">"ref/RBPCompilation_Sundararaman_MolCell2016_TableS1.xlsx"</span>, <span class="st">'Sheet2_1072_RBP_compilation'</span>)</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>RBP <span class="ot">=</span> <span class="st">'white'</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>RBP[this_anno<span class="sc">$</span>gene_name <span class="sc">%in%</span> rbp.comp<span class="sc">$</span>Gene.Symbol] <span class="ot">=</span> <span class="st">'black'</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_tree, <span class="at">colors=</span><span class="fu">cbind</span>(these_mods,cell_anno), <span class="at">dendroLabels =</span> F, <span class="at">cex.colorLabels =</span> <span class="fl">0.4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="576"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-13_c6b12e671e62bb274a685070c6866bfa">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/supplement/FigS5B_Dendro_isoExpr_wCellAnno_wRBP.pdf"</span>, <span class="at">width=</span><span class="dv">8</span>,<span class="at">height=</span><span class="dv">6</span>)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_tree, <span class="at">colors=</span><span class="fu">cbind</span>(these_mods,cell_anno), <span class="at">dendroLabels =</span> F, <span class="at">cex.colorLabels =</span> <span class="fl">0.8</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
<section id="figs5c" class="level3">
<h3 class="anchored" data-anchor-id="figs5c">FigS5C</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-14_908640768288da11009af0b687961877">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>this_tree <span class="ot">=</span> net.isoFr<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]]</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>this_anno <span class="ot">=</span> datAnno[<span class="fu">match</span>(<span class="fu">rownames</span>(datExpr.isoFr), datAnno<span class="sc">$</span>transcript_id),]</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>these_mods <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">isoFr=</span>net.isoFr<span class="sc">$</span>colors)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>GeneExpr <span class="ot">=</span> net.geneExpr<span class="sc">$</span>colors[this_anno<span class="sc">$</span>gene_id]</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>IsoExpr <span class="ot">=</span> net.isoExpr<span class="sc">$</span>colors[this_anno<span class="sc">$</span>transcript_id]</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>cell_anno <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="st">'grey'</span>, <span class="at">nrow=</span><span class="fu">nrow</span>(this_anno), <span class="at">ncol =</span> <span class="fu">length</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))))</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cell_anno) <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster)) {</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  marker_genes <span class="ot">=</span> celltypemarkers <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Ensembl) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>  cell_anno[this_anno<span class="sc">$</span>ensg <span class="sc">%in%</span> marker_genes, this_cell] <span class="ot">=</span> <span class="st">"red"</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>rbp.comp <span class="ot">=</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="st">"ref/RBPCompilation_Sundararaman_MolCell2016_TableS1.xlsx"</span>, <span class="st">'Sheet2_1072_RBP_compilation'</span>)</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>RBP <span class="ot">=</span> <span class="st">'white'</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>RBP[this_anno<span class="sc">$</span>gene_name <span class="sc">%in%</span> rbp.comp<span class="sc">$</span>Gene.Symbol] <span class="ot">=</span> <span class="st">'black'</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_tree, <span class="at">colors=</span><span class="fu">cbind</span>(these_mods,cell_anno), <span class="at">dendroLabels =</span> F, <span class="at">cex.colorLabels =</span> <span class="fl">0.4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="576"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-15_7cbb679d996495c6f90c95a6ebc2dded">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/supplement/FigS5C_Dendro_isoUsage_wCellAnno_wRBP.pdf"</span>, <span class="at">width=</span><span class="dv">8</span>,<span class="at">height=</span><span class="dv">6</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_tree, <span class="at">colors=</span><span class="fu">cbind</span>(these_mods,cell_anno), <span class="at">dendroLabels =</span> F, <span class="at">cex.colorLabels =</span> <span class="fl">0.8</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
</section>
<section id="compare-networks" class="level2">
<h2 class="anchored" data-anchor-id="compare-networks">Compare Networks</h2>
<section id="cell-type-enrichment" class="level3">
<h3 class="anchored" data-anchor-id="cell-type-enrichment">Cell-type enrichment</h3>
<section id="fishers-exact" class="level4">
<h4 class="anchored" data-anchor-id="fishers-exact">Fisher’s exact</h4>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-16_b292930db7c84640a4fc177cc3c296f8">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>networks <span class="ot">=</span> <span class="fu">list</span>(<span class="st">"geneExpr"</span> <span class="ot">=</span> net.geneExpr<span class="sc">$</span>module.number[datAnno<span class="sc">$</span>gene_id], </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>                <span class="st">"isoExpr"</span> <span class="ot">=</span> net.isoExpr<span class="sc">$</span>module.number[datAnno<span class="sc">$</span>transcript_id], </span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>                <span class="st">"isoUsage"</span> <span class="ot">=</span> net.isoFr<span class="sc">$</span>module.number[datAnno<span class="sc">$</span>transcript_id])</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_net <span class="cf">in</span> <span class="fu">names</span>(networks)) {</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  all_mods <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(networks[[this_net]][networks[[this_net]] <span class="sc">!=</span> <span class="dv">0</span>]))</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_mod <span class="cf">in</span> all_mods) {</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>    mod_genes <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(datAnno<span class="sc">$</span>ensg[networks[[this_net]]<span class="sc">==</span>this_mod]))</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>    mod_gene.bg <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(datAnno<span class="sc">$</span>ensg[<span class="sc">!</span><span class="fu">is.na</span>(networks[[this_net]])]))</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster)) {</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>      marker_genes <span class="ot">=</span> celltypemarkers <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> </span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(Ensembl) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>      enrichment <span class="ot">=</span> <span class="fu">ORA</span>(mod_genes, marker_genes, mod_gene.bg, celltypemarkers.bg)</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>      df_fisher.celltype <span class="ot">=</span> <span class="fu">rbind</span>(df_fisher.celltype, </span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">data.frame</span>(<span class="at">net =</span> this_net, <span class="at">mod =</span> this_mod, <span class="at">cell=</span>this_cell,</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>                                            <span class="fu">as.data.frame</span>(<span class="fu">t</span>(enrichment))))</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(col <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"OR"</span>, <span class="st">"Fisher.p"</span>)) df_fisher.celltype[,col] <span class="ot">=</span> <span class="fu">as.numeric</span>(df_fisher.celltype[,col])</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype<span class="sc">$</span>Fisher.p[df_fisher.celltype<span class="sc">$</span>OR<span class="sc">&lt;</span><span class="dv">1</span>] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype<span class="sc">$</span>Fisher.q <span class="ot">=</span> <span class="fu">p.adjust</span>(df_fisher.celltype<span class="sc">$</span>Fisher.p,<span class="st">'fdr'</span>)</span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype<span class="sc">$</span>label <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype<span class="sc">$</span>label[df_fisher.celltype<span class="sc">$</span>Fisher.q<span class="sc">&lt;</span>.<span class="dv">05</span>] <span class="ot">=</span> <span class="fu">signif</span>((df_fisher.celltype<span class="sc">$</span>OR[df_fisher.celltype<span class="sc">$</span>Fisher.q<span class="sc">&lt;</span>.<span class="dv">05</span>]),<span class="dv">1</span>)</span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype<span class="sc">$</span>net <span class="ot">=</span> <span class="fu">factor</span>(df_fisher.celltype<span class="sc">$</span>net, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"geneExpr"</span>, <span class="st">"isoExpr"</span>, <span class="st">"isoUsage"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-17_e61e4f0f0f805185ced7c6d833df2ee6">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order cell types</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>order.celltype <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"vRG"</span>, <span class="st">"oRG"</span>, <span class="st">"PgS"</span>, <span class="st">"PgG2M"</span>, <span class="st">"IP"</span>, <span class="st">"ExN"</span>, <span class="st">"ExM"</span>, <span class="st">"ExM-U"</span>, <span class="st">"ExDp1"</span>, <span class="st">"ExDp2"</span>, <span class="st">"InMGE"</span>, <span class="st">"InCGE"</span>, <span class="st">"OPC"</span>, <span class="st">"End"</span>, <span class="st">"Per"</span>, <span class="st">"Mic"</span>)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Order modules</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>geneExpr_order.celltype <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">21</span>,<span class="dv">9</span>,<span class="dv">16</span>,<span class="dv">15</span>,<span class="dv">1</span>, <span class="co"># Progenitors</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>                            <span class="dv">5</span>,<span class="dv">12</span>,<span class="dv">6</span>,<span class="dv">8</span>,<span class="dv">10</span>, <span class="co">#Exc Neurons</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>                            <span class="dv">4</span>,<span class="dv">11</span>, <span class="co">#Exc Neurons 2</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>                            <span class="dv">7</span>,<span class="dv">19</span>,<span class="dv">14</span>, <span class="co">#In Neurons</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>                            <span class="dv">3</span>,<span class="dv">13</span>,<span class="dv">17</span>,<span class="dv">18</span>,<span class="dv">20</span>,<span class="dv">22</span>,<span class="dv">23</span>) <span class="co"># Other </span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>isoExpr_order.celltype <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">15</span>,<span class="dv">41</span>,<span class="dv">4</span>,<span class="dv">3</span>,<span class="dv">11</span>,<span class="dv">21</span>, <span class="co"># Progenitors</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>                           <span class="dv">12</span>,<span class="dv">19</span>,<span class="dv">23</span>, <span class="co"># Sub progenitors</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>                           <span class="dv">16</span>, <span class="co"># ExN</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>                           <span class="dv">1</span>,<span class="dv">30</span>,<span class="dv">6</span>,<span class="dv">20</span>,<span class="dv">14</span>,<span class="dv">38</span>,<span class="dv">5</span>,<span class="dv">8</span>,<span class="dv">46</span>,<span class="dv">34</span>,<span class="dv">49</span>,<span class="dv">45</span>,<span class="dv">31</span>, <span class="co"># Neurons </span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>                           <span class="dv">47</span>,<span class="dv">25</span>,<span class="dv">26</span>, <span class="co"># Sub neurons</span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>                           <span class="dv">17</span>,<span class="dv">42</span>,<span class="dv">9</span>,<span class="dv">29</span>, <span class="co"># ExM</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>                           <span class="dv">35</span>,<span class="dv">40</span>,<span class="dv">53</span>,<span class="dv">27</span>, <span class="co"># Sub neurons 2</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>                           <span class="dv">48</span>,<span class="dv">39</span>,<span class="dv">13</span>, <span class="co"># In neurons</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>                           <span class="dv">7</span>,<span class="dv">10</span>,<span class="dv">18</span>,<span class="dv">22</span>,<span class="dv">24</span>,<span class="dv">28</span>,<span class="dv">32</span>,<span class="dv">33</span>,<span class="dv">36</span>,<span class="dv">37</span>,<span class="dv">43</span>,<span class="dv">44</span>,<span class="dv">50</span>,<span class="dv">51</span>,<span class="dv">52</span>) <span class="co"># Other</span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>isoUsage_order.celltype <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">12</span>,<span class="dv">11</span>, <span class="co"># Progenitors to neurons</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>                            <span class="dv">1</span>,<span class="dv">27</span>, <span class="co"># Non-migrating</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>                            <span class="dv">2</span>,<span class="dv">8</span>,<span class="dv">14</span>, <span class="co"># Neurons</span></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>                            <span class="dv">20</span>,<span class="dv">15</span>,<span class="dv">24</span>,<span class="dv">10</span>,<span class="dv">29</span>, <span class="co"># Exc neurons</span></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>                            <span class="dv">23</span>, <span class="co"># In neurons</span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>                            <span class="dv">19</span>, <span class="co"># OPC</span></span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>                            <span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">7</span>,<span class="dv">9</span>,<span class="dv">13</span>,<span class="dv">16</span>,<span class="dv">17</span>,<span class="dv">18</span>,<span class="dv">21</span>,<span class="dv">22</span>,<span class="dv">25</span>,<span class="dv">26</span>,<span class="dv">28</span>) <span class="co"># Other</span></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype_geneExpr <span class="ot">=</span> df_fisher.celltype  <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"geneExpr"</span>)</span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype_geneExpr<span class="sc">$</span>mod <span class="ot">=</span> <span class="fu">factor</span>(df_fisher.celltype_geneExpr<span class="sc">$</span>mod, <span class="at">levels=</span>geneExpr_order.celltype)</span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype_isoExpr <span class="ot">=</span> df_fisher.celltype  <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoExpr"</span>)</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype_isoExpr<span class="sc">$</span>mod <span class="ot">=</span> <span class="fu">factor</span>(df_fisher.celltype_isoExpr<span class="sc">$</span>mod, <span class="at">levels=</span>isoExpr_order.celltype)</span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype_isoUsage <span class="ot">=</span> df_fisher.celltype  <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoUsage"</span>)</span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype_isoUsage<span class="sc">$</span>mod <span class="ot">=</span> <span class="fu">factor</span>(df_fisher.celltype_isoUsage<span class="sc">$</span>mod, <span class="at">levels=</span>isoUsage_order.celltype)</span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Cell type aggregation</span></span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype <span class="ot">=</span> df_fisher.celltype <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">cell.type =</span> <span class="fu">case_when</span>(cell <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'vRG'</span>, <span class="st">'oRG'</span>, <span class="st">'PgS'</span>, <span class="st">'PgG2M'</span>, <span class="st">'IP'</span>) <span class="sc">~</span> <span class="st">'Progenitor'</span>,</span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a>                                             cell <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'ExN'</span>, <span class="st">'ExM'</span>, <span class="st">'ExM-U'</span>, <span class="st">'ExDp1'</span>, <span class="st">'ExDp2'</span>) <span class="sc">~</span> <span class="st">'Exc Neuron'</span>,</span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a>                                             cell <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'InMGE'</span>, <span class="st">'InCGE'</span>) <span class="sc">~</span> <span class="st">'Inh Neuron'</span>,</span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a>                                             cell <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'OPC'</span>, <span class="st">'End'</span>, <span class="st">'Per'</span>, <span class="st">'Mic'</span>) <span class="sc">~</span> <span class="st">'Other'</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="figs6a" class="level3">
<h3 class="anchored" data-anchor-id="figs6a">FigS6A</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-18_53cc09df05653f82e6b632b827e5593d">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>CellType_ColorBar <span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher.celltype <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"geneExpr"</span>,mod<span class="sc">==</span><span class="dv">1</span>),<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(cell, <span class="at">levels=</span>order.celltype))) <span class="sc">+</span> </span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(<span class="dv">1</span>),<span class="at">fill=</span>cell.type)) <span class="sc">+</span> </span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_point(aes(y=factor(1), shape=data.type),position=position_dodge2(width=1),size=0.5) + scale_shape_manual(values = c(1:9)) + </span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_x_discrete(labels=sapply(strsplit(levels(df_fisher_rbp$target), "_"), "[[",1)) +</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>,<span class="at">vjust=</span><span class="dv">1</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">size=</span><span class="dv">6</span>), </span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.size=</span><span class="fu">unit</span>(<span class="fl">0.3</span>,<span class="st">'cm'</span>), <span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>), <span class="at">legend.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>), </span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">21</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>),<span class="st">"pt"</span>), <span class="at">legend.position=</span><span class="fu">c</span>(<span class="fl">0.5</span>,<span class="sc">-</span><span class="fl">2.5</span>), <span class="at">legend.box=</span><span class="st">"horizontal"</span>, <span class="at">legend.direction=</span><span class="st">"horizontal"</span>, </span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major=</span><span class="fu">element_blank</span>(), <span class="at">panel.grid.minor=</span><span class="fu">element_blank</span>(), <span class="at">panel.border=</span><span class="fu">element_blank</span>(), <span class="at">axis.line=</span><span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.2</span>)) <span class="sc">+</span> </span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">''</span>, <span class="at">y=</span><span class="st">''</span>) <span class="sc">+</span> paletteer<span class="sc">::</span><span class="fu">scale_fill_paletteer_d</span>(<span class="st">"rcartocolor::Vivid"</span>) <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">order=</span><span class="dv">1</span>, <span class="at">nrow=</span><span class="dv">1</span>))</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>plot_gE <span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher.celltype_geneExpr,<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(mod, <span class="at">levels=</span><span class="fu">rev</span>(<span class="fu">levels</span>(mod))), <span class="at">x=</span><span class="fu">factor</span>(cell, <span class="at">levels=</span>order.celltype), <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.q))) <span class="sc">+</span> </span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>net, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="fl">2.5</span>)<span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>,<span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> </span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>), <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">'Cell Type'</span>, <span class="at">y=</span><span class="st">'Module'</span>)</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(plot_gE,CellType_ColorBar, <span class="at">align=</span><span class="st">"v"</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">axis=</span><span class="st">"lr"</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">9</span>,<span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="576"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-19_fda21882f53a0428ea195dd032c35a06">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/supplement/FigS6A_geneExpr_ModOrdered.pdf"</span>, <span class="at">height=</span><span class="dv">6</span>, <span class="at">width=</span><span class="dv">6</span>)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(plot_gE,CellType_ColorBar, <span class="at">align=</span><span class="st">"v"</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">axis=</span><span class="st">"lr"</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">9</span>,<span class="dv">1</span>))</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
<section id="figs6b" class="level3">
<h3 class="anchored" data-anchor-id="figs6b">FigS6B</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-20_be3b8f42428188dbfe0940bbbcf8e5d2">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>plot_iE <span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher.celltype_isoExpr,<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(mod, <span class="at">levels=</span><span class="fu">rev</span>(<span class="fu">levels</span>(mod))), <span class="at">x=</span><span class="fu">factor</span>(cell, <span class="at">levels=</span>order.celltype), <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.q))) <span class="sc">+</span> </span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>net, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="fl">2.5</span>)<span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>,<span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> </span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>), <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">'Cell Type'</span>, <span class="at">y=</span><span class="st">'Module'</span>)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(plot_iE,CellType_ColorBar, <span class="at">align=</span><span class="st">"v"</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">axis=</span><span class="st">"lr"</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">9</span>,<span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-21_e78b01c4bebcaa962d9b3a403d984d2b">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/supplement/FigS6B_isoExpr_ModOrdered.pdf"</span>, <span class="at">height=</span><span class="dv">6</span>, <span class="at">width=</span><span class="dv">6</span>)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(plot_iE,CellType_ColorBar, <span class="at">align=</span><span class="st">"v"</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">axis=</span><span class="st">"lr"</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">9</span>,<span class="dv">1</span>))</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
<section id="figs6c" class="level3">
<h3 class="anchored" data-anchor-id="figs6c">FigS6C</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-22_a542fce3506e7b1c5d659b9ca4bd61f1">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>plot_iU <span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher.celltype_isoUsage,<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(mod, <span class="at">levels=</span><span class="fu">rev</span>(<span class="fu">levels</span>(mod))), <span class="at">x=</span><span class="fu">factor</span>(cell, <span class="at">levels=</span>order.celltype), <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.q))) <span class="sc">+</span> </span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>net, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="fl">2.5</span>)<span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>,<span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> </span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>), <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">'Cell Type'</span>, <span class="at">y=</span><span class="st">'Module'</span>)</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(plot_iU,CellType_ColorBar, <span class="at">align=</span><span class="st">"v"</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">axis=</span><span class="st">"lr"</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">9</span>,<span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid" width="576"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-23_d93d3dbc0631b7463ca714ce0bc4a4e4">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/supplement/FigS6C_isoUsage_ModOrdered.pdf"</span>, <span class="at">height=</span><span class="dv">6</span>, <span class="at">width=</span><span class="dv">6</span>)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(plot_iU,CellType_ColorBar, <span class="at">align=</span><span class="st">"v"</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">axis=</span><span class="st">"lr"</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">9</span>,<span class="dv">1</span>))</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
<section id="rare-var-enrichments" class="level3">
<h3 class="anchored" data-anchor-id="rare-var-enrichments">Rare Var Enrichments</h3>
<section id="gene-level" class="level4">
<h4 class="anchored" data-anchor-id="gene-level">Gene level</h4>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-24_801e6eb56c8e79c2af381ad1dc304c9c">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>geneAnno <span class="ot">=</span> rtracklayer<span class="sc">::</span><span class="fu">import</span>(<span class="st">"ref/gencode.v33lift37.annotation.gtf.gz"</span>) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(type<span class="sc">==</span><span class="st">'gene'</span>)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>geneAnno<span class="sc">$</span>ensg <span class="ot">=</span> <span class="fu">substr</span>(geneAnno<span class="sc">$</span>gene_id,<span class="dv">1</span>,<span class="dv">15</span>)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>fu<span class="ot">=</span>openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>((<span class="st">'https://static-content.springer.com/esm/art%3A10.1038%2Fs41588-022-01104-0/MediaObjects/41588_2022_1104_MOESM3_ESM.xlsx'</span>),<span class="st">'Supplementary Table 11'</span>)</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>fu<span class="sc">$</span>p_TADA_ASD[fu<span class="sc">$</span>p_TADA_ASD<span class="sc">==</span><span class="dv">0</span>] <span class="ot">=</span> <span class="fu">min</span>(fu<span class="sc">$</span>p_TADA_ASD[fu<span class="sc">$</span>p_TADA_ASD <span class="sc">&gt;</span><span class="dv">0</span>],<span class="at">na.rm=</span>T)</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>fu<span class="sc">$</span>p_TADA_NDD[fu<span class="sc">$</span>p_TADA_NDD<span class="sc">==</span><span class="dv">0</span>] <span class="ot">=</span> <span class="fu">min</span>(fu<span class="sc">$</span>p_TADA_NDD[fu<span class="sc">$</span>p_TADA_NDD <span class="sc">&gt;</span><span class="dv">0</span>],<span class="at">na.rm=</span>T)</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>geneAnno.logit <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">ASD_fuTADA=</span> <span class="sc">-</span><span class="fu">log10</span>(fu<span class="sc">$</span>p_TADA_ASD)[<span class="fu">match</span>(geneAnno<span class="sc">$</span>ensg, fu<span class="sc">$</span>gene_id)])</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>geneAnno.logit<span class="sc">$</span>NDD_fuTADA <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(fu<span class="sc">$</span>p_TADA_NDD)[<span class="fu">match</span>(geneAnno<span class="sc">$</span>ensg, fu<span class="sc">$</span>gene_id)]</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>SCZ.schema <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/SCHEMA_gene_results.tsv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: One or more parsing issues, see `problems()` for details</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 18321 Columns: 26
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (8): gene_id, group, OR (PTV), OR (Class I), OR (Class II), OR (PTV) up...
dbl (16): Case PTV, Ctrl PTV, Case mis3, Ctrl mis3, Case mis2, Ctrl mis2, P ...
lgl  (2): De novo mis3, De novo mis2

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>geneAnno.logit<span class="sc">$</span>SCZ_schema <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(SCZ.schema<span class="sc">$</span><span class="st">`</span><span class="at">P meta</span><span class="st">`</span>[<span class="fu">match</span>(geneAnno<span class="sc">$</span>ensg, SCZ.schema<span class="sc">$</span>gene_id)])</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>BIP.bipex <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/BipEx_gene_results.tsv'</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(group<span class="sc">==</span><span class="st">"Bipolar Disorder"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 119958 Columns: 20
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (4): gene_id, group, damaging_missense_fisher_gnom_non_psych_OR, ptv_fi...
dbl (16): n_cases, n_controls, damaging_missense_case_count, damaging_missen...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>geneAnno.logit<span class="sc">$</span>BIP.bipex <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(BIP.bipex<span class="sc">$</span>ptv_fisher_gnom_non_psych_pval[<span class="fu">match</span>(geneAnno<span class="sc">$</span>ensg,BIP.bipex<span class="sc">$</span>gene_id)])</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>EPI.epi25 <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/Epi25_gene_results.tsv'</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(group<span class="sc">==</span><span class="st">"EPI"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 71456 Columns: 12
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): gene_id, group
dbl (9): xcase_lof, xctrl_lof, pval_lof, xcase_mpc, xctrl_mpc, pval_mpc, xca...
lgl (1): pval_infrIndel

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>geneAnno.logit<span class="sc">$</span>EPI.epi25 <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(EPI.epi25<span class="sc">$</span>pval[<span class="fu">match</span>(geneAnno<span class="sc">$</span>ensg,EPI.epi25<span class="sc">$</span>gene_id)])</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>networks <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">net=</span><span class="st">"isoExpr"</span>, <span class="at">module =</span> net.isoExpr<span class="sc">$</span>module.number, <span class="at">transcript_id =</span> <span class="fu">names</span>(net.isoExpr<span class="sc">$</span>colors))</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>networks <span class="ot">=</span> <span class="fu">rbind</span>(networks, <span class="fu">data.frame</span>(<span class="at">net=</span><span class="st">'isoUsage'</span>, <span class="at">module =</span> net.isoFr<span class="sc">$</span>module.number, <span class="at">transcript_id =</span> <span class="fu">names</span>(net.isoFr<span class="sc">$</span>colors)))</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>networks<span class="sc">$</span>gene_id <span class="ot">=</span> datAnno<span class="sc">$</span>gene_id[<span class="fu">match</span>(networks<span class="sc">$</span>transcript_id, datAnno<span class="sc">$</span>transcript_id)]</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>networks <span class="ot">=</span> <span class="fu">rbind</span>(networks, <span class="fu">data.frame</span>(<span class="at">net=</span><span class="st">'geneExpr'</span>, <span class="at">module =</span> net.geneExpr<span class="sc">$</span>module.number, <span class="at">transcript_id =</span> <span class="cn">NA</span>, <span class="at">gene_id=</span><span class="fu">names</span>(net.geneExpr<span class="sc">$</span>colors)))</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>networks<span class="sc">$</span>ensg <span class="ot">=</span> <span class="fu">substr</span>(networks<span class="sc">$</span>gene_id,<span class="dv">1</span>,<span class="dv">15</span>)</span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>df_logit.gene <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>binaryVec <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="at">times=</span><span class="fu">nrow</span>(geneAnno))</span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(binaryVec) <span class="ot">=</span> geneAnno<span class="sc">$</span>ensg</span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_net <span class="cf">in</span> <span class="fu">unique</span>(networks<span class="sc">$</span>net)) {</span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>  all_mods <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(networks<span class="sc">$</span>module[networks<span class="sc">$</span>net <span class="sc">==</span>this_net <span class="sc">&amp;</span> networks<span class="sc">$</span>module<span class="sc">!=</span><span class="dv">0</span>]))</span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb42-16"><a href="#cb42-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_mod <span class="cf">in</span> all_mods) {</span>
<span id="cb42-17"><a href="#cb42-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-18"><a href="#cb42-18" aria-hidden="true" tabindex="-1"></a>    this_binary_vec <span class="ot">=</span> binaryVec</span>
<span id="cb42-19"><a href="#cb42-19" aria-hidden="true" tabindex="-1"></a>    this_binary_vec[<span class="fu">names</span>(this_binary_vec) <span class="sc">%in%</span> networks<span class="sc">$</span>ensg[networks<span class="sc">$</span>net<span class="sc">==</span>this_net]] <span class="ot">=</span> <span class="dv">0</span> </span>
<span id="cb42-20"><a href="#cb42-20" aria-hidden="true" tabindex="-1"></a>    this_binary_vec[<span class="fu">names</span>(this_binary_vec) <span class="sc">%in%</span> networks<span class="sc">$</span>ensg[networks<span class="sc">$</span>net<span class="sc">==</span>this_net <span class="sc">&amp;</span> networks<span class="sc">$</span>module<span class="sc">==</span>this_mod]] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb42-21"><a href="#cb42-21" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-22"><a href="#cb42-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(this_rare_var <span class="cf">in</span> <span class="fu">colnames</span>(geneAnno.logit)) {</span>
<span id="cb42-23"><a href="#cb42-23" aria-hidden="true" tabindex="-1"></a>      this_glm <span class="ot">=</span> <span class="fu">summary</span>(<span class="fu">glm</span>(this_binary_vec <span class="sc">~</span> geneAnno.logit[,this_rare_var] <span class="sc">+</span> width <span class="sc">+</span> <span class="fu">log10</span>(width), <span class="at">data=</span>geneAnno, <span class="at">family=</span><span class="st">'binomial'</span>))</span>
<span id="cb42-24"><a href="#cb42-24" aria-hidden="true" tabindex="-1"></a>      df_logit.gene <span class="ot">=</span> <span class="fu">rbind</span>(df_logit.gene, <span class="fu">data.frame</span>(<span class="at">net =</span> this_net, <span class="at">mod =</span> this_mod, <span class="at">set =</span> this_rare_var, <span class="fu">t</span>(this_glm<span class="sc">$</span>coefficients[<span class="dv">2</span>,])))</span>
<span id="cb42-25"><a href="#cb42-25" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb42-26"><a href="#cb42-26" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb42-27"><a href="#cb42-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb42-28"><a href="#cb42-28" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>df_logit.gene<span class="sc">$</span>OR <span class="ot">=</span> <span class="fu">exp</span>(df_logit.gene<span class="sc">$</span>Estimate)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>df_logit.gene<span class="sc">$</span>P <span class="ot">=</span> df_logit.gene<span class="sc">$</span>Pr...z..</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>df_logit.gene<span class="sc">$</span>P[df_logit.gene<span class="sc">$</span>OR <span class="sc">&lt;</span> <span class="dv">1</span>] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>df_logit.gene<span class="sc">$</span>Q <span class="ot">=</span> <span class="fu">p.adjust</span>(df_logit.gene<span class="sc">$</span>P,<span class="st">'fdr'</span>)</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>df_logit.gene<span class="sc">$</span>label <span class="ot">=</span> <span class="fu">signif</span>((df_logit.gene<span class="sc">$</span>OR),<span class="dv">3</span>)</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>df_logit.gene<span class="sc">$</span>label[df_logit.gene<span class="sc">$</span>Q<span class="sc">&gt;</span>.<span class="dv">05</span>] <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>df_logit.gene<span class="sc">$</span>mod <span class="ot">=</span> <span class="fu">factor</span>(df_logit.gene<span class="sc">$</span>mod,<span class="at">levels=</span><span class="fu">c</span>(<span class="dv">100</span><span class="sc">:</span><span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-25_1db44e2d3e766f9bf6c9e01582604ca6">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>df_logit.gene_geneExpr <span class="ot">=</span> df_logit.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"geneExpr"</span>)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>df_logit.gene_geneExpr<span class="sc">$</span>mod <span class="ot">=</span> <span class="fu">factor</span>(df_logit.gene_geneExpr<span class="sc">$</span>mod, <span class="at">levels=</span>geneExpr_order.celltype)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>df_logit.gene_isoExpr <span class="ot">=</span> df_logit.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoExpr"</span>)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>df_logit.gene_isoExpr<span class="sc">$</span>mod <span class="ot">=</span> <span class="fu">factor</span>(df_logit.gene_isoExpr<span class="sc">$</span>mod, <span class="at">levels=</span>isoExpr_order.celltype)</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>df_logit.gene_isoUsage <span class="ot">=</span> df_logit.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoUsage"</span>)</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>df_logit.gene_isoUsage<span class="sc">$</span>mod <span class="ot">=</span> <span class="fu">factor</span>(df_logit.gene_isoUsage<span class="sc">$</span>mod, <span class="at">levels=</span>isoUsage_order.celltype)</span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Module aggregation</span></span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>df_logit.gene_geneExpr <span class="ot">=</span> df_logit.gene_geneExpr <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">mod.group =</span> <span class="fu">case_when</span>(mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">21</span>,<span class="dv">9</span>,<span class="dv">16</span>,<span class="dv">15</span>,<span class="dv">1</span>) <span class="sc">~</span> <span class="st">'Progenitor'</span>,</span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>                                                                                 mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">12</span>,<span class="dv">6</span>,<span class="dv">8</span>,<span class="dv">10</span>,<span class="dv">4</span>,<span class="dv">11</span>) <span class="sc">~</span> <span class="st">'Exc Neuron'</span>,</span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>                                                                                 mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">19</span>) <span class="sc">~</span> <span class="st">'Inh Neuron'</span>,</span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>                                                                                 mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">14</span>,<span class="dv">3</span>,<span class="dv">13</span>,<span class="dv">17</span>,<span class="dv">18</span>,<span class="dv">20</span>,<span class="dv">22</span>,<span class="dv">23</span>) <span class="sc">~</span> <span class="st">'Other'</span>))</span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a>df_logit.gene_isoExpr <span class="ot">=</span> df_logit.gene_isoExpr <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">mod.group =</span> <span class="fu">case_when</span>(mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">15</span>,<span class="dv">41</span>,<span class="dv">4</span>,<span class="dv">3</span>,<span class="dv">11</span>,<span class="dv">21</span>,<span class="dv">12</span>,<span class="dv">19</span>,<span class="dv">23</span>,<span class="dv">26</span>) <span class="sc">~</span> <span class="st">'Progenitor'</span>,</span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>                                                                                 mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">16</span>,<span class="dv">1</span>,<span class="dv">30</span>,<span class="dv">6</span>,<span class="dv">20</span>,<span class="dv">14</span>,<span class="dv">38</span>,<span class="dv">5</span>,<span class="dv">8</span>,<span class="dv">46</span>,<span class="dv">34</span>,<span class="dv">49</span>,<span class="dv">45</span>,<span class="dv">31</span>,<span class="dv">47</span>,<span class="dv">25</span>,<span class="dv">17</span>,<span class="dv">42</span>,<span class="dv">9</span>,<span class="dv">29</span>,<span class="dv">35</span>,<span class="dv">40</span>,<span class="dv">53</span>,<span class="dv">27</span>,<span class="dv">48</span>) <span class="sc">~</span> <span class="st">'Exc Neuron'</span>,</span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>                                                                                 mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">39</span>,<span class="dv">13</span>,<span class="dv">7</span>) <span class="sc">~</span> <span class="st">'Inh Neuron'</span>,</span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>                                                                                 mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">18</span>,<span class="dv">22</span>,<span class="dv">24</span>,<span class="dv">28</span>,<span class="dv">32</span>,<span class="dv">33</span>,<span class="dv">36</span>,<span class="dv">37</span>,<span class="dv">43</span>,<span class="dv">44</span>,<span class="dv">50</span>,<span class="dv">51</span>,<span class="dv">52</span>) <span class="sc">~</span> <span class="st">'Other'</span>))</span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a>df_logit.gene_isoUsage <span class="ot">=</span> df_logit.gene_isoUsage <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">mod.group =</span> <span class="fu">case_when</span>(mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">27</span>,<span class="dv">19</span>) <span class="sc">~</span> <span class="st">'Progenitor'</span>,</span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a>                                                                                 mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">8</span>,<span class="dv">14</span>,<span class="dv">20</span>,<span class="dv">15</span>,<span class="dv">24</span>,<span class="dv">10</span>,<span class="dv">29</span>) <span class="sc">~</span> <span class="st">'Exc Neuron'</span>,</span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a>                                                                                 mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">23</span>) <span class="sc">~</span> <span class="st">'Inh Neuron'</span>,</span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a>                                                                                 mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">7</span>,<span class="dv">9</span>,<span class="dv">13</span>,<span class="dv">16</span>,<span class="dv">17</span>,<span class="dv">18</span>,<span class="dv">21</span>,<span class="dv">22</span>,<span class="dv">25</span>,<span class="dv">26</span>,<span class="dv">28</span>) <span class="sc">~</span> <span class="st">'Other'</span>,</span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a>                                                                                 mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">12</span>,<span class="dv">11</span>,<span class="dv">1</span>) <span class="sc">~</span> <span class="st">'Prog/Neur'</span>))</span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Order of Rare Vars</span></span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a>order.rarevar <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"NDD_fuTADA"</span>, <span class="st">"ASD_fuTADA"</span>, <span class="st">"SCZ_schema"</span>, <span class="st">"BIP.bipex"</span>, <span class="st">"EPI.epi25"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-26_5d3928e415c704c679697ceb897b522b">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>ModGroup_ColorBar.gE <span class="ot">=</span> <span class="fu">ggplot</span>(df_logit.gene_geneExpr,<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(mod, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">levels</span>(mod))))) <span class="sc">+</span> </span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(<span class="dv">1</span>),<span class="at">fill=</span>mod.group)) <span class="sc">+</span> </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_point(aes(y=factor(1), shape=data.type),position=position_dodge2(width=1),size=0.5) + scale_shape_manual(values = c(1:9)) + </span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">0</span>,<span class="at">vjust=</span><span class="dv">1</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">size=</span><span class="dv">5</span>), </span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>        <span class="co"># legend.key.size=unit(0.3,'cm'), legend.text=element_text(size=3), legend.title=element_text(size=4), </span></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="sc">-</span><span class="dv">10</span>,<span class="dv">5</span>,<span class="dv">5</span>),<span class="st">"pt"</span>), <span class="at">legend.position=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">3</span>,<span class="fl">0.8</span>), </span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>        <span class="co"># legend.box="vertical", legend.direction="vertical", </span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major=</span><span class="fu">element_blank</span>(), <span class="at">panel.grid.minor=</span><span class="fu">element_blank</span>(), <span class="at">panel.border=</span><span class="fu">element_blank</span>(), <span class="at">axis.line=</span><span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.2</span>)) <span class="sc">+</span> </span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels=</span><span class="fu">rev</span>(<span class="fu">levels</span>(df_logit.gene_geneExpr<span class="sc">$</span>mod))) <span class="sc">+</span></span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">'Module'</span>, <span class="at">y=</span><span class="st">''</span>) <span class="sc">+</span> paletteer<span class="sc">::</span><span class="fu">scale_fill_paletteer_d</span>(<span class="st">"rcartocolor::Vivid"</span>) <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">reverse=</span><span class="cn">TRUE</span>)) <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>legend.gE <span class="ot">=</span> <span class="fu">get_legend</span>(ModGroup_ColorBar.gE <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="fu">c</span>(<span class="fl">2.5</span>,<span class="fl">0.5</span>), <span class="at">legend.key.size=</span><span class="fu">unit</span>(<span class="fl">0.3</span>,<span class="st">'cm'</span>), <span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>), <span class="at">legend.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">4</span>),</span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">legend.box=</span><span class="st">"horizontal"</span>, <span class="at">legend.direction=</span><span class="st">"horizontal"</span>))</span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a>plot_logit.gE <span class="ot">=</span> <span class="fu">ggplot</span>(df_logit.gene_geneExpr,<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(mod, <span class="at">levels=</span><span class="fu">rev</span>(<span class="fu">levels</span>(mod))), <span class="at">x=</span><span class="fu">factor</span>(set, <span class="at">levels=</span>order.rarevar), <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Q))) <span class="sc">+</span> </span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_x_discrete(labels=sapply(strsplit(levels(df_fisher_rbp$target), "_"), "[[",1)) + </span></span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> </span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>net, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> </span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>,<span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> </span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">1</span>, <span class="at">size=</span><span class="dv">4</span>), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>),</span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(), <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="sc">-</span><span class="dv">10</span>),<span class="st">"pt"</span>),</span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.size=</span><span class="fu">unit</span>(<span class="fl">0.25</span>,<span class="st">'cm'</span>), <span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>), <span class="at">legend.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>)) <span class="sc">+</span> </span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">'Rare Variant Set'</span>, <span class="at">y=</span><span class="st">''</span>)</span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a>ModGroup_ColorBar.iE <span class="ot">=</span> <span class="fu">ggplot</span>(df_logit.gene_isoExpr,<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(mod, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">levels</span>(mod))))) <span class="sc">+</span> </span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(<span class="dv">1</span>),<span class="at">fill=</span>mod.group)) <span class="sc">+</span> </span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_point(aes(y=factor(1), shape=data.type),position=position_dodge2(width=1),size=0.5) + scale_shape_manual(values = c(1:9)) + </span></span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-32"><a href="#cb47-32" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-33"><a href="#cb47-33" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">0</span>,<span class="at">vjust=</span><span class="dv">1</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">size=</span><span class="dv">4</span>), </span>
<span id="cb47-34"><a href="#cb47-34" aria-hidden="true" tabindex="-1"></a>        <span class="co"># legend.key.size=unit(0.3,'cm'), legend.text=element_text(size=6), legend.title=element_text(size=6), </span></span>
<span id="cb47-35"><a href="#cb47-35" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="sc">-</span><span class="dv">10</span>,<span class="dv">5</span>,<span class="dv">5</span>),<span class="st">"pt"</span>), <span class="at">legend.position=</span><span class="fu">c</span>(<span class="fl">0.5</span>,<span class="sc">-</span><span class="fl">2.5</span>), </span>
<span id="cb47-36"><a href="#cb47-36" aria-hidden="true" tabindex="-1"></a>        <span class="co"># legend.box="vertical", legend.direction="vertical", </span></span>
<span id="cb47-37"><a href="#cb47-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major=</span><span class="fu">element_blank</span>(), <span class="at">panel.grid.minor=</span><span class="fu">element_blank</span>(), <span class="at">panel.border=</span><span class="fu">element_blank</span>(), <span class="at">axis.line=</span><span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.2</span>)) <span class="sc">+</span> </span>
<span id="cb47-38"><a href="#cb47-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels=</span><span class="fu">rev</span>(<span class="fu">levels</span>(df_logit.gene_geneExpr<span class="sc">$</span>mod))) <span class="sc">+</span></span>
<span id="cb47-39"><a href="#cb47-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">'Module'</span>, <span class="at">y=</span><span class="st">''</span>) <span class="sc">+</span> paletteer<span class="sc">::</span><span class="fu">scale_fill_paletteer_d</span>(<span class="st">"rcartocolor::Vivid"</span>) <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">order=</span><span class="dv">1</span>, <span class="at">nrow=</span><span class="dv">1</span>)) <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb47-40"><a href="#cb47-40" aria-hidden="true" tabindex="-1"></a>legend.iE <span class="ot">=</span> <span class="fu">get_legend</span>(ModGroup_ColorBar.iE <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="fu">c</span>(<span class="fl">2.5</span>,<span class="fl">0.5</span>), <span class="at">legend.key.size=</span><span class="fu">unit</span>(<span class="fl">0.3</span>,<span class="st">'cm'</span>), <span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>), <span class="at">legend.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">4</span>),</span>
<span id="cb47-41"><a href="#cb47-41" aria-hidden="true" tabindex="-1"></a>                                                    <span class="at">legend.box=</span><span class="st">"horizontal"</span>, <span class="at">legend.direction=</span><span class="st">"horizontal"</span>))</span>
<span id="cb47-42"><a href="#cb47-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-43"><a href="#cb47-43" aria-hidden="true" tabindex="-1"></a>plot_logit.iE <span class="ot">=</span> <span class="fu">ggplot</span>(df_logit.gene_isoExpr,<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(mod, <span class="at">levels=</span><span class="fu">rev</span>(<span class="fu">levels</span>(mod))), <span class="at">x=</span><span class="fu">factor</span>(set, <span class="at">levels=</span>order.rarevar), <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Q))) <span class="sc">+</span> </span>
<span id="cb47-44"><a href="#cb47-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_x_discrete(labels=sapply(strsplit(levels(df_fisher_rbp$target), "_"), "[[",1)) + </span></span>
<span id="cb47-45"><a href="#cb47-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>net, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>,<span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> </span>
<span id="cb47-46"><a href="#cb47-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">1</span>, <span class="at">size=</span><span class="dv">4</span>), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>),</span>
<span id="cb47-47"><a href="#cb47-47" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(), <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="sc">-</span><span class="dv">10</span>),<span class="st">"pt"</span>),</span>
<span id="cb47-48"><a href="#cb47-48" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.size=</span><span class="fu">unit</span>(<span class="fl">0.25</span>,<span class="st">'cm'</span>), <span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>), <span class="at">legend.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>)) <span class="sc">+</span> </span>
<span id="cb47-49"><a href="#cb47-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">'Rare Variant Set'</span>, <span class="at">y=</span><span class="st">''</span>)</span>
<span id="cb47-50"><a href="#cb47-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-51"><a href="#cb47-51" aria-hidden="true" tabindex="-1"></a>ModGroup_ColorBar.iU <span class="ot">=</span> <span class="fu">ggplot</span>(df_logit.gene_isoUsage,<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(mod, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">levels</span>(mod))))) <span class="sc">+</span> </span>
<span id="cb47-52"><a href="#cb47-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(<span class="dv">1</span>),<span class="at">fill=</span>mod.group)) <span class="sc">+</span> </span>
<span id="cb47-53"><a href="#cb47-53" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_point(aes(y=factor(1), shape=data.type),position=position_dodge2(width=1),size=0.5) + scale_shape_manual(values = c(1:9)) + </span></span>
<span id="cb47-54"><a href="#cb47-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb47-55"><a href="#cb47-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-56"><a href="#cb47-56" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-57"><a href="#cb47-57" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">0</span>,<span class="at">vjust=</span><span class="dv">1</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">size=</span><span class="dv">4</span>), </span>
<span id="cb47-58"><a href="#cb47-58" aria-hidden="true" tabindex="-1"></a>        <span class="co"># legend.key.size=unit(0.3,'cm'), legend.text=element_text(size=6), legend.title=element_text(size=6), </span></span>
<span id="cb47-59"><a href="#cb47-59" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="sc">-</span><span class="dv">10</span>,<span class="dv">5</span>,<span class="dv">5</span>),<span class="st">"pt"</span>), <span class="at">legend.position=</span><span class="fu">c</span>(<span class="fl">0.5</span>,<span class="sc">-</span><span class="fl">2.5</span>), </span>
<span id="cb47-60"><a href="#cb47-60" aria-hidden="true" tabindex="-1"></a>        <span class="co"># legend.box="vertical", legend.direction="vertical", </span></span>
<span id="cb47-61"><a href="#cb47-61" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major=</span><span class="fu">element_blank</span>(), <span class="at">panel.grid.minor=</span><span class="fu">element_blank</span>(), <span class="at">panel.border=</span><span class="fu">element_blank</span>(), <span class="at">axis.line=</span><span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.2</span>)) <span class="sc">+</span> </span>
<span id="cb47-62"><a href="#cb47-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels=</span><span class="fu">rev</span>(<span class="fu">levels</span>(df_logit.gene_geneExpr<span class="sc">$</span>mod))) <span class="sc">+</span></span>
<span id="cb47-63"><a href="#cb47-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">'Module'</span>, <span class="at">y=</span><span class="st">''</span>) <span class="sc">+</span> paletteer<span class="sc">::</span><span class="fu">scale_fill_paletteer_d</span>(<span class="st">"rcartocolor::Vivid"</span>) <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">order=</span><span class="dv">1</span>, <span class="at">nrow=</span><span class="dv">1</span>)) <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb47-64"><a href="#cb47-64" aria-hidden="true" tabindex="-1"></a>legend.iU <span class="ot">=</span> <span class="fu">get_legend</span>(ModGroup_ColorBar.iU <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="fu">c</span>(<span class="fl">2.5</span>,<span class="fl">0.5</span>), <span class="at">legend.key.size=</span><span class="fu">unit</span>(<span class="fl">0.3</span>,<span class="st">'cm'</span>), <span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>), <span class="at">legend.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">4</span>),</span>
<span id="cb47-65"><a href="#cb47-65" aria-hidden="true" tabindex="-1"></a>                                                    <span class="at">legend.box=</span><span class="st">"horizontal"</span>, <span class="at">legend.direction=</span><span class="st">"horizontal"</span>))</span>
<span id="cb47-66"><a href="#cb47-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-67"><a href="#cb47-67" aria-hidden="true" tabindex="-1"></a>plot_logit.iU <span class="ot">=</span> <span class="fu">ggplot</span>(df_logit.gene_isoUsage,<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(mod, <span class="at">levels=</span><span class="fu">rev</span>(<span class="fu">levels</span>(mod))), <span class="at">x=</span><span class="fu">factor</span>(set, <span class="at">levels=</span>order.rarevar), <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Q))) <span class="sc">+</span> </span>
<span id="cb47-68"><a href="#cb47-68" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_x_discrete(labels=sapply(strsplit(levels(df_fisher_rbp$target), "_"), "[[",1)) + </span></span>
<span id="cb47-69"><a href="#cb47-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>net, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>,<span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> </span>
<span id="cb47-70"><a href="#cb47-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">1</span>, <span class="at">size=</span><span class="dv">4</span>), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>),</span>
<span id="cb47-71"><a href="#cb47-71" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(), <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="sc">-</span><span class="dv">10</span>),<span class="st">"pt"</span>),</span>
<span id="cb47-72"><a href="#cb47-72" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.size=</span><span class="fu">unit</span>(<span class="fl">0.25</span>,<span class="st">'cm'</span>), <span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>), <span class="at">legend.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>)) <span class="sc">+</span> </span>
<span id="cb47-73"><a href="#cb47-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">'Rare Variant Set'</span>, <span class="at">y=</span><span class="st">''</span>)</span>
<span id="cb47-74"><a href="#cb47-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-75"><a href="#cb47-75" aria-hidden="true" tabindex="-1"></a>FigS5_C_geneExpr <span class="ot">=</span> <span class="fu">plot_grid</span>(ModGroup_ColorBar.gE,plot_logit.gE, legend.gE, <span class="at">align=</span><span class="st">"h"</span>, <span class="at">nrow=</span><span class="dv">2</span>, <span class="at">axis=</span><span class="st">"bt"</span>, <span class="at">rel_widths=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>), <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">20</span>,<span class="dv">1</span>))</span>
<span id="cb47-76"><a href="#cb47-76" aria-hidden="true" tabindex="-1"></a>FigS5_C_isoExpr <span class="ot">=</span> <span class="fu">plot_grid</span>(ModGroup_ColorBar.iE,plot_logit.iE, legend.iE, <span class="at">align=</span><span class="st">"h"</span>, <span class="at">nrow=</span><span class="dv">2</span>, <span class="at">axis=</span><span class="st">"bt"</span>, <span class="at">rel_widths=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>), <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">20</span>,<span class="dv">1</span>))</span>
<span id="cb47-77"><a href="#cb47-77" aria-hidden="true" tabindex="-1"></a>FigS5_C_isoUsage <span class="ot">=</span> <span class="fu">plot_grid</span>(ModGroup_ColorBar.iU,plot_logit.iU, legend.iU, <span class="at">align=</span><span class="st">"h"</span>, <span class="at">nrow=</span><span class="dv">2</span>, <span class="at">axis=</span><span class="st">"bt"</span>, <span class="at">rel_widths=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>), <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">20</span>,<span class="dv">1</span>))</span>
<span id="cb47-78"><a href="#cb47-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-79"><a href="#cb47-79" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(FigS5_C_geneExpr,FigS5_C_isoExpr,FigS5_C_isoUsage, <span class="at">align=</span><span class="st">"h"</span>, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">axis=</span><span class="st">"bt"</span>, <span class="at">rel_widths=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid" width="720"></p>
</div>
</div>
</section>
</section>
<section id="gwas-enrichments" class="level3">
<h3 class="anchored" data-anchor-id="gwas-enrichments">GWAS Enrichments</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-27_23bec68c076196d8aa225b2e062a40d1">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>df_gwas <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>gwas <span class="ot">=</span> <span class="fu">dir</span>(<span class="st">"data/LDSC_cell_type_specific/networks/isoExpr_cts_result/"</span>, <span class="at">pattern=</span><span class="st">"txt"</span>)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_gwas <span class="cf">in</span> gwas) {</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>  this_ldsc <span class="ot">=</span> <span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">"data/LDSC_cell_type_specific/networks/isoExpr_cts_result/"</span>,this_gwas),<span class="at">header =</span> T)</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(this_ldsc)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">'module'</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>  this_ldsc<span class="sc">$</span>gwas <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".cell_type_results.txt"</span>, <span class="st">""</span>, this_gwas)</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>  this_ldsc<span class="sc">$</span>net <span class="ot">=</span> <span class="st">'isoExpr'</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>  df_gwas <span class="ot">=</span> <span class="fu">rbind</span>(df_gwas, this_ldsc)</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>gwas <span class="ot">=</span> <span class="fu">dir</span>(<span class="st">"data/LDSC_cell_type_specific/networks/isoUsage_cts_result/"</span>, <span class="at">pattern=</span><span class="st">"txt"</span>)</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_gwas <span class="cf">in</span> gwas) {</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>  this_ldsc <span class="ot">=</span> <span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">"data/LDSC_cell_type_specific/networks/isoUsage_cts_result/"</span>,this_gwas),<span class="at">header =</span> T)</span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(this_ldsc)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">'module'</span></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>  this_ldsc<span class="sc">$</span>gwas <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".cell_type_results.txt"</span>, <span class="st">""</span>, this_gwas)</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>  this_ldsc<span class="sc">$</span>net <span class="ot">=</span> <span class="st">'isoUsage'</span></span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>  df_gwas <span class="ot">=</span> <span class="fu">rbind</span>(df_gwas, this_ldsc)</span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a>gwas <span class="ot">=</span> <span class="fu">dir</span>(<span class="st">"data/LDSC_cell_type_specific/networks/geneExpr_cts_result/"</span>, <span class="at">pattern=</span><span class="st">"txt"</span>)</span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_gwas <span class="cf">in</span> gwas) {</span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a>  this_ldsc <span class="ot">=</span> <span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">"data/LDSC_cell_type_specific/networks/geneExpr_cts_result/"</span>,this_gwas),<span class="at">header =</span> T)</span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(this_ldsc)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">'module'</span></span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a>  this_ldsc<span class="sc">$</span>gwas <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".cell_type_results.txt"</span>, <span class="st">""</span>, this_gwas)</span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a>  this_ldsc<span class="sc">$</span>net <span class="ot">=</span> <span class="st">'geneExpr'</span></span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a>  df_gwas <span class="ot">=</span> <span class="fu">rbind</span>(df_gwas, this_ldsc)</span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a>df_gwas<span class="sc">$</span>fdr <span class="ot">=</span> <span class="fu">p.adjust</span>(df_gwas<span class="sc">$</span>Coefficient_P_value,<span class="st">'fdr'</span>)</span>
<span id="cb48-31"><a href="#cb48-31" aria-hidden="true" tabindex="-1"></a>df_gwas<span class="sc">$</span>mod <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">"M"</span>, <span class="st">""</span>, <span class="fu">unlist</span>(<span class="fu">lapply</span>(<span class="fu">strsplit</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(<span class="fu">strsplit</span>(df_gwas<span class="sc">$</span>module, <span class="st">"[.]"</span>),<span class="st">'['</span>,<span class="dv">2</span>)), <span class="st">'_'</span>),<span class="st">'['</span>,<span class="dv">1</span>))))</span>
<span id="cb48-32"><a href="#cb48-32" aria-hidden="true" tabindex="-1"></a>df_gwas<span class="sc">$</span>Zscore <span class="ot">=</span> df_gwas<span class="sc">$</span>Coefficient<span class="sc">/</span>df_gwas<span class="sc">$</span>Coefficient_std_error</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="rbp-enrichments" class="level3">
<h3 class="anchored" data-anchor-id="rbp-enrichments">RBP Enrichments</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-28_ee22ccb0510d65dc16beade3ce84ec23">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>networks <span class="ot">=</span> <span class="fu">list</span>(<span class="st">"net.isoExpr"</span> <span class="ot">=</span> net.isoExpr<span class="sc">$</span>colors[datAnno<span class="sc">$</span>transcript_id], </span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>                <span class="st">"net.isoFr"</span> <span class="ot">=</span> net.isoFr<span class="sc">$</span>colors[datAnno<span class="sc">$</span>transcript_id],</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>                <span class="st">"net.geneExpr"</span> <span class="ot">=</span> net.geneExpr<span class="sc">$</span>colors[datAnno<span class="sc">$</span>gene_id])</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_net <span class="cf">in</span> <span class="fu">names</span>(networks)) {</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>  all_mods <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(networks[[this_net]][networks[[this_net]] <span class="sc">!=</span> <span class="st">'grey'</span>]))</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_mod <span class="cf">in</span> all_mods) {</span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(this_net <span class="sc">==</span> <span class="st">"net.geneExpr"</span>) {</span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>      modGenes <span class="ot">=</span> <span class="fu">substr</span>(<span class="fu">unique</span>(<span class="fu">na.omit</span>(<span class="fu">names</span>(networks[[this_net]][networks[[this_net]] <span class="sc">==</span> this_mod]))), <span class="dv">1</span>,<span class="dv">15</span>)</span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a>      modGeneg.bg <span class="ot">=</span> <span class="fu">substr</span>(<span class="fu">unique</span>(<span class="fu">na.omit</span>(<span class="fu">names</span>(networks[[this_net]]))),<span class="dv">1</span>,<span class="dv">15</span>)</span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a>      modTx <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(<span class="fu">names</span>(networks[[this_net]][networks[[this_net]] <span class="sc">==</span> this_mod])))</span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>      modGenes <span class="ot">=</span> <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg[<span class="fu">match</span>(modTx, datAnno<span class="sc">$</span>transcript_id)])</span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a>      modGeneg.bg <span class="ot">=</span> <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg[<span class="fu">match</span>(<span class="fu">names</span>(networks[[this_net]]), datAnno<span class="sc">$</span>transcript_id)])</span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a>    } </span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(this_dataset <span class="cf">in</span> <span class="fu">unique</span>(rbp_targets<span class="sc">$</span>dataset.id)) {</span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a>      this_rbp <span class="ot">=</span> rbp_targets <span class="sc">%&gt;%</span> <span class="fu">filter</span>(dataset.id <span class="sc">==</span> this_dataset) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">target =</span> <span class="fu">paste0</span>(RBP, <span class="st">"_"</span>, data.type, <span class="st">"_"</span>, cell.type)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(target) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()  <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a>      target_genes <span class="ot">=</span> rbp_targets <span class="sc">%&gt;%</span> <span class="fu">filter</span>(dataset.id <span class="sc">==</span> this_dataset) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ENSG) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">"Human"</span>,this_rbp)) {</span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a>        this_or <span class="ot">=</span> <span class="fu">ORA</span>(modGenes, target_genes, <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg), <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg))</span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a>        this_or <span class="ot">=</span> <span class="fu">ORA</span>(modGenes, target_genes, <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg), human_mouse_bg)</span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a>      df_fisher_rbp <span class="ot">=</span> <span class="fu">rbind</span>(df_fisher_rbp, <span class="fu">data.frame</span>(<span class="at">net =</span> this_net, <span class="at">mod =</span> this_mod, <span class="at">dataset =</span> this_dataset, <span class="at">target =</span> this_rbp, <span class="fu">t</span>(this_or)))</span>
<span id="cb49-30"><a href="#cb49-30" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb49-31"><a href="#cb49-31" aria-hidden="true" tabindex="-1"></a>  }}</span>
<span id="cb49-32"><a href="#cb49-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-33"><a href="#cb49-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Order of curated RBPs</span></span>
<span id="cb49-34"><a href="#cb49-34" aria-hidden="true" tabindex="-1"></a>order.brainRBPs <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">"data/RBP_Data/curatedRBPs_order.xlsx"</span>) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span>
<span id="cb49-35"><a href="#cb49-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Order Encode by Target Regions</span></span>
<span id="cb49-36"><a href="#cb49-36" aria-hidden="true" tabindex="-1"></a>order.Encode.TarReg <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">"data/RBP_Data/ENCODE_vanNostrand_NatMeth2016_Fig2a_order.xlsx"</span>, <span class="at">sheet=</span><span class="dv">2</span>)</span>
<span id="cb49-37"><a href="#cb49-37" aria-hidden="true" tabindex="-1"></a>order.Encode.TarReg <span class="ot">=</span> order.Encode.TarReg <span class="sc">%&gt;%</span> <span class="fu">filter</span>(name <span class="sc">%in%</span> df_fisher_rbp<span class="sc">$</span>target)</span>
<span id="cb49-38"><a href="#cb49-38" aria-hidden="true" tabindex="-1"></a>order.RBPs <span class="ot">=</span> <span class="fu">rbind</span>(order.brainRBPs, order.Encode.TarReg)</span>
<span id="cb49-39"><a href="#cb49-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-40"><a href="#cb49-40" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp<span class="sc">$</span>OR <span class="ot">=</span> <span class="fu">as.numeric</span>(df_fisher_rbp<span class="sc">$</span>OR)</span>
<span id="cb49-41"><a href="#cb49-41" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp<span class="sc">$</span>Fisher.p[df_fisher_rbp<span class="sc">$</span>OR<span class="sc">&lt;</span><span class="dv">1</span>] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb49-42"><a href="#cb49-42" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp<span class="sc">$</span>Fisher.p <span class="ot">=</span> <span class="fu">p.adjust</span>(<span class="fu">as.numeric</span>(df_fisher_rbp<span class="sc">$</span>Fisher.p),<span class="st">'fdr'</span>)</span>
<span id="cb49-43"><a href="#cb49-43" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp<span class="sc">$</span>label <span class="ot">=</span> <span class="fu">signif</span>(df_fisher_rbp<span class="sc">$</span>OR,<span class="dv">2</span>)</span>
<span id="cb49-44"><a href="#cb49-44" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp<span class="sc">$</span>label[df_fisher_rbp<span class="sc">$</span>Fisher.p<span class="sc">&gt;</span>.<span class="dv">001</span>] <span class="ot">=</span> <span class="st">''</span></span>
<span id="cb49-45"><a href="#cb49-45" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp<span class="sc">$</span>target.region <span class="ot">=</span> order.RBPs<span class="sc">$</span>target.region[<span class="fu">match</span>(df_fisher_rbp<span class="sc">$</span>target, order.RBPs<span class="sc">$</span>name)]</span>
<span id="cb49-46"><a href="#cb49-46" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp<span class="sc">$</span>data.type <span class="ot">=</span> rbp_targets<span class="sc">$</span>data.type[<span class="fu">match</span>(df_fisher_rbp<span class="sc">$</span>dataset, rbp_targets<span class="sc">$</span>dataset.id)]</span>
<span id="cb49-47"><a href="#cb49-47" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp<span class="sc">$</span>cell.type <span class="ot">=</span> rbp_targets<span class="sc">$</span>cell.type[<span class="fu">match</span>(df_fisher_rbp<span class="sc">$</span>dataset, rbp_targets<span class="sc">$</span>dataset.id)]</span>
<span id="cb49-48"><a href="#cb49-48" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp<span class="ot">&lt;-</span>df_fisher_rbp <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(<span class="fu">data.frame</span>(<span class="at">mod.num =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, <span class="at">mod=</span><span class="fu">labels2colors</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "mod"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp<span class="sc">$</span>target <span class="ot">=</span> <span class="fu">factor</span>(df_fisher_rbp<span class="sc">$</span>target, <span class="at">levels=</span>order.RBPs<span class="sc">$</span>name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="tables4d" class="level3">
<h3 class="anchored" data-anchor-id="tables4d">TableS4D</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-29_86017008613a8649c7b3ee3a46634d81">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>TableS4D <span class="ot">&lt;-</span> df_fisher_rbp <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">module.color =</span> mod, <span class="at">mod =</span> <span class="fu">paste0</span>(net,<span class="st">"M."</span>,mod.num,<span class="st">"_"</span>,mod))</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(TableS4D, <span class="at">file=</span><span class="st">"output/tables/TableS4D_fisherRBP.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figs6b-1" class="level3">
<h3 class="anchored" data-anchor-id="figs6b-1">FigS6B</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-30_8286f8ae79f3b9ea22d6aea98a650978">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order of modules</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp.geneExpr <span class="ot">=</span> df_fisher_rbp  <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"net.geneExpr"</span>)</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp.geneExpr<span class="sc">$</span>mod.num <span class="ot">=</span> <span class="fu">factor</span>(df_fisher_rbp.geneExpr<span class="sc">$</span>mod.num, <span class="at">levels=</span>geneExpr_order.celltype)</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp.isoExpr <span class="ot">=</span> df_fisher_rbp  <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"net.isoExpr"</span>)</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp.isoExpr<span class="sc">$</span>mod.num <span class="ot">=</span> <span class="fu">factor</span>(df_fisher_rbp.isoExpr<span class="sc">$</span>mod.num, <span class="at">levels=</span>isoExpr_order.celltype)</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp.isoUsage <span class="ot">=</span> df_fisher_rbp  <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"net.isoFr"</span>)</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp.isoUsage<span class="sc">$</span>mod.num <span class="ot">=</span> <span class="fu">factor</span>(df_fisher_rbp.isoUsage<span class="sc">$</span>mod.num, <span class="at">levels=</span>isoUsage_order.celltype)</span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Color bar for RBP TargetRegions</span></span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>RBP_ColorBar <span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher_rbp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"net.isoFr"</span>,mod.num<span class="sc">==</span><span class="dv">1</span>),<span class="fu">aes</span>(<span class="at">x=</span>target, <span class="at">label=</span>target.region)) <span class="sc">+</span> </span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_tile(aes(y=factor(1),fill=data.type)) + </span></span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(<span class="dv">1</span>),<span class="at">fill=</span>target.region)) <span class="sc">+</span> </span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_point(aes(y=factor(1), shape=target.region),position=position_dodge2(width=1),size=0.5) + scale_shape_manual(values = c(1:9)) + </span></span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(<span class="dv">1</span>), <span class="at">shape=</span>data.type),<span class="at">position=</span><span class="fu">position_dodge2</span>(<span class="at">width=</span><span class="dv">1</span>),<span class="at">size=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>)) <span class="sc">+</span> </span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_x_discrete(labels=df_fisher$RBP) +</span></span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">levels</span>(df_fisher_rbp<span class="sc">$</span>target), <span class="st">"_"</span>), <span class="st">"[["</span>,<span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb53-18"><a href="#cb53-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb53-19"><a href="#cb53-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb53-20"><a href="#cb53-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>,<span class="at">vjust=</span><span class="dv">1</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">size=</span><span class="dv">5</span>), </span>
<span id="cb53-21"><a href="#cb53-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.size=</span><span class="fu">unit</span>(<span class="fl">0.3</span>,<span class="st">'cm'</span>), <span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>), <span class="at">legend.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>), </span>
<span id="cb53-22"><a href="#cb53-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">21</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>),<span class="st">"pt"</span>), <span class="at">legend.position=</span><span class="fu">c</span>(<span class="fl">0.5</span>,<span class="sc">-</span><span class="fl">2.5</span>), <span class="at">legend.box=</span><span class="st">"horizontal"</span>, <span class="at">legend.direction=</span><span class="st">"horizontal"</span>, </span>
<span id="cb53-23"><a href="#cb53-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major=</span><span class="fu">element_blank</span>(), <span class="at">panel.grid.minor=</span><span class="fu">element_blank</span>(), <span class="at">panel.border=</span><span class="fu">element_blank</span>(), <span class="at">axis.line=</span><span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.2</span>)) <span class="sc">+</span> </span>
<span id="cb53-24"><a href="#cb53-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">''</span>, <span class="at">y=</span><span class="st">''</span>) <span class="sc">+</span> paletteer<span class="sc">::</span><span class="fu">scale_fill_paletteer_d</span>(<span class="st">"rcartocolor::Vivid"</span>) <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">order=</span><span class="dv">1</span>, <span class="at">nrow=</span><span class="dv">1</span>))</span>
<span id="cb53-25"><a href="#cb53-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-26"><a href="#cb53-26" aria-hidden="true" tabindex="-1"></a>plot_geneExpr <span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher_rbp.geneExpr,<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(mod.num, <span class="at">levels=</span><span class="fu">rev</span>(<span class="fu">levels</span>(mod.num))), <span class="at">x=</span>target, <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p))) <span class="sc">+</span> </span>
<span id="cb53-27"><a href="#cb53-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">levels</span>(df_fisher_rbp<span class="sc">$</span>target), <span class="st">"_"</span>), <span class="st">"[["</span>,<span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb53-28"><a href="#cb53-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> </span>
<span id="cb53-29"><a href="#cb53-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># facet_wrap(~net, scales='free') + </span></span>
<span id="cb53-30"><a href="#cb53-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>,<span class="at">high=</span><span class="st">'red'</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">200</span>)) <span class="sc">+</span> </span>
<span id="cb53-31"><a href="#cb53-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x=</span><span class="fu">element_blank</span>(), <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(), <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(), <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>()) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">''</span>, <span class="at">y=</span><span class="st">''</span>)</span>
<span id="cb53-32"><a href="#cb53-32" aria-hidden="true" tabindex="-1"></a>col1_gE <span class="ot">=</span> <span class="fu">plot_grid</span>(plot_geneExpr,RBP_ColorBar, <span class="at">align=</span><span class="st">"v"</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">axis=</span><span class="st">"lr"</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">9</span>,<span class="dv">1</span>))</span>
<span id="cb53-33"><a href="#cb53-33" aria-hidden="true" tabindex="-1"></a>col2_gE <span class="ot">=</span> <span class="fu">plot_grid</span>(ModGroup_ColorBar.gE <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="sc">-</span><span class="dv">20</span>,<span class="dv">5</span>,<span class="dv">5</span>),<span class="st">"pt"</span>)),<span class="cn">NULL</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">9</span>,<span class="dv">1</span>))</span>
<span id="cb53-34"><a href="#cb53-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-35"><a href="#cb53-35" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(col2_gE, col1_gE, <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">rel_widths=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">20</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-31_4bdad13d043b2dcec7350db69159c7a5">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/supplement/FigS6B_geneExpr_ModOrdered.pdf"</span>, <span class="at">height=</span><span class="dv">6</span>, <span class="at">width=</span><span class="dv">20</span>)</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(col2_gE, col1_gE, <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">rel_widths=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">20</span>))</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-32_174b276ed23a23a45cc9daaa2cdfbb4b">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>plot_isoExpr <span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher_rbp.isoExpr,<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(mod.num, <span class="at">levels=</span><span class="fu">rev</span>(<span class="fu">levels</span>(mod.num))), <span class="at">x=</span>target, <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p))) <span class="sc">+</span> </span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">levels</span>(df_fisher_rbp<span class="sc">$</span>target), <span class="st">"_"</span>), <span class="st">"[["</span>,<span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> </span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># facet_wrap(~net, scales='free') + </span></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>,<span class="at">high=</span><span class="st">'red'</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">200</span>)) <span class="sc">+</span> </span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x=</span><span class="fu">element_blank</span>(), <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(), <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(), <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>()) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">''</span>, <span class="at">y=</span><span class="st">''</span>)</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>col1_iE <span class="ot">=</span> <span class="fu">plot_grid</span>(plot_isoExpr,RBP_ColorBar, <span class="at">align=</span><span class="st">"v"</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">axis=</span><span class="st">"lr"</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">9</span>,<span class="dv">1</span>))</span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>col2_iE <span class="ot">=</span> <span class="fu">plot_grid</span>(ModGroup_ColorBar.iE <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="sc">-</span><span class="dv">20</span>,<span class="dv">5</span>,<span class="dv">5</span>),<span class="st">"pt"</span>)),<span class="cn">NULL</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">9</span>,<span class="dv">1</span>))</span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(col2_iE, col1_iE, <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">rel_widths=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">20</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-33_1579aa23310ade2298bce10e725ddae4">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/supplement/FigS6B_isoExpr_ModOrdered.pdf"</span>, <span class="at">height=</span><span class="dv">6</span>, <span class="at">width=</span><span class="dv">20</span>)</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(col2_iE, col1_iE, <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">rel_widths=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">20</span>))</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-34_72e7f0ef71fc3f3ba9b206647495602c">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>plot_isoUsage <span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher_rbp.isoUsage,<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(mod.num, <span class="at">levels=</span><span class="fu">rev</span>(<span class="fu">levels</span>(mod.num))), <span class="at">x=</span>target, <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p))) <span class="sc">+</span> </span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">levels</span>(df_fisher_rbp<span class="sc">$</span>target), <span class="st">"_"</span>), <span class="st">"[["</span>,<span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> </span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># facet_wrap(~net, scales='free') + </span></span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>,<span class="at">high=</span><span class="st">'red'</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">200</span>)) <span class="sc">+</span> </span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x=</span><span class="fu">element_blank</span>(), <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(), <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(), <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>()) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">''</span>, <span class="at">y=</span><span class="st">''</span>)</span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>col1_iU <span class="ot">=</span> <span class="fu">plot_grid</span>(plot_isoUsage,RBP_ColorBar, <span class="at">align=</span><span class="st">"v"</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">axis=</span><span class="st">"lr"</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">9</span>,<span class="dv">1</span>))</span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>col2_iU <span class="ot">=</span> <span class="fu">plot_grid</span>(ModGroup_ColorBar.iU <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="sc">-</span><span class="dv">20</span>,<span class="dv">5</span>,<span class="dv">5</span>),<span class="st">"pt"</span>)),<span class="cn">NULL</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">9</span>,<span class="dv">1</span>))</span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(col2_iU, col1_iU, <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">rel_widths=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">20</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-35_8246aae19da18e288cfd601d07cd3b16">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/supplement/FigS6B_isoUsage_ModOrdered.pdf"</span>, <span class="at">height=</span><span class="dv">6</span>, <span class="at">width=</span><span class="dv">20</span>)</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(col2_iU, col1_iU, <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">rel_widths=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">20</span>))</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
<section id="fig4b" class="level3">
<h3 class="anchored" data-anchor-id="fig4b">Fig4B</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-36_730d6cb17c9cab1eab05c63ee1666024">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>Fig4B_density.rbp <span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher_rbp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Fisher.p <span class="sc">&lt;=</span> .<span class="dv">001</span>), <span class="fu">aes</span>(<span class="at">x=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p), <span class="at">fill=</span>net)) <span class="sc">+</span> </span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha=</span><span class="fl">0.25</span>, <span class="at">size=</span><span class="fl">0.2</span>, <span class="at">position=</span><span class="st">"identity"</span>, <span class="at">stat=</span><span class="st">'bin'</span>, <span class="at">binwidth=</span><span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.grid.major=</span><span class="fu">element_blank</span>(), <span class="at">panel.grid.minor=</span><span class="fu">element_blank</span>(), <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>), <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>),</span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">legend.position=</span><span class="st">"top"</span>, <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.25</span>,<span class="st">'cm'</span>), <span class="at">legend.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y=</span><span class="st">'RBP enrichments'</span>, <span class="at">x=</span><span class="st">'-log10(pValue)'</span>)</span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>Fig4B_density.rbp</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="288"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-37_36e79e67b38f29003e5e0ae34105b5e5">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/Fig4/Fig4B_density_RBPEnrich.pdf"</span>, <span class="at">height=</span><span class="dv">2</span>, <span class="at">width=</span><span class="dv">3</span>)</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>Fig4B_density.rbp</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-38_501bf6538a3ba41b48cc33252319d65f">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>Fig4B_density.celltype <span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher.celltype <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Fisher.q <span class="sc">&lt;=</span> .<span class="dv">001</span>), <span class="fu">aes</span>(<span class="at">x=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.q), <span class="at">fill=</span>net)) <span class="sc">+</span> </span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha=</span><span class="fl">0.25</span>, <span class="at">size=</span><span class="fl">0.2</span>, <span class="at">position=</span><span class="st">"identity"</span>, <span class="at">stat=</span><span class="st">'bin'</span>, <span class="at">binwidth=</span><span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.grid.major=</span><span class="fu">element_blank</span>(), <span class="at">panel.grid.minor=</span><span class="fu">element_blank</span>(), <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>), <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>),</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">legend.position=</span><span class="st">"top"</span>, <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.25</span>,<span class="st">'cm'</span>), <span class="at">legend.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y=</span><span class="st">'Cell Type enrichments'</span>, <span class="at">x=</span><span class="st">'-log10(pValue)'</span>)  </span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>Fig4B_density.celltype</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-38-1.png" class="img-fluid" width="288"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-39_a1869e23a193cd54440d164d3d4531ff">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/Fig4/Fig4B_density_CellTypeEnrich.pdf"</span>, <span class="at">height=</span><span class="dv">2</span>, <span class="at">width=</span><span class="dv">3</span>)</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>Fig4B_density.celltype</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
</section>
<section id="annotate-individual-modules" class="level2">
<h2 class="anchored" data-anchor-id="annotate-individual-modules">Annotate Individual Modules</h2>
<section id="fig4cd-and-figs7" class="level3">
<h3 class="anchored" data-anchor-id="fig4cd-and-figs7">Fig4C,D and FigS7</h3>
</section>
<section id="isousage-network" class="level3">
<h3 class="anchored" data-anchor-id="isousage-network">IsoUsage Network</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-40_411ec1eebeaafb516e2973300f46b2ca">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>this_datExpr <span class="ot">=</span> datExpr.isoFr</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>this_net <span class="ot">=</span> net.isoFr</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>this_net<span class="sc">$</span>module <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"M"</span>, this_net<span class="sc">$</span>module.number, <span class="st">"."</span>, <span class="fu">labels2colors</span>(this_net<span class="sc">$</span>module.number))</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>fileBaseNet<span class="ot">=</span><span class="st">"datExpr.localIF_batchCorrected_103k_byTargetRegion_Reordered_RBPModColor"</span></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Generate MEs and kME table for Network</span></span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a>MEs <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(<span class="fu">t</span>(this_datExpr), <span class="at">colors=</span>this_net<span class="sc">$</span>module.number)</span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a>kMEtable <span class="ot">=</span> <span class="fu">signedKME</span>(<span class="fu">t</span>(this_datExpr), <span class="at">datME =</span>MEs<span class="sc">$</span>eigengenes,<span class="at">corFnc =</span> <span class="st">"bicor"</span>)</span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a>this_anno <span class="ot">=</span> datAnno[<span class="fu">match</span>(<span class="fu">rownames</span>(this_datExpr), datAnno<span class="sc">$</span>transcript_id),]</span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">=</span> <span class="fu">grep</span>(<span class="st">"^TALON"</span>, this_anno<span class="sc">$</span>transcript_name)</span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>transcript_name[idx] <span class="ot">=</span> <span class="fu">paste0</span>(this_anno<span class="sc">$</span>gene_name[idx], <span class="st">'_'</span>, this_anno<span class="sc">$</span>transcript_name[idx])</span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-14"><a href="#cb74-14" aria-hidden="true" tabindex="-1"></a>cell_anno <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="st">'grey'</span>, <span class="at">nrow=</span><span class="fu">nrow</span>(this_anno), <span class="at">ncol =</span> <span class="fu">length</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))))</span>
<span id="cb74-15"><a href="#cb74-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cell_anno) <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))</span>
<span id="cb74-16"><a href="#cb74-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster)) {</span>
<span id="cb74-17"><a href="#cb74-17" aria-hidden="true" tabindex="-1"></a>  marker_genes <span class="ot">=</span> celltypemarkers <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Ensembl) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb74-18"><a href="#cb74-18" aria-hidden="true" tabindex="-1"></a>  cell_anno[this_anno<span class="sc">$</span>ensg <span class="sc">%in%</span> marker_genes, this_cell] <span class="ot">=</span> <span class="st">"red"</span></span>
<span id="cb74-19"><a href="#cb74-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb74-20"><a href="#cb74-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-21"><a href="#cb74-21" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">=</span> <span class="fu">tibble</span>(<span class="at">transcript_id =</span> <span class="fu">rownames</span>(kMEtable), <span class="at">module=</span>this_net<span class="sc">$</span>module, <span class="at">module.color =</span> this_net<span class="sc">$</span>colors, <span class="at">module.number=</span>this_net<span class="sc">$</span>module.number, kMEtable)</span>
<span id="cb74-22"><a href="#cb74-22" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(transcript_id,module,module.color,module.number), <span class="at">names_to =</span> <span class="st">"kME_to_module"</span>, <span class="at">values_to =</span> <span class="st">'kME'</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">kMEtoMod =</span> <span class="fu">gsub</span>(<span class="st">"kME"</span>, <span class="st">""</span>, kME_to_module)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module<span class="sc">!=</span><span class="dv">0</span>, module <span class="sc">!=</span> <span class="st">"grey"</span>, kMEtoMod <span class="sc">!=</span><span class="st">'grey'</span>, kMEtoMod <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>kME_to_module)</span>
<span id="cb74-23"><a href="#cb74-23" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(this_anno)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "transcript_id"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(kMEtoMod <span class="sc">==</span> module.number) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">10</span>)</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="fu">paste0</span>(<span class="st">"output/figures/Fig4/"</span>, fileBaseNet, <span class="st">".pdf"</span>),<span class="at">width=</span><span class="dv">15</span>,<span class="at">height=</span><span class="dv">8</span>)</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_net<span class="sc">$</span>tree, <span class="at">colors=</span><span class="fu">cbind</span>((this_net<span class="sc">$</span>colors),cell_anno), <span class="at">dendroLabels =</span> F)</span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df_fisher.celltype <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoUsage"</span>), <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(cell, <span class="at">levels=</span>order.celltype),<span class="at">y=</span><span class="fu">factor</span>(mod,<span class="at">levels=</span><span class="fu">c</span>(<span class="dv">100</span><span class="sc">:</span><span class="dv">1</span>)), <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p))) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>, <span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="sc">-</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">0</span>)) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>,<span class="at">y=</span><span class="st">""</span>)</span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">ncol</span>(MEs<span class="sc">$</span>eigengenes))) {</span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a><span class="co"># for(i in 2:2) {</span></span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(i)</span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a>  this_mod <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"ME"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(MEs<span class="sc">$</span>eigengenes)[i])</span>
<span id="cb76-13"><a href="#cb76-13" aria-hidden="true" tabindex="-1"></a>  mod_color <span class="ot">=</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module.number<span class="sc">==</span>this_mod,kMEtoMod<span class="sc">==</span>this_mod) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(module.color) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">as.character</span>()</span>
<span id="cb76-14"><a href="#cb76-14" aria-hidden="true" tabindex="-1"></a>  mod_genes <span class="ot">=</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module.number<span class="sc">==</span>this_mod,kMEtoMod<span class="sc">==</span>this_mod) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="sc">-</span>kME) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensg) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb76-15"><a href="#cb76-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span>this_mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"grey"</span>, <span class="dv">0</span>, <span class="st">"M0.grey"</span>)) {</span>
<span id="cb76-16"><a href="#cb76-16" aria-hidden="true" tabindex="-1"></a>    this_plot <span class="ot">=</span> <span class="fu">data.frame</span>(datMeta, <span class="at">eig=</span>MEs<span class="sc">$</span>eigengenes[,i])</span>
<span id="cb76-17"><a href="#cb76-17" aria-hidden="true" tabindex="-1"></a>    g1 <span class="ot">=</span> <span class="fu">ggplot</span>(this_plot, <span class="fu">aes</span>(<span class="at">x=</span>Region, <span class="at">y=</span>eig, <span class="at">color=</span>Subject)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">"Module "</span>,this_mod, <span class="st">": "</span>, <span class="fu">labels2colors</span>(<span class="fu">as.numeric</span>(this_mod)), <span class="st">" n="</span>, <span class="fu">length</span>(mod_genes)))</span>
<span id="cb76-18"><a href="#cb76-18" aria-hidden="true" tabindex="-1"></a>    g2 <span class="ot">=</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module.number<span class="sc">==</span>this_mod, kMEtoMod<span class="sc">==</span>this_mod) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">25</span>) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(transcript_name, kME), <span class="at">y=</span>kME, <span class="at">fill=</span>novelty)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> colorVector)</span>
<span id="cb76-19"><a href="#cb76-19" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb76-20"><a href="#cb76-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-21"><a href="#cb76-21" aria-hidden="true" tabindex="-1"></a>    g3<span class="ot">=</span><span class="fu">ggplot</span>(df_fisher.celltype <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoUsage"</span>,mod<span class="sc">==</span>this_mod),</span>
<span id="cb76-22"><a href="#cb76-22" aria-hidden="true" tabindex="-1"></a>              <span class="co"># aes(x=reorder(cell, -Fisher.p), </span></span>
<span id="cb76-23"><a href="#cb76-23" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(cell, <span class="at">levels=</span>order.celltype), </span>
<span id="cb76-24"><a href="#cb76-24" aria-hidden="true" tabindex="-1"></a>                  <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p), <span class="at">fill=</span>OR)) <span class="sc">+</span> </span>
<span id="cb76-25"><a href="#cb76-25" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> </span>
<span id="cb76-26"><a href="#cb76-26" aria-hidden="true" tabindex="-1"></a>      <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>)</span>
<span id="cb76-27"><a href="#cb76-27" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb76-28"><a href="#cb76-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add pathway enrichments</span></span>
<span id="cb76-29"><a href="#cb76-29" aria-hidden="true" tabindex="-1"></a>    path <span class="ot">=</span> gprofiler2<span class="sc">::</span><span class="fu">gost</span>(<span class="at">query=</span>mod_genes,<span class="at">ordered_query =</span> T,<span class="at">correction_method =</span> <span class="st">'fdr'</span>,</span>
<span id="cb76-30"><a href="#cb76-30" aria-hidden="true" tabindex="-1"></a>                            <span class="at">custom_bg =</span> <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg), <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">"GO"</span>,<span class="st">"KEGG"</span>, <span class="st">"REACTOME"</span>))</span>
<span id="cb76-31"><a href="#cb76-31" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(path)) {</span>
<span id="cb76-32"><a href="#cb76-32" aria-hidden="true" tabindex="-1"></a>      df_path <span class="ot">=</span> <span class="fu">as_tibble</span>(path<span class="sc">$</span>result)</span>
<span id="cb76-33"><a href="#cb76-33" aria-hidden="true" tabindex="-1"></a>      df_path <span class="ot">&lt;-</span> df_path <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term_size <span class="sc">&gt;</span> <span class="dv">5</span>, term_size <span class="sc">&lt;</span> <span class="dv">1000</span>)</span>
<span id="cb76-34"><a href="#cb76-34" aria-hidden="true" tabindex="-1"></a>      g4 <span class="ot">&lt;-</span> df_path <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(source) <span class="sc">%&gt;%</span> <span class="fu">slice_min</span>(p_value,<span class="at">n=</span><span class="dv">5</span>,<span class="at">with_ties =</span> F) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb76-35"><a href="#cb76-35" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(term_name, <span class="sc">-</span>p_value), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p_value), <span class="at">fill=</span>source)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb76-36"><a href="#cb76-36" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>)  <span class="sc">+</span> <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> <span class="cf">function</span>(x) <span class="fu">str_wrap</span>(x, <span class="at">width=</span><span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb76-37"><a href="#cb76-37" aria-hidden="true" tabindex="-1"></a>        <span class="fu">facet_grid</span>(source<span class="sc">~</span>., <span class="at">space =</span> <span class="st">'free'</span>, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>)) <span class="sc">+</span></span>
<span id="cb76-38"><a href="#cb76-38" aria-hidden="true" tabindex="-1"></a>        <span class="fu">coord_flip</span>()</span>
<span id="cb76-39"><a href="#cb76-39" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb76-40"><a href="#cb76-40" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(<span class="st">"No pathway enrichment for this module"</span>)</span>
<span id="cb76-41"><a href="#cb76-41" aria-hidden="true" tabindex="-1"></a>      g4<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb76-42"><a href="#cb76-42" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb76-43"><a href="#cb76-43" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb76-44"><a href="#cb76-44" aria-hidden="true" tabindex="-1"></a>    g5<span class="ot">=</span><span class="fu">ggplot</span>(df_fisher_rbp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"net.isoFr"</span>,mod.num<span class="sc">==</span>this_mod),</span>
<span id="cb76-45"><a href="#cb76-45" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x=</span>target, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p), <span class="at">fill=</span>OR)) <span class="sc">+</span> </span>
<span id="cb76-46"><a href="#cb76-46" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> </span>
<span id="cb76-47"><a href="#cb76-47" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">"Enrichment</span><span class="sc">\n</span><span class="st">(-log10 q-value)"</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>)<span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"RBP Enrichment"</span>) <span class="sc">+</span> </span>
<span id="cb76-48"><a href="#cb76-48" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">levels</span>(df_fisher_rbp<span class="sc">$</span>target), <span class="st">"_"</span>), <span class="st">"[["</span>,<span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb76-49"><a href="#cb76-49" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">"grey60"</span>, <span class="at">high=</span>mod_color) <span class="sc">+</span> </span>
<span id="cb76-50"><a href="#cb76-50" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>), <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(), <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">0</span>,<span class="dv">5</span>),<span class="st">"pt"</span>))</span>
<span id="cb76-51"><a href="#cb76-51" aria-hidden="true" tabindex="-1"></a>    <span class="co"># + coord_flip()</span></span>
<span id="cb76-52"><a href="#cb76-52" aria-hidden="true" tabindex="-1"></a>    Bar <span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher_rbp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"net.isoFr"</span>,mod.num<span class="sc">==</span>this_mod),<span class="fu">aes</span>(<span class="at">x=</span>target, <span class="at">label=</span>target.region)) <span class="sc">+</span> </span>
<span id="cb76-53"><a href="#cb76-53" aria-hidden="true" tabindex="-1"></a>      <span class="co"># geom_tile(aes(y=factor(1),fill=data.type)) + </span></span>
<span id="cb76-54"><a href="#cb76-54" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(<span class="dv">1</span>),<span class="at">fill=</span>target.region)) <span class="sc">+</span> </span>
<span id="cb76-55"><a href="#cb76-55" aria-hidden="true" tabindex="-1"></a>      <span class="co"># geom_point(aes(y=factor(1), shape=target.region),position=position_dodge2(width=1),size=0.5) + scale_shape_manual(values = c(1:9)) + </span></span>
<span id="cb76-56"><a href="#cb76-56" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(<span class="dv">1</span>), <span class="at">shape=</span>data.type),<span class="at">position=</span><span class="fu">position_dodge2</span>(<span class="at">width=</span><span class="dv">1</span>),<span class="at">size=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>)) <span class="sc">+</span> </span>
<span id="cb76-57"><a href="#cb76-57" aria-hidden="true" tabindex="-1"></a>      <span class="co"># scale_x_discrete(labels=df_fisher$RBP) +</span></span>
<span id="cb76-58"><a href="#cb76-58" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">levels</span>(df_fisher_rbp<span class="sc">$</span>target), <span class="st">"_"</span>), <span class="st">"[["</span>,<span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb76-59"><a href="#cb76-59" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb76-60"><a href="#cb76-60" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb76-61"><a href="#cb76-61" aria-hidden="true" tabindex="-1"></a>            <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb76-62"><a href="#cb76-62" aria-hidden="true" tabindex="-1"></a>            <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>,<span class="at">vjust=</span><span class="dv">1</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">size=</span><span class="dv">5</span>), </span>
<span id="cb76-63"><a href="#cb76-63" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.key.size=</span><span class="fu">unit</span>(<span class="fl">0.25</span>,<span class="st">'cm'</span>), <span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>), <span class="at">legend.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>), </span>
<span id="cb76-64"><a href="#cb76-64" aria-hidden="true" tabindex="-1"></a>            <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">20</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>),<span class="st">"pt"</span>), <span class="at">legend.position=</span><span class="fu">c</span>(<span class="fl">0.5</span>,<span class="sc">-</span><span class="dv">4</span>), <span class="at">legend.box=</span><span class="st">"horizontal"</span>, <span class="at">legend.direction=</span><span class="st">"horizontal"</span>, </span>
<span id="cb76-65"><a href="#cb76-65" aria-hidden="true" tabindex="-1"></a>            <span class="at">panel.grid.major=</span><span class="fu">element_blank</span>(), <span class="at">panel.grid.minor=</span><span class="fu">element_blank</span>(), <span class="at">panel.border=</span><span class="fu">element_blank</span>(), <span class="at">axis.line=</span><span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.2</span>)) <span class="sc">+</span> </span>
<span id="cb76-66"><a href="#cb76-66" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">x=</span><span class="st">''</span>, <span class="at">y=</span><span class="st">''</span>) <span class="sc">+</span> paletteer<span class="sc">::</span><span class="fu">scale_fill_paletteer_d</span>(<span class="st">"rcartocolor::Vivid"</span>) <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">order=</span><span class="dv">1</span>, <span class="at">nrow=</span><span class="dv">1</span>))</span>
<span id="cb76-67"><a href="#cb76-67" aria-hidden="true" tabindex="-1"></a>    g5.Bar <span class="ot">=</span> <span class="fu">plot_grid</span>(g5,Bar, <span class="at">align=</span><span class="st">"v"</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">axis=</span><span class="st">"lr"</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">1</span>))</span>
<span id="cb76-68"><a href="#cb76-68" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb76-69"><a href="#cb76-69" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Rare Var</span></span>
<span id="cb76-70"><a href="#cb76-70" aria-hidden="true" tabindex="-1"></a>    g6<span class="ot">=</span><span class="fu">ggplot</span>(df_logit.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoUsage"</span>,mod<span class="sc">==</span>this_mod),<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(set, <span class="sc">-</span>Q), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Q), <span class="at">fill=</span>OR)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> </span>
<span id="cb76-71"><a href="#cb76-71" aria-hidden="true" tabindex="-1"></a>      <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>)<span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Rare Var Enrichment"</span>)</span>
<span id="cb76-72"><a href="#cb76-72" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb76-73"><a href="#cb76-73" aria-hidden="true" tabindex="-1"></a>    <span class="co"># GWAS</span></span>
<span id="cb76-74"><a href="#cb76-74" aria-hidden="true" tabindex="-1"></a>    g7<span class="ot">=</span><span class="fu">ggplot</span>(df_gwas <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoUsage"</span>,mod<span class="sc">==</span>this_mod),<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(gwas, <span class="sc">-</span>fdr), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(fdr), <span class="at">fill=</span>Zscore)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> <span class="fu">scale_fill_gradient2</span>()<span class="sc">+</span></span>
<span id="cb76-75"><a href="#cb76-75" aria-hidden="true" tabindex="-1"></a>      <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"GWAS Enrichment"</span>)</span>
<span id="cb76-76"><a href="#cb76-76" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb76-77"><a href="#cb76-77" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb76-78"><a href="#cb76-78" aria-hidden="true" tabindex="-1"></a>    gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(<span class="at">grobs=</span><span class="fu">list</span>(g1,g2,g3,g4,g5.Bar,g6,g7),<span class="at">layout_matrix=</span><span class="fu">rbind</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">7</span>,<span class="dv">7</span>,<span class="dv">7</span>),</span>
<span id="cb76-79"><a href="#cb76-79" aria-hidden="true" tabindex="-1"></a>                                                                                 <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">7</span>,<span class="dv">7</span>,<span class="dv">7</span>), </span>
<span id="cb76-80"><a href="#cb76-80" aria-hidden="true" tabindex="-1"></a>                                                                                 <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">7</span>,<span class="dv">7</span>,<span class="dv">7</span>), </span>
<span id="cb76-81"><a href="#cb76-81" aria-hidden="true" tabindex="-1"></a>                                                                                 <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">7</span>,<span class="dv">7</span>,<span class="dv">7</span>),</span>
<span id="cb76-82"><a href="#cb76-82" aria-hidden="true" tabindex="-1"></a>                                                                                 <span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>),</span>
<span id="cb76-83"><a href="#cb76-83" aria-hidden="true" tabindex="-1"></a>                                                                                 <span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>)))</span>
<span id="cb76-84"><a href="#cb76-84" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb76-85"><a href="#cb76-85" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 290 × 12
# Groups:   module [29]
   transcr…¹ module modul…² modul…³   kME kMEto…⁴ gene_id trans…⁵ length gene_…⁶
   &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt;  
 1 ENST0000… M1.tu… turquo…       1 0.996 1       ENSG00… PKM-202   2305 PKM    
 2 ENST0000… M1.tu… turquo…       1 0.992 1       ENSG00… CTNNA1…   3754 CTNNA1 
 3 ENST0000… M1.tu… turquo…       1 0.992 1       ENSG00… SMARCE…   5150 SMARCE1
 4 ENST0000… M1.tu… turquo…       1 0.992 1       ENSG00… USP5-2…   3083 USP5   
 5 ENST0000… M1.tu… turquo…       1 0.990 1       ENSG00… TRIM9-…   5284 TRIM9  
 6 ENST0000… M1.tu… turquo…       1 0.989 1       ENSG00… TWF1-2…   3045 TWF1   
 7 ENST0000… M1.tu… turquo…       1 0.988 1       ENSG00… SSR1-2…   9661 SSR1   
 8 ENST0000… M1.tu… turquo…       1 0.987 1       ENSG00… RPN2-2…   2227 RPN2   
 9 ENST0000… M1.tu… turquo…       1 0.987 1       ENSG00… DPF2-2…   3714 DPF2   
10 ENST0000… M1.tu… turquo…       1 0.986 1       ENSG00… DYNC1I…   2572 DYNC1I2
# … with 280 more rows, 2 more variables: ensg &lt;chr&gt;, novelty &lt;chr&gt;, and
#   abbreviated variable names ¹​transcript_id, ²​module.color, ³​module.number,
#   ⁴​kMEtoMod, ⁵​transcript_name, ⁶​gene_name
[1] 1
[1] 2
[1] 3
[1] 4
[1] 5
[1] 6
[1] 7
[1] 8
[1] 9
[1] 10
[1] 11
[1] 12
[1] 13
[1] 14
[1] 15
[1] 16
[1] 17
[1] 18
[1] 19
[1] 20
[1] 21
[1] 22
[1] 23
[1] 24
[1] 25
[1] 26
[1] 27
[1] 28
[1] 29
[1] 30
quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
<section id="isoexpr-network" class="level3">
<h3 class="anchored" data-anchor-id="isoexpr-network">IsoExpr Network</h3>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-41_1735ed31449184002b9c8e1f1ec63155">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>this_datExpr <span class="ot">=</span> datExpr.isoExpr</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>this_net <span class="ot">=</span> net.isoExpr</span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>this_net<span class="sc">$</span>module <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"M"</span>, this_net<span class="sc">$</span>module.number, <span class="st">"."</span>, <span class="fu">labels2colors</span>(this_net<span class="sc">$</span>module.number))</span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>fileBaseNet<span class="ot">=</span><span class="st">"datExpr.isoExpr_batchCorrected_92k_byTargetRegion_Reordered_RBPModColor"</span></span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Generate MEs and kME table for Network</span></span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a>MEs <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(<span class="fu">t</span>(this_datExpr), <span class="at">colors=</span>this_net<span class="sc">$</span>module.number)</span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a>kMEtable <span class="ot">=</span> <span class="fu">signedKME</span>(<span class="fu">t</span>(this_datExpr), <span class="at">datME =</span>MEs<span class="sc">$</span>eigengenes,<span class="at">corFnc =</span> <span class="st">"bicor"</span>)</span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-10"><a href="#cb80-10" aria-hidden="true" tabindex="-1"></a>this_anno <span class="ot">=</span> datAnno[<span class="fu">match</span>(<span class="fu">rownames</span>(this_datExpr), datAnno<span class="sc">$</span>transcript_id),]</span>
<span id="cb80-11"><a href="#cb80-11" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">=</span> <span class="fu">grep</span>(<span class="st">"^TALON"</span>, this_anno<span class="sc">$</span>transcript_name)</span>
<span id="cb80-12"><a href="#cb80-12" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>transcript_name[idx] <span class="ot">=</span> <span class="fu">paste0</span>(this_anno<span class="sc">$</span>gene_name[idx], <span class="st">'_'</span>, this_anno<span class="sc">$</span>transcript_name[idx])</span>
<span id="cb80-13"><a href="#cb80-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-14"><a href="#cb80-14" aria-hidden="true" tabindex="-1"></a>cell_anno <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="st">'grey'</span>, <span class="at">nrow=</span><span class="fu">nrow</span>(this_anno), <span class="at">ncol =</span> <span class="fu">length</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))))</span>
<span id="cb80-15"><a href="#cb80-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cell_anno) <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))</span>
<span id="cb80-16"><a href="#cb80-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster)) {</span>
<span id="cb80-17"><a href="#cb80-17" aria-hidden="true" tabindex="-1"></a>  marker_genes <span class="ot">=</span> celltypemarkers <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Ensembl) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb80-18"><a href="#cb80-18" aria-hidden="true" tabindex="-1"></a>  cell_anno[this_anno<span class="sc">$</span>ensg <span class="sc">%in%</span> marker_genes, this_cell] <span class="ot">=</span> <span class="st">"red"</span></span>
<span id="cb80-19"><a href="#cb80-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb80-20"><a href="#cb80-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-21"><a href="#cb80-21" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">=</span> <span class="fu">tibble</span>(<span class="at">transcript_id =</span> <span class="fu">rownames</span>(kMEtable), <span class="at">module=</span>this_net<span class="sc">$</span>module, <span class="at">module.color =</span> this_net<span class="sc">$</span>colors, <span class="at">module.number=</span>this_net<span class="sc">$</span>module.number, kMEtable)</span>
<span id="cb80-22"><a href="#cb80-22" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(transcript_id,module,module.color,module.number), <span class="at">names_to =</span> <span class="st">"kME_to_module"</span>, <span class="at">values_to =</span> <span class="st">'kME'</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">kMEtoMod =</span> <span class="fu">gsub</span>(<span class="st">"kME"</span>, <span class="st">""</span>, kME_to_module)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module<span class="sc">!=</span><span class="dv">0</span>, module <span class="sc">!=</span> <span class="st">"grey"</span>, kMEtoMod <span class="sc">!=</span><span class="st">'grey'</span>, kMEtoMod <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>kME_to_module)</span>
<span id="cb80-23"><a href="#cb80-23" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(this_anno)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "transcript_id"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(kMEtoMod <span class="sc">==</span> module.number) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">10</span>)</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="fu">paste0</span>(<span class="st">"output/figures/Fig4/"</span>, fileBaseNet, <span class="st">".pdf"</span>),<span class="at">width=</span><span class="dv">15</span>,<span class="at">height=</span><span class="dv">8</span>)</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_net<span class="sc">$</span>tree, <span class="at">colors=</span><span class="fu">cbind</span>((this_net<span class="sc">$</span>colors),cell_anno), <span class="at">dendroLabels =</span> F)</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df_fisher.celltype <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoExpr"</span>), <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(cell, <span class="at">levels=</span>order.celltype),<span class="at">y=</span><span class="fu">factor</span>(mod,<span class="at">levels=</span><span class="fu">c</span>(<span class="dv">100</span><span class="sc">:</span><span class="dv">1</span>)), <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p))) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>, <span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="sc">-</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">0</span>)) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>,<span class="at">y=</span><span class="st">""</span>)</span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">ncol</span>(MEs<span class="sc">$</span>eigengenes))) {</span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># for(i in 2:2) {</span></span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(i)</span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a>  this_mod <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"ME"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(MEs<span class="sc">$</span>eigengenes)[i])</span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a>  mod_color <span class="ot">=</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module.number<span class="sc">==</span>this_mod,kMEtoMod<span class="sc">==</span>this_mod) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(module.color) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">as.character</span>()</span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a>  mod_genes <span class="ot">=</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module.number<span class="sc">==</span>this_mod,kMEtoMod<span class="sc">==</span>this_mod) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="sc">-</span>kME) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensg) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span>this_mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"grey"</span>, <span class="dv">0</span>, <span class="st">"M0.grey"</span>)) {</span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>    this_plot <span class="ot">=</span> <span class="fu">data.frame</span>(datMeta, <span class="at">eig=</span>MEs<span class="sc">$</span>eigengenes[,i])</span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a>    g1 <span class="ot">=</span> <span class="fu">ggplot</span>(this_plot, <span class="fu">aes</span>(<span class="at">x=</span>Region, <span class="at">y=</span>eig, <span class="at">color=</span>Subject)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">"Module "</span>,this_mod, <span class="st">": "</span>, <span class="fu">labels2colors</span>(<span class="fu">as.numeric</span>(this_mod)), <span class="st">" n="</span>, <span class="fu">length</span>(mod_genes)))</span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a>    g2 <span class="ot">=</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module.number<span class="sc">==</span>this_mod, kMEtoMod<span class="sc">==</span>this_mod) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">25</span>) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(transcript_name, kME), <span class="at">y=</span>kME, <span class="at">fill=</span>novelty)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> colorVector)</span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb82-20"><a href="#cb82-20" aria-hidden="true" tabindex="-1"></a>    g3<span class="ot">=</span><span class="fu">ggplot</span>(df_fisher.celltype <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoExpr"</span>,mod<span class="sc">==</span>this_mod),</span>
<span id="cb82-21"><a href="#cb82-21" aria-hidden="true" tabindex="-1"></a>              <span class="co"># aes(x=reorder(cell, -Fisher.p), </span></span>
<span id="cb82-22"><a href="#cb82-22" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(cell, <span class="at">levels=</span>order.celltype), </span>
<span id="cb82-23"><a href="#cb82-23" aria-hidden="true" tabindex="-1"></a>                  <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p), <span class="at">fill=</span>OR)) <span class="sc">+</span> </span>
<span id="cb82-24"><a href="#cb82-24" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> </span>
<span id="cb82-25"><a href="#cb82-25" aria-hidden="true" tabindex="-1"></a>      <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>)</span>
<span id="cb82-26"><a href="#cb82-26" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb82-27"><a href="#cb82-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add pathway enrichments</span></span>
<span id="cb82-28"><a href="#cb82-28" aria-hidden="true" tabindex="-1"></a>    path <span class="ot">=</span> gprofiler2<span class="sc">::</span><span class="fu">gost</span>(<span class="at">query=</span>mod_genes,<span class="at">ordered_query =</span> T,<span class="at">correction_method =</span> <span class="st">'fdr'</span>,</span>
<span id="cb82-29"><a href="#cb82-29" aria-hidden="true" tabindex="-1"></a>                            <span class="at">custom_bg =</span> <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg), <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">"GO"</span>,<span class="st">"KEGG"</span>, <span class="st">"REACTOME"</span>))</span>
<span id="cb82-30"><a href="#cb82-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(path)) {</span>
<span id="cb82-31"><a href="#cb82-31" aria-hidden="true" tabindex="-1"></a>      df_path <span class="ot">=</span> <span class="fu">as_tibble</span>(path<span class="sc">$</span>result)</span>
<span id="cb82-32"><a href="#cb82-32" aria-hidden="true" tabindex="-1"></a>      df_path <span class="ot">&lt;-</span> df_path <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term_size <span class="sc">&gt;</span> <span class="dv">5</span>, term_size <span class="sc">&lt;</span> <span class="dv">1000</span>)</span>
<span id="cb82-33"><a href="#cb82-33" aria-hidden="true" tabindex="-1"></a>      g4 <span class="ot">&lt;-</span> df_path <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(source) <span class="sc">%&gt;%</span> <span class="fu">slice_min</span>(p_value,<span class="at">n=</span><span class="dv">5</span>,<span class="at">with_ties =</span> F) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb82-34"><a href="#cb82-34" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(term_name, <span class="sc">-</span>p_value), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p_value), <span class="at">fill=</span>source)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb82-35"><a href="#cb82-35" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>)  <span class="sc">+</span> <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> <span class="cf">function</span>(x) <span class="fu">str_wrap</span>(x, <span class="at">width=</span><span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb82-36"><a href="#cb82-36" aria-hidden="true" tabindex="-1"></a>        <span class="fu">facet_grid</span>(source<span class="sc">~</span>., <span class="at">space =</span> <span class="st">'free'</span>, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>)) <span class="sc">+</span></span>
<span id="cb82-37"><a href="#cb82-37" aria-hidden="true" tabindex="-1"></a>        <span class="fu">coord_flip</span>()</span>
<span id="cb82-38"><a href="#cb82-38" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb82-39"><a href="#cb82-39" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(<span class="st">"No pathway enrichment for this module"</span>)</span>
<span id="cb82-40"><a href="#cb82-40" aria-hidden="true" tabindex="-1"></a>      g4<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb82-41"><a href="#cb82-41" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb82-42"><a href="#cb82-42" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb82-43"><a href="#cb82-43" aria-hidden="true" tabindex="-1"></a>    <span class="co"># RBP</span></span>
<span id="cb82-44"><a href="#cb82-44" aria-hidden="true" tabindex="-1"></a>    g5<span class="ot">=</span><span class="fu">ggplot</span>(df_fisher_rbp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"net.isoExpr"</span>,mod.num<span class="sc">==</span>this_mod),</span>
<span id="cb82-45"><a href="#cb82-45" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x=</span>target, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p), <span class="at">fill=</span>OR)) <span class="sc">+</span> </span>
<span id="cb82-46"><a href="#cb82-46" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> </span>
<span id="cb82-47"><a href="#cb82-47" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">"Enrichment</span><span class="sc">\n</span><span class="st">(-log10 q-value)"</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>)<span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"RBP Enrichment"</span>) <span class="sc">+</span> </span>
<span id="cb82-48"><a href="#cb82-48" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">levels</span>(df_fisher_rbp<span class="sc">$</span>target), <span class="st">"_"</span>), <span class="st">"[["</span>,<span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb82-49"><a href="#cb82-49" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">"grey60"</span>, <span class="at">high=</span>mod_color) <span class="sc">+</span> </span>
<span id="cb82-50"><a href="#cb82-50" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>), <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(), <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">0</span>,<span class="dv">5</span>),<span class="st">"pt"</span>))</span>
<span id="cb82-51"><a href="#cb82-51" aria-hidden="true" tabindex="-1"></a>    <span class="co"># + coord_flip()</span></span>
<span id="cb82-52"><a href="#cb82-52" aria-hidden="true" tabindex="-1"></a>    Bar <span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher_rbp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"net.isoExpr"</span>,mod.num<span class="sc">==</span>this_mod),<span class="fu">aes</span>(<span class="at">x=</span>target, <span class="at">label=</span>target.region)) <span class="sc">+</span> </span>
<span id="cb82-53"><a href="#cb82-53" aria-hidden="true" tabindex="-1"></a>      <span class="co"># geom_tile(aes(y=factor(1),fill=data.type)) + </span></span>
<span id="cb82-54"><a href="#cb82-54" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(<span class="dv">1</span>),<span class="at">fill=</span>target.region)) <span class="sc">+</span> </span>
<span id="cb82-55"><a href="#cb82-55" aria-hidden="true" tabindex="-1"></a>      <span class="co"># geom_point(aes(y=factor(1), shape=target.region),position=position_dodge2(width=1),size=0.5) + scale_shape_manual(values = c(1:9)) + </span></span>
<span id="cb82-56"><a href="#cb82-56" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(<span class="dv">1</span>), <span class="at">shape=</span>data.type),<span class="at">position=</span><span class="fu">position_dodge2</span>(<span class="at">width=</span><span class="dv">1</span>),<span class="at">size=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>)) <span class="sc">+</span> </span>
<span id="cb82-57"><a href="#cb82-57" aria-hidden="true" tabindex="-1"></a>      <span class="co"># scale_x_discrete(labels=df_fisher$RBP) +</span></span>
<span id="cb82-58"><a href="#cb82-58" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">levels</span>(df_fisher_rbp<span class="sc">$</span>target), <span class="st">"_"</span>), <span class="st">"[["</span>,<span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb82-59"><a href="#cb82-59" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb82-60"><a href="#cb82-60" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb82-61"><a href="#cb82-61" aria-hidden="true" tabindex="-1"></a>            <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb82-62"><a href="#cb82-62" aria-hidden="true" tabindex="-1"></a>            <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>,<span class="at">vjust=</span><span class="dv">1</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">size=</span><span class="dv">5</span>), </span>
<span id="cb82-63"><a href="#cb82-63" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.key.size=</span><span class="fu">unit</span>(<span class="fl">0.25</span>,<span class="st">'cm'</span>), <span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>), <span class="at">legend.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>), </span>
<span id="cb82-64"><a href="#cb82-64" aria-hidden="true" tabindex="-1"></a>            <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">20</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>),<span class="st">"pt"</span>), <span class="at">legend.position=</span><span class="fu">c</span>(<span class="fl">0.5</span>,<span class="sc">-</span><span class="dv">4</span>), <span class="at">legend.box=</span><span class="st">"horizontal"</span>, <span class="at">legend.direction=</span><span class="st">"horizontal"</span>, </span>
<span id="cb82-65"><a href="#cb82-65" aria-hidden="true" tabindex="-1"></a>            <span class="at">panel.grid.major=</span><span class="fu">element_blank</span>(), <span class="at">panel.grid.minor=</span><span class="fu">element_blank</span>(), <span class="at">panel.border=</span><span class="fu">element_blank</span>(), <span class="at">axis.line=</span><span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.2</span>)) <span class="sc">+</span> </span>
<span id="cb82-66"><a href="#cb82-66" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">x=</span><span class="st">''</span>, <span class="at">y=</span><span class="st">''</span>) <span class="sc">+</span> paletteer<span class="sc">::</span><span class="fu">scale_fill_paletteer_d</span>(<span class="st">"rcartocolor::Vivid"</span>) <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">order=</span><span class="dv">1</span>, <span class="at">nrow=</span><span class="dv">1</span>))</span>
<span id="cb82-67"><a href="#cb82-67" aria-hidden="true" tabindex="-1"></a>    g5.Bar <span class="ot">=</span> <span class="fu">plot_grid</span>(g5,Bar, <span class="at">align=</span><span class="st">"v"</span>, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">axis=</span><span class="st">"lr"</span>, <span class="at">rel_heights=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">1</span>))</span>
<span id="cb82-68"><a href="#cb82-68" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb82-69"><a href="#cb82-69" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Rare Var</span></span>
<span id="cb82-70"><a href="#cb82-70" aria-hidden="true" tabindex="-1"></a>    g6<span class="ot">=</span><span class="fu">ggplot</span>(df_logit.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoExpr"</span>,mod<span class="sc">==</span>this_mod),<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(set, <span class="sc">-</span>Q), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Q), <span class="at">fill=</span>OR)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> </span>
<span id="cb82-71"><a href="#cb82-71" aria-hidden="true" tabindex="-1"></a>      <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>)<span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Rare Var Enrichment"</span>)</span>
<span id="cb82-72"><a href="#cb82-72" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb82-73"><a href="#cb82-73" aria-hidden="true" tabindex="-1"></a>    <span class="co"># GWAS</span></span>
<span id="cb82-74"><a href="#cb82-74" aria-hidden="true" tabindex="-1"></a>    g7<span class="ot">=</span><span class="fu">ggplot</span>(df_gwas <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoExpr"</span>,mod<span class="sc">==</span>this_mod),<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(gwas, <span class="sc">-</span>fdr), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(fdr), <span class="at">fill=</span>Zscore)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> <span class="fu">scale_fill_gradient2</span>()<span class="sc">+</span></span>
<span id="cb82-75"><a href="#cb82-75" aria-hidden="true" tabindex="-1"></a>      <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"GWAS Enrichment"</span>)</span>
<span id="cb82-76"><a href="#cb82-76" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb82-77"><a href="#cb82-77" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb82-78"><a href="#cb82-78" aria-hidden="true" tabindex="-1"></a>    gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(<span class="at">grobs=</span><span class="fu">list</span>(g1,g2,g3,g4,g5.Bar,g6,g7),<span class="at">layout_matrix=</span><span class="fu">rbind</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">7</span>,<span class="dv">7</span>,<span class="dv">7</span>),</span>
<span id="cb82-79"><a href="#cb82-79" aria-hidden="true" tabindex="-1"></a>                                                                                     <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">7</span>,<span class="dv">7</span>,<span class="dv">7</span>), </span>
<span id="cb82-80"><a href="#cb82-80" aria-hidden="true" tabindex="-1"></a>                                                                                     <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">7</span>,<span class="dv">7</span>,<span class="dv">7</span>), </span>
<span id="cb82-81"><a href="#cb82-81" aria-hidden="true" tabindex="-1"></a>                                                                                     <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">7</span>,<span class="dv">7</span>,<span class="dv">7</span>),</span>
<span id="cb82-82"><a href="#cb82-82" aria-hidden="true" tabindex="-1"></a>                                                                                     <span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>),</span>
<span id="cb82-83"><a href="#cb82-83" aria-hidden="true" tabindex="-1"></a>                                                                                     <span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">5</span>)))</span>
<span id="cb82-84"><a href="#cb82-84" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb82-85"><a href="#cb82-85" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'
Detected custom background input, domain scope is set to 'custom'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 530 × 12
# Groups:   module [53]
   transcr…¹ module modul…² modul…³   kME kMEto…⁴ gene_id trans…⁵ length gene_…⁶
   &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt;  
 1 TALONT00… M1.tu… turquo…       1 0.995 1       ENSG00… ABLIM3…   4338 ABLIM3 
 2 ENST0000… M1.tu… turquo…       1 0.994 1       ENSG00… DYNC1I…   2947 DYNC1I1
 3 ENST0000… M1.tu… turquo…       1 0.992 1       ENSG00… NCDN-2…   3677 NCDN   
 4 ENST0000… M1.tu… turquo…       1 0.992 1       ENSG00… HS3ST5…   3901 HS3ST5 
 5 ENST0000… M1.tu… turquo…       1 0.991 1       ENSG00… SULT4A…   2478 SULT4A1
 6 ENST0000… M1.tu… turquo…       1 0.991 1       ENSG00… JPH4-2…   4265 JPH4   
 7 ENST0000… M1.tu… turquo…       1 0.990 1       ENSG00… SRD5A1…   7035 SRD5A1 
 8 TALONT00… M1.tu… turquo…       1 0.990 1       ENSG00… CDH13_…   3659 CDH13  
 9 ENST0000… M1.tu… turquo…       1 0.990 1       ENSG00… SATB2-…   1663 SATB2-…
10 ENST0000… M1.tu… turquo…       1 0.990 1       ENSG00… SOSTDC…   1758 SOSTDC1
# … with 520 more rows, 2 more variables: ensg &lt;chr&gt;, novelty &lt;chr&gt;, and
#   abbreviated variable names ¹​transcript_id, ²​module.color, ³​module.number,
#   ⁴​kMEtoMod, ⁵​transcript_name, ⁶​gene_name
[1] 1
[1] 2
[1] 3
[1] 4
[1] 5
[1] 6
[1] 7
[1] 8
[1] 9
[1] 10
[1] 11
[1] 12
[1] 13
[1] 14
[1] 15
[1] 16
[1] 17
[1] 18
[1] 19
[1] 20
[1] 21
[1] 22
[1] 23
[1] 24
[1] 25
[1] 26
[1] 27
[1] 28
[1] 29
[1] 30
[1] 31
[1] 32
[1] 33
[1] 34
[1] 35
[1] 36
[1] 37
[1] 38
[1] 39
[1] 40
[1] 41
[1] 42
[1] 43
[1] 44
[1] 45
[1] 46
[1] 47
[1] 48
[1] 49
[1] 50
[1] 51
[1] 52
[1] 53
[1] 54
quartz_off_screen 
                2 </code></pre>
</div>
</div>
<section id="hub-transcript-models" class="level4">
<h4 class="anchored" data-anchor-id="hub-transcript-models">Hub Transcript Models</h4>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-42_7ef0563d5c0ffb22eced3663b5c6a3dc">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(GenomicFeatures)</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggtranscript)</span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Isoseq Annotations</span></span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a>isoseq<span class="ot">=</span><span class="st">"data/sqanti/cp_vz_0.75_min_7_recovery_talon_corrected.gtf.cds.gff.gz"</span></span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a>isoseq_txdb<span class="ot">=</span><span class="fu">makeTxDbFromGFF</span>(isoseq, <span class="at">format=</span><span class="st">"gtf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Import genomic features from the file as a GRanges object ... OK
Prepare the 'metadata' data frame ... OK
Make the TxDb object ... </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .get_cds_IDX(mcols0$type, mcols0$phase): some CDS phases are missing
or not between 0 and 2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>OK</code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>isoseq_transcript<span class="ot">=</span><span class="fu">exonsBy</span>(isoseq_txdb,<span class="at">by=</span><span class="st">"tx"</span>,<span class="at">use.names=</span>T)</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>gr.isoseq <span class="ot">=</span> rtracklayer<span class="sc">::</span><span class="fu">import</span>(isoseq) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">=</span> <span class="fu">read_table</span>(<span class="st">"data/cp_vz_0.75_min_7_recovery_talon_abundance_filtered.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  annot_gene_id = col_character(),
  annot_transcript_id = col_character(),
  annot_gene_name = col_character(),
  annot_transcript_name = col_character(),
  gene_novelty = col_character(),
  transcript_novelty = col_character(),
  ISM_subtype = col_character()
)
ℹ Use `spec()` for the full column specifications.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>cts<span class="sc">$</span>novelty2 <span class="ot">=</span> <span class="fu">as.character</span>(cts<span class="sc">$</span>transcript_novelty)</span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>cts<span class="sc">$</span>novelty2[<span class="fu">which</span>(cts<span class="sc">$</span>novelty2<span class="sc">==</span><span class="st">"ISM"</span> <span class="sc">&amp;</span> cts<span class="sc">$</span>ISM_subtype<span class="sc">==</span><span class="st">"Prefix"</span>)] <span class="ot">=</span> <span class="st">"ISM_Prefix"</span></span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>cts<span class="sc">$</span>novelty2[<span class="fu">which</span>(cts<span class="sc">$</span>novelty2<span class="sc">==</span><span class="st">"ISM"</span> <span class="sc">&amp;</span> cts<span class="sc">$</span>ISM_subtype<span class="sc">==</span><span class="st">"Suffix"</span>)] <span class="ot">=</span> <span class="st">"ISM_Suffix"</span></span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a>cts<span class="sc">$</span>novelty2[cts<span class="sc">$</span>novelty2 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Antisense"</span>, <span class="st">"Genomic"</span>, <span class="st">"Intergenic"</span>, <span class="st">"ISM"</span>)] <span class="ot">=</span> <span class="st">"Other"</span></span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>cts<span class="sc">$</span>novelty2 <span class="ot">=</span> <span class="fu">factor</span>(cts<span class="sc">$</span>novelty2,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"Known"</span>, <span class="st">"ISM_Prefix"</span>, <span class="st">"ISM_Suffix"</span>, <span class="st">"NIC"</span>, <span class="st">"NNC"</span>, <span class="st">"Other"</span>))</span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a>cts<span class="sc">$</span>counts <span class="ot">=</span> <span class="fu">rowSums</span>(cts[,<span class="fu">c</span>(<span class="dv">12</span><span class="sc">:</span><span class="dv">35</span>)])</span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a>cts<span class="sc">$</span>cpm <span class="ot">=</span> cts<span class="sc">$</span>counts <span class="sc">/</span> (<span class="fu">sum</span>(cts<span class="sc">$</span>counts)<span class="sc">/</span><span class="dv">1000000</span>)</span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-9"><a href="#cb92-9" aria-hidden="true" tabindex="-1"></a>gr.isoseq.old <span class="ot">=</span> gr.isoseq</span>
<span id="cb92-10"><a href="#cb92-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-11"><a href="#cb92-11" aria-hidden="true" tabindex="-1"></a>gr.isoseq <span class="ot">&lt;-</span> gr.isoseq.old <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(cts, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"transcript_id"</span> <span class="ot">=</span> <span class="st">"annot_transcript_id"</span>))</span>
<span id="cb92-12"><a href="#cb92-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-13"><a href="#cb92-13" aria-hidden="true" tabindex="-1"></a>isoseq.gene.names <span class="ot">=</span> rtracklayer<span class="sc">::</span><span class="fu">import</span>(<span class="st">"data/cp_vz_0.75_min_7_recovery_talon.gtf.gz"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb92-14"><a href="#cb92-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb92-15"><a href="#cb92-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(type <span class="sc">==</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb92-16"><a href="#cb92-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id, gene_name)</span>
<span id="cb92-17"><a href="#cb92-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-18"><a href="#cb92-18" aria-hidden="true" tabindex="-1"></a>gr.isoseq <span class="ot">=</span> gr.isoseq <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(isoseq.gene.names)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "gene_id"</code></pre>
</div>
</div>
</section>
<section id="smarce1" class="level4">
<h4 class="anchored" data-anchor-id="smarce1">SMARCE1</h4>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-43_f603dfc98c9a5961ba597ae087652620">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>this_gene<span class="ot">=</span><span class="st">"SMARCE1"</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>these_exons <span class="ot">&lt;-</span> gr.isoseq <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(annot_gene_name <span class="sc">==</span> this_gene <span class="sc">&amp;</span> type <span class="sc">==</span> <span class="st">"exon"</span> <span class="sc">&amp;</span> (counts <span class="sc">&gt;</span> <span class="dv">100</span> <span class="sc">|</span> novelty2<span class="sc">==</span><span class="st">"Known"</span>))</span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>this_cds <span class="ot">&lt;-</span> gr.isoseq <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(annot_gene_name <span class="sc">==</span> this_gene <span class="sc">&amp;</span> type <span class="sc">==</span> <span class="st">"CDS"</span> <span class="sc">&amp;</span> (counts <span class="sc">&gt;</span> <span class="dv">100</span> <span class="sc">|</span> novelty2<span class="sc">==</span><span class="st">"Known"</span>))</span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>g1<span class="ot">&lt;-</span>these_exons <span class="sc">%&gt;%</span> </span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">xstart =</span> start,</span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">xend =</span> end,</span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">reorder</span>(transcript_id, counts)</span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_range</span>(</span>
<span id="cb94-13"><a href="#cb94-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">fill =</span> <span class="fu">log2</span>(<span class="dv">1</span><span class="sc">+</span>cpm), <span class="at">group=</span>novelty2), <span class="at">height=</span>.<span class="dv">25</span>) <span class="sc">+</span> </span>
<span id="cb94-14"><a href="#cb94-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_range</span>(<span class="at">data=</span>this_cds, <span class="fu">aes</span>(<span class="at">fill =</span> <span class="fu">log2</span>(<span class="dv">1</span><span class="sc">+</span>cpm), <span class="at">group=</span>novelty2)) <span class="sc">+</span> </span>
<span id="cb94-15"><a href="#cb94-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_intron</span>(</span>
<span id="cb94-16"><a href="#cb94-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">to_intron</span>(these_exons, <span class="st">"transcript_id"</span>),</span>
<span id="cb94-17"><a href="#cb94-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">strand =</span> strand),<span class="at">arrow.min.intron.length =</span> <span class="dv">500</span>,</span>
<span id="cb94-18"><a href="#cb94-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">arrow =</span> grid<span class="sc">::</span><span class="fu">arrow</span>(<span class="at">ends =</span> <span class="st">"last"</span>, <span class="at">length =</span> grid<span class="sc">::</span><span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">"inches"</span>)),</span>
<span id="cb94-19"><a href="#cb94-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">color=</span><span class="st">'grey60'</span>,</span>
<span id="cb94-20"><a href="#cb94-20" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> <span class="fu">facet_grid</span>(novelty2<span class="sc">~</span>.,<span class="at">scale=</span><span class="st">'free'</span>,<span class="at">space=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">y=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(this_gene,<span class="at">subtitle =</span> <span class="fu">unique</span>(these_exons<span class="sc">$</span>seqnames)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span>.<span class="dv">5</span>), <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span>.<span class="dv">5</span>), <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>))</span>
<span id="cb94-21"><a href="#cb94-21" aria-hidden="true" tabindex="-1"></a>g1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-44_00433c666e6f1f308934ea7282126396">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/Fig4/Fig4C_subplot_ggtranscript_SMARCE1.pdf"</span>, <span class="at">width=</span><span class="dv">10</span>, <span class="at">height=</span><span class="dv">20</span>)</span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(g1)</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
<section id="elavl1" class="level4">
<h4 class="anchored" data-anchor-id="elavl1">ELAVL1</h4>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-45_84a88ae010310dd6e4e949658cfaed15">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>this_gene<span class="ot">=</span><span class="st">"ELAVL1"</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>these_exons <span class="ot">&lt;-</span> gr.isoseq <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(annot_gene_name <span class="sc">==</span> this_gene <span class="sc">&amp;</span> type <span class="sc">==</span> <span class="st">"exon"</span> <span class="sc">&amp;</span> (counts <span class="sc">&gt;</span> <span class="dv">100</span> <span class="sc">|</span> novelty2<span class="sc">==</span><span class="st">"Known"</span>))</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a>this_cds <span class="ot">&lt;-</span> gr.isoseq <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(annot_gene_name <span class="sc">==</span> this_gene <span class="sc">&amp;</span> type <span class="sc">==</span> <span class="st">"CDS"</span> <span class="sc">&amp;</span> (counts <span class="sc">&gt;</span> <span class="dv">100</span> <span class="sc">|</span> novelty2<span class="sc">==</span><span class="st">"Known"</span>))</span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>g1<span class="ot">&lt;-</span>these_exons <span class="sc">%&gt;%</span> </span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">xstart =</span> start,</span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">xend =</span> end,</span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">reorder</span>(transcript_id, counts)</span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_range</span>(</span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">fill =</span> <span class="fu">log2</span>(<span class="dv">1</span><span class="sc">+</span>cpm), <span class="at">group=</span>novelty2), <span class="at">height=</span>.<span class="dv">25</span>) <span class="sc">+</span> </span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_range</span>(<span class="at">data=</span>this_cds, <span class="fu">aes</span>(<span class="at">fill =</span> <span class="fu">log2</span>(<span class="dv">1</span><span class="sc">+</span>cpm), <span class="at">group=</span>novelty2)) <span class="sc">+</span> </span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_intron</span>(</span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">to_intron</span>(these_exons, <span class="st">"transcript_id"</span>),</span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">strand =</span> strand),<span class="at">arrow.min.intron.length =</span> <span class="dv">500</span>,</span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">arrow =</span> grid<span class="sc">::</span><span class="fu">arrow</span>(<span class="at">ends =</span> <span class="st">"last"</span>, <span class="at">length =</span> grid<span class="sc">::</span><span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">"inches"</span>)),</span>
<span id="cb97-19"><a href="#cb97-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">color=</span><span class="st">'grey60'</span>,</span>
<span id="cb97-20"><a href="#cb97-20" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> <span class="fu">facet_grid</span>(novelty2<span class="sc">~</span>.,<span class="at">scale=</span><span class="st">'free'</span>,<span class="at">space=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">y=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(this_gene,<span class="at">subtitle =</span> <span class="fu">unique</span>(these_exons<span class="sc">$</span>seqnames)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span>.<span class="dv">5</span>), <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span>.<span class="dv">5</span>), <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>))</span>
<span id="cb97-21"><a href="#cb97-21" aria-hidden="true" tabindex="-1"></a>g1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-46_af727c24c6ccf5f9c9061343e2cf47a0">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/Fig4/Fig4D_subplot_ggtranscript_ELAVL1.pdf"</span>, <span class="at">width=</span><span class="dv">10</span>, <span class="at">height=</span><span class="dv">4</span>)</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(g1)</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
<section id="ddx3x" class="level4">
<h4 class="anchored" data-anchor-id="ddx3x">DDX3X</h4>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-47_bd7d139a9e100dc6275b7d8a0e5010e2">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>this_gene<span class="ot">=</span><span class="st">"DDX3X"</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>these_exons <span class="ot">&lt;-</span> gr.isoseq <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(annot_gene_name <span class="sc">==</span> this_gene <span class="sc">&amp;</span> type <span class="sc">==</span> <span class="st">"exon"</span> <span class="sc">&amp;</span> (counts <span class="sc">&gt;</span> <span class="dv">100</span> <span class="sc">|</span> novelty2<span class="sc">==</span><span class="st">"Known"</span>))</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>this_cds <span class="ot">&lt;-</span> gr.isoseq <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(annot_gene_name <span class="sc">==</span> this_gene <span class="sc">&amp;</span> type <span class="sc">==</span> <span class="st">"CDS"</span> <span class="sc">&amp;</span> (counts <span class="sc">&gt;</span> <span class="dv">100</span> <span class="sc">|</span> novelty2<span class="sc">==</span><span class="st">"Known"</span>))</span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>g1<span class="ot">&lt;-</span>these_exons <span class="sc">%&gt;%</span> </span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">xstart =</span> start,</span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">xend =</span> end,</span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">reorder</span>(transcript_id, counts)</span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb100-12"><a href="#cb100-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_range</span>(</span>
<span id="cb100-13"><a href="#cb100-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">fill =</span> <span class="fu">log2</span>(<span class="dv">1</span><span class="sc">+</span>cpm), <span class="at">group=</span>novelty2), <span class="at">height=</span>.<span class="dv">25</span>) <span class="sc">+</span> </span>
<span id="cb100-14"><a href="#cb100-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_range</span>(<span class="at">data=</span>this_cds, <span class="fu">aes</span>(<span class="at">fill =</span> <span class="fu">log2</span>(<span class="dv">1</span><span class="sc">+</span>cpm), <span class="at">group=</span>novelty2)) <span class="sc">+</span> </span>
<span id="cb100-15"><a href="#cb100-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_intron</span>(</span>
<span id="cb100-16"><a href="#cb100-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">to_intron</span>(these_exons, <span class="st">"transcript_id"</span>),</span>
<span id="cb100-17"><a href="#cb100-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">strand =</span> strand),<span class="at">arrow.min.intron.length =</span> <span class="dv">500</span>,</span>
<span id="cb100-18"><a href="#cb100-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">arrow =</span> grid<span class="sc">::</span><span class="fu">arrow</span>(<span class="at">ends =</span> <span class="st">"last"</span>, <span class="at">length =</span> grid<span class="sc">::</span><span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">"inches"</span>)),</span>
<span id="cb100-19"><a href="#cb100-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">color=</span><span class="st">'grey60'</span>,</span>
<span id="cb100-20"><a href="#cb100-20" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> <span class="fu">facet_grid</span>(novelty2<span class="sc">~</span>.,<span class="at">scale=</span><span class="st">'free'</span>,<span class="at">space=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">y=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(this_gene,<span class="at">subtitle =</span> <span class="fu">unique</span>(these_exons<span class="sc">$</span>seqnames)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span>.<span class="dv">5</span>), <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span>.<span class="dv">5</span>), <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>))</span>
<span id="cb100-21"><a href="#cb100-21" aria-hidden="true" tabindex="-1"></a>g1 </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_Networks_Enrichments_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-48_a1d765212a51ced83de1d34129d37c4d">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"output/figures/supplement/FigS7A_subplot_ggtranscript_DDX3X.pdf"</span>, <span class="at">width=</span><span class="dv">10</span>, <span class="at">height=</span><span class="dv">20</span>)</span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(g1)</span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
<div class="cell" data-hash="Figure4_Networks_Enrichments_cache/html/unnamed-chunk-49_86a8949af04f39339bc29978ae7177b9">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.1.3 (2022-03-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggtranscript_0.99.9    GenomicFeatures_1.46.5 AnnotationDbi_1.56.2  
 [4] Biobase_2.54.0         GenomicRanges_1.46.1   GenomeInfoDb_1.30.1   
 [7] IRanges_2.28.0         S4Vectors_0.32.4       BiocGenerics_0.40.0   
[10] paletteer_1.5.0        cowplot_1.1.1          gridExtra_2.3         
[13] gprofiler2_0.2.1       readxl_1.4.1           openxlsx_4.2.5        
[16] biomaRt_2.50.3         WGCNA_1.72-1           fastcluster_1.2.3     
[19] dynamicTreeCut_1.63-1  edgeR_3.36.0           limma_3.50.3          
[22] forcats_0.5.2          stringr_1.5.0          dplyr_1.0.10          
[25] purrr_0.3.4            readr_2.1.2            tidyr_1.2.0           
[28] tibble_3.1.8           ggplot2_3.3.6          tidyverse_1.3.2       

loaded via a namespace (and not attached):
  [1] backports_1.4.1             Hmisc_4.7-1                
  [3] BiocFileCache_2.2.1         lazyeval_0.2.2             
  [5] splines_4.1.3               BiocParallel_1.28.3        
  [7] digest_0.6.29               foreach_1.5.2              
  [9] htmltools_0.5.3             GO.db_3.14.0               
 [11] fansi_1.0.3                 magrittr_2.0.3             
 [13] checkmate_2.1.0             memoise_2.0.1              
 [15] googlesheets4_1.0.1         cluster_2.1.4              
 [17] doParallel_1.0.17           tzdb_0.3.0                 
 [19] Biostrings_2.62.0           modelr_0.1.9               
 [21] matrixStats_0.62.0          vroom_1.5.7                
 [23] prettyunits_1.1.1           jpeg_0.1-9                 
 [25] colorspace_2.0-3            ggrepel_0.9.2              
 [27] blob_1.2.3                  rvest_1.0.3                
 [29] rappdirs_0.3.3              haven_2.5.1                
 [31] xfun_0.32                   prismatic_1.1.1            
 [33] crayon_1.5.1                RCurl_1.98-1.8             
 [35] jsonlite_1.8.0              impute_1.68.0              
 [37] survival_3.4-0              iterators_1.0.14           
 [39] glue_1.6.2                  gtable_0.3.1               
 [41] gargle_1.2.0                zlibbioc_1.40.0            
 [43] XVector_0.34.0              DelayedArray_0.20.0        
 [45] scales_1.2.1                DBI_1.1.3                  
 [47] Rcpp_1.0.9                  viridisLite_0.4.1          
 [49] progress_1.2.2              htmlTable_2.4.1            
 [51] foreign_0.8-82              bit_4.0.4                  
 [53] preprocessCore_1.56.0       Formula_1.2-4              
 [55] htmlwidgets_1.5.4           httr_1.4.4                 
 [57] RColorBrewer_1.1-3          ellipsis_0.3.2             
 [59] pkgconfig_2.0.3             XML_3.99-0.10              
 [61] farver_2.1.1                nnet_7.3-17                
 [63] dbplyr_2.2.1                deldir_1.0-6               
 [65] locfit_1.5-9.6              utf8_1.2.2                 
 [67] tidyselect_1.2.0            labeling_0.4.2             
 [69] rlang_1.0.6                 munsell_0.5.0              
 [71] cellranger_1.1.0            tools_4.1.3                
 [73] cachem_1.0.6                cli_3.4.1                  
 [75] generics_0.1.3              RSQLite_2.2.16             
 [77] broom_1.0.1                 evaluate_0.16              
 [79] fastmap_1.1.0               yaml_2.3.5                 
 [81] rematch2_2.1.2              knitr_1.40                 
 [83] bit64_4.0.5                 fs_1.5.2                   
 [85] zip_2.2.0                   KEGGREST_1.34.0            
 [87] xml2_1.3.3                  compiler_4.1.3             
 [89] rstudioapi_0.14             plotly_4.10.0              
 [91] filelock_1.0.2              curl_4.3.2                 
 [93] png_0.1-7                   reprex_2.0.2               
 [95] stringi_1.7.8               lattice_0.20-45            
 [97] Matrix_1.4-1                vctrs_0.5.1                
 [99] pillar_1.8.1                lifecycle_1.0.3            
[101] data.table_1.14.2           bitops_1.0-7               
[103] rtracklayer_1.54.0          BiocIO_1.4.0               
[105] R6_2.5.1                    latticeExtra_0.6-30        
[107] codetools_0.2-18            assertthat_0.2.1           
[109] SummarizedExperiment_1.24.0 rjson_0.2.21               
[111] withr_2.5.0                 GenomicAlignments_1.30.0   
[113] Rsamtools_2.10.0            GenomeInfoDbData_1.2.7     
[115] parallel_4.1.3              hms_1.1.2                  
[117] grid_4.1.3                  rpart_4.1.16               
[119] rmarkdown_2.16              MatrixGenerics_1.6.0       
[121] googledrive_2.0.0           lubridate_1.8.0            
[123] base64enc_0.1-3             restfulr_0.0.15            
[125] interp_1.1-3               </code></pre>
</div>
</div>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>