<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.38">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Connor Jops">
<meta name="dcterms.date" content="2022-09-19">

<title>Dev Brain scIso-Seq CP/VZ - Figure 3 - IsoformSwitchAnalyzeR</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-sidebar floating">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Figure 3 - IsoformSwitchAnalyzeR</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Dev Brain scIso-Seq CP/VZ</a> 
    </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/index.html" class="sidebar-item-text sidebar-link">Index</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure1_BulkTxomeAnalysis.html" class="sidebar-item-text sidebar-link">Figure 1 - BulkTxomeAnalysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure1_TableS1.html" class="sidebar-item-text sidebar-link">Figure 1 - generate Table S1</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_isoFraction.html" class="sidebar-item-text sidebar-link">Figure 2 - Isoform Fraction analysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_plotNovelGenes.html" class="sidebar-item-text sidebar-link">Figure 2 - Plot Novel Genes</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_PlotGene.html" class="sidebar-item-text sidebar-link">Figure 2 - Plot One Gene</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_novel_exons.html" class="sidebar-item-text sidebar-link">Figure 2 - Novel Exons</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_IsoformSwitchAnalyzeR.html" class="sidebar-item-text sidebar-link active">Figure 3 - IsoformSwitchAnalyzeR</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_DTU_enrichments.html" class="sidebar-item-text sidebar-link">Figure 3 - DTU enrichments</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_APA.html" class="sidebar-item-text sidebar-link">Figure 3 - APA</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#build-switch-list" id="toc-build-switch-list" class="nav-link active" data-scroll-target="#build-switch-list">Build switch list</a>
  <ul class="collapse">
  <li><a href="#make-design" id="toc-make-design" class="nav-link" data-scroll-target="#make-design">Make design</a></li>
  <li><a href="#extract-expresssion" id="toc-extract-expresssion" class="nav-link" data-scroll-target="#extract-expresssion">Extract Expresssion</a></li>
  <li><a href="#create-pre-filtered-switchanalyzerlist" id="toc-create-pre-filtered-switchanalyzerlist" class="nav-link" data-scroll-target="#create-pre-filtered-switchanalyzerlist">Create pre-filtered switchAnalyzeRlist</a></li>
  </ul></li>
  <li><a href="#run-deseq2" id="toc-run-deseq2" class="nav-link" data-scroll-target="#run-deseq2">Run DESeq2</a>
  <ul class="collapse">
  <li><a href="#dte" id="toc-dte" class="nav-link" data-scroll-target="#dte">DTE</a></li>
  <li><a href="#dge" id="toc-dge" class="nav-link" data-scroll-target="#dge">DGE</a></li>
  </ul></li>
  <li><a href="#switch-analysis-part-1" id="toc-switch-analysis-part-1" class="nav-link" data-scroll-target="#switch-analysis-part-1">Switch analysis part 1</a></li>
  <li><a href="#table-s3" id="toc-table-s3" class="nav-link" data-scroll-target="#table-s3">Table S3</a>
  <ul class="collapse">
  <li><a href="#what-genes-does-isoformswitchanalyzer-think-are-significant" id="toc-what-genes-does-isoformswitchanalyzer-think-are-significant" class="nav-link" data-scroll-target="#what-genes-does-isoformswitchanalyzer-think-are-significant">What genes does IsoformSwitchAnalyzeR think are significant?</a></li>
  <li><a href="#what-transcripts-does-isoformswitchanalyzer-think-are-significant" id="toc-what-transcripts-does-isoformswitchanalyzer-think-are-significant" class="nav-link" data-scroll-target="#what-transcripts-does-isoformswitchanalyzer-think-are-significant">What transcripts does IsoformSwitchAnalyzeR think are significant?</a></li>
  <li><a href="#create-our-table-s3" id="toc-create-our-table-s3" class="nav-link" data-scroll-target="#create-our-table-s3">Create our Table S3</a></li>
  </ul></li>
  <li><a href="#fig3" id="toc-fig3" class="nav-link" data-scroll-target="#fig3">Fig3</a>
  <ul class="collapse">
  <li><a href="#fig3a-volcano-plots" id="toc-fig3a-volcano-plots" class="nav-link" data-scroll-target="#fig3a-volcano-plots">Fig3a: Volcano Plots</a></li>
  <li><a href="#fig3a_2-pie-chart" id="toc-fig3a_2-pie-chart" class="nav-link" data-scroll-target="#fig3a_2-pie-chart">Fig3a_2: Pie Chart</a></li>
  <li><a href="#fig3b-dge-vs-dtu-plots" id="toc-fig3b-dge-vs-dtu-plots" class="nav-link" data-scroll-target="#fig3b-dge-vs-dtu-plots">Fig3b: DGE vs DTU plots</a></li>
  <li><a href="#fig3c-venn-diagrams" id="toc-fig3c-venn-diagrams" class="nav-link" data-scroll-target="#fig3c-venn-diagrams">Fig3c: Venn Diagrams</a></li>
  </ul></li>
  <li><a href="#switch-analysis-part-2-functional-consequences" id="toc-switch-analysis-part-2-functional-consequences" class="nav-link" data-scroll-target="#switch-analysis-part-2-functional-consequences">Switch analysis part 2 (functional consequences)</a></li>
  <li><a href="#functional-analysis-of-isoform-switches" id="toc-functional-analysis-of-isoform-switches" class="nav-link" data-scroll-target="#functional-analysis-of-isoform-switches">Functional Analysis of Isoform Switches:</a>
  <ul class="collapse">
  <li><a href="#alternative-tss-isoform-switches" id="toc-alternative-tss-isoform-switches" class="nav-link" data-scroll-target="#alternative-tss-isoform-switches">Alternative TSS isoform switches</a></li>
  </ul></li>
  <li><a href="#plot-interesting-disease-genes" id="toc-plot-interesting-disease-genes" class="nav-link" data-scroll-target="#plot-interesting-disease-genes">Plot interesting disease genes</a></li>
  <li><a href="#plot-genes-with-top-switching-novel-isoforms" id="toc-plot-genes-with-top-switching-novel-isoforms" class="nav-link" data-scroll-target="#plot-genes-with-top-switching-novel-isoforms">Plot genes with top switching novel isoforms</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Figure 3 - IsoformSwitchAnalyzeR</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Connor Jops </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">September 19, 2022</p>
    </div>
  </div>
    
  </div>
  

</header>

<p><a href="https://bioconductor.org/packages/3.14/bioc/vignettes/IsoformSwitchAnalyzeR/inst/doc/IsoformSwitchAnalyzeR.html" class="uri">https://bioconductor.org/packages/3.14/bioc/vignettes/IsoformSwitchAnalyzeR/inst/doc/IsoformSwitchAnalyzeR.html</a></p>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-1_1f4580290c845650f09ad6ca5f7a4d0d">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(IsoformSwitchAnalyzeR)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(rtracklayer)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(DESeq2)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggVennDiagram)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(tidyverse)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-2_18e36b244f390aeb3c09ec1243e68367">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">packageVersion</span>(<span class="st">'IsoformSwitchAnalyzeR'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] '1.20.0'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>colorVector <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Known"</span> <span class="ot">=</span> <span class="st">"#009E73"</span>,</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ISM"</span>   <span class="ot">=</span> <span class="st">"#0072B2"</span>,</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ISM_Prefix"</span> <span class="ot">=</span> <span class="st">"#005996"</span>,</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ISM_Suffix"</span> <span class="ot">=</span> <span class="st">"#378bcc"</span>,</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"NIC"</span>   <span class="ot">=</span> <span class="st">"#D55E00"</span>,</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"NNC"</span>   <span class="ot">=</span> <span class="st">"#E69F00"</span>,</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Other"</span> <span class="ot">=</span> <span class="st">"#000000"</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>colorVector_ismSplit <span class="ot">=</span> colorVector[<span class="sc">-</span><span class="dv">2</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="build-switch-list" class="level2">
<h2 class="anchored" data-anchor-id="build-switch-list">Build switch list</h2>
<section id="make-design" class="level3">
<h3 class="anchored" data-anchor-id="make-design">Make design</h3>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-3_5a8e23b6250a4b00a5cc1ae44385fbd1">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>myDesign <span class="ot">=</span> <span class="fu">tribble</span>(</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span>sampleID, <span class="sc">~</span>condition, <span class="sc">~</span>donor,</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"VZ_209"</span>, <span class="st">"VZ"</span>, <span class="st">"209"</span>,</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"VZ_334"</span>, <span class="st">"VZ"</span>, <span class="st">"334"</span>,</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"VZ_336"</span>, <span class="st">"VZ"</span>, <span class="st">"336"</span>,</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"CP_209"</span>, <span class="st">"CP"</span>, <span class="st">"209"</span>,</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"CP_334"</span>, <span class="st">"CP"</span>, <span class="st">"334"</span>,</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">"CP_336"</span>, <span class="st">"CP"</span>, <span class="st">"336"</span>,</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">across</span>(<span class="fu">c</span>(condition, donor), as_factor)</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="extract-expresssion" class="level3">
<h3 class="anchored" data-anchor-id="extract-expresssion">Extract Expresssion</h3>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-4_d502c84aa565d7c566ea90071d52bf9d">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">=</span> <span class="fu">read_table</span>(<span class="st">"data/cp_vz_0.75_min_7_recovery_talon_abundance_filtered.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  annot_gene_id = col_character(),
  annot_transcript_id = col_character(),
  annot_gene_name = col_character(),
  annot_transcript_name = col_character(),
  gene_novelty = col_character(),
  transcript_novelty = col_character(),
  ISM_subtype = col_character()
)
ℹ Use `spec()` for the full column specifications.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">=</span> cts <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">VZ_209 =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">matches</span>(<span class="st">"209_.*_VZ"</span>))),</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">VZ_334 =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">matches</span>(<span class="st">"334_.*_VZ"</span>))),</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">VZ_336 =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">matches</span>(<span class="st">"336_.*_VZ"</span>))),</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">CP_209 =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">matches</span>(<span class="st">"209_.*_CP"</span>))),</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">CP_334 =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">matches</span>(<span class="st">"334_.*_CP"</span>))),</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">CP_336 =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">matches</span>(<span class="st">"336_.*_CP"</span>))),</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">.keep =</span> <span class="st">"unused"</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(<span class="st">"gene_ID"</span>, <span class="st">"transcript_ID"</span>, <span class="st">"annot_transcript_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_id =</span> <span class="st">"annot_gene_id"</span>,</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">transcript_id =</span> <span class="st">"annot_transcript_id"</span>,</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_name =</span> <span class="st">"annot_gene_name"</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_novelty =</span> <span class="fu">as.factor</span>(gene_novelty),</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">transcript_novelty =</span> <span class="fu">as.factor</span>(transcript_novelty),</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">ISM_subtype =</span> ISM_subtype <span class="sc">%&gt;%</span> <span class="fu">na_if</span>(<span class="st">"None"</span>) <span class="sc">%&gt;%</span> <span class="fu">as.factor</span>()</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>talonExpression <span class="ot">=</span> cts <span class="sc">%&gt;%</span></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(transcript_id, <span class="fu">starts_with</span>(<span class="fu">c</span>(<span class="st">"VZ"</span>, <span class="st">"CP"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">isoform_id =</span> <span class="st">"transcript_id"</span>)</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>talonExpression</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 214,516 × 7
   isoform_id          VZ_209 VZ_334 VZ_336 CP_209 CP_334 CP_336
   &lt;chr&gt;                &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 ENST00000494149.2_2      0      0      2      0      0      0
 2 ENST00000424587.2        0      0      0      0      0      2
 3 ENST00000445840.1        1      1      0      0      0      0
 4 ENST00000457540.1_1      0      1      1      0      0      0
 5 ENST00000414273.1_1      0      0      1      0      0      0
 6 ENST00000655765.1_2      0      2      0      2      1      2
 7 ENST00000457084.1_2      0      2      0      0      0      0
 8 ENST00000591702.1_2      1      0      2      2      0      0
 9 ENST00000644482.1_1      3      0      0      0      0      0
10 ENST00000445118.7_3      1      0      0      8      0      0
# … with 214,506 more rows</code></pre>
</div>
</div>
</section>
<section id="create-pre-filtered-switchanalyzerlist" class="level3">
<h3 class="anchored" data-anchor-id="create-pre-filtered-switchanalyzerlist">Create pre-filtered switchAnalyzeRlist</h3>
<p>IsoformSwitchAnalyzeR will make assumptions based on whether file extension is .gtf(.gz) or .gff(.gz)… was necessary to symlink <code>_corrected.gtf.cds.gff</code>.</p>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-5_3c662992dd9cbff15ebb13d609aead82">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>rdata_path  <span class="ot">=</span> <span class="st">"data/working/talonSwitchList_preFilter.rds"</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(rdata_path)) { </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  talonSwitchList <span class="ot">&lt;-</span> <span class="fu">importRdata</span>(</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">isoformCountMatrix   =</span> talonExpression,</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">designMatrix         =</span> myDesign,</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">isoformExonAnnoation =</span> <span class="st">'data/cp_vz_0.75_min_7_recovery_talon.gtf.gz'</span>,</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">isoformNtFasta       =</span> <span class="st">'data/sqanti/cp_vz_0.75_min_7_recovery_talon_corrected.fasta.gz'</span>,</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">addAnnotatedORFs     =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">fixStringTieAnnotationProblem =</span> <span class="cn">FALSE</span> <span class="co"># otherwise will mess up gene_ids</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>  talonSwitchList <span class="ot">&lt;-</span> <span class="fu">addORFfromGTF</span>(</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">switchAnalyzeRlist     =</span> talonSwitchList,</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">pathToGTF              =</span> <span class="st">'data/sqanti/cp_vz_0.75_min_7_recovery_talon_corrected.gtf.cds.gtf.gz'</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>  talonSwitchList <span class="ot">&lt;-</span> <span class="fu">preFilter</span>(</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">switchAnalyzeRlist         =</span> talonSwitchList,</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">geneExpressionCutoff       =</span> <span class="dv">1</span>,     <span class="co"># default</span></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">isoformExpressionCutoff    =</span> <span class="dv">0</span>,     <span class="co"># default</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">IFcutoff                   =</span> <span class="fl">0.01</span>,  <span class="co"># default</span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">removeSingleIsoformGenes   =</span> <span class="cn">TRUE</span>,  <span class="co"># default</span></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduceToSwitchingGenes     =</span> <span class="cn">FALSE</span>, <span class="co"># default (we didn't run DEXSeq yet)</span></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">keepIsoformInAllConditions =</span> <span class="cn">TRUE</span>   <span class="co"># we only have 2 conditions so doesn't matter</span></span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(talonSwitchList, <span class="at">file =</span> rdata_path)</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>  talonSwitchList <span class="ot">=</span> <span class="fu">readRDS</span>(rdata_path)</span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(talonSwitchList)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>This switchAnalyzeRlist list contains:
 102319 isoforms from 10809 genes
 1 comparison from 2 conditions (in total 6 samples)

Feature analyzed:
[1] "ntSequence, ORFs"</code></pre>
</div>
</div>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-6_863e549c5a9403e883907d4800ef8bfe">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">switchPlot</span>(</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  talonSwitchList,</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene=</span><span class="st">'KMT2E'</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in switchPlot(talonSwitchList, gene = "KMT2E"): We recomend running the
isoform switching analysis before doing the transcript plot. See
?detectIsoformSwitching for more details</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="run-deseq2" class="level2">
<h2 class="anchored" data-anchor-id="run-deseq2">Run DESeq2</h2>
<section id="dte" class="level3">
<h3 class="anchored" data-anchor-id="dte">DTE</h3>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-7_1ad773d0132ccbf962a8720329040db4">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>cts_preFilter <span class="ot">=</span> talonExpression <span class="sc">%&gt;%</span> <span class="fu">filter</span>(isoform_id <span class="sc">%in%</span> talonSwitchList<span class="sc">$</span>isoformFeatures<span class="sc">$</span>isoform_id)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>cts_preFilter</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 102,319 × 7
   isoform_id           VZ_209 VZ_334 VZ_336 CP_209 CP_334 CP_336
   &lt;chr&gt;                 &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 ENST00000669922.1_2       4     10      4      6      9     10
 2 ENST00000659124.1_2       4      0      0      9      0      0
 3 ENST00000441765.6_3       1      0      0      0      1      0
 4 ENST00000668541.1_2       0      0      0      5      1      5
 5 ENST00000327044.7_3     283    672    708    804    820    665
 6 ENST00000428771.6_3       0      1      0      0      2      0
 7 ENST00000304952.10_3      0     25     41      3      5     16
 8 ENST00000481869.1_2       0      6      4      2      2      6
 9 ENST00000484667.2_2       0      4      5      0      0      9
10 ENST00000379370.7_2       4     12      7      9     34      8
# … with 102,309 more rows</code></pre>
</div>
</div>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-8_658746ead696c52d683493dab68ee788">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">=</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(cts_preFilter),</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(myDesign),</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> donor <span class="sc">+</span> condition,</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">tidy =</span> T</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>converting counts to integer mode</code></pre>
</div>
</div>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-9_7c0cd880ebb1a7802f5d0f579dca32f5">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># takes 30sec</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">system.time</span>({</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  dds  <span class="ot">=</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>   user  system elapsed 
 12.054   0.378  12.886 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>DTE_results <span class="ot">=</span> DESeq2<span class="sc">::</span><span class="fu">results</span>(dds)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>DTE_results</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): condition CP vs VZ 
Wald test p-value: condition CP vs VZ 
DataFrame with 102319 rows and 6 columns
                      baseMean log2FoldChange     lfcSE       stat    pvalue
                     &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;
ENST00000669922.1_2   6.794078      0.3497446  0.926990  0.3772904 0.7059578
ENST00000659124.1_2   2.340796      0.1825893  1.972493  0.0925678 0.9262469
ENST00000441765.6_3   0.406806     -0.0633991  4.033013 -0.0157200 0.9874577
ENST00000668541.1_2   1.695927      3.8310438  2.135802  1.7937258 0.0728569
ENST00000327044.7_3 615.853668      0.3351470  0.261315  1.2825405 0.1996531
...                        ...            ...       ...        ...       ...
TALONT003163865        1.27255        2.91192   2.53106   1.150474  0.249949
TALONT003172883        1.32847        2.46363   2.50056   0.985233  0.324510
TALONT003175107        1.61344        3.01946   2.33696   1.292045  0.196342
TALONT003196342        1.10300        2.77499   2.66225   1.042347  0.297251
TALONT003225592        1.41593        1.80728   2.62236   0.689179  0.490711
                         padj
                    &lt;numeric&gt;
ENST00000669922.1_2  0.925055
ENST00000659124.1_2        NA
ENST00000441765.6_3        NA
ENST00000668541.1_2        NA
ENST00000327044.7_3  0.570881
...                       ...
TALONT003163865            NA
TALONT003172883            NA
TALONT003175107            NA
TALONT003196342            NA
TALONT003225592            NA</code></pre>
</div>
</div>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-10_205d9a56fcee27cdb3f473459590937e">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>DTE_results <span class="ot">=</span> DTE_results <span class="sc">%&gt;%</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="st">"isoform_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">padj =</span> <span class="fu">replace_na</span>(padj, <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="dge" class="level3">
<h3 class="anchored" data-anchor-id="dge">DGE</h3>
<p>Mike note – lets calculate DGE for all genes (expressed) not just those in talonSwitchList_part1, which filters out genes with only 1 isoform, etc Should be ~25k</p>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-11_a114ae7a7c60792bcb267cf685467eae">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>cts_gene <span class="ot">=</span> <span class="fu">read_table</span>(<span class="st">"data/cp_vz_talon_abundance.tsv.gz"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">VZ_209 =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">matches</span>(<span class="st">"209_.*_VZ"</span>))),</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">VZ_334 =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">matches</span>(<span class="st">"334_.*_VZ"</span>))),</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">VZ_336 =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">matches</span>(<span class="st">"336_.*_VZ"</span>))),</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">CP_209 =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">matches</span>(<span class="st">"209_.*_CP"</span>))),</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">CP_334 =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">matches</span>(<span class="st">"334_.*_CP"</span>))),</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">CP_336 =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">matches</span>(<span class="st">"336_.*_CP"</span>))),</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">.keep =</span> <span class="st">"unused"</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(annot_gene_id, annot_gene_name, gene_novelty) <span class="sc">%&gt;%</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">starts_with</span>(<span class="fu">c</span>(<span class="st">"VZ"</span>, <span class="st">"CP"</span>)), sum), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_id =</span> <span class="st">"annot_gene_id"</span>,</span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_name =</span> <span class="st">"annot_gene_name"</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_novelty =</span> <span class="fu">as.factor</span>(gene_novelty))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  annot_gene_id = col_character(),
  annot_transcript_id = col_character(),
  annot_gene_name = col_character(),
  annot_transcript_name = col_character(),
  gene_novelty = col_character(),
  transcript_novelty = col_character(),
  ISM_subtype = col_character()
)
ℹ Use `spec()` for the full column specifications.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>cts_gene_preFilter <span class="ot">=</span> cts_gene <span class="sc">%&gt;%</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene_id <span class="sc">%in%</span> <span class="fu">unique</span>(cts<span class="sc">$</span>gene_id)) <span class="sc">%&gt;%</span> <span class="co"># Filtering for those with at least one detected isoform</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id, <span class="fu">starts_with</span>(<span class="fu">c</span>(<span class="st">"VZ"</span>, <span class="st">"CP"</span>)))</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>cts_gene_preFilter</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 24,554 × 7
   gene_id              VZ_209 VZ_334 VZ_336 CP_209 CP_334 CP_336
   &lt;chr&gt;                 &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 ENSG00000000003.15_4    443    993    812    335    483    305
 2 ENSG00000000419.12_5     30    119    108    388    123    311
 3 ENSG00000000457.14_6    149    300    222    192    149    148
 4 ENSG00000000460.17_6    103    423    275     99    132     76
 5 ENSG00000000938.13_5      0      3     10     14      4     11
 6 ENSG00000000971.16_3     13     17     23     21     26      9
 7 ENSG00000001036.14_4    206    243    415    163    105    205
 8 ENSG00000001084.13_9     49     73     51     54     65     44
 9 ENSG00000001167.14_3    184    401    249    436    525    286
10 ENSG00000001460.18_6      5      9     15     44     21     42
# … with 24,544 more rows</code></pre>
</div>
</div>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-12_68de5f3eeb60b1a10434b422c744dd36">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>dds_gene <span class="ot">=</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(cts_gene_preFilter),</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(myDesign),</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> donor <span class="sc">+</span> condition,</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">tidy =</span> T</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>converting counts to integer mode</code></pre>
</div>
</div>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-13_8565d05d26620e29b45f774abf8d31fd">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># takes 10sec</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">system.time</span>({</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  dds_gene  <span class="ot">=</span> <span class="fu">DESeq</span>(dds_gene)</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>   user  system elapsed 
  3.582   0.118   3.735 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>DGE_results <span class="ot">=</span> DESeq2<span class="sc">::</span><span class="fu">results</span>(dds_gene)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>DGE_results</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): condition CP vs VZ 
Wald test p-value: condition CP vs VZ 
DataFrame with 24554 rows and 6 columns
                      baseMean log2FoldChange     lfcSE      stat      pvalue
                     &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
ENSG00000000003.15_4 547.45338      -1.127697  0.224974  -5.01257 5.37067e-07
ENSG00000000419.12_5 168.65442       1.554264  0.394700   3.93784 8.22188e-05
ENSG00000000457.14_6 189.49516      -0.588598  0.272137  -2.16287 3.05513e-02
ENSG00000000460.17_6 172.14571      -1.381243  0.295422  -4.67548 2.93261e-06
ENSG00000000938.13_5   6.44186       1.487234  1.237223   1.20207 2.29335e-01
...                        ...            ...       ...       ...         ...
TALONG000171040        7.30690        6.13371   1.58942   3.85908 0.000113814
TALONG000171269        1.51623        3.77994   2.59371   1.45735 0.145019484
TALONG000180651        3.39053        4.85165   1.95562   2.48088 0.013106017
TALONG000182469        2.31534        4.58042   2.20513   2.07717 0.037785814
TALONG000183977        1.57984        3.87094   2.55344   1.51597 0.129526613
                            padj
                       &lt;numeric&gt;
ENSG00000000003.15_4 6.36749e-06
ENSG00000000419.12_5 6.41440e-04
ENSG00000000457.14_6 1.04347e-01
ENSG00000000460.17_6 3.06182e-05
ENSG00000000938.13_5 4.59846e-01
...                          ...
TALONG000171040      0.000856288
TALONG000171269               NA
TALONG000180651      0.053168102
TALONG000182469      0.123652258
TALONG000183977               NA</code></pre>
</div>
</div>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-14_ce753639f59e98bbbef8a74bd73e4c15">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>DGE_results <span class="ot">=</span> DGE_results <span class="sc">%&gt;%</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="st">"gene_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">padj =</span> <span class="fu">replace_na</span>(padj, <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="switch-analysis-part-1" class="level2">
<h2 class="anchored" data-anchor-id="switch-analysis-part-1">Switch analysis part 1</h2>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-15_5584731ae30add6adcf3b9c449e0b906">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>rdata_path  <span class="ot">=</span> <span class="st">"data/working/talonSwitchList_part1.rds"</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(rdata_path)) {</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  <span class="do">### DEXSeq DTU - takes 15min</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1 <span class="ot">&lt;-</span> <span class="fu">isoformSwitchTestDEXSeq</span>(</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">switchAnalyzeRlist         =</span> talonSwitchList,</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduceToSwitchingGenes     =</span> <span class="cn">FALSE</span></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>  <span class="do">### Add DTE/DGE to switchList</span></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>  idx <span class="ot">=</span> <span class="fu">match</span>(talonSwitchList_part1<span class="sc">$</span>isoformFeatures<span class="sc">$</span>isoform_id, DTE_results<span class="sc">$</span>isoform_id)</span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1<span class="sc">$</span>isoformFeatures<span class="sc">$</span>iso_q_value <span class="ot">=</span> DTE_results<span class="sc">$</span>padj[idx]</span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a>  idx <span class="ot">=</span> <span class="fu">match</span>(talonSwitchList_part1<span class="sc">$</span>isoformFeatures<span class="sc">$</span>gene_id, DGE_results<span class="sc">$</span>gene_id)</span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1<span class="sc">$</span>isoformFeatures<span class="sc">$</span>gene_q_value <span class="ot">=</span> DGE_results<span class="sc">$</span>padj[idx]</span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>  <span class="do">### Extract AA sequences</span></span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1<span class="sc">$</span>aaSequence <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a>  isoformFeatures_part1 <span class="ot">=</span> talonSwitchList_part1<span class="sc">$</span>isoformFeatures</span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="ot">=</span> isoformFeatures_part1 <span class="sc">%&gt;%</span></span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a>      <span class="at">isoform_switch_q_value =</span> <span class="fu">if_else</span>(<span class="fu">any</span>(</span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a>        <span class="co"># our actual filtering criteria - genes with DTU, DTE, or DGE</span></span>
<span id="cb46-25"><a href="#cb46-25" aria-hidden="true" tabindex="-1"></a>        (isoform_switch_q_value <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> dIF <span class="sc">&gt;</span> <span class="fl">0.1</span>) <span class="sc">|</span> iso_q_value <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">|</span> gene_q_value <span class="sc">&lt;</span> <span class="fl">0.05</span></span>
<span id="cb46-26"><a href="#cb46-26" aria-hidden="true" tabindex="-1"></a>      ), <span class="dv">0</span>, <span class="dv">1</span>),</span>
<span id="cb46-27"><a href="#cb46-27" aria-hidden="true" tabindex="-1"></a>      <span class="at">dIF =</span> <span class="dv">1</span></span>
<span id="cb46-28"><a href="#cb46-28" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb46-29"><a href="#cb46-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb46-30"><a href="#cb46-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb46-31"><a href="#cb46-31" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1 <span class="ot">&lt;-</span> <span class="fu">extractSequence</span>(</span>
<span id="cb46-32"><a href="#cb46-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">switchAnalyzeRlist =</span> talonSwitchList_part1,</span>
<span id="cb46-33"><a href="#cb46-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">pathToOutput       =</span> <span class="st">"data/working/IsoformSwitchAnalyzeR/"</span>,</span>
<span id="cb46-34"><a href="#cb46-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">extractNTseq       =</span> <span class="cn">TRUE</span>,</span>
<span id="cb46-35"><a href="#cb46-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">extractAAseq       =</span> <span class="cn">TRUE</span>,</span>
<span id="cb46-36"><a href="#cb46-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">removeShortAAseq   =</span> <span class="cn">TRUE</span>,</span>
<span id="cb46-37"><a href="#cb46-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">removeLongAAseq    =</span> <span class="cn">FALSE</span>,</span>
<span id="cb46-38"><a href="#cb46-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">onlySwitchingGenes =</span> <span class="cn">TRUE</span></span>
<span id="cb46-39"><a href="#cb46-39" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb46-40"><a href="#cb46-40" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="ot">=</span> isoformFeatures_part1</span>
<span id="cb46-41"><a href="#cb46-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-42"><a href="#cb46-42" aria-hidden="true" tabindex="-1"></a>  <span class="do">### Save to file</span></span>
<span id="cb46-43"><a href="#cb46-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(talonSwitchList_part1, <span class="at">file =</span> rdata_path)</span>
<span id="cb46-44"><a href="#cb46-44" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb46-45"><a href="#cb46-45" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1 <span class="ot">=</span> <span class="fu">readRDS</span>(rdata_path)</span>
<span id="cb46-46"><a href="#cb46-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb46-47"><a href="#cb46-47" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(talonSwitchList_part1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>This switchAnalyzeRlist list contains:
 102319 isoforms from 10809 genes
 1 comparison from 2 conditions (in total 6 samples)

Switching features:
  Comparison Isoforms Switches Genes
1   VZ vs CP     1860     1073  1237

Feature analyzed:
[1] "Isoform Switch Identification, ntSequence, ORFs, aaSequence"</code></pre>
</div>
</div>
<p>WARNING: <code>extractSequence()</code> silently caches its result in <code>switchList$aaSequence</code>. If this object exists, <code>extractSequence()</code> will return the same result, no matter if you adjust <code>alpha</code>, <code>dIFcutoff</code>, or even re-run <code>isoformSwitchTestDEXSeq()</code>.</p>
<p>What are the exact isoforms <code>extractSequence()</code> will return sequences for? (Slightly fewer if <code>removeShortAAseq = TRUE</code>)</p>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-16_859ab078b5b6e619fc372a76bceee87d">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>orf_isoforms <span class="ot">=</span> talonSwitchList_part1<span class="sc">$</span>orfAnalysis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(orfTransciptStart) <span class="sc">%&gt;%</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(isoform_id)</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">any</span>(isoform_switch_q_value <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(dIF) <span class="sc">&gt;</span> <span class="fl">0.1</span>)) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(isoform_id <span class="sc">%in%</span> orf_isoforms)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 12,026 × 30
   iso_ref       gene_…¹ isofo…² gene_id condi…³ condi…⁴ gene_…⁵ gene_…⁶ iso_b…⁷
   &lt;chr&gt;         &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  
 1 isoComp_0000… geneCo… ENST00… ENSG00… VZ      CP      TSPAN6  protei… protei…
 2 isoComp_0000… geneCo… TALONT… ENSG00… VZ      CP      TSPAN6  protei… &lt;NA&gt;   
 3 isoComp_0000… geneCo… TALONT… ENSG00… VZ      CP      TSPAN6  protei… &lt;NA&gt;   
 4 isoComp_0000… geneCo… TALONT… ENSG00… VZ      CP      TSPAN6  protei… &lt;NA&gt;   
 5 isoComp_0000… geneCo… TALONT… ENSG00… VZ      CP      TSPAN6  protei… &lt;NA&gt;   
 6 isoComp_0000… geneCo… TALONT… ENSG00… VZ      CP      TSPAN6  protei… &lt;NA&gt;   
 7 isoComp_0000… geneCo… TALONT… ENSG00… VZ      CP      TSPAN6  protei… &lt;NA&gt;   
 8 isoComp_0000… geneCo… ENST00… ENSG00… VZ      CP      C1orf1… protei… protei…
 9 isoComp_0000… geneCo… ENST00… ENSG00… VZ      CP      C1orf1… protei… protei…
10 isoComp_0000… geneCo… ENST00… ENSG00… VZ      CP      C1orf1… protei… protei…
# … with 12,016 more rows, 21 more variables: gene_overall_mean &lt;dbl&gt;,
#   gene_value_1 &lt;dbl&gt;, gene_value_2 &lt;dbl&gt;, gene_stderr_1 &lt;dbl&gt;,
#   gene_stderr_2 &lt;dbl&gt;, gene_log2_fold_change &lt;dbl&gt;, gene_q_value &lt;dbl&gt;,
#   iso_overall_mean &lt;dbl&gt;, iso_value_1 &lt;dbl&gt;, iso_value_2 &lt;dbl&gt;,
#   iso_stderr_1 &lt;dbl&gt;, iso_stderr_2 &lt;dbl&gt;, iso_log2_fold_change &lt;dbl&gt;,
#   iso_q_value &lt;dbl&gt;, IF_overall &lt;dbl&gt;, IF1 &lt;dbl&gt;, IF2 &lt;dbl&gt;, dIF &lt;dbl&gt;,
#   isoform_switch_q_value &lt;dbl&gt;, gene_switch_q_value &lt;dbl&gt;, PTC &lt;lgl&gt;, and …</code></pre>
</div>
</div>
</section>
<section id="table-s3" class="level2">
<h2 class="anchored" data-anchor-id="table-s3">Table S3</h2>
<section id="what-genes-does-isoformswitchanalyzer-think-are-significant" class="level3">
<h3 class="anchored" data-anchor-id="what-genes-does-isoformswitchanalyzer-think-are-significant">What genes does IsoformSwitchAnalyzeR think are significant?</h3>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-17_d886d5dcf61a7f4e41c819e87d4580e3">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">extractTopSwitches</span>(</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1, <span class="at">filterForConsequences =</span> <span class="cn">FALSE</span>, <span class="at">n=</span><span class="cn">Inf</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span>  <span class="fu">as_tibble</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1,237 × 7
   gene_ref          gene_id             gene_…¹ condi…² condi…³ gene_sw…⁴  Rank
   &lt;chr&gt;             &lt;chr&gt;               &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;       &lt;dbl&gt; &lt;int&gt;
 1 geneComp_00001085 ENSG00000073584.20… SMARCE1 VZ      CP      1.03e-211     1
 2 geneComp_00000969 ENSG00000070087.14… PFN2    VZ      CP      6.32e-118     2
 3 geneComp_00008384 ENSG00000155849.15… ELMO1   VZ      CP      3.74e-116     3
 4 geneComp_00000525 ENSG00000044115.21… CTNNA1  VZ      CP      3.84e-103     4
 5 geneComp_00005804 ENSG00000133884.10… DPF2    VZ      CP      4.94e- 95     5
 6 geneComp_00020536 ENSG00000263001.6_6 GTF2I   VZ      CP      9.35e- 94     6
 7 geneComp_00001542 ENSG00000087274.17… ADD1    VZ      CP      2.19e- 85     7
 8 geneComp_00003418 ENSG00000111667.13… USP5    VZ      CP      6.38e- 83     8
 9 geneComp_00004817 ENSG00000124783.14… SSR1    VZ      CP      2.45e- 80     9
10 geneComp_00000441 ENSG00000033627.16… ATP6V0… VZ      CP      4.61e- 74    10
# … with 1,227 more rows, and abbreviated variable names ¹​gene_name,
#   ²​condition_1, ³​condition_2, ⁴​gene_switch_q_value</code></pre>
</div>
</div>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-18_2025f471d36f25954b3130c9619f0f53">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_ref, gene_id, gene_name, condition_1, condition_2, gene_switch_q_value) <span class="sc">%&gt;%</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">any</span>(isoform_switch_q_value <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(dIF) <span class="sc">&gt;</span> <span class="fl">0.1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>() <span class="sc">%&gt;%</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(gene_switch_q_value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'gene_ref', 'gene_id', 'gene_name',
'condition_1', 'condition_2'. You can override using the `.groups` argument.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1,237 × 6
# Groups:   gene_ref, gene_id, gene_name, condition_1, condition_2 [1,237]
   gene_ref          gene_id              gene_name conditio…¹ condi…² gene_sw…³
   &lt;chr&gt;             &lt;chr&gt;                &lt;chr&gt;     &lt;chr&gt;      &lt;chr&gt;       &lt;dbl&gt;
 1 geneComp_00001085 ENSG00000073584.20_8 SMARCE1   VZ         CP      1.03e-211
 2 geneComp_00000969 ENSG00000070087.14_5 PFN2      VZ         CP      6.32e-118
 3 geneComp_00008384 ENSG00000155849.15_4 ELMO1     VZ         CP      3.74e-116
 4 geneComp_00000525 ENSG00000044115.21_7 CTNNA1    VZ         CP      3.84e-103
 5 geneComp_00005804 ENSG00000133884.10_4 DPF2      VZ         CP      4.94e- 95
 6 geneComp_00020536 ENSG00000263001.6_6  GTF2I     VZ         CP      9.35e- 94
 7 geneComp_00001542 ENSG00000087274.17_6 ADD1      VZ         CP      2.19e- 85
 8 geneComp_00003418 ENSG00000111667.13_4 USP5      VZ         CP      6.38e- 83
 9 geneComp_00004817 ENSG00000124783.14_6 SSR1      VZ         CP      2.45e- 80
10 geneComp_00000441 ENSG00000033627.16_4 ATP6V0A1  VZ         CP      4.61e- 74
# … with 1,227 more rows, and abbreviated variable names ¹​condition_1,
#   ²​condition_2, ³​gene_switch_q_value</code></pre>
</div>
</div>
</section>
<section id="what-transcripts-does-isoformswitchanalyzer-think-are-significant" class="level3">
<h3 class="anchored" data-anchor-id="what-transcripts-does-isoformswitchanalyzer-think-are-significant">What transcripts does IsoformSwitchAnalyzeR think are significant?</h3>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-19_b8578b0bdab6a126220d3d963f19a542">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">extractTopSwitches</span>(</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1, <span class="at">extractGenes =</span> <span class="cn">FALSE</span>, <span class="at">filterForConsequences =</span> <span class="cn">FALSE</span>, <span class="at">n=</span><span class="cn">Inf</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span>  <span class="fu">as_tibble</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .fun(piece, ...): Less than n genes genes with significant switches
were found. Returning those.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1,860 × 12
   iso_ref    gene_…¹ isofo…² gene_id gene_…³ condi…⁴ condi…⁵   IF1   IF2    dIF
   &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;
 1 isoComp_0… geneCo… ENST00… ENSG00… SMARCE1 VZ      CP      0.57  0.164 -0.405
 2 isoComp_0… geneCo… ENST00… ENSG00… PFN2    VZ      CP      0.25  0.021 -0.228
 3 isoComp_0… geneCo… ENST00… ENSG00… ELMO1   VZ      CP      0.404 0.024 -0.38 
 4 isoComp_0… geneCo… ENST00… ENSG00… CTNNA1  VZ      CP      0.541 0.129 -0.413
 5 isoComp_0… geneCo… ENST00… ENSG00… DPF2    VZ      CP      0.082 0.553  0.47 
 6 isoComp_0… geneCo… ENST00… ENSG00… GTF2I   VZ      CP      0.128 0.017 -0.111
 7 isoComp_0… geneCo… TALONT… ENSG00… ADD1    VZ      CP      0.31  0.059 -0.251
 8 isoComp_0… geneCo… ENST00… ENSG00… CTNNA1  VZ      CP      0.124 0.32   0.197
 9 isoComp_0… geneCo… ENST00… ENSG00… USP5    VZ      CP      0.314 0.693  0.379
10 isoComp_0… geneCo… ENST00… ENSG00… USP5    VZ      CP      0.561 0.185 -0.376
# … with 1,850 more rows, 2 more variables: isoform_switch_q_value &lt;dbl&gt;,
#   Rank &lt;int&gt;, and abbreviated variable names ¹​gene_ref, ²​isoform_id,
#   ³​gene_name, ⁴​condition_1, ⁵​condition_2</code></pre>
</div>
</div>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-20_7bf3e3d853fcadf1b400fc7dccd69ec7">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(isoform_switch_q_value <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(dIF) <span class="sc">&gt;</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(iso_ref, gene_ref, isoform_id, gene_id, gene_name, condition_1, condition_2, IF1, IF2, dIF, isoform_switch_q_value) <span class="sc">%&gt;%</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(isoform_switch_q_value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1,860 × 11
   iso_ref  gene_…¹ isofo…² gene_id gene_…³ condi…⁴ condi…⁵    IF1    IF2    dIF
   &lt;chr&gt;    &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;    &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 isoComp… geneCo… ENST00… ENSG00… SMARCE1 VZ      CP      0.570  0.164  -0.405
 2 isoComp… geneCo… ENST00… ENSG00… PFN2    VZ      CP      0.250  0.0213 -0.228
 3 isoComp… geneCo… ENST00… ENSG00… ELMO1   VZ      CP      0.404  0.0237 -0.380
 4 isoComp… geneCo… ENST00… ENSG00… CTNNA1  VZ      CP      0.541  0.129  -0.413
 5 isoComp… geneCo… ENST00… ENSG00… DPF2    VZ      CP      0.0823 0.553   0.470
 6 isoComp… geneCo… ENST00… ENSG00… GTF2I   VZ      CP      0.128  0.0167 -0.111
 7 isoComp… geneCo… TALONT… ENSG00… ADD1    VZ      CP      0.310  0.059  -0.251
 8 isoComp… geneCo… ENST00… ENSG00… CTNNA1  VZ      CP      0.124  0.320   0.197
 9 isoComp… geneCo… ENST00… ENSG00… USP5    VZ      CP      0.314  0.693   0.379
10 isoComp… geneCo… ENST00… ENSG00… USP5    VZ      CP      0.561  0.185  -0.376
# … with 1,850 more rows, 1 more variable: isoform_switch_q_value &lt;dbl&gt;, and
#   abbreviated variable names ¹​gene_ref, ²​isoform_id, ³​gene_name,
#   ⁴​condition_1, ⁵​condition_2</code></pre>
</div>
</div>
</section>
<section id="create-our-table-s3" class="level3">
<h3 class="anchored" data-anchor-id="create-our-table-s3">Create our Table S3</h3>
<p>Note: isoformSwitchAnalyzer uses both q-value (&lt;.05) and effect size cutoff |dIF&gt;.1| for DTU calling. Lets remove the dIF effect size cutoff here so that we can directly compare DGE, DTE, and DTU genes in the Venn Diagram below</p>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-21_6c64abde22331c9b6efd4ec740efc48f">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>tableS3 <span class="ot">=</span> talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(isoform_id, gene_id, gene_name, condition_1, condition_2) <span class="sc">%&gt;%</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>    talonSwitchList_part1<span class="sc">$</span>isoformSwitchAnalysis <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(isoform_id, dIF, pvalue, padj)</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">DTU_dIF    =</span> <span class="st">"dIF"</span>,</span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">DTU_pval   =</span> <span class="st">"pvalue"</span>,</span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">DTU_qval   =</span> <span class="st">"padj"</span></span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">DTU =</span> DTU_qval <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="co"># &amp; abs(DTU_dIF) &gt; 0.1</span></span>
<span id="cb60-14"><a href="#cb60-14" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb60-15"><a href="#cb60-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb60-16"><a href="#cb60-16" aria-hidden="true" tabindex="-1"></a>    DTE_results <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(isoform_id, log2FoldChange, pvalue, padj)</span>
<span id="cb60-17"><a href="#cb60-17" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb60-18"><a href="#cb60-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb60-19"><a href="#cb60-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">DTE_log2FC =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb60-20"><a href="#cb60-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">DTE_pval   =</span> <span class="st">"pvalue"</span>,</span>
<span id="cb60-21"><a href="#cb60-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">DTE_qval   =</span> <span class="st">"padj"</span></span>
<span id="cb60-22"><a href="#cb60-22" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb60-23"><a href="#cb60-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb60-24"><a href="#cb60-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">DTE =</span> DTE_qval <span class="sc">&lt;</span> <span class="fl">0.05</span></span>
<span id="cb60-25"><a href="#cb60-25" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb60-26"><a href="#cb60-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb60-27"><a href="#cb60-27" aria-hidden="true" tabindex="-1"></a>    DGE_results <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id, log2FoldChange, pvalue, padj)</span>
<span id="cb60-28"><a href="#cb60-28" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb60-29"><a href="#cb60-29" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb60-30"><a href="#cb60-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">DGE_log2FC =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb60-31"><a href="#cb60-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">DGE_pval   =</span> <span class="st">"pvalue"</span>,</span>
<span id="cb60-32"><a href="#cb60-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">DGE_qval   =</span> <span class="st">"padj"</span></span>
<span id="cb60-33"><a href="#cb60-33" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb60-34"><a href="#cb60-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb60-35"><a href="#cb60-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">DGE =</span> DGE_qval <span class="sc">&lt;</span> <span class="fl">0.05</span></span>
<span id="cb60-36"><a href="#cb60-36" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(isoform_id)`
Joining with `by = join_by(isoform_id)`
Joining with `by = join_by(gene_id)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>tableS3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 102,319 × 17
   isoform_id   gene_id gene_…¹ condi…² condi…³  DTU_dIF DTU_pval DTU_qval DTU  
   &lt;chr&gt;        &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;      &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;lgl&gt;
 1 ENST0000037… ENSG00… TSPAN6  VZ      CP      -0.347   4.89e-14 9.84e-12 TRUE 
 2 TALONT00074… ENSG00… TSPAN6  VZ      CP       0.140   1.42e- 8 1.23e- 6 TRUE 
 3 TALONT00074… ENSG00… TSPAN6  VZ      CP       0.196   1.36e- 6 7.89e- 5 TRUE 
 4 TALONT00074… ENSG00… TSPAN6  VZ      CP       0.00667 6.29e- 1 9.24e- 1 FALSE
 5 TALONT00074… ENSG00… TSPAN6  VZ      CP      -0.00633 4.95e- 1 8.74e- 1 FALSE
 6 TALONT00074… ENSG00… TSPAN6  VZ      CP      -0.00867 1.89e- 1 6.47e- 1 FALSE
 7 TALONT00074… ENSG00… TSPAN6  VZ      CP       0.017   1.48e- 3 3.08e- 2 TRUE 
 8 ENST0000037… ENSG00… DPM1    VZ      CP      -0.0317  8.43e- 1 9.76e- 1 FALSE
 9 ENST0000037… ENSG00… DPM1    VZ      CP      -0.0273  2.34e- 1 7.00e- 1 FALSE
10 ENST0000037… ENSG00… DPM1    VZ      CP       0.0993  1.48e- 1 5.89e- 1 FALSE
# … with 102,309 more rows, 8 more variables: DTE_log2FC &lt;dbl&gt;, DTE_pval &lt;dbl&gt;,
#   DTE_qval &lt;dbl&gt;, DTE &lt;lgl&gt;, DGE_log2FC &lt;dbl&gt;, DGE_pval &lt;dbl&gt;,
#   DGE_qval &lt;dbl&gt;, DGE &lt;lgl&gt;, and abbreviated variable names ¹​gene_name,
#   ²​condition_1, ³​condition_2</code></pre>
</div>
</div>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-22_8e9dbb251aac62d8c47986d4ff0dfa82">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>tableS3 <span class="sc">%&gt;%</span> <span class="fu">write_tsv</span>(<span class="st">"output/tables/TableS3_v3.tsv.gz"</span>)</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>tableS3b_geneLevel <span class="ot">=</span> tableS3 <span class="sc">%&gt;%</span></span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_name, gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">DTU =</span> <span class="fu">any</span>(DTU), <span class="at">DTE =</span> <span class="fu">any</span>(DTE), <span class="at">DGE =</span> <span class="fu">any</span>(DGE),</span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">DTU_qval_min =</span> <span class="fu">min</span>(DTU_qval), <span class="at">DTU_pval_min =</span> <span class="fu">min</span>(DTU_pval),</span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">DTE_qval_min =</span> <span class="fu">min</span>(DTE_qval), <span class="at">DTE_pval_min =</span> <span class="fu">min</span>(DTE_pval),</span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">DGE_pval =</span> <span class="fu">min</span>(DGE_pval), <span class="at">DGE_qval =</span> <span class="fu">min</span>(DGE_qval)</span>
<span id="cb64-10"><a href="#cb64-10" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'gene_name'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>tableS3b_geneLevel <span class="sc">%&gt;%</span> <span class="fu">write_tsv</span>(<span class="st">"output/tables/TableS3b_geneLevel.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Note: log2fc in switchlist are based on the calculated RPKM values, not raw counts, so they don’t agree with DESEq2 log2fc columns.</p>
<p>Sanity check: The isoform_switch_q_value column in switchList<span class="math inline">\(isoformFeatures comes from the padj column in switchList\)</span>isoformSwitchAnalysis (which is the DEXSeq results). gene_switch_q_value is just the minimum padj of any isoform for that gene.</p>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-23_c031c1665c974ea4dcd5dbcf3e21dfc1">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">left_join</span>(</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>(),</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1<span class="sc">$</span>isoformSwitchAnalysis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>(),</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span> <span class="st">"isoform_id"</span></span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(isoform_switch_q_value <span class="sc">!=</span> padj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 0 × 39
# … with 39 variables: iso_ref.x &lt;chr&gt;, gene_ref.x &lt;chr&gt;, isoform_id &lt;chr&gt;,
#   gene_id &lt;chr&gt;, condition_1.x &lt;chr&gt;, condition_2.x &lt;chr&gt;, gene_name &lt;chr&gt;,
#   gene_biotype &lt;chr&gt;, iso_biotype &lt;chr&gt;, gene_overall_mean &lt;dbl&gt;,
#   gene_value_1 &lt;dbl&gt;, gene_value_2 &lt;dbl&gt;, gene_stderr_1 &lt;dbl&gt;,
#   gene_stderr_2 &lt;dbl&gt;, gene_log2_fold_change &lt;dbl&gt;, gene_q_value &lt;dbl&gt;,
#   iso_overall_mean &lt;dbl&gt;, iso_value_1 &lt;dbl&gt;, iso_value_2 &lt;dbl&gt;,
#   iso_stderr_1 &lt;dbl&gt;, iso_stderr_2 &lt;dbl&gt;, iso_log2_fold_change &lt;dbl&gt;, …</code></pre>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">left_join</span>(</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>(),</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1<span class="sc">$</span>isoformSwitchAnalysis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>(),</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span> <span class="st">"isoform_id"</span></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene_switch_q_value <span class="sc">!=</span> <span class="fu">min</span>(padj))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 0 × 39
# Groups:   gene_id [0]
# … with 39 variables: iso_ref.x &lt;chr&gt;, gene_ref.x &lt;chr&gt;, isoform_id &lt;chr&gt;,
#   gene_id &lt;chr&gt;, condition_1.x &lt;chr&gt;, condition_2.x &lt;chr&gt;, gene_name &lt;chr&gt;,
#   gene_biotype &lt;chr&gt;, iso_biotype &lt;chr&gt;, gene_overall_mean &lt;dbl&gt;,
#   gene_value_1 &lt;dbl&gt;, gene_value_2 &lt;dbl&gt;, gene_stderr_1 &lt;dbl&gt;,
#   gene_stderr_2 &lt;dbl&gt;, gene_log2_fold_change &lt;dbl&gt;, gene_q_value &lt;dbl&gt;,
#   iso_overall_mean &lt;dbl&gt;, iso_value_1 &lt;dbl&gt;, iso_value_2 &lt;dbl&gt;,
#   iso_stderr_1 &lt;dbl&gt;, iso_stderr_2 &lt;dbl&gt;, iso_log2_fold_change &lt;dbl&gt;, …</code></pre>
</div>
</div>
</section>
</section>
<section id="fig3" class="level2">
<h2 class="anchored" data-anchor-id="fig3">Fig3</h2>
<section id="fig3a-volcano-plots" class="level3">
<h3 class="anchored" data-anchor-id="fig3a-volcano-plots">Fig3a: Volcano Plots</h3>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-24_cfa2607a4cb4f925c5333119992b90f7">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>cts<span class="sc">$</span>novelty2 <span class="ot">=</span> <span class="fu">as.character</span>(cts<span class="sc">$</span>transcript_novelty)</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>cts<span class="sc">$</span>novelty2[<span class="fu">which</span>(cts<span class="sc">$</span>novelty2<span class="sc">==</span><span class="st">"ISM"</span> <span class="sc">&amp;</span> cts<span class="sc">$</span>ISM_subtype<span class="sc">==</span><span class="st">"Prefix"</span>)] <span class="ot">=</span> <span class="st">"ISM_Prefix"</span></span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>cts<span class="sc">$</span>novelty2[<span class="fu">which</span>(cts<span class="sc">$</span>novelty2<span class="sc">==</span><span class="st">"ISM"</span> <span class="sc">&amp;</span> cts<span class="sc">$</span>ISM_subtype<span class="sc">==</span><span class="st">"Suffix"</span>)] <span class="ot">=</span> <span class="st">"ISM_Suffix"</span></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>cts<span class="sc">$</span>novelty2[cts<span class="sc">$</span>novelty2 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Antisense"</span>, <span class="st">"Genomic"</span>, <span class="st">"Intergenic"</span>, <span class="st">"ISM"</span>)] <span class="ot">=</span> <span class="st">"Other"</span></span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>cts<span class="sc">$</span>novelty2 <span class="ot">=</span> <span class="fu">factor</span>(cts<span class="sc">$</span>novelty2,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"Known"</span>, <span class="st">"ISM_Prefix"</span>, <span class="st">"ISM_Suffix"</span>, <span class="st">"NIC"</span>, <span class="st">"NNC"</span>, <span class="st">"Other"</span>))</span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a>data_to_label1 <span class="ot">=</span> talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>((<span class="sc">-</span><span class="fu">log10</span>(isoform_switch_q_value) <span class="sc">&gt;</span> <span class="dv">50</span> <span class="sc">&amp;</span> (<span class="fu">abs</span>(dIF) <span class="sc">&gt;</span> .<span class="dv">1</span>)) <span class="sc">|</span> </span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>           (<span class="fu">abs</span>(dIF) <span class="sc">&gt;</span> .<span class="dv">5</span> <span class="sc">&amp;</span> isoform_switch_q_value <span class="sc">&lt;</span> <span class="fl">1e-10</span>))</span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a>data_to_label2 <span class="ot">=</span> talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> </span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(isoform_switch_q_value <span class="sc">&lt;</span> .<span class="dv">05</span>, gene_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"SRRM1"</span>, <span class="st">"PTBP2"</span>, <span class="st">"ELAVL2"</span>, <span class="st">"ELAVL4"</span>, <span class="st">"RBFOX2"</span>,<span class="st">"NF1"</span>, <span class="st">"TSC2"</span>, <span class="st">"UBE3A"</span>, <span class="st">"KMT2E"</span>, <span class="st">"KMT5B"</span>, <span class="st">"SMARCD3"</span>, <span class="st">"SMARCA1"</span>, <span class="st">"FOXP2"</span>, <span class="st">"GRIA3"</span>, <span class="st">"VAMP1"</span>, <span class="st">"GAD1"</span>, <span class="st">"NLGN4X"</span>, <span class="st">"NRXN1"</span>)) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene_name) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(<span class="at">n =</span> <span class="dv">1</span>, <span class="fu">abs</span>(dIF))</span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-13"><a href="#cb71-13" aria-hidden="true" tabindex="-1"></a>data_to_label <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(data_to_label1,data_to_label2)</span>
<span id="cb71-14"><a href="#cb71-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-15"><a href="#cb71-15" aria-hidden="true" tabindex="-1"></a>Fig3a <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="at">data=</span>talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(cts <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">isoform_id=</span>transcript_id, <span class="at">novelty=</span>novelty2)), </span>
<span id="cb71-16"><a href="#cb71-16" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>dIF, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(isoform_switch_q_value))) <span class="sc">+</span></span>
<span id="cb71-17"><a href="#cb71-17" aria-hidden="true" tabindex="-1"></a>     <span class="fu">geom_point</span>(</span>
<span id="cb71-18"><a href="#cb71-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">color=</span>novelty), <span class="co"># default cutoff</span></span>
<span id="cb71-19"><a href="#cb71-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">size=</span><span class="dv">2</span>, <span class="at">alpha=</span>.<span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb71-20"><a href="#cb71-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype=</span><span class="st">'dashed'</span>,<span class="at">color=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="co"># default cutoff</span></span>
<span id="cb71-21"><a href="#cb71-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">'difference in isoform fraction (dIF)'</span>, <span class="at">y=</span><span class="st">'-log10 ( Isoform Switch Q Value )'</span>) <span class="sc">+</span></span>
<span id="cb71-22"><a href="#cb71-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values=</span>colorVector_ismSplit) <span class="sc">+</span> </span>
<span id="cb71-23"><a href="#cb71-23" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span>  data_to_label <span class="sc">%&gt;%</span> <span class="fu">filter</span>(dIF <span class="sc">&lt;</span> <span class="dv">0</span>),<span class="fu">aes</span>(<span class="at">label=</span>gene_name, <span class="at">segment.size  =</span> .<span class="dv">1</span>),<span class="at">size=</span><span class="dv">3</span>,<span class="at">force =</span> <span class="dv">10</span>, <span class="at">max.overlaps =</span> <span class="dv">50</span>,<span class="at">nudge_y =</span> <span class="dv">10</span>, <span class="at">nudge_x =</span> <span class="sc">-</span>.<span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb71-24"><a href="#cb71-24" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span>  data_to_label <span class="sc">%&gt;%</span> <span class="fu">filter</span>(dIF <span class="sc">&gt;</span> <span class="dv">0</span>),<span class="fu">aes</span>(<span class="at">label=</span>gene_name, <span class="at">segment.size  =</span> .<span class="dv">1</span>),<span class="at">size=</span><span class="dv">3</span>,<span class="at">force =</span> <span class="dv">10</span>, <span class="at">max.overlaps =</span> <span class="dv">50</span>,<span class="at">nudge_y =</span> <span class="dv">10</span>, <span class="at">nudge_x =</span> .<span class="dv">2</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>) <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>,<span class="at">lty=</span><span class="dv">1</span>,<span class="at">color=</span><span class="st">'darkgrey'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(isoform_id)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>Fig3a</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 4 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_text_repel()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(Fig3a, <span class="at">file=</span><span class="st">"output/figures/Fig3/Fig3a.pdf"</span>,<span class="at">width=</span><span class="dv">6</span>,<span class="at">height=</span><span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 4 rows containing missing values (`geom_point()`).
Removed 1 rows containing missing values (`geom_text_repel()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>Fig3a.fg <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="at">data=</span>talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(cts <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">isoform_id=</span>transcript_id, <span class="at">novelty=</span>novelty2)), </span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>dIF, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(isoform_switch_q_value))) <span class="sc">+</span></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>     <span class="fu">geom_point</span>(</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">color=</span>novelty), <span class="at">color=</span><span class="cn">NA</span>, <span class="co"># default cutoff</span></span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">size=</span><span class="dv">2</span>, <span class="at">alpha=</span>.<span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype=</span><span class="st">'dashed'</span>,<span class="at">color=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="co"># default cutoff</span></span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">'difference in isoform fraction (dIF)'</span>, <span class="at">y=</span><span class="st">'-log10 ( Isoform Switch Q Value )'</span>) <span class="sc">+</span></span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values=</span>colorVector_ismSplit) <span class="sc">+</span> </span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a> ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span>  data_to_label <span class="sc">%&gt;%</span> <span class="fu">filter</span>(dIF <span class="sc">&lt;</span> <span class="dv">0</span>),<span class="fu">aes</span>(<span class="at">label=</span>gene_name, <span class="at">segment.size   =</span> .<span class="dv">1</span>),<span class="at">size=</span><span class="dv">3</span>,<span class="at">force =</span> <span class="dv">10</span>, <span class="at">max.overlaps =</span> <span class="dv">50</span>,<span class="at">nudge_y =</span> <span class="dv">10</span>, <span class="at">nudge_x =</span> <span class="sc">-</span>.<span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span>  data_to_label <span class="sc">%&gt;%</span> <span class="fu">filter</span>(dIF <span class="sc">&gt;</span> <span class="dv">0</span>),<span class="fu">aes</span>(<span class="at">label=</span>gene_name, <span class="at">segment.size  =</span> .<span class="dv">1</span>),<span class="at">size=</span><span class="dv">3</span>,<span class="at">force =</span> <span class="dv">10</span>, <span class="at">max.overlaps =</span> <span class="dv">50</span>,<span class="at">nudge_y =</span> <span class="dv">10</span>, <span class="at">nudge_x =</span> .<span class="dv">2</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>) <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>,<span class="at">lty=</span><span class="dv">1</span>,<span class="at">color=</span><span class="st">'darkgrey'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(isoform_id)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>Fig3a.fg</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 102319 rows containing missing values (`geom_point()`).
Removed 1 rows containing missing values (`geom_text_repel()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-24-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(Fig3a.fg, <span class="at">file=</span><span class="st">"output/figures/Fig3/Fig3a_fg.pdf"</span>,<span class="at">width=</span><span class="dv">6</span>,<span class="at">height=</span><span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 102319 rows containing missing values (`geom_point()`).
Removed 1 rows containing missing values (`geom_text_repel()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>Fig3a.bg <span class="ot">=</span><span class="fu">ggplot</span>(<span class="at">data=</span>talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(cts <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">isoform_id=</span>transcript_id, <span class="at">novelty=</span>novelty2)), </span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>dIF, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(isoform_switch_q_value))) <span class="sc">+</span></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>     <span class="fu">geom_point</span>(</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">color=</span>novelty), <span class="co"># default cutoff</span></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">size=</span><span class="dv">2</span>, <span class="at">alpha=</span>.<span class="dv">5</span>)  <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values=</span>colorVector_ismSplit) <span class="sc">+</span> <span class="fu">theme_void</span>()  <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(isoform_id)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(Fig3a.bg, <span class="at">file=</span><span class="st">"output/figures/Fig3/Fig3a_bg.jpg"</span>,<span class="at">width=</span><span class="dv">6</span>,<span class="at">height=</span><span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 4 rows containing missing values (`geom_point()`).</code></pre>
</div>
</div>
</section>
<section id="fig3a_2-pie-chart" class="level3">
<h3 class="anchored" data-anchor-id="fig3a_2-pie-chart">Fig3a_2: Pie Chart</h3>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-25_5e6ac3fc17300b665a0fa991aba62291">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>this_df <span class="ot">=</span> talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> </span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(cts <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">isoform_id=</span>transcript_id, <span class="at">novelty=</span>novelty2)) <span class="sc">%&gt;%</span></span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(isoform_switch_q_value <span class="sc">&lt;</span> .<span class="dv">05</span>) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(novelty) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">switches=</span><span class="fu">n_distinct</span>(isoform_id))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(isoform_id)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>this_df<span class="ot">&lt;-</span>this_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(novelty)) <span class="sc">%&gt;%</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">prop =</span> switches <span class="sc">/</span> <span class="fu">sum</span>(switches) <span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ypos =</span> <span class="fu">cumsum</span>(prop)<span class="sc">-</span> <span class="fl">0.5</span><span class="sc">*</span>prop )</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>Fig3a_part2 <span class="ot">=</span> <span class="fu">ggplot</span>(this_df, <span class="fu">aes</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span>prop, <span class="at">fill=</span>novelty)) <span class="sc">+</span></span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">width=</span><span class="dv">1</span>, <span class="at">color=</span><span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_polar</span>(<span class="st">"y"</span>,<span class="at">start=</span><span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>, <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span>.<span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>colorVector_ismSplit) <span class="sc">+</span> </span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">y=</span>ypos, <span class="at">label =</span> <span class="fu">paste0</span>(novelty,<span class="st">"</span><span class="sc">\n</span><span class="st">(N="</span>, switches, <span class="st">")"</span>)), <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a>  Fig3a_part2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(Fig3a_part2, <span class="at">file=</span><span class="st">"output/figures/Fig3/Fig3a_part2.pdf"</span>,<span class="at">width=</span><span class="dv">4</span>,<span class="at">height=</span><span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="fig3b-dge-vs-dtu-plots" class="level3">
<h3 class="anchored" data-anchor-id="fig3b-dge-vs-dtu-plots">Fig3b: DGE vs DTU plots</h3>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-26_58d5dbcec83214428840ff6334a60082">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>this_df <span class="ot">=</span> talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> </span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(cts <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">isoform_id=</span>transcript_id, <span class="at">novelty=</span>novelty2))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(isoform_id)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>this_df<span class="sc">$</span>gene_q_value <span class="ot">=</span> DGE_results<span class="sc">$</span>padj[<span class="fu">match</span>(this_df<span class="sc">$</span>gene_id, DGE_results<span class="sc">$</span>gene_id)]</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>Fig3b <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="at">data=</span>this_df, </span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>gene_log2_fold_change, <span class="at">y=</span>dIF)) <span class="sc">+</span></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>     <span class="fu">geom_point</span>(</span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">color=</span><span class="fu">abs</span>(dIF) <span class="sc">&gt;</span> <span class="fl">0.1</span> <span class="sc">&amp;</span> isoform_switch_q_value <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="co"># default cutoff</span></span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">shape =</span> <span class="fu">abs</span>(gene_log2_fold_change) <span class="sc">&gt;</span> .<span class="dv">1</span> <span class="sc">&amp;</span> gene_q_value <span class="sc">&lt;</span> .<span class="dv">05</span>),</span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">size=</span><span class="dv">1</span>, <span class="at">alpha=</span>.<span class="dv">5</span>,</span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">'dashed'</span>) <span class="sc">+</span> <span class="co"># default cutoff</span></span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">'dashed'</span>) <span class="sc">+</span> <span class="co"># default cutoff</span></span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">y=</span><span class="st">'difference in isoform fraction (dIF)'</span>, <span class="at">x=</span><span class="st">'Gene log2FC'</span>) <span class="sc">+</span></span>
<span id="cb94-13"><a href="#cb94-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()  <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">5</span>,<span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb94-14"><a href="#cb94-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="st">'Signficant</span><span class="sc">\n</span><span class="st">Isoform Switch'</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">'black'</span>,<span class="st">'red'</span>)) <span class="sc">+</span></span>
<span id="cb94-15"><a href="#cb94-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="st">'Signficant</span><span class="sc">\n</span><span class="st">DGE'</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"FALSE"</span><span class="ot">=</span><span class="dv">1</span>, <span class="st">"TRUE"</span><span class="ot">=</span><span class="dv">16</span>)) <span class="sc">+</span></span>
<span id="cb94-16"><a href="#cb94-16" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span> talonSwitchList_part1<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">abs</span>(dIF) <span class="sc">&gt;</span> .<span class="dv">5</span> <span class="sc">&amp;</span> isoform_switch_q_value <span class="sc">&lt;</span> <span class="fl">0.05</span>), <span class="fu">aes</span>(<span class="at">label=</span>gene_name),<span class="at">size=</span><span class="dv">3</span>, <span class="at">max.overlaps =</span> <span class="dv">20</span>)</span>
<span id="cb94-17"><a href="#cb94-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-18"><a href="#cb94-18" aria-hidden="true" tabindex="-1"></a>Fig3b</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 31 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_text_repel()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 4 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(Fig3b, <span class="at">file=</span><span class="st">"output/figures/Fig3/Fig3b.pdf"</span>,<span class="at">width=</span><span class="dv">8</span>,<span class="at">height=</span><span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 31 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_text_repel()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 2 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="do">## DGE Volcano plot colored by DTU</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>this_df <span class="ot">=</span> DGE_results <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(cts_gene <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id, gene_name)) <span class="sc">%&gt;%</span> </span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tableS3 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id, DTU) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">DTU =</span> <span class="fu">any</span>(DTU)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(gene_id)`
Joining with `by = join_by(gene_id)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>this_df<span class="sc">$</span>DTU[<span class="fu">is.na</span>(this_df<span class="sc">$</span>DTU)] <span class="ot">=</span> F</span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>this_df<span class="sc">$</span>DTU <span class="ot">=</span> <span class="fu">factor</span>(this_df<span class="sc">$</span>DTU, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"TRUE"</span>, <span class="st">"FALSE"</span>))</span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(this_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU<span class="sc">==</span><span class="cn">FALSE</span>), <span class="fu">aes</span>(<span class="at">x=</span>log2FoldChange, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(padj))) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">alpha=</span>.<span class="dv">25</span>, <span class="at">color=</span><span class="st">'grey'</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data=</span>this_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(DTU<span class="sc">==</span><span class="cn">TRUE</span>), <span class="at">alpha=</span>.<span class="dv">5</span>, <span class="at">color=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-26-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="fig3c-venn-diagrams" class="level3">
<h3 class="anchored" data-anchor-id="fig3c-venn-diagrams">Fig3c: Venn Diagrams</h3>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-27_f82dc879745c0a1623c7f89256f8701d">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>gene_overlaps <span class="ot">=</span> tableS3 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">DTE =</span> <span class="fu">any</span>(DTE), <span class="at">DGE=</span><span class="fu">any</span>(DGE), <span class="at">DTU=</span><span class="fu">any</span>(DTU)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>gene_id)</span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Sanity check: 2,679 of 10809 genes with DTU (iso_q_value&lt; .05)</span></span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>switches<span class="ot">=</span><span class="fu">extractTopSwitches</span>(</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part1, <span class="at">extractGenes =</span> <span class="cn">TRUE</span>, <span class="at">dIFcutoff =</span> <span class="dv">0</span>, <span class="at">filterForConsequences =</span> <span class="cn">FALSE</span>, <span class="at">n=</span><span class="cn">Inf</span></span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span>  <span class="fu">as_tibble</span>()</span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(gene_overlaps<span class="sc">$</span>DTU)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
 8130  2679 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene-level DGE:  4475 of 24554 -- note different backgrounds</span></span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(DGE_results<span class="sc">$</span>padj <span class="sc">&lt;</span> .<span class="dv">05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
20079  4475 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Fisher's Exact Test</span></span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="at">file =</span> <span class="st">"code/fisher_overlap.R"</span>)</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ORA</span>(tableS3<span class="sc">$</span>gene_id[tableS3<span class="sc">$</span>DTU_qval<span class="sc">&lt;</span>.<span class="dv">05</span>],DGE_results<span class="sc">$</span>gene_id[DGE_results<span class="sc">$</span>padj<span class="sc">&lt;</span>.<span class="dv">05</span>], tableS3<span class="sc">$</span>gene_id, DGE_results<span class="sc">$</span>gene_id)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                    OR               Fisher p                 -95%CI 
    "2.00059991005475" "1.21479676549559e-46"     "1.81968255413102" 
                +95%CI                Overlap         Reference List 
    "2.19910780950278"                 "1010"                 "2898" 
            Input List             Background         % List Overlap 
                "2679"                "10809"                 "34.9" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 1411 isoform-switching genes without even nominally significant evidence of DGE</span></span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">intersect</span>(tableS3<span class="sc">$</span>gene_id[tableS3<span class="sc">$</span>DTU_qval<span class="sc">&lt;</span>.<span class="dv">05</span>], DGE_results<span class="sc">$</span>gene_id[DGE_results<span class="sc">$</span>pvalue<span class="sc">&gt;</span>.<span class="dv">05</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1411</code></pre>
</div>
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>Fig3c <span class="ot">=</span> <span class="fu">ggVennDiagram</span>(<span class="fu">list</span>(<span class="at">DTU =</span> <span class="fu">which</span>(gene_overlaps<span class="sc">$</span>DTU), </span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">DGE =</span> <span class="fu">which</span>(gene_overlaps<span class="sc">$</span>DGE),</span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">DTE =</span> <span class="fu">which</span>(gene_overlaps<span class="sc">$</span>DTE))) <span class="sc">+</span> </span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">"grey"</span>,<span class="at">high =</span> <span class="st">"red"</span>)</span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>Fig3c</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(Fig3c, <span class="at">file=</span><span class="st">"output/figures/Fig3/Fig3c.pdf"</span>,<span class="at">width=</span><span class="dv">8</span>,<span class="at">height=</span><span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="switch-analysis-part-2-functional-consequences" class="level2">
<h2 class="anchored" data-anchor-id="switch-analysis-part-2-functional-consequences">Switch analysis part 2 (functional consequences)</h2>
<p>IUPred and SignalP are difficult to run. Non-webserver IUPred2A/3 only takes a single sequence at a time, so requires a wrapper. Although “V5 is supported” for SignalP (I guess through the webserver?) v5.0 and v6.0 no longer seem to produce the expected output (<code>-f summary</code>) of v4.1.</p>
<p>Currently, trying to import from the <code>webtools_v3</code> crashes R.</p>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-28_37145ca628ebaa26aaf0d64579e987d8">
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>rdata_path  <span class="ot">=</span> <span class="st">"data/working/talonSwitchList_part2.rds"</span></span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(rdata_path)) {</span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part2 <span class="ot">&lt;-</span> <span class="fu">isoformSwitchAnalysisPart2</span>(</span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">switchAnalyzeRlist        =</span> talonSwitchList_part1, </span>
<span id="cb115-5"><a href="#cb115-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">n                         =</span> <span class="dv">10</span>, <span class="co"># number of PDF plots to generate</span></span>
<span id="cb115-6"><a href="#cb115-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">removeNoncodinORFs        =</span> <span class="cn">FALSE</span>,</span>
<span id="cb115-7"><a href="#cb115-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">pathToCPC2resultFile      =</span> <span class="st">"data/working/IsoformSwitchAnalyzeR_webtools_v4/result_cpc2.txt"</span>,</span>
<span id="cb115-8"><a href="#cb115-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">pathToPFAMresultFile      =</span> <span class="st">"data/working/IsoformSwitchAnalyzeR_webtools_v4/result_pfam_scan.txt"</span>,</span>
<span id="cb115-9"><a href="#cb115-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># might want to disable smoothing through `analyzeIUPred2A` since IUPred3 now</span></span>
<span id="cb115-10"><a href="#cb115-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># performs smoothing:</span></span>
<span id="cb115-11"><a href="#cb115-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">pathToIUPred2AresultFile  =</span> <span class="st">"data/working/IsoformSwitchAnalyzeR_webtools_v4/result_iupred3.txt.gz"</span>,</span>
<span id="cb115-12"><a href="#cb115-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">pathToSignalPresultFile   =</span> <span class="st">"data/working/IsoformSwitchAnalyzeR_webtools_v4/result_signalP.txt"</span>,</span>
<span id="cb115-13"><a href="#cb115-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">pathToOutput              =</span> <span class="st">"output/IsoformSwitchAnalyzeR/"</span>,</span>
<span id="cb115-14"><a href="#cb115-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">outputPlots               =</span> T</span>
<span id="cb115-15"><a href="#cb115-15" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb115-16"><a href="#cb115-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># todo - split this into individual functions because `analyzeSwitchConsequences`</span></span>
<span id="cb115-17"><a href="#cb115-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># is still filtering on dIF in addition to qval</span></span>
<span id="cb115-18"><a href="#cb115-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-19"><a href="#cb115-19" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part2 <span class="ot">&lt;-</span> <span class="fu">analyzeSwitchConsequences</span>(talonSwitchList_part2, <span class="at">onlySigIsoforms =</span> T, <span class="at">dIFcutoff =</span> <span class="dv">0</span>, <span class="at">consequencesToAnalyze =</span> <span class="fu">c</span>(<span class="st">'tss'</span>,<span class="st">'tts'</span>,<span class="st">'last_exon'</span>,<span class="st">'isoform_length'</span>,<span class="st">'exon_number'</span>,<span class="st">'intron_structure'</span>,<span class="st">'ORF_length'</span>, <span class="st">'5_utr_seq_similarity'</span>, <span class="st">'5_utr_length'</span>, <span class="st">'3_utr_seq_similarity'</span>, <span class="st">'3_utr_length'</span>,<span class="st">'coding_potential'</span>,<span class="st">'ORF_seq_similarity'</span>,<span class="st">'NMD_status'</span>,<span class="st">'domains_identified'</span>,<span class="st">'signal_peptide_identified'</span>))</span>
<span id="cb115-20"><a href="#cb115-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-21"><a href="#cb115-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-22"><a href="#cb115-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(talonSwitchList_part2, <span class="at">file=</span>rdata_path)</span>
<span id="cb115-23"><a href="#cb115-23" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb115-24"><a href="#cb115-24" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part2 <span class="ot">=</span> <span class="fu">readRDS</span>(rdata_path)</span>
<span id="cb115-25"><a href="#cb115-25" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot a gene:</p>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-29_b076ce67247edc2566956064326904db">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="st">"output/figures/Fig3/Fig3_SMARCC2.pdf"</span>, <span class="at">height=</span><span class="dv">4</span>, <span class="at">width=</span><span class="dv">8</span>)</span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a><span class="fu">switchPlot</span>(</span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part2,<span class="at">dIFcutoff =</span> .<span class="dv">1</span>,<span class="at">logYaxis =</span> <span class="cn">TRUE</span>,</span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene=</span><span class="st">'SMARCC2'</span></span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="st">"output/figures/Fig4/FigS4_NF1.pdf"</span>, <span class="at">height=</span><span class="dv">4</span>, <span class="at">width=</span><span class="dv">6</span>)</span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a><span class="fu">switchPlot</span>(</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part2,<span class="at">dIFcutoff =</span> .<span class="dv">1</span>,<span class="at">logYaxis =</span> <span class="cn">TRUE</span>,</span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene=</span><span class="st">'NF1'</span></span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="fu">switchPlot</span>(</span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part2,</span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene=</span><span class="st">'TALONG000088362'</span></span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="functional-analysis-of-isoform-switches" class="level2">
<h2 class="anchored" data-anchor-id="functional-analysis-of-isoform-switches">Functional Analysis of Isoform Switches:</h2>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-30_c41e991853593394f2860b2e254c05e0">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="st">'output/figures/Fig3/Fig3e.pdf'</span>, <span class="at">width=</span><span class="dv">7</span>,<span class="at">height=</span><span class="dv">5</span>)</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>switch_consequences <span class="ot">&lt;-</span> <span class="fu">extractConsequenceEnrichment</span>(</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>    talonSwitchList_part2,<span class="at">dIFcutoff =</span> <span class="dv">0</span>,<span class="at">countGenes =</span> F,</span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">returnResult =</span> T <span class="co"># if TRUE returns a data.frame with the summary statistics</span></span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(switch_consequences, <span class="at">file=</span><span class="st">"output/tables/TableS3_switchConsequences_summary_v2.tsv"</span>)</span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(talonSwitchList_part2<span class="sc">$</span>switchConsequence, <span class="at">file=</span><span class="st">"output/tables/TableS3_switchConsequences_data.tsv"</span>)</span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-4"><a href="#cb123-4" aria-hidden="true" tabindex="-1"></a>consequence <span class="ot">=</span> <span class="fu">extractConsequenceSummary</span>(</span>
<span id="cb123-5"><a href="#cb123-5" aria-hidden="true" tabindex="-1"></a>    talonSwitchList_part2, <span class="at">includeCombined =</span> T,</span>
<span id="cb123-6"><a href="#cb123-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">dIFcutoff=</span><span class="dv">0</span>, <span class="at">returnResult =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="fu">extractSplicingEnrichment</span>(</span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a>    talonSwitchList_part2,<span class="at">dIFcutoff =</span> <span class="dv">0</span>,<span class="at">onlySigIsoforms =</span> T,</span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">countGenes =</span> F,</span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">returnResult =</span> <span class="cn">TRUE</span> <span class="co"># if TRUE returns a data.frame with the summary statistics</span></span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-30-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>  condition_1 condition_2                            AStype nUp nDown    propUp
1          VZ          CP     A3 gain (paired with A3 loss) 603   472 0.5609302
2          VZ          CP     A5 gain (paired with A5 loss) 526   466 0.5302419
3          VZ          CP ATSS gain (paired with ATSS loss) 868   679 0.5610860
4          VZ          CP ATTS gain (paired with ATTS loss) 690   544 0.5591572
5          VZ          CP               ES (paired with EI) 776  1153 0.4022810
6          VZ          CP     IR gain (paired with IR loss) 266   265 0.5009416
7          VZ          CP   MEE gain (paired with MEE loss)   4     3 0.5714286
8          VZ          CP             MES (paired with MEI) 364   587 0.3827550
  propUpCiLo propUpCiHi   propUpPval   propUpQval Significant Comparison
1  0.5306659  0.5908602 7.202013e-05 1.152322e-04        TRUE VZ\nvs\nCP
2  0.4986297  0.5616742 6.097909e-02 8.130545e-02       FALSE VZ\nvs\nCP
3  0.5359374  0.5860023 1.705724e-06 4.548597e-06        TRUE VZ\nvs\nCP
4  0.5309374  0.5870945 3.592843e-05 7.185686e-05        TRUE VZ\nvs\nCP
5  0.3803020  0.4245576 8.845503e-18 7.076402e-17        TRUE VZ\nvs\nCP
6  0.4575691  0.5443036 1.000000e+00 1.000000e+00       FALSE VZ\nvs\nCP
7  0.1840516  0.9010117 1.000000e+00 1.000000e+00       FALSE VZ\nvs\nCP
8  0.3517411  0.4144974 4.775090e-13 1.910036e-12        TRUE VZ\nvs\nCP</code></pre>
</div>
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="fu">extractSplicingSummary</span>(talonSwitchList_part2,</span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">splicingToAnalyze =</span> <span class="st">'all'</span>,<span class="at">dIFcutoff =</span> <span class="dv">0</span>,</span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">onlySigIsoforms =</span> T,<span class="at">asFractionTotal =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-30-3.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="alternative-tss-isoform-switches" class="level3">
<h3 class="anchored" data-anchor-id="alternative-tss-isoform-switches">Alternative TSS isoform switches</h3>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-31_fa5c2cc1069d3ea83366f608596e35d1">
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>TSS_switches <span class="ot">=</span>talonSwitchList_part2<span class="sc">$</span>switchConsequence <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(featureCompared <span class="sc">==</span> <span class="st">'tss'</span>, isoformsDifferent<span class="sc">==</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb127-3"><a href="#cb127-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name, isoformUpregulated, isoformDownregulated) <span class="sc">%&gt;%</span></span>
<span id="cb127-4"><a href="#cb127-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(talonSwitchList_part2<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">isoformUpregulated=</span>isoform_id, dIF, isoform_switch_q_value)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(isoformUpregulated)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Top 10 TSS switches by isoform fraction</span></span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a>TSS_switches<span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="sc">-</span><span class="fu">abs</span>(dIF))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2,452 × 5
   gene_name isoformUpregulated   isoformDownregulated   dIF isoform_switch_q_…¹
   &lt;chr&gt;     &lt;chr&gt;                &lt;chr&gt;                &lt;dbl&gt;               &lt;dbl&gt;
 1 S100A16   ENST00000368705.2_2  ENST00000368706.9_3  0.811            5.86e- 3
 2 CPVL      TALONT000753261      ENST00000265394.10_3 0.769            7.02e-16
 3 ZNF358    ENST00000596712.1_3  ENST00000597229.2_2  0.766            7.57e- 4
 4 PLAUR     ENST00000340093.8_5  TALONT000368020      0.697            5.55e- 8
 5 ELMO1     ENST00000396045.7_2  ENST00000310758.8_2  0.654            1.67e-39
 6 ELMO1     ENST00000396045.7_2  ENST00000448602.5_2  0.654            1.67e-39
 7 ELMO1     ENST00000396045.7_2  TALONT000761865      0.654            1.67e-39
 8 BIN1      ENST00000316724.10_4 ENST00000348750.8_3  0.624            1.51e- 5
 9 BIN1      ENST00000316724.10_4 ENST00000393040.7_3  0.624            1.51e- 5
10 BIN1      ENST00000316724.10_4 ENST00000409400.1_4  0.624            1.51e- 5
# … with 2,442 more rows, and abbreviated variable name ¹​isoform_switch_q_value</code></pre>
</div>
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(top_genes <span class="cf">in</span> TSS_switches <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="sc">-</span><span class="fu">abs</span>(dIF)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name) <span class="sc">%&gt;%</span> </span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">10</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()) {</span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a><span class="fu">switchPlot</span>(</span>
<span id="cb131-4"><a href="#cb131-4" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part2,</span>
<span id="cb131-5"><a href="#cb131-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene=</span>top_genes)</span>
<span id="cb131-6"><a href="#cb131-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-7"><a href="#cb131-7" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-8.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-9.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-10.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Top 10 TSS switches by q-value</span></span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>TSS_switches<span class="sc">%&gt;%</span> <span class="fu">arrange</span>(isoform_switch_q_value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2,452 × 5
   gene_name isoformUpregulated   isoformDownregulated   dIF isoform_switch_q_…¹
   &lt;chr&gt;     &lt;chr&gt;                &lt;chr&gt;                &lt;dbl&gt;               &lt;dbl&gt;
 1 CTNNA1    ENST00000540387.5_1  ENST00000302763.12_2 0.197            1.04e-83
 2 PKM       TALONT000323027      TALONT000323086      0.453            9.72e-73
 3 TPM1      ENST00000334895.10_2 ENST00000358278.7_3  0.475            4.54e-65
 4 TPM1      ENST00000334895.10_2 ENST00000558868.5_1  0.475            4.54e-65
 5 TPM1      ENST00000334895.10_2 ENST00000560959.5_1  0.475            4.54e-65
 6 KIF2A     ENST00000407818.7_2  ENST00000401507.7_2  0.276            1.96e-54
 7 KIF2A     ENST00000407818.7_2  TALONT000664196      0.276            1.96e-54
 8 ABAT      ENST00000396600.6_1  ENST00000268251.13_2 0.317            4.11e-51
 9 ABAT      ENST00000396600.6_1  ENST00000425191.6_1  0.317            4.11e-51
10 ABAT      ENST00000396600.6_1  TALONT000412117      0.317            4.11e-51
# … with 2,442 more rows, and abbreviated variable name ¹​isoform_switch_q_value</code></pre>
</div>
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(top_genes <span class="cf">in</span> TSS_switches <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(isoform_switch_q_value) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name) <span class="sc">%&gt;%</span> </span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">10</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()) {</span>
<span id="cb134-3"><a href="#cb134-3" aria-hidden="true" tabindex="-1"></a><span class="fu">switchPlot</span>(</span>
<span id="cb134-4"><a href="#cb134-4" aria-hidden="true" tabindex="-1"></a>  talonSwitchList_part2,</span>
<span id="cb134-5"><a href="#cb134-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene=</span>top_genes)</span>
<span id="cb134-6"><a href="#cb134-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-11.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-12.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-13.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-14.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-15.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-16.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-17.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-18.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-19.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-31-20.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(TSS_switches, <span class="at">file=</span><span class="st">"output/tables/TableS3_tss_switches.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="plot-interesting-disease-genes" class="level2">
<h2 class="anchored" data-anchor-id="plot-interesting-disease-genes">Plot interesting disease genes</h2>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-32_4488435d2bb122f620be0aa04d417cd8">
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a>pdf_path <span class="ot">=</span> <span class="st">"output/figures/switch_plots/switch_plots_diseaseGenes.pdf"</span></span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(pdf_path)) {</span>
<span id="cb136-3"><a href="#cb136-3" aria-hidden="true" tabindex="-1"></a>  disease_genes <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">'ref/ASD+SCZ+DDD_2022.csv'</span>)</span>
<span id="cb136-4"><a href="#cb136-4" aria-hidden="true" tabindex="-1"></a>  interesting_genes <span class="ot">=</span> <span class="fu">extractTopSwitches</span>(talonSwitchList_part2, <span class="at">filterForConsequences =</span> T, <span class="at">n =</span> <span class="cn">Inf</span>)</span>
<span id="cb136-5"><a href="#cb136-5" aria-hidden="true" tabindex="-1"></a>  interesting_genes <span class="ot">=</span> interesting_genes<span class="sc">$</span>gene_name</span>
<span id="cb136-6"><a href="#cb136-6" aria-hidden="true" tabindex="-1"></a>  interesting_genes <span class="ot">=</span> interesting_genes[interesting_genes <span class="sc">%in%</span> disease_genes<span class="sc">$</span>Gene]</span>
<span id="cb136-7"><a href="#cb136-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-8"><a href="#cb136-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="at">file=</span>pdf_path)</span>
<span id="cb136-9"><a href="#cb136-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_gene <span class="cf">in</span> interesting_genes) {</span>
<span id="cb136-10"><a href="#cb136-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">switchPlot</span>(</span>
<span id="cb136-11"><a href="#cb136-11" aria-hidden="true" tabindex="-1"></a>    talonSwitchList_part2,</span>
<span id="cb136-12"><a href="#cb136-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene=</span>this_gene</span>
<span id="cb136-13"><a href="#cb136-13" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb136-14"><a href="#cb136-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(this_gene)</span>
<span id="cb136-15"><a href="#cb136-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb136-16"><a href="#cb136-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb136-17"><a href="#cb136-17" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "DPF2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "GTF2I"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ATP6V0A1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SMARCC2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "TBR1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ABAT"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "AP1S2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "GPHN"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "NRCAM"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ELAVL2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CADM1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ZEB2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PRICKLE1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "MYH10"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PROM1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "DNM1L"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PKN2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ROBO2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "AMPH"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CACNB1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CDC42"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "TSC2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CELF2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "AKAP9"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SATB1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "POMGNT1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "NCDN"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "FGF12"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "MAP7"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ZNF711"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "FOXP1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "NF1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "APBA2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PRUNE2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "GNAO1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "KLHL5"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_segment()`).
Removed 1 rows containing missing values (`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PBX1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SLC38A10"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ACTN4"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PHLDB1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "FAM174B"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CAMTA1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PHF14"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PPARGC1A"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "DPYSL2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "BRINP3"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ARHGEF7"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ATL1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ANK3"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "NTRK2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CDIP1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "FOXP2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "HNRNPK"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "HOPX"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ZMYND8"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_segment()`).
Removed 1 rows containing missing values (`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ST7"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "TRAPPC9"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "MRTFB"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "EXOC6"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SPOP"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "UBE3A"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "RTN4"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PDE1C"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "HNRNPF"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "NEO1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "RANBP9"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SLC16A9"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "TRIM46"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CAMK2B"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "NCKAP5"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PLAUR"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "TRIM58"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ZMYND11"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SET"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "FGF13"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "TCF4"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PIK3R1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "KLHL35"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SETBP1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "TBC1D5"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "VAMP2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "NADK"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "AFF3"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SPEN"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "GCA"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PSD3"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "EXOC6B"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PPP3CA"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_segment()`).
Removed 1 rows containing missing values (`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SNX5"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SLC1A2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ABLIM2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "JADE1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "STX1A"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "DNMT3A"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ZNF138"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "RAP1GDS1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "IP6K2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CADM2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PSMC6"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "NELFCD"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "TCF20"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CAMK4"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "NIPA2"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values (`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values (`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CACNB4"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "TOP3B"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PCDH11X"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PAK1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "TAOK2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "FADS1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "RPP14"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "RAB11FIP5"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ABL1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "F8"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CPEB4"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PCDHA6"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ANKS1B"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "MAP1A"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "DAGLA"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
<section id="plot-genes-with-top-switching-novel-isoforms" class="level2">
<h2 class="anchored" data-anchor-id="plot-genes-with-top-switching-novel-isoforms">Plot genes with top switching novel isoforms</h2>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-33_4c6f445e9268d002e90b70d2914a11f1">
<div class="sourceCode cell-code" id="cb261"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb261-1"><a href="#cb261-1" aria-hidden="true" tabindex="-1"></a>pdf_path <span class="ot">=</span> <span class="st">"output/figures/switch_plots/Fig3_switch_plots_novelIsoforms.pdf"</span></span>
<span id="cb261-2"><a href="#cb261-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(pdf_path)) {</span>
<span id="cb261-3"><a href="#cb261-3" aria-hidden="true" tabindex="-1"></a>  disease_genes <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">'ref/ASD+SCZ+DDD_2022.csv'</span>)</span>
<span id="cb261-4"><a href="#cb261-4" aria-hidden="true" tabindex="-1"></a>  switches <span class="ot">=</span> talonSwitchList_part2<span class="sc">$</span>isoformFeatures <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(cts <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">isoform_id=</span>transcript_id, <span class="at">novelty=</span>novelty2))</span>
<span id="cb261-5"><a href="#cb261-5" aria-hidden="true" tabindex="-1"></a>  interesting_genes <span class="ot">&lt;-</span> switches <span class="sc">%&gt;%</span> <span class="fu">filter</span>(novelty<span class="sc">!=</span><span class="st">"Known"</span> <span class="sc">&amp;</span> isoform_switch_q_value <span class="sc">&lt;</span> <span class="fl">10e-15</span> ) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">arrange</span>(isoform_switch_q_value) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb261-6"><a href="#cb261-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb261-7"><a href="#cb261-7" aria-hidden="true" tabindex="-1"></a>  interesting_genes <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">c</span>(interesting_genes[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>], interesting_genes[interesting_genes<span class="sc">%in%</span> disease_genes<span class="sc">$</span>Gene]))</span>
<span id="cb261-8"><a href="#cb261-8" aria-hidden="true" tabindex="-1"></a>  interesting_genes</span>
<span id="cb261-9"><a href="#cb261-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb261-10"><a href="#cb261-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="at">file =</span> pdf_path)</span>
<span id="cb261-11"><a href="#cb261-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(this_gene <span class="cf">in</span> interesting_genes) {</span>
<span id="cb261-12"><a href="#cb261-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">switchPlot</span>(</span>
<span id="cb261-13"><a href="#cb261-13" aria-hidden="true" tabindex="-1"></a>      talonSwitchList_part2,</span>
<span id="cb261-14"><a href="#cb261-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">gene=</span>this_gene</span>
<span id="cb261-15"><a href="#cb261-15" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb261-16"><a href="#cb261-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(this_gene)</span>
<span id="cb261-17"><a href="#cb261-17" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb261-18"><a href="#cb261-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb261-19"><a href="#cb261-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(isoform_id)`</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ADD1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PKM"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SMARCC2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "KIF1B"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "MYO6"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "INTS10"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SEPTIN2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "VCAN"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "FNBP1L"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ARVCF"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ELAVL2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "MYH10"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CLTC"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ROBO2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "EIF4A2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CADM1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "AKAP9"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "POMGNT1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ZEB2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "EIF4G1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "NCDN"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PSAP"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PTK2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "NF1"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "APBA2"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "KMT2E"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
<p>this is a disgusting hack</p>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-34_14ac7f47104ce67ac6ad48775a1dcbd9">
<div class="sourceCode cell-code" id="cb290"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb290-1"><a href="#cb290-1" aria-hidden="true" tabindex="-1"></a>strp <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">str_extract</span>(x, <span class="st">"^[^</span><span class="sc">\\</span><span class="st">.]*"</span>)</span>
<span id="cb290-2"><a href="#cb290-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb290-3"><a href="#cb290-3" aria-hidden="true" tabindex="-1"></a>strippedSwitchList <span class="ot">=</span> <span class="cf">function</span>(switchList) {</span>
<span id="cb290-4"><a href="#cb290-4" aria-hidden="true" tabindex="-1"></a>  switchList<span class="sc">$</span>isoformFeatures<span class="sc">$</span>isoform_id <span class="ot">&lt;-</span> <span class="fu">strp</span>(switchList<span class="sc">$</span>isoformFeatures<span class="sc">$</span>isoform_id)</span>
<span id="cb290-5"><a href="#cb290-5" aria-hidden="true" tabindex="-1"></a>  switchList[[<span class="st">"exons"</span>]]<span class="sc">@</span>elementMetadata<span class="sc">@</span>listData[[<span class="st">"isoform_id"</span>]] <span class="ot">&lt;-</span> <span class="fu">strp</span>(switchList[[<span class="st">"exons"</span>]]<span class="sc">@</span>elementMetadata<span class="sc">@</span>listData[[<span class="st">"isoform_id"</span>]])</span>
<span id="cb290-6"><a href="#cb290-6" aria-hidden="true" tabindex="-1"></a>  switchList<span class="sc">$</span>isoformCountMatrix<span class="sc">$</span>isoform_id <span class="ot">&lt;-</span> <span class="fu">strp</span>(switchList<span class="sc">$</span>isoformCountMatrix<span class="sc">$</span>isoform_id)</span>
<span id="cb290-7"><a href="#cb290-7" aria-hidden="true" tabindex="-1"></a>  switchList<span class="sc">$</span>isoformRepExpression<span class="sc">$</span>isoform_id <span class="ot">&lt;-</span> <span class="fu">strp</span>(switchList<span class="sc">$</span>isoformRepExpression<span class="sc">$</span>isoform_id)</span>
<span id="cb290-8"><a href="#cb290-8" aria-hidden="true" tabindex="-1"></a>  switchList<span class="sc">$</span>orfAnalysis<span class="sc">$</span>isoform_id <span class="ot">&lt;-</span> <span class="fu">strp</span>(switchList<span class="sc">$</span>orfAnalysis<span class="sc">$</span>isoform_id)</span>
<span id="cb290-9"><a href="#cb290-9" aria-hidden="true" tabindex="-1"></a>  switchList<span class="sc">$</span>isoformSwitchAnalysis<span class="sc">$</span>isoform_id <span class="ot">&lt;-</span> <span class="fu">strp</span>(switchList<span class="sc">$</span>isoformSwitchAnalysis<span class="sc">$</span>isoform_id)</span>
<span id="cb290-10"><a href="#cb290-10" aria-hidden="true" tabindex="-1"></a>  switchList<span class="sc">$</span>domainAnalysis<span class="sc">$</span>isoform_id <span class="ot">&lt;-</span> <span class="fu">strp</span>(switchList<span class="sc">$</span>domainAnalysis<span class="sc">$</span>isoform_id)</span>
<span id="cb290-11"><a href="#cb290-11" aria-hidden="true" tabindex="-1"></a>  switchList<span class="sc">$</span>idrAnalysis<span class="sc">$</span>isoform_id <span class="ot">&lt;-</span> <span class="fu">strp</span>(switchList<span class="sc">$</span>idrAnalysis<span class="sc">$</span>isoform_id)</span>
<span id="cb290-12"><a href="#cb290-12" aria-hidden="true" tabindex="-1"></a>  switchList<span class="sc">$</span>signalPeptideAnalysis<span class="sc">$</span>isoform_id <span class="ot">&lt;-</span> <span class="fu">strp</span>(switchList<span class="sc">$</span>signalPeptideAnalysis<span class="sc">$</span>isoform_id)</span>
<span id="cb290-13"><a href="#cb290-13" aria-hidden="true" tabindex="-1"></a>  switchList<span class="sc">$</span>AlternativeSplicingAnalysis<span class="sc">$</span>isoform_id <span class="ot">&lt;-</span> <span class="fu">strp</span>(switchList<span class="sc">$</span>AlternativeSplicingAnalysis<span class="sc">$</span>isoform_id)</span>
<span id="cb290-14"><a href="#cb290-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb290-15"><a href="#cb290-15" aria-hidden="true" tabindex="-1"></a>  switchList<span class="sc">$</span>switchConsequence<span class="sc">$</span>isoformUpregulated <span class="ot">&lt;-</span> <span class="fu">strp</span>(switchList<span class="sc">$</span>switchConsequence<span class="sc">$</span>isoformUpregulated)</span>
<span id="cb290-16"><a href="#cb290-16" aria-hidden="true" tabindex="-1"></a>  switchList<span class="sc">$</span>switchConsequence<span class="sc">$</span>isoformDownregulated <span class="ot">&lt;-</span> <span class="fu">strp</span>(switchList<span class="sc">$</span>switchConsequence<span class="sc">$</span>isoformDownregulated)</span>
<span id="cb290-17"><a href="#cb290-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(switchList)</span>
<span id="cb290-18"><a href="#cb290-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb290-19"><a href="#cb290-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb290-20"><a href="#cb290-20" aria-hidden="true" tabindex="-1"></a>talonSwitchList_part2_strp <span class="ot">=</span> <span class="fu">strippedSwitchList</span>(talonSwitchList_part2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-35_5cca6ac12d5077dd9421c06e830bb2db">
<div class="sourceCode cell-code" id="cb291"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb291-1"><a href="#cb291-1" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_gray</span>(<span class="at">base_size =</span> <span class="dv">18</span>))</span>
<span id="cb291-2"><a href="#cb291-2" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_update</span>(</span>
<span id="cb291-3"><a href="#cb291-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">"black"</span>),</span>
<span id="cb291-4"><a href="#cb291-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="dv">1</span>), <span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb291-5"><a href="#cb291-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>export at 8 x 6in</p>
<div class="cell" data-hash="Figure3_IsoformSwitchAnalyzeR_cache/html/unnamed-chunk-36_b085d1aa57d870c010b71c31fdb70522">
<div class="sourceCode cell-code" id="cb292"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb292-1"><a href="#cb292-1" aria-hidden="true" tabindex="-1"></a><span class="fu">switchPlot</span>(talonSwitchList_part2_strp, <span class="at">gene =</span> <span class="st">"KMT2E"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb293"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb293-1"><a href="#cb293-1" aria-hidden="true" tabindex="-1"></a><span class="fu">switchPlot</span>(talonSwitchList_part2_strp, <span class="at">gene =</span> <span class="st">"UBE3A"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure3_IsoformSwitchAnalyzeR_files/figure-html/unnamed-chunk-36-2.png" class="img-fluid" width="672"></p>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>