<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Michael Gandal">
<meta name="dcterms.date" content="2023-01-19">

<title>Dev Brain Iso-Seq - Figure 4 - Combined Networks</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-sidebar floating">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Figure 4 - Combined Networks</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Dev Brain Iso-Seq</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/gandallab/Dev_Brain_IsoSeq" title="" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure1_BulkTxomeAnalysis.html" class="sidebar-item-text sidebar-link">Figure 1 - BulkTxomeAnalysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure1_TableS1.html" class="sidebar-item-text sidebar-link">Figure 1 - Table S1</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../code/Fig1_MetaCoverage.html" class="sidebar-item-text sidebar-link">Figure 1 - MetaCoverage</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_isoFraction.html" class="sidebar-item-text sidebar-link">Figure 2 - Isoform Fraction analysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_plotNovelGenes.html" class="sidebar-item-text sidebar-link">Figure 2 - Plot Novel Genes</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_PlotGene.html" class="sidebar-item-text sidebar-link">Figure 2 - Plot One Gene</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_novel_exons.html" class="sidebar-item-text sidebar-link">Figure 2 - Novel Exons</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_NF1Locus.html" class="sidebar-item-text sidebar-link">Figure 2 - NF1 Locus</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_PeptideSupport.html" class="sidebar-item-text sidebar-link">Figure 2 - Peptide Support</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure2_externalSupport.html" class="sidebar-item-text sidebar-link">Figure 2 - External Support</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_IsoformSwitchAnalyzeR.html" class="sidebar-item-text sidebar-link">Figure 3 - IsoformSwitchAnalyzeR</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_DTU_enrichments.html" class="sidebar-item-text sidebar-link">Figure 3 - DTU enrichments</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_RBPEnrichments.html" class="sidebar-item-text sidebar-link">Figure 3 - RBP Enrichments</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure3_APA.html" class="sidebar-item-text sidebar-link">Figure 3 - APA</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure4_combinedNetworks.html" class="sidebar-item-text sidebar-link active">Figure 4 - Combined Networks</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure4_Networks_Enrichments.html" class="sidebar-item-text sidebar-link">Figure 4 - Networks Enrichments</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_SeuratClustering.html" class="sidebar-item-text sidebar-link">Figure 5 - Seurat Clustering</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_MarkerGenesFigs.html" class="sidebar-item-text sidebar-link">Figure 5 - Marker Genes</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_TableS5D.html" class="sidebar-item-text sidebar-link">Figure 5 - Table S5D</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_PseudotimeAnalysis.html" class="sidebar-item-text sidebar-link">Figure 5 - Pseudotime Analysis</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure5_Sankey.html" class="sidebar-item-text sidebar-link">Figure 5 - Sankey</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure6_rareVariant.html" class="sidebar-item-text sidebar-link">Figure 6 - Rare Variant</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure6_VEPConsequences.html" class="sidebar-item-text sidebar-link">Figure 6 - VEP Consequences</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/Figure6_TrackPlots.html" class="sidebar-item-text sidebar-link">Figure 6 - Track Plots</a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/FigureS9_cor_plots.html" class="sidebar-item-text sidebar-link">Figure S9 - Correlation plots</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../analysis/FigureS9_GH.html" class="sidebar-item-text sidebar-link">Figure S9 - G, H</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#load-packages-and-data" id="toc-load-packages-and-data" class="nav-link active" data-scroll-target="#load-packages-and-data">Load Packages and Data</a>
  <ul class="collapse">
  <li><a href="#tables4a" id="toc-tables4a" class="nav-link" data-scroll-target="#tables4a">TableS4A</a></li>
  </ul></li>
  <li><a href="#network-dendrogram-plots" id="toc-network-dendrogram-plots" class="nav-link" data-scroll-target="#network-dendrogram-plots">Network Dendrogram Plots</a>
  <ul class="collapse">
  <li><a href="#fig4a-plot-isoexpr" id="toc-fig4a-plot-isoexpr" class="nav-link" data-scroll-target="#fig4a-plot-isoexpr">Fig4A: Plot IsoExpr</a></li>
  <li><a href="#figs4-supplement-plot-isofr" id="toc-figs4-supplement-plot-isofr" class="nav-link" data-scroll-target="#figs4-supplement-plot-isofr">FigS4: Supplement (Plot IsoFr)</a></li>
  </ul></li>
  <li><a href="#compare-networks" id="toc-compare-networks" class="nav-link" data-scroll-target="#compare-networks">Compare networks</a>
  <ul class="collapse">
  <li><a href="#cell-type-enrichment" id="toc-cell-type-enrichment" class="nav-link" data-scroll-target="#cell-type-enrichment">Cell-type enrichment</a></li>
  <li><a href="#rare-var-enrichments" id="toc-rare-var-enrichments" class="nav-link" data-scroll-target="#rare-var-enrichments">Rare Var Enrichments</a></li>
  <li><a href="#gwas-enrichments" id="toc-gwas-enrichments" class="nav-link" data-scroll-target="#gwas-enrichments">GWAS Enrichments</a></li>
  <li><a href="#rbp-enrichments" id="toc-rbp-enrichments" class="nav-link" data-scroll-target="#rbp-enrichments">RBP Enrichments</a></li>
  </ul></li>
  <li><a href="#annotate-individual-modules" id="toc-annotate-individual-modules" class="nav-link" data-scroll-target="#annotate-individual-modules">Annotate Individual Modules</a>
  <ul class="collapse">
  <li><a href="#isoexpr-network" id="toc-isoexpr-network" class="nav-link" data-scroll-target="#isoexpr-network">IsoExpr Network</a></li>
  <li><a href="#isousage-network" id="toc-isousage-network" class="nav-link" data-scroll-target="#isousage-network">IsoUsage Network</a></li>
  </ul></li>
  <li><a href="#combined-network-plots" id="toc-combined-network-plots" class="nav-link" data-scroll-target="#combined-network-plots">Combined Network Plots</a>
  <ul class="collapse">
  <li><a href="#network-cluster" id="toc-network-cluster" class="nav-link" data-scroll-target="#network-cluster">Network Cluster</a></li>
  <li><a href="#module-umap" id="toc-module-umap" class="nav-link" data-scroll-target="#module-umap">Module UMAP</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Figure 4 - Combined Networks</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Michael Gandal </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 19, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="load-packages-and-data" class="level2">
<h2 class="anchored" data-anchor-id="load-packages-and-data">Load Packages and Data</h2>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-1_75ccc345989f5523e6452ad2c7a52768">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">'code/fisher_overlap.R'</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(edgeR)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(WGCNA)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(biomaRt)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>colorVector <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Known"</span> <span class="ot">=</span> <span class="st">"#009E73"</span>,</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ISM"</span>   <span class="ot">=</span> <span class="st">"#0072B2"</span>,</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ISM_Prefix"</span> <span class="ot">=</span> <span class="st">"#005996"</span>,</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ISM_Suffix"</span> <span class="ot">=</span> <span class="st">"#378bcc"</span>,</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">"NIC"</span>   <span class="ot">=</span> <span class="st">"#D55E00"</span>,</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">"NNC"</span>   <span class="ot">=</span> <span class="st">"#E69F00"</span>,</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Other"</span> <span class="ot">=</span> <span class="st">"#000000"</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>colorVector_ismSplit <span class="ot">=</span> colorVector[<span class="sc">-</span><span class="dv">2</span>]</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"ref/EWCE/CellTypeData_DamonNeuralFetalOnly.rda"</span>)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>celltypemarkers <span class="ot">&lt;-</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="st">'https://www.cell.com/cms/10.1016/j.neuron.2019.06.011/attachment/508ae008-b926-487a-b871-844a12acc1f8/mmc5'</span>, <span class="at">sheet=</span><span class="st">'Cluster enriched genes'</span>) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>celltypemarkers_tableS5 <span class="ot">=</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="st">'https://ars.els-cdn.com/content/image/1-s2.0-S0896627319305616-mmc6.xlsx'</span>,<span class="at">sheet=</span><span class="dv">2</span>)</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>celltypemarkers.bg <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"ref/polioudakis_neuron2020/single_cell_background_MJG221228.csv"</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensembl_gene_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>celltypemarkers.bg <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">c</span>(celltypemarkers.bg, celltypemarkers<span class="sc">$</span>Ensembl))</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">"data/cp_vz_0.75_min_7_recovery_talon_abundance_filtered.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 214516 Columns: 35
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (7): annot_gene_id, annot_transcript_id, annot_gene_name, annot_transcr...
dbl (28): gene_ID, transcript_ID, n_exons, length, 209_1_VZ, 209_2_VZ, 209_3...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>datExpr.isoCounts <span class="ot">=</span> <span class="fu">as.data.frame</span>(cts[,<span class="dv">12</span><span class="sc">:</span><span class="dv">35</span>])</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(datExpr.isoCounts) <span class="ot">=</span> cts<span class="sc">$</span>annot_transcript_id</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>datMeta <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">sample=</span><span class="fu">colnames</span>(datExpr.isoCounts))</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>datMeta<span class="sc">$</span>Region <span class="ot">=</span> <span class="fu">substr</span>(datMeta<span class="sc">$</span>sample, <span class="dv">7</span>,<span class="dv">9</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>datMeta<span class="sc">$</span>Subject <span class="ot">=</span> <span class="fu">substr</span>(datMeta<span class="sc">$</span>sample, <span class="dv">1</span>,<span class="dv">3</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>datMeta<span class="sc">$</span>batch <span class="ot">=</span> <span class="fu">substr</span>(datMeta<span class="sc">$</span>sample, <span class="dv">5</span>,<span class="dv">5</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>datAnno <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">gene_id =</span> cts<span class="sc">$</span>annot_gene_id, <span class="at">transcript_id=</span>cts<span class="sc">$</span>annot_transcript_id,</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">transcript_name=</span>cts<span class="sc">$</span>annot_transcript_name, <span class="at">length=</span>cts<span class="sc">$</span>length,</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">gene_name =</span> cts<span class="sc">$</span>annot_gene_name, <span class="at">ensg =</span> <span class="fu">substr</span>(cts<span class="sc">$</span>annot_gene_id,<span class="dv">1</span>,<span class="dv">15</span>), <span class="at">novelty=</span>cts<span class="sc">$</span>transcript_novelty)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>rbp_targets <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"ref/RBP_targets.csv"</span>)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>rbp_targets<span class="sc">$</span>regulation[<span class="fu">is.na</span>(rbp_targets<span class="sc">$</span>regulation)]<span class="ot">=</span><span class="st">""</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>mart <span class="ot">=</span> <span class="fu">useMart</span>(<span class="st">"ENSEMBL_MART_ENSEMBL"</span>,<span class="st">"mmusculus_gene_ensembl"</span>)</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>f <span class="ot">=</span> <span class="fu">listFilters</span>(mart); a <span class="ot">=</span> <span class="fu">listAttributes</span>(mart)</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>featuresToGet <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>, <span class="st">"external_gene_name"</span>, <span class="st">"hsapiens_homolog_ensembl_gene"</span>, <span class="st">"hsapiens_homolog_associated_gene_name"</span>,<span class="st">"hsapiens_homolog_orthology_type"</span>)</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>mouseHumanHomologs <span class="ot">=</span> <span class="fu">getBM</span>(<span class="at">attributes =</span> featuresToGet,<span class="at">mart =</span> mart)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>human_mouse_bg <span class="ot">=</span> mouseHumanHomologs <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hsapiens_homolog_orthology_type <span class="sc">==</span> <span class="st">"ortholog_one2one"</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">"hsapiens_homolog_ensembl_gene"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a><span class="do">## Load 3 Networks</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>datExpr.isoFr <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">'data/working/WGCNA/final/datExpr.localIF_batchCorrected_103k.rds'</span>)</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>net.isoFr <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">'data/working/WGCNA/final/WGCNA_isoformFraction_top103k_signed_sft14_net.rds'</span>)</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>net.isoFr<span class="sc">$</span>colors <span class="ot">=</span> <span class="fu">labels2colors</span>(net.isoFr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(net.isoFr<span class="sc">$</span>colors) <span class="ot">=</span> <span class="fu">names</span>(net.isoFr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>net.isoFr<span class="sc">$</span>module.number <span class="ot">=</span> net.isoFr<span class="sc">$</span>cut2<span class="sc">$</span>labels</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>net.isoFr<span class="sc">$</span>MEs <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(<span class="fu">t</span>(datExpr.isoFr), <span class="at">colors=</span>net.isoFr<span class="sc">$</span>colors)</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>net.isoFr<span class="sc">$</span>kMEtable <span class="ot">=</span> <span class="fu">signedKME</span>(<span class="fu">t</span>(datExpr.isoFr), <span class="at">datME =</span>net.isoFr<span class="sc">$</span>MEs<span class="sc">$</span>eigengenes,<span class="at">corFnc =</span> <span class="st">"bicor"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bicor(datExpr, datME, , use = "p"): bicor: zero MAD in variable 'x'.
Pearson correlation was used for individual columns with zero (or missing) MAD.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>datExpr.isoExpr <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="at">file=</span><span class="st">'data/working/WGCNA/final/datExpr.isoExpr_batchCorrected_92k.rds'</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>net.isoExpr <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">'data/working/WGCNA/final/WGCNA_isoExpression_top92k_signed_sft9_net.rds'</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>net.isoExpr<span class="sc">$</span>colors <span class="ot">=</span> <span class="fu">labels2colors</span>(net.isoExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(net.isoExpr<span class="sc">$</span>colors) <span class="ot">=</span> <span class="fu">names</span>(net.isoExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>net.isoExpr<span class="sc">$</span>module.number <span class="ot">=</span> net.isoExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>net.isoExpr<span class="sc">$</span>MEs <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(<span class="fu">t</span>(datExpr.isoExpr), <span class="at">colors=</span>net.isoExpr<span class="sc">$</span>colors)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>net.isoExpr<span class="sc">$</span>kMEtable <span class="ot">=</span> <span class="fu">signedKME</span>(<span class="fu">t</span>(datExpr.isoExpr), <span class="at">datME =</span>net.isoExpr<span class="sc">$</span>MEs<span class="sc">$</span>eigengenes,<span class="at">corFnc =</span> <span class="st">"bicor"</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>datExpr.geneExpr <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="at">file=</span><span class="st">'data/working/WGCNA/final/datExpr.geneExpr_batchCorrected_16k.rds'</span>)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>net.geneExpr <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="at">file=</span><span class="st">'data/working/WGCNA/final/WGCNA_geneExpression_top16k_signed_sft14_net.rds'</span>)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>net.geneExpr<span class="sc">$</span>colors <span class="ot">=</span> <span class="fu">labels2colors</span>(net.geneExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(net.geneExpr<span class="sc">$</span>colors) <span class="ot">=</span> <span class="fu">names</span>(net.geneExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>net.geneExpr<span class="sc">$</span>module.number <span class="ot">=</span> net.geneExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>net.geneExpr<span class="sc">$</span>MEs <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(<span class="fu">t</span>(datExpr.geneExpr), <span class="at">colors=</span>net.geneExpr<span class="sc">$</span>colors)</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>net.geneExpr<span class="sc">$</span>kMEtable <span class="ot">=</span> <span class="fu">signedKME</span>(<span class="fu">t</span>(datExpr.geneExpr), <span class="at">datME =</span>net.geneExpr<span class="sc">$</span>MEs<span class="sc">$</span>eigengenes,<span class="at">corFnc =</span> <span class="st">"bicor"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="tables4a" class="level3">
<h3 class="anchored" data-anchor-id="tables4a">TableS4A</h3>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-2_1b9fe105663b8645d203f90fee04609a">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>tidyMods1 <span class="ot">=</span> <span class="fu">tibble</span>(<span class="at">network =</span> <span class="st">"isoExpr"</span>, <span class="at">transcript_id =</span> <span class="fu">rownames</span>(net.isoExpr<span class="sc">$</span>kMEtable), <span class="at">module.color=</span>net.isoExpr<span class="sc">$</span>colors, <span class="at">module.number =</span> net.isoExpr<span class="sc">$</span>module.number, net.isoExpr<span class="sc">$</span>kMEtable) <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(network, transcript_id,module.color,module.number), <span class="at">names_to =</span> <span class="st">"kME_to_module"</span>, <span class="at">values_to =</span> <span class="st">'kME'</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">kMEtoMod =</span> <span class="fu">gsub</span>(<span class="st">"kME"</span>, <span class="st">""</span>, kME_to_module)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module.color<span class="sc">==</span>kMEtoMod) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>kMEtoMod, <span class="sc">-</span>kME_to_module)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>tidyMods2 <span class="ot">=</span> <span class="fu">tibble</span>(<span class="at">network =</span> <span class="st">"isoUsage"</span>, <span class="at">transcript_id =</span> <span class="fu">rownames</span>(net.isoFr<span class="sc">$</span>kMEtable), <span class="at">module.color=</span>net.isoFr<span class="sc">$</span>colors, <span class="at">module.number =</span> net.isoFr<span class="sc">$</span>module.number, net.isoFr<span class="sc">$</span>kMEtable) <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(network, transcript_id,module.color,module.number), <span class="at">names_to =</span> <span class="st">"kME_to_module"</span>, <span class="at">values_to =</span> <span class="st">'kME'</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">kMEtoMod =</span> <span class="fu">gsub</span>(<span class="st">"kME"</span>, <span class="st">""</span>, kME_to_module)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module.color<span class="sc">==</span>kMEtoMod) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>kMEtoMod, <span class="sc">-</span>kME_to_module)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> <span class="fu">rbind</span>(tidyMods1, tidyMods2) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">module=</span><span class="fu">paste0</span>(network, <span class="st">".M"</span>, module.number, <span class="st">"_"</span>, module.color))</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(datAnno <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id, transcript_id, gene_name))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "transcript_id"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>tidyMods3 <span class="ot">=</span> <span class="fu">tibble</span>(<span class="at">network =</span> <span class="st">"geneExpr"</span>, <span class="at">gene_id =</span> <span class="fu">rownames</span>(net.geneExpr<span class="sc">$</span>kMEtable), <span class="at">module.color=</span>net.geneExpr<span class="sc">$</span>colors, <span class="at">module.number =</span> net.geneExpr<span class="sc">$</span>module.number, net.geneExpr<span class="sc">$</span>kMEtable) <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(network, gene_id,module.color,module.number), <span class="at">names_to =</span> <span class="st">"kME_to_module"</span>, <span class="at">values_to =</span> <span class="st">'kME'</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">kMEtoMod =</span> <span class="fu">gsub</span>(<span class="st">"kME"</span>, <span class="st">""</span>, kME_to_module)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module.color<span class="sc">==</span>kMEtoMod) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>kMEtoMod, <span class="sc">-</span>kME_to_module) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">module=</span><span class="fu">paste0</span>(network, <span class="st">".M"</span>, module.number, <span class="st">"_"</span>, module.color), <span class="at">transcript_id=</span><span class="cn">NA</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>tidyMods3<span class="sc">$</span>gene_name <span class="ot">=</span> datAnno<span class="sc">$</span>gene_name[<span class="fu">match</span>(tidyMods3<span class="sc">$</span>gene_id, datAnno<span class="sc">$</span>gene_id)]</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">=</span> <span class="fu">rbind</span>(tidyMods, tidyMods3)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">hub.rank =</span> <span class="fu">rank</span>(<span class="sc">-</span>kME)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">hub =</span> (hub.rank <span class="sc">&lt;=</span> <span class="dv">25</span>), <span class="at">ensg =</span> <span class="fu">substr</span>(gene_id,<span class="dv">1</span>,<span class="dv">15</span>))</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(tidyMods1, tidyMods2,tidyMods3)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(tidyMods,<span class="at">file=</span><span class="st">'output/tables/TableS4A_networks.tsv'</span>)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(tidyMods <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensg, module),<span class="at">file=</span><span class="st">'output/tables/TableS4A_networks_forLDSC.tsv'</span>)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="co"># write_tsv(tidyMods %&gt;% dplyr::select(gene_name, module),file='~/Downloads/TableS4A_networks_forTransite.tsv')</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="co"># write_tsv(tidyMods %&gt;% ungroup() %&gt;% dplyr::select(gene_name) %&gt;% unique(),file='~/Downloads/TableS4A_networks_forTransite_background.tsv')</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>tidyMods_all <span class="ot">=</span> tidyMods</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="network-dendrogram-plots" class="level2">
<h2 class="anchored" data-anchor-id="network-dendrogram-plots">Network Dendrogram Plots</h2>
<section id="fig4a-plot-isoexpr" class="level3">
<h3 class="anchored" data-anchor-id="fig4a-plot-isoexpr">Fig4A: Plot IsoExpr</h3>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-3_bd34aa6a852a48e026edb428cd27c514">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>this_tree <span class="ot">=</span> net.isoExpr<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]]</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>this_anno <span class="ot">=</span> datAnno[<span class="fu">match</span>(<span class="fu">rownames</span>(datExpr.isoExpr), datAnno<span class="sc">$</span>transcript_id),]</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>these_mods <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">IsoExpr=</span>net.isoExpr<span class="sc">$</span>colors)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>IsoFR <span class="ot">=</span> net.isoFr<span class="sc">$</span>colors[this_anno<span class="sc">$</span>transcript_id]</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>GeneExpr <span class="ot">=</span> net.geneExpr<span class="sc">$</span>colors[this_anno<span class="sc">$</span>gene_id]</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>cell_anno <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="st">'white'</span>, <span class="at">nrow=</span><span class="fu">nrow</span>(this_anno), <span class="at">ncol =</span> <span class="fu">length</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))))</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cell_anno) <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster)) {</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>    marker_genes <span class="ot">=</span> celltypemarkers <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Ensembl) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>    cell_anno[this_anno<span class="sc">$</span>ensg <span class="sc">%in%</span> marker_genes, this_cell] <span class="ot">=</span> <span class="st">"red"</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>ndd <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"ref/ASD+SCZ+DDD_2022.csv"</span>)</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>ndd <span class="ot">&lt;-</span> ndd <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Set <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"ASD (Satterstrom et al. 2019)"</span>, <span class="st">"ASD (SFARI score 1or2)"</span>, <span class="st">"ASD (SFARI syndromic)"</span>, <span class="st">"DDD (Kaplanis et al. 2019)"</span>, <span class="st">"SCZ_SCHEMA genes (FDR &lt; 5%)"</span>)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Gene) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>NDD <span class="ot">=</span> <span class="st">'white'</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>NDD[this_anno<span class="sc">$</span>gene_name <span class="sc">%in%</span> ndd] <span class="ot">=</span> <span class="st">'purple'</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>rbpcolors <span class="ot">=</span> <span class="fu">table</span>(rbp_targets<span class="sc">$</span>ENSG)[this_anno<span class="sc">$</span>ensg]</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>RBP_targets <span class="ot">=</span> <span class="fu">numbers2colors</span>(rbpcolors,<span class="at">signed =</span> T,<span class="at">centered =</span> F, <span class="at">colors=</span><span class="fu">blueWhiteRed</span>(<span class="dv">100</span>)[<span class="dv">51</span><span class="sc">:</span><span class="dv">1</span>], <span class="at">naColor =</span> <span class="st">"white"</span> )</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_tree, <span class="at">colors=</span><span class="fu">cbind</span>(these_mods), <span class="at">dendroLabels =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_combinedNetworks_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a> <span class="fu">pdf</span>(<span class="at">file =</span> <span class="st">"output/figures/Fig4/Fig4_wgcna.pdf"</span>,<span class="at">width =</span> <span class="dv">8</span>, <span class="at">height=</span><span class="dv">8</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_tree, <span class="at">colors=</span><span class="fu">cbind</span>(these_mods, cell_anno), <span class="at">dendroLabels =</span> F, <span class="at">main=</span><span class="st">""</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file =</span> <span class="st">"output/figures/Fig4/Fig4A.pdf"</span>,<span class="at">width =</span> <span class="dv">8</span>, <span class="at">height=</span><span class="dv">2</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_tree,<span class="at">colors=</span><span class="cn">NULL</span>,<span class="at">dendroLabels =</span> F, <span class="at">main=</span><span class="st">""</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">jpeg</span>(<span class="at">filename =</span> <span class="st">"output/figures/Fig4/Fig4_wgcna.jpeg"</span>,<span class="at">width =</span> <span class="dv">8</span>, <span class="at">height=</span><span class="dv">8</span>, <span class="at">units=</span><span class="st">'in'</span>,<span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_tree, <span class="at">colors=</span><span class="fu">cbind</span>(these_mods, cell_anno), <span class="at">dendroLabels =</span> F, <span class="at">main=</span><span class="st">""</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
<section id="figs4-supplement-plot-isofr" class="level3">
<h3 class="anchored" data-anchor-id="figs4-supplement-plot-isofr">FigS4: Supplement (Plot IsoFr)</h3>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-4_2a78accbd69f4f543b9b18a6875ef831">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>this_tree <span class="ot">=</span> net.isoFr<span class="sc">$</span>tree</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>this_anno <span class="ot">=</span> datAnno[<span class="fu">match</span>(<span class="fu">rownames</span>(datExpr.isoFr), datAnno<span class="sc">$</span>transcript_id),]</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>these_mods <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">IsoFr=</span>net.isoFr<span class="sc">$</span>colors)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>IsoExpr <span class="ot">=</span> net.isoExpr<span class="sc">$</span>colors[this_anno<span class="sc">$</span>transcript_id]</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>GeneExpr <span class="ot">=</span> net.geneExpr<span class="sc">$</span>colors[this_anno<span class="sc">$</span>gene_id]</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>cell_anno <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="st">'white'</span>, <span class="at">nrow=</span><span class="fu">nrow</span>(this_anno), <span class="at">ncol =</span> <span class="fu">length</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))))</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cell_anno) <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster)) {</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>    marker_genes <span class="ot">=</span> celltypemarkers <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Ensembl) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>    cell_anno[this_anno<span class="sc">$</span>ensg <span class="sc">%in%</span> marker_genes, this_cell] <span class="ot">=</span> <span class="st">"red"</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>ndd <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"ref/ASD+SCZ+DDD_2022.csv"</span>)</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>ndd <span class="ot">&lt;-</span> ndd <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Set <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"ASD (Satterstrom et al. 2019)"</span>, <span class="st">"ASD (SFARI score 1or2)"</span>, <span class="st">"ASD (SFARI syndromic)"</span>, <span class="st">"DDD (Kaplanis et al. 2019)"</span>, <span class="st">"SCZ_SCHEMA genes (FDR &lt; 5%)"</span>)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Gene) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>NDD <span class="ot">=</span> <span class="st">'white'</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>these_mods<span class="sc">$</span>NDD[this_anno<span class="sc">$</span>gene_name <span class="sc">%in%</span> ndd] <span class="ot">=</span> <span class="st">'purple'</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_tree, <span class="at">colors=</span><span class="fu">cbind</span>(these_mods), <span class="at">dendroLabels =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_combinedNetworks_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="compare-networks" class="level2">
<h2 class="anchored" data-anchor-id="compare-networks">Compare networks</h2>
<section id="cell-type-enrichment" class="level3">
<h3 class="anchored" data-anchor-id="cell-type-enrichment">Cell-type enrichment</h3>
<section id="ewce" class="level4">
<h4 class="anchored" data-anchor-id="ewce"><del>EWCE</del></h4>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-5_e7d04872f298510ebb91526a0ed3d021">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="cn">FALSE</span>){</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  df_ewce<span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  datAnno<span class="sc">$</span>net.geneExpr <span class="ot">=</span> net.geneExpr<span class="sc">$</span>colors[datAnno<span class="sc">$</span>gene_id]</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  datAnno<span class="sc">$</span>net.isoExpr <span class="ot">=</span> net.isoExpr<span class="sc">$</span>colors[datAnno<span class="sc">$</span>transcript_id]</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  datAnno<span class="sc">$</span>net.isoFr <span class="ot">=</span> net.isoFr<span class="sc">$</span>colors[datAnno<span class="sc">$</span>transcript_id]</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_mod <span class="cf">in</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(datAnno<span class="sc">$</span>net.geneExpr[datAnno<span class="sc">$</span>net.geneExpr<span class="sc">!=</span><span class="st">"grey"</span>]))) {</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>    mod_genes <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(datAnno<span class="sc">$</span>gene_name[datAnno<span class="sc">$</span>net.geneExpr<span class="sc">==</span>this_mod]))</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>    mod_gene.bg <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(datAnno<span class="sc">$</span>gene_name[<span class="sc">!</span><span class="fu">is.na</span>(datAnno<span class="sc">$</span>net.geneExpr)]))</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>    res<span class="ot">=</span>EWCE<span class="sc">::</span><span class="fu">bootstrap_enrichment_test</span>(ctd, <span class="at">hits =</span> mod_genes, <span class="at">bg=</span>mod_gene.bg,<span class="at">genelistSpecies =</span> <span class="st">'human'</span>,<span class="at">sctSpecies =</span> <span class="st">'human'</span>,<span class="at">annotLevel =</span> <span class="dv">2</span>,<span class="at">verbose =</span> F,<span class="at">no_cores =</span> <span class="dv">10</span>,<span class="at">reps =</span> <span class="dv">10000</span>)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>      df_ewce <span class="ot">=</span> <span class="fu">rbind</span>(df_ewce,<span class="fu">data.frame</span>(<span class="at">net =</span> <span class="st">'net.geneExpr'</span>, <span class="at">mod=</span>this_mod, res<span class="sc">$</span>results))</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_mod <span class="cf">in</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(datAnno<span class="sc">$</span>net.isoExpr[datAnno<span class="sc">$</span>net.isoExpr<span class="sc">!=</span><span class="st">"grey"</span>]))) {</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>    mod_genes <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(datAnno<span class="sc">$</span>gene_name[datAnno<span class="sc">$</span>net.isoExpr<span class="sc">==</span>this_mod]))</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>    mod_gene.bg <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(datAnno<span class="sc">$</span>gene_name[<span class="sc">!</span><span class="fu">is.na</span>(datAnno<span class="sc">$</span>net.isoExpr)]))</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>    res<span class="ot">=</span>EWCE<span class="sc">::</span><span class="fu">bootstrap_enrichment_test</span>(ctd, <span class="at">hits =</span> mod_genes, <span class="at">bg=</span>mod_gene.bg,<span class="at">genelistSpecies =</span> <span class="st">'human'</span>,<span class="at">sctSpecies =</span> <span class="st">'human'</span>,<span class="at">annotLevel =</span> <span class="dv">2</span>,<span class="at">verbose =</span> F,<span class="at">no_cores =</span> <span class="dv">10</span>,<span class="at">reps =</span> <span class="dv">10000</span>)</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>      df_ewce <span class="ot">=</span> <span class="fu">rbind</span>(df_ewce,<span class="fu">data.frame</span>(<span class="at">net =</span> <span class="st">'net.isoExpr'</span>, <span class="at">mod=</span>this_mod, res<span class="sc">$</span>results))</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_mod <span class="cf">in</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(datAnno<span class="sc">$</span>net.isoFr[datAnno<span class="sc">$</span>net.isoFr<span class="sc">!=</span><span class="st">"grey"</span>]))) {</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>    mod_genes <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(datAnno<span class="sc">$</span>gene_name[datAnno<span class="sc">$</span>net.isoFr<span class="sc">==</span>this_mod]))</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>    mod_gene.bg <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(datAnno<span class="sc">$</span>gene_name[<span class="sc">!</span><span class="fu">is.na</span>(datAnno<span class="sc">$</span>net.isoFr)]))</span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>    res<span class="ot">=</span>EWCE<span class="sc">::</span><span class="fu">bootstrap_enrichment_test</span>(ctd, <span class="at">hits =</span> mod_genes, <span class="at">bg=</span>mod_gene.bg,<span class="at">genelistSpecies =</span> <span class="st">'human'</span>,<span class="at">sctSpecies =</span> <span class="st">'human'</span>,<span class="at">annotLevel =</span> <span class="dv">2</span>,<span class="at">verbose =</span> F,<span class="at">no_cores =</span> <span class="dv">10</span>,<span class="at">reps =</span> <span class="dv">10000</span>)</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>      df_ewce <span class="ot">=</span> <span class="fu">rbind</span>(df_ewce,<span class="fu">data.frame</span>(<span class="at">net =</span> <span class="st">'net.isoFr'</span>, <span class="at">mod=</span>this_mod, res<span class="sc">$</span>results))</span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>  df_ewce<span class="sc">$</span>label <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>  df_ewce<span class="sc">$</span>label[df_ewce<span class="sc">$</span>q<span class="sc">&lt;</span>.<span class="dv">05</span>] <span class="ot">=</span> <span class="fu">signif</span>((df_ewce<span class="sc">$</span>sd_from_mean[df_ewce<span class="sc">$</span>q<span class="sc">&lt;</span>.<span class="dv">05</span>]),<span class="dv">1</span>)</span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(df_ewce, <span class="fu">aes</span>(<span class="at">y=</span>mod, <span class="at">x=</span>CellType, <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(p<span class="fl">+1e-5</span>))) <span class="sc">+</span> <span class="fu">geom_tile</span>(<span class="at">color=</span><span class="st">'grey'</span>) <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>, <span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="sc">-</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">0</span>)) <span class="sc">+</span> <span class="fu">facet_grid</span>(net<span class="sc">~</span>., <span class="at">scales=</span><span class="st">'free'</span>,<span class="at">space =</span> <span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">""</span>)</span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>  df_ewce<span class="sc">$</span>mod2 <span class="ot">=</span> <span class="fu">paste0</span>(df_ewce<span class="sc">$</span>net, <span class="st">'.'</span>, df_ewce<span class="sc">$</span>mod)</span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>  df_ewce <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span>  <span class="fu">mutate</span>(<span class="at">ct =</span> ((sd_from_mean))) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(net) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="fu">mean</span>(ct), <span class="fu">median</span>(ct), <span class="fu">max</span>(ct))</span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="fishers-exact" class="level4">
<h4 class="anchored" data-anchor-id="fishers-exact">Fisher’s exact</h4>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-6_df0b8dd002db24b228c6d7b2d4bf100b">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>networks <span class="ot">=</span> <span class="fu">list</span>(<span class="st">"geneExpr"</span> <span class="ot">=</span> net.geneExpr<span class="sc">$</span>module.number[datAnno<span class="sc">$</span>gene_id], </span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>                <span class="st">"isoExpr"</span> <span class="ot">=</span> net.isoExpr<span class="sc">$</span>module.number[datAnno<span class="sc">$</span>transcript_id], </span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>                <span class="st">"isoUsage"</span> <span class="ot">=</span> net.isoFr<span class="sc">$</span>module.number[datAnno<span class="sc">$</span>transcript_id])</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_net <span class="cf">in</span> <span class="fu">names</span>(networks)) {</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  all_mods <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(networks[[this_net]][networks[[this_net]] <span class="sc">!=</span> <span class="dv">0</span>]))</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_mod <span class="cf">in</span> all_mods) {</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>    mod_genes <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(datAnno<span class="sc">$</span>ensg[networks[[this_net]]<span class="sc">==</span>this_mod]))</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    mod_gene.bg <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(datAnno<span class="sc">$</span>ensg[<span class="sc">!</span><span class="fu">is.na</span>(networks[[this_net]])]))</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster)) {</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>      marker_genes <span class="ot">=</span> celltypemarkers <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> </span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(Ensembl) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>      enrichment <span class="ot">=</span> <span class="fu">ORA</span>(mod_genes, marker_genes, mod_gene.bg, celltypemarkers.bg)</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>      df_fisher.celltype <span class="ot">=</span> <span class="fu">rbind</span>(df_fisher.celltype, </span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">data.frame</span>(<span class="at">net =</span> this_net, <span class="at">mod =</span> this_mod, <span class="at">cell=</span>this_cell,</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">as.data.frame</span>(<span class="fu">t</span>(enrichment))))</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(col <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"OR"</span>, <span class="st">"Fisher.p"</span>)) df_fisher.celltype[,col] <span class="ot">=</span> <span class="fu">as.numeric</span>(df_fisher.celltype[,col])</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype<span class="sc">$</span>Fisher.p[df_fisher.celltype<span class="sc">$</span>OR<span class="sc">&lt;</span><span class="dv">1</span>] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype<span class="sc">$</span>Fisher.q <span class="ot">=</span> <span class="fu">p.adjust</span>(df_fisher.celltype<span class="sc">$</span>Fisher.p,<span class="st">'fdr'</span>)</span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype<span class="sc">$</span>label <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype<span class="sc">$</span>label[df_fisher.celltype<span class="sc">$</span>Fisher.q<span class="sc">&lt;</span>.<span class="dv">05</span>] <span class="ot">=</span> <span class="fu">signif</span>((df_fisher.celltype<span class="sc">$</span>OR[df_fisher.celltype<span class="sc">$</span>Fisher.q<span class="sc">&lt;</span>.<span class="dv">05</span>]),<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="figs4" class="level5">
<h5 class="anchored" data-anchor-id="figs4">FigS4</h5>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-7_26eb4b32e8deaa2bef09f3b00ec5aafb">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>df_fisher.celltype<span class="sc">$</span>net <span class="ot">=</span> <span class="fu">factor</span>(df_fisher.celltype<span class="sc">$</span>net, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"geneExpr"</span>, <span class="st">"isoUsage"</span>, <span class="st">"isoExpr"</span>))</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>g1<span class="ot">=</span> <span class="fu">ggplot</span>(df_fisher.celltype, <span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(mod,<span class="at">levels=</span><span class="fu">c</span>(<span class="dv">100</span><span class="sc">:</span><span class="dv">1</span>)), <span class="at">x=</span>cell, <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.q))) <span class="sc">+</span> <span class="fu">geom_tile</span>(<span class="at">color=</span><span class="st">'grey'</span>) <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>, <span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="sc">-</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">0</span>)) <span class="sc">+</span> <span class="fu">facet_grid</span>(net<span class="sc">~</span>., <span class="at">scales=</span><span class="st">'free'</span>,<span class="at">space =</span> <span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">""</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>g1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_combinedNetworks_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(g1, <span class="at">file=</span><span class="st">"output/figures/Fig4/Fig4_supp_Fisher.pdf"</span>,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height=</span><span class="dv">12</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="tables4b" class="level5">
<h5 class="anchored" data-anchor-id="tables4b">TableS4B</h5>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-8_f6eec99dafe684376ae47c118066e076">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>TableS4B <span class="ot">&lt;-</span> df_fisher.celltype<span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">module.number =</span> <span class="fu">as.numeric</span>(mod), <span class="at">module.color =</span> <span class="fu">labels2colors</span>(module.number), <span class="at">mod =</span> <span class="fu">paste0</span>(net, <span class="st">"M."</span>, module.number, <span class="st">"_"</span>, module.color))</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(TableS4B, <span class="at">file=</span><span class="st">"output/tables/TableS4B_fisherCellType.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="single-cell-isoseq-overlap" class="level4">
<h4 class="anchored" data-anchor-id="single-cell-isoseq-overlap">Single Cell IsoSeq Overlap</h4>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-9_b9955e366888eb77a992c2aad73b283b">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>tableS5 <span class="ot">=</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="st">'data/single_cell/Table_S5_v2.xlsx'</span>,<span class="at">sheet =</span> <span class="st">"C"</span>) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>mod_bg <span class="ot">=</span> tidyMods_all <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(transcript_id) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">1</span>,<span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Adding missing grouping variables: `module`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>df_scOverlap <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>tidyMods_all <span class="ot">&lt;-</span> tidyMods_all <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(tableS5<span class="sc">$</span>cluster)){</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  cell_markers <span class="ot">=</span> tableS5 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Isoform.Id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_mod <span class="cf">in</span> tidyMods_all <span class="sc">%&gt;%</span> <span class="fu">filter</span>(network <span class="sc">!=</span> <span class="st">'geneExpr'</span>, module.color <span class="sc">!=</span><span class="st">'grey'</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(module) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>()) {</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>    mod_isoforms <span class="ot">=</span> tidyMods_all <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module<span class="sc">==</span>this_mod) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(transcript_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">substr</span>(<span class="dv">1</span>,<span class="dv">15</span>)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>    this_stat <span class="ot">=</span> <span class="fu">ORA</span>(mod_isoforms,cell_markers, mod_bg,mod_bg)</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>    df_scOverlap <span class="ot">=</span> <span class="fu">rbind</span>(df_scOverlap, <span class="fu">data.frame</span>(<span class="at">cell=</span>this_cell, <span class="at">mod=</span>this_mod, <span class="fu">t</span>(this_stat)))</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(col <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"OR"</span>, <span class="st">"Fisher.p"</span>)) df_scOverlap[,col] <span class="ot">=</span> <span class="fu">as.numeric</span>(df_scOverlap[,col])</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>df_scOverlap<span class="sc">$</span>Fisher.p[df_scOverlap<span class="sc">$</span>OR<span class="sc">&lt;</span><span class="dv">1</span>] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>df_scOverlap<span class="sc">$</span>Fisher.q <span class="ot">=</span> <span class="fu">p.adjust</span>(df_scOverlap<span class="sc">$</span>Fisher.p,<span class="st">'fdr'</span>)</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>df_scOverlap<span class="sc">$</span>label <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>df_scOverlap<span class="sc">$</span>label[df_scOverlap<span class="sc">$</span>Fisher.q<span class="sc">&lt;</span>.<span class="dv">05</span>] <span class="ot">=</span> <span class="fu">signif</span>((df_scOverlap<span class="sc">$</span>OR[df_scOverlap<span class="sc">$</span>Fisher.q<span class="sc">&lt;</span>.<span class="dv">05</span>]),<span class="dv">1</span>)</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>df_scOverlap<span class="sc">$</span>net <span class="ot">=</span> <span class="fu">factor</span>(<span class="fu">substr</span>(df_scOverlap<span class="sc">$</span>mod,<span class="dv">1</span>,<span class="dv">6</span>))</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df_scOverlap, <span class="fu">aes</span>(<span class="at">y=</span>mod, <span class="at">x=</span>cell, <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.q))) <span class="sc">+</span> <span class="fu">geom_tile</span>(<span class="at">color=</span><span class="st">'grey'</span>) <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>, <span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="sc">-</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">0</span>)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(net<span class="sc">~</span>., <span class="at">scales=</span><span class="st">'free'</span>,<span class="at">shrink =</span>T,<span class="at">ncol=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">coord_flip</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_combinedNetworks_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="rare-var-enrichments" class="level3">
<h3 class="anchored" data-anchor-id="rare-var-enrichments">Rare Var Enrichments</h3>
<section id="isoform-level" class="level4">
<h4 class="anchored" data-anchor-id="isoform-level"><del>Isoform level</del></h4>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-10_f3fc4ead79a7f2d57b07d8fe87637f9f">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="cn">FALSE</span>) {</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  fu<span class="ot">=</span>openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>((<span class="st">'https://static-content.springer.com/esm/art%3A10.1038%2Fs41588-022-01104-0/MediaObjects/41588_2022_1104_MOESM3_ESM.xlsx'</span>),<span class="st">'Supplementary Table 11'</span>)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>fu<span class="sc">$</span>p_TADA_ASD[fu<span class="sc">$</span>p_TADA_ASD<span class="sc">==</span><span class="dv">0</span>] <span class="ot">=</span> <span class="fu">min</span>(fu<span class="sc">$</span>p_TADA_ASD[fu<span class="sc">$</span>p_TADA_ASD <span class="sc">&gt;</span><span class="dv">0</span>],<span class="at">na.rm=</span>T)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>fu<span class="sc">$</span>p_TADA_NDD[fu<span class="sc">$</span>p_TADA_NDD<span class="sc">==</span><span class="dv">0</span>] <span class="ot">=</span> <span class="fu">min</span>(fu<span class="sc">$</span>p_TADA_NDD[fu<span class="sc">$</span>p_TADA_NDD <span class="sc">&gt;</span><span class="dv">0</span>],<span class="at">na.rm=</span>T)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>datAnno.logit <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">ASD_fuTADA=</span> <span class="sc">-</span><span class="fu">log10</span>(fu<span class="sc">$</span>p_TADA_ASD)[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg, fu<span class="sc">$</span>gene_id)])</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>datAnno.logit<span class="sc">$</span>NDD_fuTADA <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(fu<span class="sc">$</span>p_TADA_NDD)[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg, fu<span class="sc">$</span>gene_id)]</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>SCZ.schema <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/SCHEMA_gene_results.tsv'</span>)</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>datAnno.logit<span class="sc">$</span>SCZ_schema <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(SCZ.schema<span class="sc">$</span><span class="st">`</span><span class="at">P meta</span><span class="st">`</span>[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg, SCZ.schema<span class="sc">$</span>gene_id)])</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>BIP.bipex <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/BipEx_gene_results.tsv'</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(group<span class="sc">==</span><span class="st">"Bipolar Disorder"</span>)</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>datAnno.logit<span class="sc">$</span>BIP.bipex <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(BIP.bipex<span class="sc">$</span>ptv_fisher_gnom_non_psych_pval[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg,BIP.bipex<span class="sc">$</span>gene_id)])</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>EPI.epi25 <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/Epi25_gene_results.tsv'</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(group<span class="sc">==</span><span class="st">"EPI"</span>)</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>datAnno.logit<span class="sc">$</span>EPI.epi25 <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(EPI.epi25<span class="sc">$</span>pval[<span class="fu">match</span>(datAnno<span class="sc">$</span>ensg,EPI.epi25<span class="sc">$</span>gene_id)])</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>networks <span class="ot">=</span> <span class="fu">list</span>(<span class="st">"net.isoExpr"</span> <span class="ot">=</span> net.isoExpr<span class="sc">$</span>colors[datAnno<span class="sc">$</span>transcript_id], </span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>                <span class="st">"net.isoFr"</span> <span class="ot">=</span> net.isoFr<span class="sc">$</span>colors[datAnno<span class="sc">$</span>transcript_id])</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>df_logit<span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_net <span class="cf">in</span> <span class="fu">names</span>(networks)) {</span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>  all_mods <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(networks[[this_net]][networks[[this_net]] <span class="sc">!=</span> <span class="st">'grey'</span>]))</span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_mod <span class="cf">in</span> all_mods) {</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>    binaryVec <span class="ot">=</span> <span class="fu">as.numeric</span>(networks[[this_net]] <span class="sc">==</span> this_mod)</span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(this_rare_var <span class="cf">in</span> <span class="fu">colnames</span>(datAnno.logit)) {</span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a>      this_glm <span class="ot">=</span> <span class="fu">summary</span>(<span class="fu">glm</span>(binaryVec <span class="sc">~</span> datAnno.logit[,this_rare_var] <span class="sc">+</span> length <span class="sc">+</span> <span class="fu">log10</span>(length), <span class="at">data=</span>datAnno, <span class="at">family=</span><span class="st">'binomial'</span>))</span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a>    df_logit <span class="ot">=</span> <span class="fu">rbind</span>(df_logit, <span class="fu">data.frame</span>(<span class="at">net =</span> this_net, <span class="at">mod =</span> this_mod, <span class="at">set =</span> this_rare_var, <span class="fu">t</span>(this_glm<span class="sc">$</span>coefficients[<span class="dv">2</span>,])))</span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb25-39"><a href="#cb25-39" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb25-40"><a href="#cb25-40" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb25-41"><a href="#cb25-41" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>OR <span class="ot">=</span> <span class="fu">exp</span>(df_logit<span class="sc">$</span>Estimate)</span>
<span id="cb25-42"><a href="#cb25-42" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>P <span class="ot">=</span> df_logit<span class="sc">$</span>Pr...z..</span>
<span id="cb25-43"><a href="#cb25-43" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>P[df_logit<span class="sc">$</span>OR <span class="sc">&lt;</span> <span class="dv">1</span>] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb25-44"><a href="#cb25-44" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>Q <span class="ot">=</span> <span class="fu">p.adjust</span>(df_logit<span class="sc">$</span>P,<span class="st">'fdr'</span>)</span>
<span id="cb25-45"><a href="#cb25-45" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>mod <span class="ot">=</span> <span class="fu">factor</span>(df_logit<span class="sc">$</span>mod, <span class="at">levels=</span><span class="fu">labels2colors</span>(<span class="dv">100</span><span class="sc">:</span><span class="dv">1</span>))</span>
<span id="cb25-46"><a href="#cb25-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-47"><a href="#cb25-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-48"><a href="#cb25-48" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>label <span class="ot">=</span> <span class="fu">signif</span>((df_logit<span class="sc">$</span>OR),<span class="dv">3</span>)</span>
<span id="cb25-49"><a href="#cb25-49" aria-hidden="true" tabindex="-1"></a>df_logit<span class="sc">$</span>label[df_logit<span class="sc">$</span>Q<span class="sc">&gt;</span>.<span class="dv">1</span>] <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb25-50"><a href="#cb25-50" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df_logit, <span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(mod),<span class="at">x=</span>set, <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(P))) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>, <span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="sc">-</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">0</span>)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>net, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>,<span class="at">y=</span><span class="st">""</span>)</span>
<span id="cb25-51"><a href="#cb25-51" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="gene-level" class="level4">
<h4 class="anchored" data-anchor-id="gene-level">Gene level</h4>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-11_4e4ce999a0ffde1c0c393859665f021d">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>geneAnno <span class="ot">=</span> rtracklayer<span class="sc">::</span><span class="fu">import</span>(<span class="st">"ref/gencode.v33lift37.annotation.gtf.gz"</span>) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(type<span class="sc">==</span><span class="st">'gene'</span>)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>geneAnno<span class="sc">$</span>ensg <span class="ot">=</span> <span class="fu">substr</span>(geneAnno<span class="sc">$</span>gene_id,<span class="dv">1</span>,<span class="dv">15</span>)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>fu<span class="ot">=</span>openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>((<span class="st">'https://static-content.springer.com/esm/art%3A10.1038%2Fs41588-022-01104-0/MediaObjects/41588_2022_1104_MOESM3_ESM.xlsx'</span>),<span class="st">'Supplementary Table 11'</span>)</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>fu<span class="sc">$</span>p_TADA_ASD[fu<span class="sc">$</span>p_TADA_ASD<span class="sc">==</span><span class="dv">0</span>] <span class="ot">=</span> <span class="fu">min</span>(fu<span class="sc">$</span>p_TADA_ASD[fu<span class="sc">$</span>p_TADA_ASD <span class="sc">&gt;</span><span class="dv">0</span>],<span class="at">na.rm=</span>T)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>fu<span class="sc">$</span>p_TADA_NDD[fu<span class="sc">$</span>p_TADA_NDD<span class="sc">==</span><span class="dv">0</span>] <span class="ot">=</span> <span class="fu">min</span>(fu<span class="sc">$</span>p_TADA_NDD[fu<span class="sc">$</span>p_TADA_NDD <span class="sc">&gt;</span><span class="dv">0</span>],<span class="at">na.rm=</span>T)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>geneAnno.logit <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">ASD_fuTADA=</span> <span class="sc">-</span><span class="fu">log10</span>(fu<span class="sc">$</span>p_TADA_ASD)[<span class="fu">match</span>(geneAnno<span class="sc">$</span>ensg, fu<span class="sc">$</span>gene_id)])</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>geneAnno.logit<span class="sc">$</span>NDD_fuTADA <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(fu<span class="sc">$</span>p_TADA_NDD)[<span class="fu">match</span>(geneAnno<span class="sc">$</span>ensg, fu<span class="sc">$</span>gene_id)]</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>SCZ.schema <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/SCHEMA_gene_results.tsv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: One or more parsing issues, see `problems()` for details</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 18321 Columns: 26
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (8): gene_id, group, OR (PTV), OR (Class I), OR (Class II), OR (PTV) up...
dbl (16): Case PTV, Ctrl PTV, Case mis3, Ctrl mis3, Case mis2, Ctrl mis2, P ...
lgl  (2): De novo mis3, De novo mis2

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>geneAnno.logit<span class="sc">$</span>SCZ_schema <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(SCZ.schema<span class="sc">$</span><span class="st">`</span><span class="at">P meta</span><span class="st">`</span>[<span class="fu">match</span>(geneAnno<span class="sc">$</span>ensg, SCZ.schema<span class="sc">$</span>gene_id)])</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>BIP.bipex <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/BipEx_gene_results.tsv'</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(group<span class="sc">==</span><span class="st">"Bipolar Disorder"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 119958 Columns: 20
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (4): gene_id, group, damaging_missense_fisher_gnom_non_psych_OR, ptv_fi...
dbl (16): n_cases, n_controls, damaging_missense_case_count, damaging_missen...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>geneAnno.logit<span class="sc">$</span>BIP.bipex <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(BIP.bipex<span class="sc">$</span>ptv_fisher_gnom_non_psych_pval[<span class="fu">match</span>(geneAnno<span class="sc">$</span>ensg,BIP.bipex<span class="sc">$</span>gene_id)])</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>EPI.epi25 <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">'ref/risk_genes/Epi25_gene_results.tsv'</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(group<span class="sc">==</span><span class="st">"EPI"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 71456 Columns: 12
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): gene_id, group
dbl (9): xcase_lof, xctrl_lof, pval_lof, xcase_mpc, xctrl_mpc, pval_mpc, xca...
lgl (1): pval_infrIndel

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>geneAnno.logit<span class="sc">$</span>EPI.epi25 <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(EPI.epi25<span class="sc">$</span>pval[<span class="fu">match</span>(geneAnno<span class="sc">$</span>ensg,EPI.epi25<span class="sc">$</span>gene_id)])</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>networks <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">net=</span><span class="st">"isoExpr"</span>, <span class="at">module =</span> net.isoExpr<span class="sc">$</span>module.number, <span class="at">transcript_id =</span> <span class="fu">names</span>(net.isoExpr<span class="sc">$</span>colors))</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>networks <span class="ot">=</span> <span class="fu">rbind</span>(networks, <span class="fu">data.frame</span>(<span class="at">net=</span><span class="st">'isoUsage'</span>, <span class="at">module =</span> net.isoFr<span class="sc">$</span>module.number, <span class="at">transcript_id =</span> <span class="fu">names</span>(net.isoFr<span class="sc">$</span>colors)))</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>networks<span class="sc">$</span>gene_id <span class="ot">=</span> datAnno<span class="sc">$</span>gene_id[<span class="fu">match</span>(networks<span class="sc">$</span>transcript_id, datAnno<span class="sc">$</span>transcript_id)]</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>networks <span class="ot">=</span> <span class="fu">rbind</span>(networks, <span class="fu">data.frame</span>(<span class="at">net=</span><span class="st">'geneExpr'</span>, <span class="at">module =</span> net.geneExpr<span class="sc">$</span>module.number, <span class="at">transcript_id =</span> <span class="cn">NA</span>, <span class="at">gene_id=</span><span class="fu">names</span>(net.geneExpr<span class="sc">$</span>colors)))</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>networks<span class="sc">$</span>ensg <span class="ot">=</span> <span class="fu">substr</span>(networks<span class="sc">$</span>gene_id,<span class="dv">1</span>,<span class="dv">15</span>)</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>df_logit.gene <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>binaryVec <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="at">times=</span><span class="fu">nrow</span>(geneAnno))</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(binaryVec) <span class="ot">=</span> geneAnno<span class="sc">$</span>ensg</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_net <span class="cf">in</span> <span class="fu">unique</span>(networks<span class="sc">$</span>net)) {</span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>  all_mods <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(networks<span class="sc">$</span>module[networks<span class="sc">$</span>net <span class="sc">==</span>this_net <span class="sc">&amp;</span> networks<span class="sc">$</span>module<span class="sc">!=</span><span class="dv">0</span>]))</span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_mod <span class="cf">in</span> all_mods) {</span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a>    this_binary_vec <span class="ot">=</span> binaryVec</span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a>    this_binary_vec[<span class="fu">names</span>(this_binary_vec) <span class="sc">%in%</span> networks<span class="sc">$</span>ensg[networks<span class="sc">$</span>net<span class="sc">==</span>this_net]] <span class="ot">=</span> <span class="dv">0</span> </span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a>    this_binary_vec[<span class="fu">names</span>(this_binary_vec) <span class="sc">%in%</span> networks<span class="sc">$</span>ensg[networks<span class="sc">$</span>net<span class="sc">==</span>this_net <span class="sc">&amp;</span> networks<span class="sc">$</span>module<span class="sc">==</span>this_mod]] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(this_rare_var <span class="cf">in</span> <span class="fu">colnames</span>(geneAnno.logit)) {</span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>      this_glm <span class="ot">=</span> <span class="fu">summary</span>(<span class="fu">glm</span>(this_binary_vec <span class="sc">~</span> geneAnno.logit[,this_rare_var] <span class="sc">+</span> width <span class="sc">+</span> <span class="fu">log10</span>(width), <span class="at">data=</span>geneAnno, <span class="at">family=</span><span class="st">'binomial'</span>))</span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a>    df_logit.gene <span class="ot">=</span> <span class="fu">rbind</span>(df_logit.gene, <span class="fu">data.frame</span>(<span class="at">net =</span> this_net, <span class="at">mod =</span> this_mod, <span class="at">set =</span> this_rare_var, <span class="fu">t</span>(this_glm<span class="sc">$</span>coefficients[<span class="dv">2</span>,])))</span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>df_logit.gene<span class="sc">$</span>OR <span class="ot">=</span> <span class="fu">exp</span>(df_logit.gene<span class="sc">$</span>Estimate)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>df_logit.gene<span class="sc">$</span>P <span class="ot">=</span> df_logit.gene<span class="sc">$</span>Pr...z..</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>df_logit.gene<span class="sc">$</span>P[df_logit.gene<span class="sc">$</span>OR <span class="sc">&lt;</span> <span class="dv">1</span>] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>df_logit.gene<span class="sc">$</span>Q <span class="ot">=</span> <span class="fu">p.adjust</span>(df_logit.gene<span class="sc">$</span>P,<span class="st">'fdr'</span>)</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>df_logit.gene<span class="sc">$</span>label <span class="ot">=</span> <span class="fu">signif</span>((df_logit.gene<span class="sc">$</span>OR),<span class="dv">3</span>)</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>df_logit.gene<span class="sc">$</span>label[df_logit.gene<span class="sc">$</span>Q<span class="sc">&gt;</span>.<span class="dv">05</span>] <span class="ot">=</span> <span class="st">""</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="figs4-1" class="level5">
<h5 class="anchored" data-anchor-id="figs4-1">FigS4</h5>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-12_5f41c101f5c4f5d9cddffa645f450732">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>g1 <span class="ot">=</span> <span class="fu">ggplot</span>(df_logit.gene, <span class="fu">aes</span>(<span class="at">y=</span><span class="fu">factor</span>(mod,<span class="at">levels=</span><span class="fu">c</span>(<span class="dv">100</span><span class="sc">:</span><span class="dv">1</span>)),<span class="at">x=</span>set, <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(P))) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>, <span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="sc">-</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">0</span>)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>net, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">""</span>)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>g1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_combinedNetworks_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(g1,<span class="at">file=</span><span class="st">"output/figures/supplement/FigS4_rareVar_geneLogit.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
</div>
</section>
<section id="tables4c" class="level5">
<h5 class="anchored" data-anchor-id="tables4c">TableS4C</h5>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-13_ee96d00f59603b949c9025b1e9eb9d15">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>TableS4C <span class="ot">&lt;-</span> df_logit.gene <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">module=</span><span class="fu">paste0</span>(net, <span class="st">".M"</span>, mod, <span class="st">"_"</span>, <span class="fu">labels2colors</span>(mod)))</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(TableS4C, <span class="at">file=</span><span class="st">"output/tables/TableS4C_rareVarlogit.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="gwas-enrichments" class="level3">
<h3 class="anchored" data-anchor-id="gwas-enrichments">GWAS Enrichments</h3>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-14_89fd7dc3c2f8e5f097c14881213e8444">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>df_gwas <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>gwas <span class="ot">=</span> <span class="fu">dir</span>(<span class="st">"data/LDSC_cell_type_specific/networks/isoExpr_cts_result/"</span>, <span class="at">pattern=</span><span class="st">"txt"</span>)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_gwas <span class="cf">in</span> gwas) {</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>  this_ldsc <span class="ot">=</span> <span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">"data/LDSC_cell_type_specific/networks/isoExpr_cts_result/"</span>,this_gwas),<span class="at">header =</span> T)</span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(this_ldsc)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">'module'</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>  this_ldsc<span class="sc">$</span>gwas <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".cell_type_results.txt"</span>, <span class="st">""</span>, this_gwas)</span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>  this_ldsc<span class="sc">$</span>net <span class="ot">=</span> <span class="st">'isoExpr'</span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>  df_gwas <span class="ot">=</span> <span class="fu">rbind</span>(df_gwas, this_ldsc)</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>gwas <span class="ot">=</span> <span class="fu">dir</span>(<span class="st">"data/LDSC_cell_type_specific/networks/isoUsage_cts_result/"</span>, <span class="at">pattern=</span><span class="st">"txt"</span>)</span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_gwas <span class="cf">in</span> gwas) {</span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a>  this_ldsc <span class="ot">=</span> <span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">"data/LDSC_cell_type_specific/networks/isoUsage_cts_result/"</span>,this_gwas),<span class="at">header =</span> T)</span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(this_ldsc)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">'module'</span></span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>  this_ldsc<span class="sc">$</span>gwas <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".cell_type_results.txt"</span>, <span class="st">""</span>, this_gwas)</span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>  this_ldsc<span class="sc">$</span>net <span class="ot">=</span> <span class="st">'isoUsage'</span></span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a>  df_gwas <span class="ot">=</span> <span class="fu">rbind</span>(df_gwas, this_ldsc)</span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a>gwas <span class="ot">=</span> <span class="fu">dir</span>(<span class="st">"data/LDSC_cell_type_specific/networks/geneExpr_cts_result/"</span>, <span class="at">pattern=</span><span class="st">"txt"</span>)</span>
<span id="cb41-21"><a href="#cb41-21" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_gwas <span class="cf">in</span> gwas) {</span>
<span id="cb41-22"><a href="#cb41-22" aria-hidden="true" tabindex="-1"></a>  this_ldsc <span class="ot">=</span> <span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">"data/LDSC_cell_type_specific/networks/geneExpr_cts_result/"</span>,this_gwas),<span class="at">header =</span> T)</span>
<span id="cb41-23"><a href="#cb41-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(this_ldsc)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">'module'</span></span>
<span id="cb41-24"><a href="#cb41-24" aria-hidden="true" tabindex="-1"></a>  this_ldsc<span class="sc">$</span>gwas <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".cell_type_results.txt"</span>, <span class="st">""</span>, this_gwas)</span>
<span id="cb41-25"><a href="#cb41-25" aria-hidden="true" tabindex="-1"></a>  this_ldsc<span class="sc">$</span>net <span class="ot">=</span> <span class="st">'geneExpr'</span></span>
<span id="cb41-26"><a href="#cb41-26" aria-hidden="true" tabindex="-1"></a>  df_gwas <span class="ot">=</span> <span class="fu">rbind</span>(df_gwas, this_ldsc)</span>
<span id="cb41-27"><a href="#cb41-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb41-28"><a href="#cb41-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-29"><a href="#cb41-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-30"><a href="#cb41-30" aria-hidden="true" tabindex="-1"></a>df_gwas<span class="sc">$</span>fdr <span class="ot">=</span> <span class="fu">p.adjust</span>(df_gwas<span class="sc">$</span>Coefficient_P_value,<span class="st">'fdr'</span>)</span>
<span id="cb41-31"><a href="#cb41-31" aria-hidden="true" tabindex="-1"></a>df_gwas<span class="sc">$</span>mod <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">"M"</span>, <span class="st">""</span>, <span class="fu">unlist</span>(<span class="fu">lapply</span>(<span class="fu">strsplit</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(<span class="fu">strsplit</span>(df_gwas<span class="sc">$</span>module, <span class="st">"[.]"</span>),<span class="st">'['</span>,<span class="dv">2</span>)), <span class="st">'_'</span>),<span class="st">'['</span>,<span class="dv">1</span>))))</span>
<span id="cb41-32"><a href="#cb41-32" aria-hidden="true" tabindex="-1"></a>df_gwas<span class="sc">$</span>Zscore <span class="ot">=</span> df_gwas<span class="sc">$</span>Coefficient<span class="sc">/</span>df_gwas<span class="sc">$</span>Coefficient_std_error</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="rbp-enrichments" class="level3">
<h3 class="anchored" data-anchor-id="rbp-enrichments">RBP Enrichments</h3>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-15_192bd11c9880d4c34f51faf94b2c26d6">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>networks <span class="ot">=</span> <span class="fu">list</span>(<span class="st">"net.isoExpr"</span> <span class="ot">=</span> net.isoExpr<span class="sc">$</span>colors[datAnno<span class="sc">$</span>transcript_id], </span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>                <span class="st">"net.isoFr"</span> <span class="ot">=</span> net.isoFr<span class="sc">$</span>colors[datAnno<span class="sc">$</span>transcript_id],</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>                <span class="st">"net.geneExpr"</span> <span class="ot">=</span> net.geneExpr<span class="sc">$</span>colors[datAnno<span class="sc">$</span>gene_id])</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_net <span class="cf">in</span> <span class="fu">names</span>(networks)) {</span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>  all_mods <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(networks[[this_net]][networks[[this_net]] <span class="sc">!=</span> <span class="st">'grey'</span>]))</span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_mod <span class="cf">in</span> all_mods) {</span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(this_net <span class="sc">==</span> <span class="st">"net.geneExpr"</span>) {</span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>      modGenes <span class="ot">=</span> <span class="fu">substr</span>(<span class="fu">unique</span>(<span class="fu">na.omit</span>(<span class="fu">names</span>(networks[[this_net]][networks[[this_net]] <span class="sc">==</span> this_mod]))), <span class="dv">1</span>,<span class="dv">15</span>)</span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>      modGeneg.bg <span class="ot">=</span> <span class="fu">substr</span>(<span class="fu">unique</span>(<span class="fu">na.omit</span>(<span class="fu">names</span>(networks[[this_net]]))),<span class="dv">1</span>,<span class="dv">15</span>)</span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb42-16"><a href="#cb42-16" aria-hidden="true" tabindex="-1"></a>      modTx <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">na.omit</span>(<span class="fu">names</span>(networks[[this_net]][networks[[this_net]] <span class="sc">==</span> this_mod])))</span>
<span id="cb42-17"><a href="#cb42-17" aria-hidden="true" tabindex="-1"></a>      modGenes <span class="ot">=</span> <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg[<span class="fu">match</span>(modTx, datAnno<span class="sc">$</span>transcript_id)])</span>
<span id="cb42-18"><a href="#cb42-18" aria-hidden="true" tabindex="-1"></a>      modGeneg.bg <span class="ot">=</span> <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg[<span class="fu">match</span>(<span class="fu">names</span>(networks[[this_net]]), datAnno<span class="sc">$</span>transcript_id)])</span>
<span id="cb42-19"><a href="#cb42-19" aria-hidden="true" tabindex="-1"></a>    } </span>
<span id="cb42-20"><a href="#cb42-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(this_dataset <span class="cf">in</span> <span class="fu">unique</span>(rbp_targets<span class="sc">$</span>dataset)) {</span>
<span id="cb42-21"><a href="#cb42-21" aria-hidden="true" tabindex="-1"></a>    this_rbp <span class="ot">=</span> rbp_targets <span class="sc">%&gt;%</span> <span class="fu">filter</span>(dataset <span class="sc">==</span> this_dataset) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">target =</span> <span class="fu">paste0</span>(RBP, <span class="st">"_"</span>, data.type, <span class="st">"_"</span>, regulation, <span class="st">"_"</span>, msORhs)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(target) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()  <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb42-22"><a href="#cb42-22" aria-hidden="true" tabindex="-1"></a>    target_genes <span class="ot">=</span> rbp_targets <span class="sc">%&gt;%</span> <span class="fu">filter</span>(dataset <span class="sc">==</span> this_dataset) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ENSG) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb42-23"><a href="#cb42-23" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb42-24"><a href="#cb42-24" aria-hidden="true" tabindex="-1"></a> <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">"Human"</span>,this_rbp)) {</span>
<span id="cb42-25"><a href="#cb42-25" aria-hidden="true" tabindex="-1"></a>   this_or <span class="ot">=</span> <span class="fu">ORA</span>(modGenes, target_genes, <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg), <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg))</span>
<span id="cb42-26"><a href="#cb42-26" aria-hidden="true" tabindex="-1"></a> } <span class="cf">else</span> {</span>
<span id="cb42-27"><a href="#cb42-27" aria-hidden="true" tabindex="-1"></a>  this_or <span class="ot">=</span> <span class="fu">ORA</span>(modGenes, target_genes, <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg), human_mouse_bg)</span>
<span id="cb42-28"><a href="#cb42-28" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb42-29"><a href="#cb42-29" aria-hidden="true" tabindex="-1"></a> df_fisher_rbp <span class="ot">=</span> <span class="fu">rbind</span>(df_fisher_rbp, <span class="fu">data.frame</span>(<span class="at">net =</span> this_net, <span class="at">mod =</span> this_mod, <span class="at">dataset =</span> this_dataset, <span class="at">target =</span> this_rbp, <span class="fu">t</span>(this_or)))</span>
<span id="cb42-30"><a href="#cb42-30" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb42-31"><a href="#cb42-31" aria-hidden="true" tabindex="-1"></a>}}</span>
<span id="cb42-32"><a href="#cb42-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb42-33"><a href="#cb42-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-34"><a href="#cb42-34" aria-hidden="true" tabindex="-1"></a>df_fisher_rbp<span class="sc">$</span>OR <span class="ot">=</span> <span class="fu">as.numeric</span>(df_fisher_rbp<span class="sc">$</span>OR)</span>
<span id="cb42-35"><a href="#cb42-35" aria-hidden="true" tabindex="-1"></a>  df_fisher_rbp<span class="sc">$</span>Fisher.p[df_fisher_rbp<span class="sc">$</span>OR<span class="sc">&lt;</span><span class="dv">1</span>] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb42-36"><a href="#cb42-36" aria-hidden="true" tabindex="-1"></a>  df_fisher_rbp<span class="sc">$</span>Fisher.p <span class="ot">=</span> <span class="fu">p.adjust</span>(<span class="fu">as.numeric</span>(df_fisher_rbp<span class="sc">$</span>Fisher.p),<span class="st">'fdr'</span>)</span>
<span id="cb42-37"><a href="#cb42-37" aria-hidden="true" tabindex="-1"></a>  df_fisher_rbp<span class="sc">$</span>label <span class="ot">=</span> <span class="fu">signif</span>(df_fisher_rbp<span class="sc">$</span>OR,<span class="dv">2</span>)</span>
<span id="cb42-38"><a href="#cb42-38" aria-hidden="true" tabindex="-1"></a>  df_fisher_rbp<span class="sc">$</span>label[df_fisher_rbp<span class="sc">$</span>Fisher.p<span class="sc">&gt;</span>.<span class="dv">001</span>] <span class="ot">=</span> <span class="st">''</span></span>
<span id="cb42-39"><a href="#cb42-39" aria-hidden="true" tabindex="-1"></a>  df_fisher_rbp<span class="ot">&lt;-</span>df_fisher_rbp <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(<span class="fu">data.frame</span>(<span class="at">mod.num =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, <span class="at">mod=</span><span class="fu">labels2colors</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "mod"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>  df_fisher_rbp<span class="sc">$</span>mod.num <span class="ot">=</span> <span class="fu">factor</span>(df_fisher_rbp<span class="sc">$</span>mod.num,<span class="at">levels=</span><span class="fu">c</span>(<span class="dv">100</span><span class="sc">:</span><span class="dv">1</span>))</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(df_fisher_rbp,<span class="fu">aes</span>(<span class="at">y=</span>(mod.num), <span class="at">x=</span>target, <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p))) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>net, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">2</span>)<span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>,<span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>,<span class="at">hjust=</span><span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_combinedNetworks_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(df_fisher_rbp,aes(y=reorder(paste0(net, ".M", mod.num, "_", target),  -Fisher.p), x=-log10(Fisher.p), label=label, fill=net)) + </span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_bar(stat='identity') + geom_text(size=2)+  theme(axis.text.x = element_text(angle=90,hjust=1))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="annotate-individual-modules" class="level2">
<h2 class="anchored" data-anchor-id="annotate-individual-modules">Annotate Individual Modules</h2>
<section id="isoexpr-network" class="level3">
<h3 class="anchored" data-anchor-id="isoexpr-network">IsoExpr Network</h3>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-16_5a4619774ae2f5d92ecaeae315f752f8">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="cn">FALSE</span>) {</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>this_datExpr <span class="ot">=</span> datExpr.isoExpr</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>this_net <span class="ot">=</span> net.isoExpr</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>this_net<span class="sc">$</span>module <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"M"</span>, net.isoExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels, <span class="st">"."</span>, <span class="fu">labels2colors</span>(net.isoExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels))</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>this_net<span class="sc">$</span>module.number <span class="ot">=</span> net.isoExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>this_net<span class="sc">$</span>colors <span class="ot">=</span> <span class="fu">labels2colors</span>(net.isoExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>fileBaseNet<span class="ot">=</span><span class="st">"data/working/WGCNA/final/datExpr.isoExpr_batchCorrected_92k"</span></span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Generate MEs and kME table for Network</span></span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>MEs <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(<span class="fu">t</span>(this_datExpr), <span class="at">colors=</span>this_net<span class="sc">$</span>module.number)</span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>kMEtable <span class="ot">=</span> <span class="fu">signedKME</span>(<span class="fu">t</span>(this_datExpr), <span class="at">datME =</span>MEs<span class="sc">$</span>eigengenes,<span class="at">corFnc =</span> <span class="st">"bicor"</span>)</span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>this_anno <span class="ot">=</span> datAnno[<span class="fu">match</span>(<span class="fu">rownames</span>(this_datExpr), datAnno<span class="sc">$</span>transcript_id),]</span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>module <span class="ot">=</span> this_net<span class="sc">$</span>module</span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>module.color <span class="ot">=</span> this_net<span class="sc">$</span>colors</span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>module.number <span class="ot">=</span> this_net<span class="sc">$</span>module.number</span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">=</span> <span class="fu">grep</span>(<span class="st">"^TALON"</span>, this_anno<span class="sc">$</span>transcript_name)</span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>transcript_name[idx] <span class="ot">=</span> <span class="fu">paste0</span>(this_anno<span class="sc">$</span>gene_name[idx], <span class="st">'_'</span>, this_anno<span class="sc">$</span>transcript_name[idx])</span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a>cell_anno <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="st">'grey'</span>, <span class="at">nrow=</span><span class="fu">nrow</span>(this_anno), <span class="at">ncol =</span> <span class="fu">length</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))))</span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cell_anno) <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))</span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster)) {</span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a>    marker_genes <span class="ot">=</span> celltypemarkers <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Ensembl) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb46-25"><a href="#cb46-25" aria-hidden="true" tabindex="-1"></a>    cell_anno[this_anno<span class="sc">$</span>ensg <span class="sc">%in%</span> marker_genes, this_cell] <span class="ot">=</span> <span class="st">"red"</span></span>
<span id="cb46-26"><a href="#cb46-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb46-27"><a href="#cb46-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-28"><a href="#cb46-28" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">=</span> <span class="fu">tibble</span>(<span class="at">transcript_id =</span> <span class="fu">rownames</span>(kMEtable), <span class="at">module=</span>this_net<span class="sc">$</span>module, <span class="at">module.color =</span> this_net<span class="sc">$</span>colors, <span class="at">module.number=</span>this_net<span class="sc">$</span>module.number, kMEtable)</span>
<span id="cb46-29"><a href="#cb46-29" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(transcript_id,module,module.color,module.number), <span class="at">names_to =</span> <span class="st">"kME_to_module"</span>, <span class="at">values_to =</span> <span class="st">'kME'</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">kMEtoMod =</span> <span class="fu">gsub</span>(<span class="st">"kME"</span>, <span class="st">""</span>, kME_to_module)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module<span class="sc">!=</span><span class="dv">0</span>, module <span class="sc">!=</span> <span class="st">"grey"</span>, kMEtoMod <span class="sc">!=</span><span class="st">'grey'</span>, kMEtoMod <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>kME_to_module)</span>
<span id="cb46-30"><a href="#cb46-30" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(this_anno)</span>
<span id="cb46-31"><a href="#cb46-31" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(kMEtoMod <span class="sc">==</span> module.number) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">10</span>)</span>
<span id="cb46-32"><a href="#cb46-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-33"><a href="#cb46-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-34"><a href="#cb46-34" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="fu">paste0</span>(fileBaseNet, <span class="st">".pdf"</span>),<span class="at">width=</span><span class="dv">15</span>,<span class="at">height=</span><span class="dv">8</span>)</span>
<span id="cb46-35"><a href="#cb46-35" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_net<span class="sc">$</span>tree, <span class="at">colors=</span><span class="fu">cbind</span>((this_net<span class="sc">$</span>colors),cell_anno), <span class="at">dendroLabels =</span> F)</span>
<span id="cb46-36"><a href="#cb46-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-37"><a href="#cb46-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-38"><a href="#cb46-38" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df_fisher.celltype <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoExpr"</span>), <span class="fu">aes</span>(<span class="at">x=</span>cell,<span class="at">y=</span><span class="fu">factor</span>(mod,<span class="at">levels=</span><span class="fu">c</span>(<span class="dv">100</span><span class="sc">:</span><span class="dv">1</span>)), <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p))) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>, <span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="sc">-</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">0</span>)) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>,<span class="at">y=</span><span class="st">""</span>)</span>
<span id="cb46-39"><a href="#cb46-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-40"><a href="#cb46-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-41"><a href="#cb46-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-42"><a href="#cb46-42" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">ncol</span>(MEs<span class="sc">$</span>eigengenes))) {</span>
<span id="cb46-43"><a href="#cb46-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(i)</span>
<span id="cb46-44"><a href="#cb46-44" aria-hidden="true" tabindex="-1"></a>  this_mod <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"ME"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(MEs<span class="sc">$</span>eigengenes)[i])</span>
<span id="cb46-45"><a href="#cb46-45" aria-hidden="true" tabindex="-1"></a>    mod_genes <span class="ot">=</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module.number<span class="sc">==</span>this_mod,kMEtoMod<span class="sc">==</span>this_mod) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="sc">-</span>kME) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensg) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb46-46"><a href="#cb46-46" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span>this_mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"grey"</span>, <span class="dv">0</span>, <span class="st">"M0.grey"</span>)) {</span>
<span id="cb46-47"><a href="#cb46-47" aria-hidden="true" tabindex="-1"></a>  this_plot <span class="ot">=</span> <span class="fu">data.frame</span>(datMeta, <span class="at">eig=</span>MEs<span class="sc">$</span>eigengenes[,i])</span>
<span id="cb46-48"><a href="#cb46-48" aria-hidden="true" tabindex="-1"></a>  g1 <span class="ot">=</span> <span class="fu">ggplot</span>(this_plot, <span class="fu">aes</span>(<span class="at">x=</span>Region, <span class="at">y=</span>eig, <span class="at">color=</span>Subject)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">"Module "</span>,this_mod, <span class="st">": "</span>, <span class="fu">labels2colors</span>(<span class="fu">as.numeric</span>(this_mod)), <span class="st">" n="</span>, <span class="fu">length</span>(mod_genes)))</span>
<span id="cb46-49"><a href="#cb46-49" aria-hidden="true" tabindex="-1"></a>  g2 <span class="ot">=</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module.number<span class="sc">==</span>this_mod, kMEtoMod<span class="sc">==</span>this_mod) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">25</span>) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(transcript_name, kME), <span class="at">y=</span>kME, <span class="at">fill=</span>novelty)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> colorVector)</span>
<span id="cb46-50"><a href="#cb46-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-51"><a href="#cb46-51" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-52"><a href="#cb46-52" aria-hidden="true" tabindex="-1"></a>  g3<span class="ot">=</span><span class="fu">ggplot</span>(df_fisher.celltype <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoExpr"</span>,mod<span class="sc">==</span>this_mod),<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(cell, <span class="sc">-</span>Fisher.p), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p), <span class="at">fill=</span>OR)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> </span>
<span id="cb46-53"><a href="#cb46-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>)</span>
<span id="cb46-54"><a href="#cb46-54" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-55"><a href="#cb46-55" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add pathway enrichments</span></span>
<span id="cb46-56"><a href="#cb46-56" aria-hidden="true" tabindex="-1"></a>  path <span class="ot">=</span> gprofiler2<span class="sc">::</span><span class="fu">gost</span>(<span class="at">query=</span>mod_genes,<span class="at">ordered_query =</span> T,<span class="at">correction_method =</span> <span class="st">'fdr'</span>,</span>
<span id="cb46-57"><a href="#cb46-57" aria-hidden="true" tabindex="-1"></a>                        <span class="at">custom_bg =</span> <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg), <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">"GO"</span>,<span class="st">"KEGG"</span>, <span class="st">"REACTOME"</span>))</span>
<span id="cb46-58"><a href="#cb46-58" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(path)) {</span>
<span id="cb46-59"><a href="#cb46-59" aria-hidden="true" tabindex="-1"></a>    df_path <span class="ot">=</span> <span class="fu">as_tibble</span>(path<span class="sc">$</span>result)</span>
<span id="cb46-60"><a href="#cb46-60" aria-hidden="true" tabindex="-1"></a>    df_path <span class="ot">&lt;-</span> df_path <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term_size <span class="sc">&gt;</span> <span class="dv">5</span>, term_size <span class="sc">&lt;</span> <span class="dv">1000</span>)</span>
<span id="cb46-61"><a href="#cb46-61" aria-hidden="true" tabindex="-1"></a>  g4 <span class="ot">&lt;-</span> df_path <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(source) <span class="sc">%&gt;%</span> <span class="fu">slice_min</span>(p_value,<span class="at">n=</span><span class="dv">5</span>,<span class="at">with_ties =</span> F) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb46-62"><a href="#cb46-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(term_name, <span class="sc">-</span>p_value), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p_value), <span class="at">fill=</span>source)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>)  <span class="sc">+</span> </span>
<span id="cb46-63"><a href="#cb46-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(source<span class="sc">~</span>., <span class="at">space =</span> <span class="st">'free'</span>, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>)</span>
<span id="cb46-64"><a href="#cb46-64" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb46-65"><a href="#cb46-65" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"No pathway enrichment for this module"</span>)</span>
<span id="cb46-66"><a href="#cb46-66" aria-hidden="true" tabindex="-1"></a>    g4<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb46-67"><a href="#cb46-67" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb46-68"><a href="#cb46-68" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-69"><a href="#cb46-69" aria-hidden="true" tabindex="-1"></a>  <span class="co"># RBP</span></span>
<span id="cb46-70"><a href="#cb46-70" aria-hidden="true" tabindex="-1"></a>   g5<span class="ot">=</span><span class="fu">ggplot</span>(df_fisher_rbp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"net.isoExpr"</span>,mod.num<span class="sc">==</span>this_mod,Fisher.p<span class="sc">&lt;</span>.<span class="dv">05</span>),<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(target, <span class="sc">-</span>Fisher.p), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p), <span class="at">fill=</span>OR)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> </span>
<span id="cb46-71"><a href="#cb46-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>)<span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"RBP Enrichment"</span>)</span>
<span id="cb46-72"><a href="#cb46-72" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb46-73"><a href="#cb46-73" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Rare Var</span></span>
<span id="cb46-74"><a href="#cb46-74" aria-hidden="true" tabindex="-1"></a>   g6<span class="ot">=</span><span class="fu">ggplot</span>(df_logit.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoExpr"</span>,mod<span class="sc">==</span>this_mod),<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(set, <span class="sc">-</span>Q), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Q), <span class="at">fill=</span>OR)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> </span>
<span id="cb46-75"><a href="#cb46-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>)<span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Rare Var Enrichment"</span>)</span>
<span id="cb46-76"><a href="#cb46-76" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb46-77"><a href="#cb46-77" aria-hidden="true" tabindex="-1"></a>     <span class="co"># GWAS</span></span>
<span id="cb46-78"><a href="#cb46-78" aria-hidden="true" tabindex="-1"></a>   g7<span class="ot">=</span><span class="fu">ggplot</span>(df_gwas <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoExpr"</span>,mod<span class="sc">==</span>this_mod),<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(gwas, <span class="sc">-</span>fdr), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(fdr), <span class="at">fill=</span>Zscore)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> <span class="fu">scale_fill_gradient2</span>()<span class="sc">+</span></span>
<span id="cb46-79"><a href="#cb46-79" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"GWAS Enrichment"</span>)</span>
<span id="cb46-80"><a href="#cb46-80" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb46-81"><a href="#cb46-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-82"><a href="#cb46-82" aria-hidden="true" tabindex="-1"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(<span class="at">grobs=</span><span class="fu">list</span>(g1,g2,g3,g4,g5,g6,g7),<span class="at">layout_matrix=</span><span class="fu">rbind</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">5</span>),<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">5</span>), <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">7</span>), <span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">4</span>,<span class="dv">7</span>)))</span>
<span id="cb46-83"><a href="#cb46-83" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb46-84"><a href="#cb46-84" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb46-85"><a href="#cb46-85" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb46-86"><a href="#cb46-86" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="isousage-network" class="level3">
<h3 class="anchored" data-anchor-id="isousage-network">IsoUsage Network</h3>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-17_256484be720afcb404782f6e8dc5e512">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="cn">FALSE</span>) {</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>this_datExpr <span class="ot">=</span> datExpr.isoFr</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>this_net <span class="ot">=</span> net.isoFr</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>this_net<span class="sc">$</span>module <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"M"</span>, this_net<span class="sc">$</span>module.number, <span class="st">"."</span>, <span class="fu">labels2colors</span>(this_net<span class="sc">$</span>module.number))</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>fileBaseNet<span class="ot">=</span><span class="st">"data/working/WGCNA/final/datExpr.localIF_batchCorrected_103k"</span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Generate MEs and kME table for Network</span></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>MEs <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(<span class="fu">t</span>(this_datExpr), <span class="at">colors=</span>this_net<span class="sc">$</span>module.number)</span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>kMEtable <span class="ot">=</span> <span class="fu">signedKME</span>(<span class="fu">t</span>(this_datExpr), <span class="at">datME =</span>MEs<span class="sc">$</span>eigengenes,<span class="at">corFnc =</span> <span class="st">"bicor"</span>)</span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>this_anno <span class="ot">=</span> datAnno[<span class="fu">match</span>(<span class="fu">rownames</span>(this_datExpr), datAnno<span class="sc">$</span>transcript_id),]</span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">=</span> <span class="fu">grep</span>(<span class="st">"^TALON"</span>, this_anno<span class="sc">$</span>transcript_name)</span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>transcript_name[idx] <span class="ot">=</span> <span class="fu">paste0</span>(this_anno<span class="sc">$</span>gene_name[idx], <span class="st">'_'</span>, this_anno<span class="sc">$</span>transcript_name[idx])</span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a>cell_anno <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="st">'grey'</span>, <span class="at">nrow=</span><span class="fu">nrow</span>(this_anno), <span class="at">ncol =</span> <span class="fu">length</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))))</span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cell_anno) <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster))</span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(this_cell <span class="cf">in</span> <span class="fu">unique</span>(celltypemarkers<span class="sc">$</span>Cluster)) {</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>    marker_genes <span class="ot">=</span> celltypemarkers <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Cluster <span class="sc">==</span> this_cell) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Ensembl) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a>    cell_anno[this_anno<span class="sc">$</span>ensg <span class="sc">%in%</span> marker_genes, this_cell] <span class="ot">=</span> <span class="st">"red"</span></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">=</span> <span class="fu">tibble</span>(<span class="at">transcript_id =</span> <span class="fu">rownames</span>(kMEtable), <span class="at">module=</span>this_net<span class="sc">$</span>module, <span class="at">module.color =</span> this_net<span class="sc">$</span>colors, <span class="at">module.number=</span>this_net<span class="sc">$</span>module.number, kMEtable)</span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(transcript_id,module,module.color,module.number), <span class="at">names_to =</span> <span class="st">"kME_to_module"</span>, <span class="at">values_to =</span> <span class="st">'kME'</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">kMEtoMod =</span> <span class="fu">gsub</span>(<span class="st">"kME"</span>, <span class="st">""</span>, kME_to_module)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module<span class="sc">!=</span><span class="dv">0</span>, module <span class="sc">!=</span> <span class="st">"grey"</span>, kMEtoMod <span class="sc">!=</span><span class="st">'grey'</span>, kMEtoMod <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>kME_to_module)</span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(this_anno)</span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(kMEtoMod <span class="sc">==</span> module) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">10</span>)</span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="fu">paste0</span>(fileBaseNet, <span class="st">".pdf"</span>),<span class="at">width=</span><span class="dv">15</span>,<span class="at">height=</span><span class="dv">8</span>)</span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(this_net<span class="sc">$</span>tree, <span class="at">colors=</span><span class="fu">cbind</span>((this_net<span class="sc">$</span>colors),cell_anno), <span class="at">dendroLabels =</span> F)</span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-32"><a href="#cb47-32" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df_fisher.celltype <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoUsage"</span>), <span class="fu">aes</span>(<span class="at">x=</span>cell,<span class="at">y=</span><span class="fu">factor</span>(mod,<span class="at">levels=</span><span class="fu">c</span>(<span class="dv">100</span><span class="sc">:</span><span class="dv">1</span>)), <span class="at">label=</span>label, <span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p))) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">'white'</span>, <span class="at">high=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="sc">-</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">0</span>)) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>,<span class="at">y=</span><span class="st">""</span>)</span>
<span id="cb47-33"><a href="#cb47-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-34"><a href="#cb47-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-35"><a href="#cb47-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-36"><a href="#cb47-36" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">ncol</span>(MEs<span class="sc">$</span>eigengenes))) {</span>
<span id="cb47-37"><a href="#cb47-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(i)</span>
<span id="cb47-38"><a href="#cb47-38" aria-hidden="true" tabindex="-1"></a>  this_mod <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"ME"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(MEs<span class="sc">$</span>eigengenes)[i])</span>
<span id="cb47-39"><a href="#cb47-39" aria-hidden="true" tabindex="-1"></a>  mod_genes <span class="ot">=</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module.number<span class="sc">==</span>this_mod,kMEtoMod<span class="sc">==</span>this_mod) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="sc">-</span>kME) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ensg) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span>
<span id="cb47-40"><a href="#cb47-40" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span>this_mod <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"grey"</span>, <span class="dv">0</span>, <span class="st">"M0.grey"</span>)) {</span>
<span id="cb47-41"><a href="#cb47-41" aria-hidden="true" tabindex="-1"></a>  this_plot <span class="ot">=</span> <span class="fu">data.frame</span>(datMeta, <span class="at">eig=</span>MEs<span class="sc">$</span>eigengenes[,i])</span>
<span id="cb47-42"><a href="#cb47-42" aria-hidden="true" tabindex="-1"></a>  g1 <span class="ot">=</span> <span class="fu">ggplot</span>(this_plot, <span class="fu">aes</span>(<span class="at">x=</span>Region, <span class="at">y=</span>eig, <span class="at">color=</span>Subject)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">"Module "</span>,this_mod, <span class="st">": "</span>, <span class="fu">labels2colors</span>(<span class="fu">as.numeric</span>(this_mod)), <span class="st">" n="</span>, <span class="fu">length</span>(mod_genes)))</span>
<span id="cb47-43"><a href="#cb47-43" aria-hidden="true" tabindex="-1"></a>  g2 <span class="ot">=</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module.number<span class="sc">==</span>this_mod, kMEtoMod<span class="sc">==</span>this_mod) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">25</span>) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(transcript_name, kME), <span class="at">y=</span>kME, <span class="at">fill=</span>novelty)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> colorVector)</span>
<span id="cb47-44"><a href="#cb47-44" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-45"><a href="#cb47-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-46"><a href="#cb47-46" aria-hidden="true" tabindex="-1"></a>  g3<span class="ot">=</span><span class="fu">ggplot</span>(df_fisher.celltype <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoExpr"</span>,mod<span class="sc">==</span>this_mod),<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(cell, <span class="sc">-</span>Fisher.p), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p), <span class="at">fill=</span>OR)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> </span>
<span id="cb47-47"><a href="#cb47-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>)</span>
<span id="cb47-48"><a href="#cb47-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-49"><a href="#cb47-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add pathway enrichments</span></span>
<span id="cb47-50"><a href="#cb47-50" aria-hidden="true" tabindex="-1"></a>  path <span class="ot">=</span> gprofiler2<span class="sc">::</span><span class="fu">gost</span>(<span class="at">query=</span>mod_genes,<span class="at">ordered_query =</span> T,<span class="at">correction_method =</span> <span class="st">'fdr'</span>,</span>
<span id="cb47-51"><a href="#cb47-51" aria-hidden="true" tabindex="-1"></a>                        <span class="at">custom_bg =</span> <span class="fu">unique</span>(datAnno<span class="sc">$</span>ensg), <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">"GO"</span>,<span class="st">"KEGG"</span>, <span class="st">"REACTOME"</span>))</span>
<span id="cb47-52"><a href="#cb47-52" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(path)) {</span>
<span id="cb47-53"><a href="#cb47-53" aria-hidden="true" tabindex="-1"></a>    df_path <span class="ot">=</span> <span class="fu">as_tibble</span>(path<span class="sc">$</span>result)</span>
<span id="cb47-54"><a href="#cb47-54" aria-hidden="true" tabindex="-1"></a>    df_path <span class="ot">&lt;-</span> df_path <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term_size <span class="sc">&gt;</span> <span class="dv">5</span>, term_size <span class="sc">&lt;</span> <span class="dv">1000</span>)</span>
<span id="cb47-55"><a href="#cb47-55" aria-hidden="true" tabindex="-1"></a>  g4 <span class="ot">&lt;-</span> df_path <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(source) <span class="sc">%&gt;%</span> <span class="fu">slice_min</span>(p_value,<span class="at">n=</span><span class="dv">5</span>,<span class="at">with_ties =</span> F) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb47-56"><a href="#cb47-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(term_name, <span class="sc">-</span>p_value), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p_value), <span class="at">fill=</span>source)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>)  <span class="sc">+</span> </span>
<span id="cb47-57"><a href="#cb47-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(source<span class="sc">~</span>., <span class="at">space =</span> <span class="st">'free'</span>, <span class="at">scales=</span><span class="st">'free'</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>)</span>
<span id="cb47-58"><a href="#cb47-58" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb47-59"><a href="#cb47-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"No pathway enrichment for this module"</span>)</span>
<span id="cb47-60"><a href="#cb47-60" aria-hidden="true" tabindex="-1"></a>    g4<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb47-61"><a href="#cb47-61" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb47-62"><a href="#cb47-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-63"><a href="#cb47-63" aria-hidden="true" tabindex="-1"></a>  <span class="co"># RBP</span></span>
<span id="cb47-64"><a href="#cb47-64" aria-hidden="true" tabindex="-1"></a>   g5<span class="ot">=</span><span class="fu">ggplot</span>(df_fisher_rbp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"net.isoFr"</span>,mod.num<span class="sc">==</span>this_mod,Fisher.p<span class="sc">&lt;</span>.<span class="dv">05</span>),<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(target, <span class="sc">-</span>Fisher.p), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Fisher.p), <span class="at">fill=</span>OR)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> </span>
<span id="cb47-65"><a href="#cb47-65" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>)<span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"RBP Enrichment"</span>)</span>
<span id="cb47-66"><a href="#cb47-66" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb47-67"><a href="#cb47-67" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Rare Var</span></span>
<span id="cb47-68"><a href="#cb47-68" aria-hidden="true" tabindex="-1"></a>   g6<span class="ot">=</span><span class="fu">ggplot</span>(df_logit.gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoUsage"</span>,mod<span class="sc">==</span>this_mod),<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(set, <span class="sc">-</span>Q), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(Q), <span class="at">fill=</span>OR)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> </span>
<span id="cb47-69"><a href="#cb47-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>)<span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Rare Var Enrichment"</span>)</span>
<span id="cb47-70"><a href="#cb47-70" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb47-71"><a href="#cb47-71" aria-hidden="true" tabindex="-1"></a>     <span class="co"># GWAS</span></span>
<span id="cb47-72"><a href="#cb47-72" aria-hidden="true" tabindex="-1"></a>   g7<span class="ot">=</span><span class="fu">ggplot</span>(df_gwas <span class="sc">%&gt;%</span> <span class="fu">filter</span>(net<span class="sc">==</span><span class="st">"isoUsage"</span>,mod<span class="sc">==</span>this_mod),<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">reorder</span>(gwas, <span class="sc">-</span>fdr), <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(fdr), <span class="at">fill=</span>Zscore)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>,<span class="at">position =</span> <span class="fu">position_dodge2</span>()) <span class="sc">+</span> <span class="fu">scale_fill_gradient2</span>()<span class="sc">+</span></span>
<span id="cb47-73"><a href="#cb47-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>) <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>),<span class="at">lty=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">'red'</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"GWAS Enrichment"</span>)</span>
<span id="cb47-74"><a href="#cb47-74" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb47-75"><a href="#cb47-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-76"><a href="#cb47-76" aria-hidden="true" tabindex="-1"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(<span class="at">grobs=</span><span class="fu">list</span>(g1,g2,g3,g4,g5,g6,g7),<span class="at">layout_matrix=</span><span class="fu">rbind</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">5</span>),<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">5</span>), <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">7</span>), <span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">4</span>,<span class="dv">7</span>)))</span>
<span id="cb47-77"><a href="#cb47-77" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb47-78"><a href="#cb47-78" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-79"><a href="#cb47-79" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb47-80"><a href="#cb47-80" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="combined-network-plots" class="level2">
<h2 class="anchored" data-anchor-id="combined-network-plots">Combined Network Plots</h2>
<section id="network-cluster" class="level3">
<h3 class="anchored" data-anchor-id="network-cluster">Network Cluster</h3>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-18_8cf3afa387a5c8d995d1b115cebfacf9">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">=</span> <span class="fu">data.frame</span>(net.isoFr<span class="sc">$</span>MEs<span class="sc">$</span>eigengenes)</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df1) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"isoUsage_"</span>, <span class="fu">gsub</span>(<span class="st">"ME"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(df1)))</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>df2 <span class="ot">=</span> <span class="fu">data.frame</span>(net.isoExpr<span class="sc">$</span>MEs<span class="sc">$</span>eigengenes)</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df2) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"isoExpr_"</span>,  <span class="fu">gsub</span>(<span class="st">"ME"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(df2)))</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>df3 <span class="ot">=</span> <span class="fu">data.frame</span>(net.geneExpr<span class="sc">$</span>MEs<span class="sc">$</span>eigengenes)</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df3) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"geneExpr_"</span>,  <span class="fu">gsub</span>(<span class="st">"ME"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(df3)))</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">cbind</span>(df1,df2,df3)</span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">=</span> <span class="fu">hclust</span>(<span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span><span class="fu">bicor</span>(df)),<span class="at">method =</span> <span class="st">'average'</span>)</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Hierarchically cluster the sample-eigenegene correlations</span></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a><span class="co"># and save the dendrogram representation</span></span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>dist_mat <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span><span class="fu">bicor</span>((df)))</span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>clust <span class="ot">&lt;-</span> <span class="fu">hclust</span>(dist_mat,<span class="at">method =</span> <span class="st">'average'</span>)</span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>dend <span class="ot">&lt;-</span> <span class="fu">as.dendrogram</span>(clust)</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the mapping between network names and colors</span></span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a>labelColors <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">"geneExpr"</span> <span class="ot">=</span> <span class="st">"#d62828"</span>,</span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a>    <span class="st">"isoExpr"</span> <span class="ot">=</span> <span class="st">"#219ebc"</span>,</span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a>    <span class="st">"isoUsage"</span> <span class="ot">=</span> <span class="st">"#ff9914"</span>)</span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate over the dendrogram leafs and manually change the node</span></span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a><span class="co"># parameters according to the label name (the column name)</span></span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a>colLab <span class="ot">&lt;-</span> <span class="cf">function</span>(n) {</span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.leaf</span>(n)) {</span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb48-31"><a href="#cb48-31" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Grab the attributes of the leaf/node</span></span>
<span id="cb48-32"><a href="#cb48-32" aria-hidden="true" tabindex="-1"></a>        a <span class="ot">&lt;-</span> <span class="fu">attributes</span>(n)</span>
<span id="cb48-33"><a href="#cb48-33" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb48-34"><a href="#cb48-34" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Print the name so that later on you can arrange the heatmap</span></span>
<span id="cb48-35"><a href="#cb48-35" aria-hidden="true" tabindex="-1"></a>        <span class="co"># to match the order in the dendrogram</span></span>
<span id="cb48-36"><a href="#cb48-36" aria-hidden="true" tabindex="-1"></a>        <span class="fu">cat</span>(<span class="fu">as.character</span>(a<span class="sc">$</span>label))</span>
<span id="cb48-37"><a href="#cb48-37" aria-hidden="true" tabindex="-1"></a>        <span class="fu">cat</span>(<span class="st">", "</span>)</span>
<span id="cb48-38"><a href="#cb48-38" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb48-39"><a href="#cb48-39" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Extract the name of the network (this regex extracts dev_gene from dev_gene_blahblah)</span></span>
<span id="cb48-40"><a href="#cb48-40" aria-hidden="true" tabindex="-1"></a>        net_name <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(a<span class="sc">$</span>label, <span class="st">"[:alnum:]+"</span>)</span>
<span id="cb48-41"><a href="#cb48-41" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb48-42"><a href="#cb48-42" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Assign the text color according to the mapping in labelColors</span></span>
<span id="cb48-43"><a href="#cb48-43" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (net_name <span class="sc">%in%</span> <span class="fu">names</span>(labelColors)) {</span>
<span id="cb48-44"><a href="#cb48-44" aria-hidden="true" tabindex="-1"></a>            labCol <span class="ot">&lt;-</span> labelColors[[net_name]] </span>
<span id="cb48-45"><a href="#cb48-45" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb48-46"><a href="#cb48-46" aria-hidden="true" tabindex="-1"></a>            labCol <span class="ot">&lt;-</span> <span class="st">"#5603ad"</span></span>
<span id="cb48-47"><a href="#cb48-47" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb48-48"><a href="#cb48-48" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb48-49"><a href="#cb48-49" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Assign the node color to match the WGCNA color (saddlebrown, turquoise, etc.)</span></span>
<span id="cb48-50"><a href="#cb48-50" aria-hidden="true" tabindex="-1"></a>        mod_color <span class="ot">&lt;-</span> <span class="fu">str_match</span>(a<span class="sc">$</span>label,<span class="st">"[^_]+_(.*)"</span>)[<span class="dv">2</span>]</span>
<span id="cb48-51"><a href="#cb48-51" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb48-52"><a href="#cb48-52" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Add the new node attributes onto the existing node attributes</span></span>
<span id="cb48-53"><a href="#cb48-53" aria-hidden="true" tabindex="-1"></a>        <span class="fu">attr</span>(n, <span class="st">"nodePar"</span>) <span class="ot">&lt;-</span> <span class="fu">list</span>(a<span class="sc">$</span>nodePar,</span>
<span id="cb48-54"><a href="#cb48-54" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">cex =</span> <span class="dv">1</span>,</span>
<span id="cb48-55"><a href="#cb48-55" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pch =</span> <span class="dv">15</span>,</span>
<span id="cb48-56"><a href="#cb48-56" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">lab.col =</span> labCol, <span class="co"># This is the color of the leaf label (the text)</span></span>
<span id="cb48-57"><a href="#cb48-57" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">col =</span> mod_color)  <span class="co"># This is the color of the leaf (the square)</span></span>
<span id="cb48-58"><a href="#cb48-58" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb48-59"><a href="#cb48-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(n)</span>
<span id="cb48-60"><a href="#cb48-60" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb48-61"><a href="#cb48-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-62"><a href="#cb48-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply the iteration function onto our saved dendrogram</span></span>
<span id="cb48-63"><a href="#cb48-63" aria-hidden="true" tabindex="-1"></a>new_dendro <span class="ot">&lt;-</span> <span class="fu">dendrapply</span>(dend, colLab)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>isoExpr_salmon, isoExpr_purple, isoUsage_orange, isoExpr_darkgrey, isoExpr_mediumpurple3, isoExpr_palevioletred3, isoExpr_floralwhite, isoExpr_green, isoUsage_white, isoExpr_skyblue, isoExpr_darkorange2, isoUsage_darkgrey, isoExpr_sienna3, isoUsage_green, isoUsage_brown, isoUsage_grey, isoExpr_paleturquoise, isoExpr_royalblue, isoUsage_pink, isoExpr_plum1, isoExpr_steelblue, isoExpr_magenta, isoExpr_white, isoUsage_royalblue, isoExpr_darkmagenta, isoExpr_plum2, isoUsage_darkred, isoUsage_salmon, isoExpr_lightcyan1, isoExpr_greenyellow, isoExpr_saddlebrown, isoExpr_darkslateblue, isoExpr_orangered4, isoUsage_purple, isoUsage_saddlebrown, isoExpr_lightyellow, isoExpr_darkturquoise, isoExpr_skyblue3, isoUsage_lightcyan, isoUsage_darkturquoise, isoUsage_lightgreen, geneExpr_cyan, geneExpr_darkturquoise, geneExpr_lightyellow, geneExpr_lightcyan, geneExpr_blue, geneExpr_grey, geneExpr_lightgreen, geneExpr_darkgreen, geneExpr_midnightblue, geneExpr_darkred, geneExpr_magenta, isoExpr_black, isoExpr_darkorange, isoExpr_yellowgreen, isoExpr_darkgreen, isoExpr_orange, isoExpr_ivory, isoExpr_violet, isoExpr_darkolivegreen, isoExpr_lightcyan, isoUsage_magenta, isoExpr_salmon4, isoExpr_lightgreen, isoExpr_tan, isoUsage_black, isoUsage_tan, isoUsage_greenyellow, isoExpr_brown, isoUsage_skyblue, isoExpr_brown4, isoExpr_red, isoExpr_cyan, isoExpr_pink, isoExpr_thistle2, isoExpr_thistle1, isoUsage_midnightblue, isoExpr_bisque4, isoUsage_red, isoUsage_cyan, isoUsage_darkorange, isoExpr_grey60, isoUsage_darkgreen, geneExpr_purple, geneExpr_greenyellow, geneExpr_yellow, geneExpr_turquoise, isoUsage_blue, isoExpr_grey, isoExpr_turquoise, geneExpr_royalblue, isoUsage_turquoise, isoExpr_blue, geneExpr_salmon, geneExpr_black, geneExpr_brown, isoUsage_grey60, isoExpr_lightsteelblue1, isoExpr_midnightblue, isoExpr_darkred, isoUsage_lightyellow, isoExpr_yellow, isoUsage_yellow, geneExpr_green, geneExpr_tan, geneExpr_grey60, geneExpr_pink, geneExpr_red, </code></pre>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the resulting dendrogram</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">repr.plot.width =</span> <span class="dv">30</span>, <span class="at">repr.plot.height =</span> <span class="dv">5</span>, <span class="at">repr.plot.res =</span> <span class="dv">600</span>)</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">cex=</span><span class="fl">0.8</span>, <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(new_dendro,</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">main =</span> <span class="st">"Clustering of module eigengenes (bicor)"</span>,</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">sub =</span> <span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_combinedNetworks_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="module-umap" class="level3">
<h3 class="anchored" data-anchor-id="module-umap">Module UMAP</h3>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-19_79f49445c82c6774abaaecd9fa721da6">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Attaching SeuratObject</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Attaching sp</code></pre>
</div>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'patchwork'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    align_plots</code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'igraph'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    as_data_frame, groups, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:purrr':

    compose, simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    crossing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tibble':

    as_data_frame</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    decompose, spectrum</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    union</code></pre>
</div>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hdWGCNA) <span class="co"># https://smorabit.github.io/hdWGCNA/#installation</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>                 <span class="co"># options(timeout = 600)</span></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>this_datExpr <span class="ot">=</span> datExpr.isoFr</span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>this_net <span class="ot">=</span> net.isoFr</span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a>this_net<span class="sc">$</span>colors <span class="ot">=</span> <span class="fu">labels2colors</span>(net.isoFr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(this_net<span class="sc">$</span>colors) <span class="ot">=</span> <span class="fu">names</span>(net.isoFr<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a>MEs <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(<span class="fu">t</span>(this_datExpr), <span class="at">colors=</span>this_net<span class="sc">$</span>colors)</span>
<span id="cb65-10"><a href="#cb65-10" aria-hidden="true" tabindex="-1"></a>kMEtable <span class="ot">=</span> <span class="fu">signedKME</span>(<span class="fu">t</span>(this_datExpr), <span class="at">datME =</span>MEs<span class="sc">$</span>eigengenes,<span class="at">corFnc =</span> <span class="st">"bicor"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bicor(datExpr, datME, , use = "p"): bicor: zero MAD in variable 'x'.
Pearson correlation was used for individual columns with zero (or missing) MAD.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>this_anno <span class="ot">=</span> datAnno[<span class="fu">match</span>(<span class="fu">rownames</span>(this_datExpr), datAnno<span class="sc">$</span>transcript_id),]</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>module <span class="ot">=</span> this_net<span class="sc">$</span>module</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>module.color <span class="ot">=</span> this_net<span class="sc">$</span>colors</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>module.number <span class="ot">=</span> this_net<span class="sc">$</span>module.number</span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">=</span> <span class="fu">grep</span>(<span class="st">"^TALON"</span>, this_anno<span class="sc">$</span>transcript_name)</span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>transcript_name[idx] <span class="ot">=</span> <span class="fu">paste0</span>(this_anno<span class="sc">$</span>gene_name[idx], <span class="st">'_'</span>, this_anno<span class="sc">$</span>transcript_name[idx])</span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a>modules_to_plot <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"turquoise"</span>, <span class="st">"blue"</span>, <span class="st">"brown"</span>, <span class="st">"pink"</span>)</span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a><span class="co">#modules_to_plot = unique(this_net$colors)</span></span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">=</span> <span class="fu">tibble</span>(<span class="at">transcript_id =</span> <span class="fu">rownames</span>(kMEtable), <span class="at">module=</span>this_net<span class="sc">$</span>colors,</span>
<span id="cb67-12"><a href="#cb67-12" aria-hidden="true" tabindex="-1"></a>                kMEtable)</span>
<span id="cb67-13"><a href="#cb67-13" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(transcript_id,module), <span class="at">names_to =</span> <span class="st">"kME_to_module"</span>, <span class="at">values_to =</span> <span class="st">'kME'</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">kMEtoMod =</span> <span class="fu">gsub</span>(<span class="st">"kME"</span>, <span class="st">""</span>, kME_to_module)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module<span class="sc">!=</span><span class="dv">0</span>, module <span class="sc">!=</span> <span class="st">"grey"</span>, kMEtoMod <span class="sc">!=</span><span class="st">'grey'</span>, kMEtoMod <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>kME_to_module)</span>
<span id="cb67-14"><a href="#cb67-14" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(datAnno)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "transcript_id"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>hubs100 <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module <span class="sc">%in%</span> modules_to_plot, kMEtoMod <span class="sc">==</span> module) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">100</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(transcript_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Adding missing grouping variables: `module`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>hubs25 <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module <span class="sc">%in%</span> modules_to_plot, kMEtoMod <span class="sc">==</span> module) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">25</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(transcript_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Adding missing grouping variables: `module`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>hubs5 <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module <span class="sc">%in%</span> modules_to_plot, kMEtoMod <span class="sc">==</span> module) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">5</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(transcript_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Adding missing grouping variables: `module`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>isoTOM <span class="ot">=</span> <span class="fu">TOMsimilarityFromExpr</span>(<span class="fu">t</span>(this_datExpr[hubs25,]),<span class="at">networkType =</span> <span class="st">'signed'</span>, <span class="at">power =</span> <span class="dv">14</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>TOM calculation: adjacency..
..will not use multithreading.
 Fraction of slow calculations: 0.000000
..connectivity..
..matrix multiplication (system BLAS)..
..normalization..
..done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(isoTOM) <span class="ot">=</span> <span class="fu">rownames</span>(isoTOM) <span class="ot">=</span> hubs25</span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>df<span class="ot">=</span>uwot<span class="sc">::</span><span class="fu">umap</span>(isoTOM,<span class="at">n_neighbors =</span> <span class="dv">25</span>,<span class="at">metric =</span> <span class="st">'cosine'</span>,<span class="at">spread =</span> <span class="dv">20</span>,<span class="at">min_dist =</span> .<span class="dv">4</span>)</span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_combinedNetworks_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>umap_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(df)</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(umap_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"UMAP1"</span>, <span class="st">"UMAP2"</span>)</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>umap_df<span class="sc">$</span>gene <span class="ot">&lt;-</span> hubs25</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>umap_df<span class="sc">$</span>module <span class="ot">&lt;-</span> this_net<span class="sc">$</span>colors[umap_df<span class="sc">$</span>gene]</span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>umap_df<span class="sc">$</span>color <span class="ot">&lt;-</span> net.isoExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels[umap_df<span class="sc">$</span>gene]</span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>umap_df<span class="sc">$</span>hub <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>    umap_df<span class="sc">$</span>gene <span class="sc">%in%</span> <span class="fu">as.character</span>(<span class="fu">unlist</span>(hubs5)), <span class="st">'hub'</span>, <span class="st">'other'</span></span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a>umap_df <span class="ot">&lt;-</span> umap_df <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(tidyMods <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module <span class="sc">==</span> kMEtoMod) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">gene=</span>transcript_id, module, kME), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"module"</span>, <span class="st">"gene"</span>))</span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a>umap_df <span class="ot">&lt;-</span> umap_df <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(datAnno <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name, transcript_id), <span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span> <span class="ot">=</span> <span class="st">"transcript_id"</span>))</span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> <span class="fu">levels</span>(<span class="fu">factor</span>(umap_df<span class="sc">$</span>module))</span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> mods[mods <span class="sc">!=</span> <span class="st">'grey'</span>]</span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-15"><a href="#cb78-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(umap_df, <span class="fu">aes</span>(<span class="at">x=</span>UMAP1, <span class="at">y=</span>UMAP2)) <span class="sc">+</span></span>
<span id="cb78-16"><a href="#cb78-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(</span>
<span id="cb78-17"><a href="#cb78-17" aria-hidden="true" tabindex="-1"></a>   <span class="at">color=</span>umap_df<span class="sc">$</span>module, <span class="co"># color each point by WGCNA module</span></span>
<span id="cb78-18"><a href="#cb78-18" aria-hidden="true" tabindex="-1"></a>   <span class="at">size=</span>umap_df<span class="sc">$</span>kME <span class="co"># size of each point based on intramodular connectivity</span></span>
<span id="cb78-19"><a href="#cb78-19" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb78-20"><a href="#cb78-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">umap_theme</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_combinedNetworks_files/figure-html/unnamed-chunk-19-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>subset_TOM <span class="ot">&lt;-</span> isoTOM[umap_df<span class="sc">$</span>gene, umap_df<span class="sc">$</span>gene[umap_df<span class="sc">$</span>hub <span class="sc">==</span> <span class="st">'hub'</span>]]</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>g1 <span class="ot">=</span> <span class="fu">graph.adjacency</span>(isoTOM[umap_df<span class="sc">$</span>gene, umap_df<span class="sc">$</span>gene], <span class="at">mode=</span><span class="st">'undirected'</span>, <span class="at">diag=</span>F,<span class="at">weighted =</span> T)</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>layoutMat <span class="ot">&lt;-</span> <span class="fu">layout.fruchterman.reingold</span>(g1)</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cairo_pdf</span>(<span class="at">file=</span><span class="st">'output/figures/Fig4/IsoUsage_hairballs.pdf'</span>, <span class="at">width=</span><span class="dv">20</span>,<span class="at">height=</span><span class="dv">20</span>,<span class="at">family=</span><span class="st">'serif'</span>)</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot.igraph</span>(g1, <span class="at">layout=</span><span class="fu">as.matrix</span>(umap_df[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)]),</span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">vertex.label =</span> <span class="fu">paste0</span>(umap_df<span class="sc">$</span>gene_name,<span class="st">"_"</span>, <span class="fu">substr</span>(umap_df<span class="sc">$</span>gene,<span class="dv">1</span>,<span class="dv">15</span>)),</span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">vertex.family=</span><span class="st">'serif'</span>,</span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">edge.curved=</span><span class="dv">0</span>, <span class="at">edge.alpha=</span>.<span class="dv">5</span>,</span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">edge.color =</span> <span class="fu">ifelse</span>(<span class="fu">E</span>(g1)<span class="sc">$</span>weight <span class="sc">&gt;</span> .<span class="dv">25</span>, <span class="st">"grey"</span>, <span class="cn">NA</span>),</span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">edge.width =</span> <span class="fu">E</span>(g1)<span class="sc">$</span>weight,</span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">vertex.label.cex=</span><span class="fl">0.5</span>, <span class="at">vertex.color =</span> umap_df<span class="sc">$</span>module,</span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">vertex.size=</span>umap_df<span class="sc">$</span>kME<span class="sc">^</span><span class="dv">5</span><span class="sc">*</span><span class="dv">4</span>)</span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>  selected_modules <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene_name =</span> umap_df<span class="sc">$</span>gene, <span class="at">module =</span> this_net<span class="sc">$</span>module.number[umap_df<span class="sc">$</span>gene],<span class="at">color=</span> this_net<span class="sc">$</span>colors[umap_df<span class="sc">$</span>gene])</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>  selected_modules <span class="ot">&lt;-</span> <span class="fu">cbind</span>(selected_modules, umap_df[,<span class="fu">c</span>(<span class="st">'UMAP1'</span>, <span class="st">'UMAP2'</span>, <span class="st">'hub'</span>, <span class="st">'kME'</span>)])</span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a>  selected_modules<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(selected_modules<span class="sc">$</span>hub <span class="sc">%in%</span> <span class="st">"hub"</span>, selected_modules<span class="sc">$</span>gene_name, <span class="st">''</span>)</span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a>  selected_modules<span class="sc">$</span>fontcolor <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(selected_modules<span class="sc">$</span>color <span class="sc">==</span> <span class="st">'black'</span>, <span class="st">'gray50'</span>, <span class="st">'black'</span>)</span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-7"><a href="#cb81-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set frome color</span></span>
<span id="cb81-8"><a href="#cb81-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># same color as module for all genes, black outline for the selected hub genes</span></span>
<span id="cb81-9"><a href="#cb81-9" aria-hidden="true" tabindex="-1"></a>  selected_modules<span class="sc">$</span>framecolor <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(selected_modules<span class="sc">$</span>hub <span class="sc">%in%</span> <span class="st">"hub"</span>, <span class="st">'black'</span>, selected_modules<span class="sc">$</span>color)</span>
<span id="cb81-10"><a href="#cb81-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-11"><a href="#cb81-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-12"><a href="#cb81-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-13"><a href="#cb81-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-14"><a href="#cb81-14" aria-hidden="true" tabindex="-1"></a>edge_df <span class="ot">&lt;-</span> subset_TOM <span class="sc">%&gt;%</span> reshape2<span class="sc">::</span><span class="fu">melt</span>()</span>
<span id="cb81-15"><a href="#cb81-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">dim</span>(edge_df))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2000    3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>    edge_df<span class="sc">$</span>color <span class="ot">&lt;-</span> future.apply<span class="sc">::</span><span class="fu">future_sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(edge_df), <span class="cf">function</span>(i){</span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>      gene1 <span class="ot">=</span> <span class="fu">as.character</span>(edge_df[i,<span class="st">'Var1'</span>])</span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>      gene2 <span class="ot">=</span> <span class="fu">as.character</span>(edge_df[i,<span class="st">'Var2'</span>])</span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>      col1 <span class="ot">&lt;-</span> selected_modules[selected_modules<span class="sc">$</span>gene_name <span class="sc">==</span> gene1, <span class="st">'color'</span>]</span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a>      col2 <span class="ot">&lt;-</span> selected_modules[selected_modules<span class="sc">$</span>gene_name <span class="sc">==</span> gene2, <span class="st">'color'</span>]</span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(col1 <span class="sc">==</span> col2){</span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a>        col <span class="ot">=</span> col1</span>
<span id="cb83-10"><a href="#cb83-10" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span>{</span>
<span id="cb83-11"><a href="#cb83-11" aria-hidden="true" tabindex="-1"></a>        col <span class="ot">=</span> <span class="st">'grey90'</span></span>
<span id="cb83-12"><a href="#cb83-12" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb83-13"><a href="#cb83-13" aria-hidden="true" tabindex="-1"></a>      col</span>
<span id="cb83-14"><a href="#cb83-14" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb83-15"><a href="#cb83-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-16"><a href="#cb83-16" aria-hidden="true" tabindex="-1"></a>  edge_df <span class="ot">&lt;-</span> edge_df <span class="sc">%&gt;%</span> <span class="fu">subset</span>(color <span class="sc">!=</span> <span class="st">'grey90'</span>)</span>
<span id="cb83-17"><a href="#cb83-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-18"><a href="#cb83-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-19"><a href="#cb83-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># subset edges:</span></span>
<span id="cb83-20"><a href="#cb83-20" aria-hidden="true" tabindex="-1"></a>  groups <span class="ot">&lt;-</span> <span class="fu">unique</span>(edge_df<span class="sc">$</span>color)</span>
<span id="cb83-21"><a href="#cb83-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-22"><a href="#cb83-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get top strongest edges</span></span>
<span id="cb83-23"><a href="#cb83-23" aria-hidden="true" tabindex="-1"></a>    temp <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(groups, <span class="cf">function</span>(cur_group){</span>
<span id="cb83-24"><a href="#cb83-24" aria-hidden="true" tabindex="-1"></a>      cur_df <span class="ot">&lt;-</span> edge_df <span class="sc">%&gt;%</span> <span class="fu">subset</span>(color <span class="sc">==</span> cur_group)</span>
<span id="cb83-25"><a href="#cb83-25" aria-hidden="true" tabindex="-1"></a>      n_edges <span class="ot">&lt;-</span> <span class="fu">nrow</span>(cur_df)</span>
<span id="cb83-26"><a href="#cb83-26" aria-hidden="true" tabindex="-1"></a>      cur_df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">top_n</span>(<span class="fu">round</span>(n_edges <span class="sc">*</span> .<span class="dv">5</span>), <span class="at">wt=</span>value)</span>
<span id="cb83-27"><a href="#cb83-27" aria-hidden="true" tabindex="-1"></a>    }))</span>
<span id="cb83-28"><a href="#cb83-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-29"><a href="#cb83-29" aria-hidden="true" tabindex="-1"></a>  edge_df <span class="ot">&lt;-</span> temp</span>
<span id="cb83-30"><a href="#cb83-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">dim</span>(edge_df))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 248   4</code></pre>
</div>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>    edge_df <span class="ot">&lt;-</span> edge_df <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(color) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">value=</span>hdWGCNA<span class="sc">::</span><span class="fu">scale01</span>(value))</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>  edge_df<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(edge_df), <span class="cf">function</span>(i){</span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>    a <span class="ot">=</span> edge_df<span class="sc">$</span>value[i]</span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if(edge_df$value[i] &lt; 0.05){a=0.05}</span></span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">alpha</span>(edge_df<span class="sc">$</span>color[i], <span class="at">alpha=</span>a)</span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a>  g <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">graph_from_data_frame</span>(</span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a>    edge_df,</span>
<span id="cb85-12"><a href="#cb85-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">directed=</span><span class="cn">FALSE</span>,</span>
<span id="cb85-13"><a href="#cb85-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertices=</span>selected_modules</span>
<span id="cb85-14"><a href="#cb85-14" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb85-15"><a href="#cb85-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-16"><a href="#cb85-16" aria-hidden="true" tabindex="-1"></a>  l <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">layout_with_fr</span>(g,)</span>
<span id="cb85-17"><a href="#cb85-17" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(</span>
<span id="cb85-18"><a href="#cb85-18" aria-hidden="true" tabindex="-1"></a>    g, <span class="at">layout=</span>l,</span>
<span id="cb85-19"><a href="#cb85-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">edge.color=</span><span class="fu">adjustcolor</span>(<span class="fu">E</span>(g)<span class="sc">$</span>color, <span class="at">alpha.f=</span>.<span class="dv">25</span>),</span>
<span id="cb85-20"><a href="#cb85-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.size=</span><span class="fu">V</span>(g)<span class="sc">$</span>size,</span>
<span id="cb85-21"><a href="#cb85-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">edge.curved=</span><span class="dv">0</span>,</span>
<span id="cb85-22"><a href="#cb85-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">edge.width=</span><span class="fl">0.5</span>,</span>
<span id="cb85-23"><a href="#cb85-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.color=</span><span class="fu">V</span>(g)<span class="sc">$</span>color,</span>
<span id="cb85-24"><a href="#cb85-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.frame.color=</span><span class="fu">V</span>(g)<span class="sc">$</span>color,</span>
<span id="cb85-25"><a href="#cb85-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.label=</span><span class="fu">V</span>(g)<span class="sc">$</span>label,</span>
<span id="cb85-26"><a href="#cb85-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.label.family=</span><span class="st">'Helvetica'</span>, <span class="co">#vertex.label.font=vertex_df$font,</span></span>
<span id="cb85-27"><a href="#cb85-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.label.font =</span> <span class="dv">3</span>,</span>
<span id="cb85-28"><a href="#cb85-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.label.color =</span> <span class="fu">V</span>(g)<span class="sc">$</span>fontcolor,</span>
<span id="cb85-29"><a href="#cb85-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.label.cex=</span>.<span class="dv">5</span></span>
<span id="cb85-30"><a href="#cb85-30" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_combinedNetworks_files/figure-html/unnamed-chunk-19-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>hub_list <span class="ot">=</span> hubs5</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(hub_list) <span class="ot">=</span> this_net<span class="sc">$</span>colors[hubs5]</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>hub_labels <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unlist</span>(hub_list))</span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>g<span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">graph_from_data_frame</span>(edge_df, <span class="at">directed=</span>F)</span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(g, <span class="at">layout=</span> <span class="fu">as.matrix</span>(umap_df[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_combinedNetworks_files/figure-html/unnamed-chunk-19-4.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="isoexpr.m11-hairball" class="level4">
<h4 class="anchored" data-anchor-id="isoexpr.m11-hairball">IsoExpr.M11 Hairball</h4>
<div class="cell" data-hash="Figure4_combinedNetworks_cache/html/unnamed-chunk-20_5e854f9b5bffaaec53779676633d2966">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>this_datExpr <span class="ot">=</span> datExpr.isoExpr</span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>this_net <span class="ot">=</span> net.isoExpr</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>this_net<span class="sc">$</span>colors <span class="ot">=</span> <span class="fu">labels2colors</span>(this_net<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(this_net<span class="sc">$</span>colors) <span class="ot">=</span> <span class="fu">names</span>(this_net<span class="sc">$</span>cut2<span class="sc">$</span>labels)</span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>MEs <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(<span class="fu">t</span>(this_datExpr), <span class="at">colors=</span>this_net<span class="sc">$</span>colors)</span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>kMEtable <span class="ot">=</span> <span class="fu">signedKME</span>(<span class="fu">t</span>(this_datExpr), <span class="at">datME =</span>MEs<span class="sc">$</span>eigengenes,<span class="at">corFnc =</span> <span class="st">"bicor"</span>)</span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a>this_anno <span class="ot">=</span> datAnno[<span class="fu">match</span>(<span class="fu">rownames</span>(this_datExpr), datAnno<span class="sc">$</span>transcript_id),]</span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>module <span class="ot">=</span> this_net<span class="sc">$</span>module</span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>module.color <span class="ot">=</span> this_net<span class="sc">$</span>colors</span>
<span id="cb87-11"><a href="#cb87-11" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>module.number <span class="ot">=</span> this_net<span class="sc">$</span>module.number</span>
<span id="cb87-12"><a href="#cb87-12" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">=</span> <span class="fu">grep</span>(<span class="st">"^TALON"</span>, this_anno<span class="sc">$</span>transcript_name)</span>
<span id="cb87-13"><a href="#cb87-13" aria-hidden="true" tabindex="-1"></a>this_anno<span class="sc">$</span>transcript_name[idx] <span class="ot">=</span> <span class="fu">paste0</span>(this_anno<span class="sc">$</span>gene_name[idx], <span class="st">'_'</span>, this_anno<span class="sc">$</span>transcript_name[idx])</span>
<span id="cb87-14"><a href="#cb87-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-15"><a href="#cb87-15" aria-hidden="true" tabindex="-1"></a>modules_to_plot <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"greenyellow"</span>)</span>
<span id="cb87-16"><a href="#cb87-16" aria-hidden="true" tabindex="-1"></a><span class="co">#modules_to_plot = unique(this_net$colors)</span></span>
<span id="cb87-17"><a href="#cb87-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-18"><a href="#cb87-18" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">=</span> <span class="fu">tibble</span>(<span class="at">transcript_id =</span> <span class="fu">rownames</span>(kMEtable), <span class="at">module=</span>this_net<span class="sc">$</span>colors,</span>
<span id="cb87-19"><a href="#cb87-19" aria-hidden="true" tabindex="-1"></a>                kMEtable)</span>
<span id="cb87-20"><a href="#cb87-20" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(transcript_id,module), <span class="at">names_to =</span> <span class="st">"kME_to_module"</span>, <span class="at">values_to =</span> <span class="st">'kME'</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">kMEtoMod =</span> <span class="fu">gsub</span>(<span class="st">"kME"</span>, <span class="st">""</span>, kME_to_module)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module<span class="sc">!=</span><span class="dv">0</span>, module <span class="sc">!=</span> <span class="st">"grey"</span>, kMEtoMod <span class="sc">!=</span><span class="st">'grey'</span>, kMEtoMod <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>kME_to_module)</span>
<span id="cb87-21"><a href="#cb87-21" aria-hidden="true" tabindex="-1"></a>tidyMods <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(datAnno)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "transcript_id"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>hubs100 <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module <span class="sc">%in%</span> modules_to_plot, kMEtoMod <span class="sc">==</span> module) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">100</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(transcript_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Adding missing grouping variables: `module`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>hubs25 <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module <span class="sc">%in%</span> modules_to_plot, kMEtoMod <span class="sc">==</span> module) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">25</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(transcript_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Adding missing grouping variables: `module`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a>hubs5 <span class="ot">&lt;-</span> tidyMods <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module <span class="sc">%in%</span> modules_to_plot, kMEtoMod <span class="sc">==</span> module) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(kME,<span class="at">n=</span><span class="dv">5</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(transcript_id) <span class="sc">%&gt;%</span> <span class="fu">pull</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Adding missing grouping variables: `module`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>isoTOM <span class="ot">=</span> <span class="fu">TOMsimilarityFromExpr</span>(<span class="fu">t</span>(this_datExpr[hubs25,]),<span class="at">networkType =</span> <span class="st">'signed'</span>, <span class="at">power =</span> <span class="dv">14</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>TOM calculation: adjacency..
..will not use multithreading.
 Fraction of slow calculations: 0.000000
..connectivity..
..matrix multiplication (system BLAS)..
..normalization..
..done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(isoTOM) <span class="ot">=</span> <span class="fu">rownames</span>(isoTOM) <span class="ot">=</span> hubs25</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>df<span class="ot">=</span>uwot<span class="sc">::</span><span class="fu">umap</span>(isoTOM,<span class="at">n_neighbors =</span> <span class="dv">25</span>,<span class="at">metric =</span> <span class="st">'cosine'</span>,<span class="at">spread =</span> <span class="dv">20</span>,<span class="at">min_dist =</span> .<span class="dv">4</span>)</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_combinedNetworks_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>umap_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(df)</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(umap_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"UMAP1"</span>, <span class="st">"UMAP2"</span>)</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>umap_df<span class="sc">$</span>gene <span class="ot">&lt;-</span> hubs25</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>umap_df<span class="sc">$</span>module <span class="ot">&lt;-</span> this_net<span class="sc">$</span>colors[umap_df<span class="sc">$</span>gene]</span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>umap_df<span class="sc">$</span>color <span class="ot">&lt;-</span> net.isoExpr<span class="sc">$</span>cut2<span class="sc">$</span>labels[umap_df<span class="sc">$</span>gene]</span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a>umap_df<span class="sc">$</span>hub <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a>    umap_df<span class="sc">$</span>gene <span class="sc">%in%</span> <span class="fu">as.character</span>(<span class="fu">unlist</span>(hubs5)), <span class="st">'hub'</span>, <span class="st">'other'</span></span>
<span id="cb98-8"><a href="#cb98-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb98-9"><a href="#cb98-9" aria-hidden="true" tabindex="-1"></a>umap_df <span class="ot">&lt;-</span> umap_df <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(tidyMods <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module <span class="sc">==</span> kMEtoMod) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">gene=</span>transcript_id, module, kME), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"module"</span>, <span class="st">"gene"</span>))</span>
<span id="cb98-10"><a href="#cb98-10" aria-hidden="true" tabindex="-1"></a>umap_df <span class="ot">&lt;-</span> umap_df <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(datAnno <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name, transcript_id), <span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span> <span class="ot">=</span> <span class="st">"transcript_id"</span>))</span>
<span id="cb98-11"><a href="#cb98-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-12"><a href="#cb98-12" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> <span class="fu">levels</span>(<span class="fu">factor</span>(umap_df<span class="sc">$</span>module))</span>
<span id="cb98-13"><a href="#cb98-13" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> mods[mods <span class="sc">!=</span> <span class="st">'grey'</span>]</span>
<span id="cb98-14"><a href="#cb98-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-15"><a href="#cb98-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(umap_df, <span class="fu">aes</span>(<span class="at">x=</span>UMAP1, <span class="at">y=</span>UMAP2)) <span class="sc">+</span></span>
<span id="cb98-16"><a href="#cb98-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(</span>
<span id="cb98-17"><a href="#cb98-17" aria-hidden="true" tabindex="-1"></a>   <span class="at">color=</span>umap_df<span class="sc">$</span>module, <span class="co"># color each point by WGCNA module</span></span>
<span id="cb98-18"><a href="#cb98-18" aria-hidden="true" tabindex="-1"></a>   <span class="at">size=</span>umap_df<span class="sc">$</span>kME <span class="co"># size of each point based on intramodular connectivity</span></span>
<span id="cb98-19"><a href="#cb98-19" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb98-20"><a href="#cb98-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">umap_theme</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure4_combinedNetworks_files/figure-html/unnamed-chunk-20-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>subset_TOM <span class="ot">&lt;-</span> isoTOM[umap_df<span class="sc">$</span>gene, umap_df<span class="sc">$</span>gene[umap_df<span class="sc">$</span>hub <span class="sc">==</span> <span class="st">'hub'</span>]]</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>g1 <span class="ot">=</span> <span class="fu">graph.adjacency</span>(isoTOM[umap_df<span class="sc">$</span>gene, umap_df<span class="sc">$</span>gene], <span class="at">mode=</span><span class="st">'undirected'</span>, <span class="at">diag=</span>F,<span class="at">weighted =</span> T)</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>layoutMat <span class="ot">&lt;-</span> <span class="fu">layout.fruchterman.reingold</span>(g1)</span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cairo_pdf</span>(<span class="at">file=</span><span class="st">'output/figures/Fig4/IsoExpr_hairballs.pdf'</span>, <span class="at">width=</span><span class="dv">20</span>,<span class="at">height=</span><span class="dv">20</span>,<span class="at">family=</span><span class="st">'serif'</span>)</span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot.igraph</span>(g1, <span class="at">layout=</span><span class="fu">as.matrix</span>(umap_df[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)]),</span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">vertex.label =</span> <span class="fu">paste0</span>(umap_df<span class="sc">$</span>gene_name,<span class="st">"_"</span>, <span class="fu">substr</span>(umap_df<span class="sc">$</span>gene,<span class="dv">1</span>,<span class="dv">15</span>)),</span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">vertex.family=</span><span class="st">'serif'</span>,</span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">edge.curved=</span><span class="dv">0</span>, <span class="at">edge.alpha=</span>.<span class="dv">5</span>,</span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">edge.color =</span> <span class="fu">ifelse</span>(<span class="fu">E</span>(g1)<span class="sc">$</span>weight <span class="sc">&gt;</span> .<span class="dv">25</span>, <span class="st">"grey"</span>, <span class="cn">NA</span>),</span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">edge.width =</span> <span class="fu">E</span>(g1)<span class="sc">$</span>weight,</span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">vertex.label.cex=</span><span class="fl">0.5</span>, <span class="at">vertex.color =</span> umap_df<span class="sc">$</span>module,</span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">vertex.size=</span>umap_df<span class="sc">$</span>kME<span class="sc">^</span><span class="dv">5</span><span class="sc">*</span><span class="dv">4</span>)</span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>